diff --git a/src/nm-policy-hosts.c b/src/nm-policy-hosts.c
index 7f9cff8..1d8b493 100644
--- a/src/nm-policy-hosts.c
+++ b/src/nm-policy-hosts.c
@@ -483,6 +483,7 @@ nm_policy_hosts_update_etc_hosts (const char *hostname,
                                   const char *ip6_addr,
                                   gboolean *out_changed)
 {
+	return TRUE;
 	char *contents = NULL;
 	char **lines = NULL;
 	GError *error = NULL;
