# Calculate format=diff
diff --git a/src/widgets/styles/qcommonstyle.cpp b/src/widgets/styles/qcommonstyle.cpp
index 7420bfb..c971836 100644
--- a/src/widgets/styles/qcommonstyle.cpp
+++ b/src/widgets/styles/qcommonstyle.cpp
@@ -5464,6 +5464,12 @@ QPixmap QCommonStyle::standardPixmap(StandardPixmap sp, const QStyleOption *opti
                 pixmap = QIcon::fromTheme(QLatin1String("go-previous"),
                                           QIcon::fromTheme(QLatin1String("back"))).pixmap(16);
                 break;
+        case SP_FileDialogContentsView:
+                pixmap = QIcon::fromTheme(QLatin1String("view_contents")).pixmap(16);
+                break;
+        case SP_FileDialogInfoView:
+                pixmap = QIcon::fromTheme(QLatin1String("view_info")).pixmap(16);
+                break;
         case SP_FileDialogDetailedView:
                 pixmap = QIcon::fromTheme(QLatin1String("view_detailed")).pixmap(16);
                 break;
@@ -5884,8 +5890,14 @@ QIcon QCommonStyle::standardIcon(StandardPixmap standardIcon, const QStyleOption
                 icon = QIcon::fromTheme(QLatin1String("view-list-details"));
                 break;
         case SP_FileDialogListView:
+                icon = QIcon::fromTheme(QLatin1String("view-list-text"));
+                break;
+        case SP_FileDialogContentsView:
                 icon = QIcon::fromTheme(QLatin1String("view-list-icons"));
                 break;
+        case SP_FileDialogInfoView:
+                icon = QIcon::fromTheme(QLatin1String("view-list-images"));
+                break;
         case SP_BrowserReload:
                 icon = QIcon::fromTheme(QLatin1String("view-refresh"));
                 break;
