--- kdecore/io/klockfile_unix.cpp	(revision 861152)
+++ kdecore/io/klockfile_unix.cpp	(working copy)
@@ -173,7 +173,9 @@ static KLockFile::LockResult lockFile(co
   if (st_buf != st_buf2 || S_ISLNK(st_buf.st_mode) || S_ISLNK(st_buf2.st_mode))
   {
      // SMBFS supports hardlinks by copying the file, as a result the above test will always fail
-     if ((st_buf.st_nlink == 1) && (st_buf2.st_nlink == 1) && (st_buf.st_ino != st_buf2.st_ino))
+     if ( (((st_buf.st_nlink == 1) && (st_buf2.st_nlink == 1)) ||
+	   ((st_buf.st_nlink == 2) && (st_buf2.st_nlink == 2))) // DF: for #156759, but no idea why
+	  && (st_buf.st_ino != st_buf2.st_ino))
      {
         linkCountSupport = testLinkCountSupport(uniqueName);
         if (!linkCountSupport)
