# Calculate format=diff
diff --git a/transport/http.py b/transport/http.py
index e2a2717..ee539c9 100644
--- a/transport/http.py
+++ b/transport/http.py
@@ -304,7 +304,7 @@ class HttpTransport(Transport):
                 # the workaround breaks things on other systems, so it is applied
                 # on an on-demand basis
                 log.warning("Activating SSL workaround")
-                CheckingHTTPSConnection.FORCE_SSL_VERSION = ssl.PROTOCOL_SSLv3
+                CheckingHTTPSConnection.FORCE_SSL_VERSION = ssl.PROTOCOL_SSLv23
               elif "ASN1_item_verify:unknown message digest algorithm" in str(e):
                 # this bug was reported but is present for now
                 # see: http://bugs.python.org/issue8484
diff --git a/transport/pyopenssl_wrapper.py b/transport/pyopenssl_wrapper.py
index 790b6d9..35fd8bd 100644
--- a/transport/pyopenssl_wrapper.py
+++ b/transport/pyopenssl_wrapper.py
@@ -11,8 +11,8 @@ import _ssl             # if we can't import it, let the error propagate
 
 from _ssl import SSLError
 from _ssl import CERT_NONE, CERT_OPTIONAL, CERT_REQUIRED
-from _ssl import PROTOCOL_SSLv3, PROTOCOL_SSLv23, PROTOCOL_TLSv1
-from _ssl import RAND_status, RAND_egd, RAND_add
+from _ssl import PROTOCOL_SSLv23, PROTOCOL_TLSv1
+from _ssl import RAND_status, RAND_add
 from _ssl import \
      SSL_ERROR_ZERO_RETURN, \
      SSL_ERROR_WANT_READ, \
@@ -39,7 +39,6 @@ _ssl_to_openssl_cert_op_remap = {
   }
   
 _ssl_to_openssl_version_remap = {
-  PROTOCOL_SSLv3: OpenSSL.SSL.SSLv3_METHOD, 
   PROTOCOL_SSLv23: OpenSSL.SSL.SSLv23_METHOD, 
   PROTOCOL_TLSv1: OpenSSL.SSL.TLSv1_METHOD,                                   
   }
