# Calculate format=diff
diff --git a/tdtd-font-maker.el b/tdtd-font-maker.el
index 30dd5fc..e0981a0 100644
--- a/tdtd-font-maker.el
+++ b/tdtd-font-maker.el
@@ -875,7 +875,7 @@
   (kill-region (point-min) (point-max))
   (insert ";;;; tdtd-font.el --- Font-lock keywords for Tony's DTD mode\n")
   (insert ";; $\Id$\n")
-  (insert ";; $\Name$\n")
+  (insert ";; $\\Name$\n")
   (insert ";;\n")
   (insert ";; Copyright (C) 1999, Tony Graham\n")
   (insert ";;\n")
