# Calculate format=diff
diff --git a/src/lightdm-gtk-greeter.c b/src/lightdm-gtk-greeter.c
index aa1c4cb..f0fe46f 100644
--- a/src/lightdm-gtk-greeter.c
+++ b/src/lightdm-gtk-greeter.c
@@ -2832,7 +2832,7 @@ main (int argc, char **argv)
     {
         Display *display = gdk_x11_display_get_xdisplay (gdk_display_get_default ());
         XGetScreenSaver (display, &timeout, &interval, &prefer_blanking, &allow_exposures);
-        XForceScreenSaver (display, ScreenSaverActive);
+        //XForceScreenSaver (display, ScreenSaverActive);
         XSetScreenSaver (display, config_get_int (NULL, CONFIG_KEY_SCREENSAVER_TIMEOUT, 60), 0,
                          ScreenSaverActive, DefaultExposures);
     }
