diff --git a/src/connection-editor/nm-connection-editor.c b/src/connection-editor/nm-connection-editor.c
index c054642..ae3aeca 100644
--- a/src/connection-editor/nm-connection-editor.c
+++ b/src/connection-editor/nm-connection-editor.c
@@ -164,10 +164,10 @@ update_sensitivity (NMConnectionEditor *editor)
 	 */
 	if (   !nm_setting_connection_get_read_only (s_con)
 	    && editor_is_initialized (editor)) {
-		if (editor->system_settings_can_modify) {
+		//if (editor->system_settings_can_modify) {
 			actionable = ce_polkit_button_get_actionable (CE_POLKIT_BUTTON (editor->ok_button));
 			authorized = ce_polkit_button_get_authorized (CE_POLKIT_BUTTON (editor->ok_button));
-		}
+		//}
 
 		if (editor->orig_scope == NM_CONNECTION_SCOPE_SYSTEM) {
 			/* If the user cannot ever be authorized to change system connections, and
