diff --git pym/cl_kernel.py pym/cl_kernel.py
index 456efad..4efd797 100644
--- pym/cl_kernel.py
+++ pym/cl_kernel.py
@@ -398,8 +398,10 @@ class cl_kernel(color_print):
                                   ' calculate=video:%s'%x11video,y,1)
 
         # if has grub2
-        if filter(lambda x:x.startswith("grub-1.99"),
-           listDirectory('/var/db/pkg/sys-boot')):
+        if filter(lambda x:x.startswith("grub-1.99") or \
+                           x.startswith("grub-2"),
+           listDirectory('/var/db/pkg/sys-boot')) and \
+           os.environ["EBUILD_PHASE"] == '':
             grubInstall = process('/usr/sbin/cl-core','--method','core_setup',
                                   '--pkg-name','grub',
                                   '--no-progress',stderr=None)
