commit cc5e5dea882cd27db72805344e0328bb50b3fc9e
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Mar 22 08:49:42 2013 +0400

    Fix prepare isohybrid for i686

diff --git pym/cl_builder.py pym/cl_builder.py
index af75022..19480cd 100644
--- pym/cl_builder.py
+++ pym/cl_builder.py
@@ -647,7 +647,11 @@ class cl_builder(color_print):
         self.printByResult(True)
         if self.clVars.Get('cl_builder_isohybrid_set') == 'on':
             self.printMessageForTest(_("Performing isohybrid"))
-            processIsoHybrid = process("/usr/bin/isohybrid","--uefi",isoFile)
+            if self.clVars.Get('os_builder_arch_machine') == 'x86_64':
+                processIsoHybrid = process("/usr/bin/isohybrid",
+                                           "--uefi",isoFile)
+            else:
+                processIsoHybrid = process("/usr/bin/isohybrid",isoFile)
             self.printByResult(processIsoHybrid.success())
         self.printMessageForTest(_("Generating the list of installed packages"))
         self.printByResult(self.createPackageList(sourceDirectory,
