commit 3229268d1f24a02ab72a15a89a8a4e100c7effc9
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Mar 25 09:41:15 2013 +0400

    Discard clt templates from prepare image

diff --git pym/cl_builder.py pym/cl_builder.py
index 19480cd..56228f2 100644
--- pym/cl_builder.py
+++ pym/cl_builder.py
@@ -127,7 +127,7 @@ class cl_builder(color_print):
         #self.clVars.Set("cl_root_path","/", True)
         self.clVars.Set("cl_action","squash", True)
         self.clVars.Set("cl_chroot_path",directory, True)
-        self.clTempl = template(self.clVars)
+        self.clTempl = template(self.clVars,cltObj=False)
         dirsFiles = self.clTempl.applyTemplates()
         self.createDevNodes(directory)
         self.clTempl.closeFiles()
@@ -144,7 +144,7 @@ class cl_builder(color_print):
             self.clVars.Get('cl_builder_squash_path'))
         self.clVars.Set("cl_chroot_path",chrootPath, True)
         self.clVars.Set("cl_root_path","..", True)
-        self.clTempl = template(self.clVars)
+        self.clTempl = template(self.clVars,cltObj=False)
         dirsFiles = self.clTempl.applyTemplates()
         self.clTempl.closeFiles()
         if self.clTempl.getError():
