commit bf7398c148201a0d573c8680f6887cb5e9b8b3c3
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Jun 25 11:25:31 2012 +0400

    Fix for using 12.0 themes

diff --git a/pym/cl_kernel.py b/pym/cl_kernel.py
index 69454e5..b5f37cb 100644
--- a/pym/cl_kernel.py
+++ b/pym/cl_kernel.py
@@ -22,7 +22,7 @@ import traceback
 from os import path
 from cl_builder import DataVarsBuilder, printNoColor, __version__, __app__
 from cl_utils import process,listDirectory,cmpVersion,reVerSplitToPV, \
-                     reVerSplit
+                     reVerSplit,getPkgInstalled
 from cl_vars_share import varsShare
 from subprocess import STDOUT,PIPE
 from cl_print import color_print
@@ -96,7 +96,20 @@ class cl_kernel(color_print):
                    mrproper=False,target="all"):
         """Run kernel compilation"""
         clVars = self.clVars
-        standardParams = ["--splash=tty1", "--unionfs",
+        for themePkg in ['media-gfx/cldx-themes',
+                         'media-gfx/cld-themes',
+                         'media-gfx/cldg-themes',
+                         'media-gfx/cls-themes',
+                         'media-gfx/cds-themes',
+                         'media-gfx/cmc-themes']:
+            installed = getPkgInstalled(themePkg,
+                                       prefix=self.clVars.Get('cl_chroot_path'))
+            if installed and installed[0]['PV'].startswith('12'):
+                themeName = "calculate"
+                break
+        else:
+            themeName = "tty1"
+        standardParams = ["--splash=%s"%themeName, "--unionfs",
                           "--all-ramdisk-modules","--disklabel",
                           "--no-save-config", "--firmware","--udev",
                           "--lspci"]
