diff --git a/setup.py b/setup.py
index f2306c1..6c4677a 100755
--- a/setup.py
+++ b/setup.py
@@ -62,13 +62,12 @@ def create_data_files(data_dirs, prefix="", replace_dirname=""):
     return data_files
 
 
-data_files += [('/usr/share/calculate/config', ['data/chroot.rc']),
-               ('/etc/init.d', ['data/calculate-builder'])]
+data_files += [('/usr/share/calculate/config', ['data/chroot.rc'])]
 
 class cl_install_data(install_data):
     def run (self):
         install_data.run(self)
-        data_file = [("/etc/init.d/calculate-builder",0755)]
+        data_file = []
         fileNames = map(lambda x: os.path.split(x[0])[1], data_file)
         listNames = map(lambda x: filter(lambda y: y, x[0].split("/")),data_file)
         data_find = {}
