commit f904ae0aa3925665a112011e8f8729bfb02ff1e9
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Jun 26 16:33:36 2012 +0400

    Fix grub configuration

diff --git a/pym/cl_kernel.py b/pym/cl_kernel.py
index b5f37cb..9dccc2b 100644
--- a/pym/cl_kernel.py
+++ b/pym/cl_kernel.py
@@ -22,12 +22,11 @@ import traceback
 from os import path
 from cl_builder import DataVarsBuilder, printNoColor, __version__, __app__
 from cl_utils import process,listDirectory,cmpVersion,reVerSplitToPV, \
-                     reVerSplit,getPkgInstalled
+                     reVerSplit,getPkgInstalled,PIPE,STDOUT
 from cl_vars_share import varsShare
 from subprocess import STDOUT,PIPE
 from cl_print import color_print
 from shutil import copy2 as copy_with_perm
-from cl_apply_template import updateConfFiles
 
 from cl_kernel_utils import KernelConfig,InitRamFs
 
@@ -402,8 +401,9 @@ class cl_kernel(color_print):
         # if has grub2
         if filter(lambda x:x.startswith("grub-1.99"),
            listDirectory('/var/db/pkg/sys-boot')):
-            clTemplate = updateConfFiles()
-            clTemplate.applyAllTemplates(package="grub")
+            grubInstall = process('/usr/sbin/cl-core','--method','core_setup',
+                                  '--pkg-name','grub',stderr=None)
+            grubInstall.success()
 
         if not filter(lambda x:x.startswith("grub-0.9"),
            listDirectory('/var/db/pkg/sys-boot')) or not path.exists(grubconf):
