commit d1fda7db0d9a21d67d88f3e4fa005d446e8434d0
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Aug 19 11:05:56 2014 +0400

    Исправлено вычисление переменных

diff --git pym/cl_fill_builder.py pym/cl_fill_builder.py
index 421b58b..e6c140f 100644
--- pym/cl_fill_builder.py
+++ pym/cl_fill_builder.py
@@ -430,8 +430,9 @@ class fillVars(object, varsShare):
             imagepath = [imagepath]
         paths = filter(lambda x:os.access(x,W_OK), imagepath)
         if paths:
+            shortname = self.Get('os_builder_linux_shortname') or ""
             return path.join(paths[0],"%(shortname)s-%(ver)s-%(march)s.iso"%
-                 {'shortname':self.Get('os_builder_linux_shortname').lower(),
+                 {'shortname':shortname.lower(),
                   'ver':self.Get('os_builder_linux_build') or 
                         self.Get('os_builder_linux_ver'),
                   'march':self.Get('os_builder_arch_machine')})
