commit 66d71469ba6f4e007190fe5ff29247f81b8076e2
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jun 22 14:41:12 2012 +0400

    Fix create urandom device

diff --git a/pym/cl_builder.py b/pym/cl_builder.py
index de9fbf5..fd3d922 100644
--- a/pym/cl_builder.py
+++ b/pym/cl_builder.py
@@ -287,7 +287,7 @@ class cl_builder(color_print):
         createdDev = []
         for node,mode,dmode,major,minor in [("urandom",0666,stat.S_IFCHR,1,9)]:
             try:
-                nodePath = path.join(devPath,node)
+                nodePath = path.join(devPath,"dev",node)
                 if not path.exists(nodePath):
                     os.mknod(nodePath,mode|dmode,os.makedev(major,minor))
                     os.chmod(nodePath,mode)
