commit 7648a7d181fc527baa1352211ad11a1b5e64ab99
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Aug 29 13:22:21 2012 +0400

    Discard progress bar for update grub during kernel building

diff --git pym/cl_kernel.py pym/cl_kernel.py
index 9dccc2b..b6e3c35 100644
--- pym/cl_kernel.py
+++ pym/cl_kernel.py
@@ -402,7 +402,8 @@ class cl_kernel(color_print):
         if filter(lambda x:x.startswith("grub-1.99"),
            listDirectory('/var/db/pkg/sys-boot')):
             grubInstall = process('/usr/sbin/cl-core','--method','core_setup',
-                                  '--pkg-name','grub',stderr=None)
+                                  '--pkg-name','grub',
+                                  '--no-progress',stderr=None)
             grubInstall.success()
 
         if not filter(lambda x:x.startswith("grub-0.9"),
