commit a49cdd85515ec142378b2ed49e7b867cc8366974
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Aug 2 15:20:24 2013 +0400

    Новый portage требует наличие layout.conf

diff --git pym/cl_builder.py pym/cl_builder.py
index 56228f2..efae85e 100644
--- pym/cl_builder.py
+++ pym/cl_builder.py
@@ -580,8 +580,12 @@ class cl_builder(color_print):
             portagePath = pathJoin(sourceDirectory,"usr/portage")
             overlayPath = pathJoin(sourceDirectory,"var/lib/layman/calculate")
             return map(lambda x: path.join("var/lib/layman/calculate",x),
-                  filter(lambda x: not x in ("distfiles", "eclass", ".git","profiles"),
+                  filter(lambda x: not x in ("metadata","distfiles", "eclass",
+                                             ".git","profiles"),
                   listDirectory(overlayPath))) + \
+                  map(lambda x:path.join(overlayPath,"metadata",x),
+                  filter(lambda x: x != "layout.conf",
+                  listDirectory(path.join(overlayPath,'metadata')))) + \
                   [pathJoin(sourceDirectory,
                     "var/lib/layman/calculate/profiles/templates/deprecated")] + \
                   map(lambda x: path.join("usr/portage",x),
