diff --git a/pym/cl_fill_builder.py b/pym/cl_fill_builder.py
index a56a518..7f3d13d 100644
--- a/pym/cl_fill_builder.py
+++ b/pym/cl_fill_builder.py
@@ -310,7 +310,7 @@ class fillVars(object, varsShare):
                                        x.split("=")[0].strip() == "os_linux_ver" and\
                                        x.split("=")[1].strip(), data))
             if verList:
-                reVer=re.compile("^((\d+\.)*\d|\d+)$",re.S)
+                reVer=re.compile("^((\d+\.)*\d+|\d+)$",re.S)
                 reRes = filter(reVer.search,verList)
                 if reRes:
                     return reRes[0]
