commit cdb5384a328bd1a7bb6d7d04afadcd1227c4a0ec
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Dec 27 18:23:09 2010 +0300

    Disable splash themes in initramfs.

diff --git a/pym/cl_kernel.py b/pym/cl_kernel.py
index 1afd1c2..a5a4544 100644
--- a/pym/cl_kernel.py
+++ b/pym/cl_kernel.py
@@ -63,7 +63,7 @@ class cl_kernel(color_print):
                    mrproper=False):
         """Run kernel compilation"""
         clVars = self.clVars
-        standardParams = ["--splash=tty1", "--unionfs",
+        standardParams = ["--no-splash", "--unionfs",
                           "--all-ramdisk-modules","--disklabel",
                           "--slowusb", "--no-save-config"]
         kernelDir = ["--kerneldir=%s"%clVars.Get('cl_kernel_src_path')]
