commit 8aef3924330d4346a510cb425cc7623937c093f5
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Thu Dec 13 12:13:51 2012 +0400

    Fix cl-builder for work with grub2

diff --git scripts/cl-builder scripts/cl-builder
index 3715f17..150e360 100644
--- scripts/cl-builder
+++ scripts/cl-builder
@@ -207,13 +207,17 @@ update_from_builder() {
 		sed -nr 's/.*\[.\]\s//p')
 	if [[ "$roottype" != "livecd" ]]
 	then
-		local curkernellink=$(readlink /boot/vmlinuz-$(cl-kernel -v \
-			--filter cl_kernel_uid | sed -nr 's/.*\[.\]\s//p'))
-		local newkernellink=vmlinuz-$(cl-kernel -v --filter cl_kernel_suffix | \
+		local osRootDev=$(cl-kernel -v --filter os_root_dev | \
 			sed -nr 's/.*\[.\]\s//p')
-
-		[[ "$curkernellink" != "$newkernellink" ]] && \
-			echo "For used the new kernel in the workspace perform 'cl-kernel --symlink'."
+		rootUUID=$(blkid $osRootDev | sed -r 's/^.*UUID="([^"]+)".*/\1/')
+		if ! diff -q <( cat /boot/grub/grub.cfg | grep linux | \
+				grep -e $rootUUID -e $osRootDev| \
+				grep -oP /boot/vmlinuz\\S+ | sort ) \
+			<( ls -1 /boot/vmlinuz* | sort ) &>/dev/null
+		then
+			echo "Detects a kernel is not used in grub.cfg"
+			echo "Run 'cl-setup-boot' for update grub"
+		fi
 	fi
 }
 
