commit eb9b2e1fe15d034ea462632ace03fd776f10ccec
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Mar 15 14:55:29 2011 +0300

    Add isohybryd action by default.

diff --git a/i18n/cl_builder_ru.mo b/i18n/cl_builder_ru.mo
index 5b0ad74..75acccf 100644
Binary files a/i18n/cl_builder_ru.mo and b/i18n/cl_builder_ru.mo differ
diff --git a/pym/cl_builder.py b/pym/cl_builder.py
index 177f45d..44dc34f 100644
--- a/pym/cl_builder.py
+++ b/pym/cl_builder.py
@@ -480,6 +480,10 @@ class cl_builder(color_print):
         self.printMessageForTest(_("Creating squash image"))
         self.targetDistr.installFrom(self.sourceDistr)
         self.printByResult(True)
+        if self.clVars.Get('cl_builder_isohybrid_set') == 'on':
+            self.printMessageForTest(_("Performing isohybrid"))
+            processIsoHybrid = process("/usr/bin/isohybrid",isoFile)
+            self.printByResult(processIsoHybrid.success())
         self.printMessageForTest(_("Creating installed package list"))
         self.printByResult(self.createPackageList(sourceDirectory,
                            isoFile[:-4]+".list"))
diff --git a/pym/cl_image_cmd.py b/pym/cl_image_cmd.py
index 28d4070..cc00376 100644
--- a/pym/cl_image_cmd.py
+++ b/pym/cl_image_cmd.py
@@ -41,6 +41,9 @@ CMD_OPTIONS = [{'shortOption':"p",
                {'longOption':'notree',
                 'help':_("exclude portage tree from image")
                },
+               {'longOption':'noisohybrid',
+                'help':_("create iso image without isohybrid")
+               },
                {'longOption':"set"},
                {'shortOption':"f",
                 'longOption':"force",
@@ -128,6 +131,9 @@ class image_cmd(share_cmd):
         if values.notree:
             self.logicObj.clVars.Set('cl_builder_tree','off',True)
 
+        if values.noisohybrid:
+            self.logicObj.clVars.Set('cl_builder_isohybrid_set','off',True)
+
         if args and args[0] == "squash":
             self.logicObj.clVars.Set('cl_builder_iso_path','/mnt/flash',True)
         self.optobj.checkVarSyntax(values)
diff --git a/pym/cl_vars_builder.py b/pym/cl_vars_builder.py
index b4b9694..f839fa9 100644
--- a/pym/cl_vars_builder.py
+++ b/pym/cl_vars_builder.py
@@ -100,6 +100,8 @@ class Data:
     # include portage directory from image
     cl_builder_tree = {'value':'on'}
 
+    cl_builder_isohybrid_set = {'value':'on'}
+
     # build system name
     os_builder_linux_name = {}
 
