commit 75dcdf30c69241440668b6b074e198ba0662cbbd
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Wed Mar 14 15:58:17 2012 +0400

    Fix window list and middle click on QuickLaunch

diff --git a/extensions/QuickLaunch@wwwdotorg.org/extension.js b/extensions/QuickLaunch@wwwdotorg.org/extension.js
index 2f68749..daab441 100644
--- a/extensions/QuickLaunch@wwwdotorg.org/extension.js
+++ b/extensions/QuickLaunch@wwwdotorg.org/extension.js
@@ -101,7 +101,7 @@ QLButton.prototype = {
     },
 
     generateMenu: function(animate) {
-        if( global.get_pointer()[2] == 512 ) {
+        if( global.get_pointer()[2] & 512 ) {
             this._focusDesktopActor();
             this.appinfo.launch([], null);
             return;
diff --git a/extensions/windowlist@o2net.cl/extension.js b/extensions/windowlist@o2net.cl/extension.js
index 67fe570..3993b18 100644
--- a/extensions/windowlist@o2net.cl/extension.js
+++ b/extensions/windowlist@o2net.cl/extension.js
@@ -120,6 +120,16 @@ AppMenuButton.prototype = {
 
     },
 
+    handleDragOver : function(source, actor, x, y, time) {
+        Main.testSource = source;
+        if (source == Main.xdndHandler) {
+            this.metaWindow.activate(global.get_current_time());
+            return DND.DragMotionResult.CONTINUE;
+        }
+
+        return DND.DragMotionResult.CONTINUE;
+    },
+
     _onTitleChange: function() {
         this._label.setText(this.metaWindow.get_title());
         //this._label.setText(this.app.get_name());
