# Calculate format=diff
diff --git a/Makefile b/Makefile
index b389275..6050828 100644
--- a/Makefile
+++ b/Makefile
@@ -452,9 +452,9 @@ KRB5_CONFIG =
 LD = /usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
 LDFLAGS = -Wl,--as-needed
 LIBICONV = 
-LIBLZ4_CFLAGS = -I/usr/local/include 
-LIBLZ4_LIBS = -L/usr/local/lib -llz4 
-LIBLZ4_PC = liblz4
+LIBLZ4_CFLAGS =
+LIBLZ4_LIBS =
+LIBLZ4_PC =
 LIBOBJS = 
 LIBP11_CFLAGS = 
 LIBP11_LIBS = 
diff --git a/config.h b/config.h
index eedf4ef..210aea8 100644
--- a/config.h
+++ b/config.h
@@ -128,7 +128,7 @@
 /* #undef HAVE_LIBSTOKEN */
 
 /* LZ4 was found */
-#define HAVE_LZ4 /**/
+/* #undef HAVE_LZ4 */
 
 /* Define to 1 if you have the <memory.h> header file. */
 #define HAVE_MEMORY_H 1
diff --git a/po/Makefile b/po/Makefile
index 13c74ed..4ea8c04 100644
--- a/po/Makefile
+++ b/po/Makefile
@@ -181,9 +181,9 @@ KRB5_CONFIG =
 LD = /usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
 LDFLAGS = -Wl,--as-needed
 LIBICONV = 
-LIBLZ4_CFLAGS = -I/usr/local/include 
-LIBLZ4_LIBS = -L/usr/local/lib -llz4 
-LIBLZ4_PC = liblz4
+LIBLZ4_CFLAGS =
+LIBLZ4_LIBS =
+LIBLZ4_PC =
 LIBOBJS = 
 LIBP11_CFLAGS = 
 LIBP11_LIBS = 
diff --git a/www/Makefile b/www/Makefile
index b4a810c..dd255db 100644
--- a/www/Makefile
+++ b/www/Makefile
@@ -271,9 +271,9 @@ KRB5_CONFIG =
 LD = /usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
 LDFLAGS = -Wl,--as-needed
 LIBICONV = 
-LIBLZ4_CFLAGS = -I/usr/local/include 
-LIBLZ4_LIBS = -L/usr/local/lib -llz4 
-LIBLZ4_PC = liblz4
+LIBLZ4_CFLAGS =
+LIBLZ4_LIBS =
+LIBLZ4_PC =
 LIBOBJS = 
 LIBP11_CFLAGS = 
 LIBP11_LIBS = 
diff --git a/www/images/Makefile b/www/images/Makefile
index 2dd6f7d..797de77 100644
--- a/www/images/Makefile
+++ b/www/images/Makefile
@@ -210,9 +210,9 @@ KRB5_CONFIG =
 LD = /usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
 LDFLAGS = -Wl,--as-needed
 LIBICONV = 
-LIBLZ4_CFLAGS = -I/usr/local/include 
-LIBLZ4_LIBS = -L/usr/local/lib -llz4 
-LIBLZ4_PC = liblz4
+LIBLZ4_CFLAGS =
+LIBLZ4_LIBS =
+LIBLZ4_PC =
 LIBOBJS = 
 LIBP11_CFLAGS = 
 LIBP11_LIBS = 
diff --git a/www/inc/Makefile b/www/inc/Makefile
index fb578a7..78272a1 100644
--- a/www/inc/Makefile
+++ b/www/inc/Makefile
@@ -210,9 +210,9 @@ KRB5_CONFIG =
 LD = /usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
 LDFLAGS = -Wl,--as-needed
 LIBICONV = 
-LIBLZ4_CFLAGS = -I/usr/local/include 
-LIBLZ4_LIBS = -L/usr/local/lib -llz4 
-LIBLZ4_PC = liblz4
+LIBLZ4_CFLAGS =
+LIBLZ4_LIBS =
+LIBLZ4_PC =
 LIBOBJS = 
 LIBP11_CFLAGS = 
 LIBP11_LIBS = 
diff --git a/www/styles/Makefile b/www/styles/Makefile
index 4b9b0d2..7481564 100644
--- a/www/styles/Makefile
+++ b/www/styles/Makefile
@@ -210,9 +210,9 @@ KRB5_CONFIG =
 LD = /usr/x86_64-pc-linux-gnu/bin/ld -m elf_x86_64
 LDFLAGS = -Wl,--as-needed
 LIBICONV = 
-LIBLZ4_CFLAGS = -I/usr/local/include 
-LIBLZ4_LIBS = -L/usr/local/lib -llz4 
-LIBLZ4_PC = liblz4
+LIBLZ4_CFLAGS =
+LIBLZ4_LIBS =
+LIBLZ4_PC =
 LIBOBJS = 
 LIBP11_CFLAGS = 
 LIBP11_LIBS = 
