# Calculate format=diff
diff --git a/microblogs/twitter/twittereditaccount.cpp b/microblogs/twitter/twittereditaccount.cpp
index efea8d5..9f50534 100644
--- a/microblogs/twitter/twittereditaccount.cpp
+++ b/microblogs/twitter/twittereditaccount.cpp
@@ -40,8 +40,8 @@ along with this program; if not, see http://www.gnu.org/licenses/
 #include <QCheckBox>
 #include <KInputDialog>
 
-const char * twitterConsumerKey = "VyXMf0O7CvciiUQjliYtYg";
-const char * twitterConsumerSecret = "uD2HvsOBjzt1Vs6SnouFtuxDeHmvOOVwmn3fBVyCw0";
+const char * twitterConsumerKey = "irARiVrv825Fl2oZaYiQA";
+const char * twitterConsumerSecret = "qYZa8Dszrow6vry4iwtSm1zTidnW7ii8IQztgXWZY";
 
 TwitterEditAccountWidget::TwitterEditAccountWidget(TwitterMicroBlog *microblog,
                                                     TwitterAccount* account, QWidget* parent)
