# Calculate format=diff
diff --git a/src/eap_peer/Makefile b/src/eap_peer/Makefile
index 12e5f2c..584fc66 100644
--- src/eap_peer/Makefile
+++ src/eap_peer/Makefile
@@ -37,13 +37,15 @@ OBJS_both += ../utils/wpa_debug.o
 OBJS_both += ../utils/base64.o
 OBJS_both += ../utils/wpabuf.o
 OBJS_both += ../utils/eloop.o
-OBJS_both += ../crypto/md5.o
 ifneq ($(CONFIG_TLS), openssl)
 OBJS_both += ../crypto/sha1.o
+OBJS_both += ../crypto/md5.o
 endif
 OBJS_both += ../crypto/sha1-tlsprf.o
 OBJS_both += ../crypto/aes-encblock.o
+ifneq ($(CONFIG_TLS), openssl)
 OBJS_both += ../crypto/aes-wrap.o
+endif
 OBJS_both += ../crypto/aes-ctr.o
 OBJS_both += ../crypto/aes-eax.o
 OBJS_both += ../crypto/aes-omac1.o
