diff --git consolegui/application/more.py consolegui/application/more.py
index 0fd9128..dc6d9be 100755
--- consolegui/application/more.py
+++ consolegui/application/more.py
@@ -19,7 +19,7 @@ from PySide import QtGui, QtCore
 import os, time, urllib2
 from calculate.core.client.function import create_obj
 from calculate.lib.utils.files import readLinesFile
-import Image
+from PIL import Image 
 import pwd, sys
 from calculate.lib.utils.files import listDirectory
 
