commit ce678fb31a5c7eee18737c0821179bead5d7040b
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Thu Aug 2 15:57:33 2012 +0400

    Fix syntax bug

diff --git a/consolegui/application/MainFrameResult.py b/consolegui/application/MainFrameResult.py
index 8f6f21f..cc9816a 100755
--- a/consolegui/application/MainFrameResult.py
+++ b/consolegui/application/MainFrameResult.py
@@ -141,11 +141,13 @@ class MainFrameRes(QtGui.QWidget):
         if not hasattr (self, 'mytable'):
             if table.body:
                 len_body = len(table.body[0])
-            else: len_body = 1
+            else:
+                len_body = 1
             if table.head:
                 len_head = len(table.head[0])
-            else: len_head = 1
-                self.mytable = SelectedTableWidget(len_body, len_head, self,
+            else:
+                len_head = 1
+            self.mytable = SelectedTableWidget(len_body, len_head, self,
                                                    readonly = True)
         list_head = table.head[0]
         self.mytable.setHorizontalHeaderLabels(list_head)
@@ -832,4 +834,4 @@ class MessageDialog(QtGui.QWidget):
         if self.pass_edit.text() != self.pass_edit2.text():
             self.send_button.setDisabled(True)
         else:
-            self.send_button.setEnabled(True)
\ No newline at end of file
+            self.send_button.setEnabled(True)
