diff --git consolegui/application/CertificateClass.py consolegui/application/CertificateClass.py
index c177dc3..c4a71ff 100755
--- consolegui/application/CertificateClass.py
+++ consolegui/application/CertificateClass.py
@@ -43,7 +43,8 @@ class CertClass (QtGui.QWidget):
         
         self.sendlayout.addWidget(LabelWordWrap(_('Port'), self), 1,0)
         
-        self.send_port = QtGui.QLineEdit('8888', self)
+        cl_core_port = self.ClientObj.VarsApi.Get('core.cl_core_port')
+        self.send_port = QtGui.QLineEdit(cl_core_port, self)
         self.send_port.setValidator(QtGui.QIntValidator(self))
         self.sendlayout.addWidget(self.send_port, 1, 1, 1, 2)
 
diff --git consolegui/application/ConnectionTabs.py consolegui/application/ConnectionTabs.py
index 5edde6b..dad842c 100644
--- consolegui/application/ConnectionTabs.py
+++ consolegui/application/ConnectionTabs.py
@@ -46,6 +46,7 @@ class SelectedMethodWgt(QtGui.QWidget):
         self.VarsGui.flIniFile()
 
         self.ClientObj = ApiClient(app, self)
+        self.port = args.port or self.VarsGui.GetInteger('core.cl_core_port')
         self.MethodWidget = self.ClientObj.MainWidget
         self.MethodWidget.topmenu.hide()
         layout = QtGui.QHBoxLayout(self)
@@ -78,7 +79,7 @@ class SelectedMethodWgt(QtGui.QWidget):
         self.show()
 
     def connect(self):
-        self.MethodWidget.connect_to_localhost(self.args.host, self.args.port)
+        self.MethodWidget.connect_to_localhost(self.args.host, self.port)
         if not self.ClientObj.client:
             return 1
         self.ClientObj.sid = get_sid(self.ClientObj.client)
@@ -226,7 +227,7 @@ def parse():
         '--method', type=str, dest='method', 
         help=_('call method'))
     parser.add_argument( 
-        '-p', '--port', type=int, default = '8888', dest='port',
+        '-p', '--port', type=int, dest='port',
         help=_('port number'))      
     parser.add_argument( 
         '--host', type=str, default = 'localhost', dest='host',
@@ -343,6 +344,8 @@ class ToolTabWidget(QtGui.QTabWidget):
 
         if not host or host == 'localhost':
             host = '127.0.0.1'
+        if not port:
+            port = self.VarsGui.GetInteger('core.cl_core_port')
         self.FirstWidget.connect_to_localhost(host, port)
         if self.FirstWidget.ClientObj.client:
             if host in ['127.0.0.1', 'localhost']:
diff --git consolegui/application/MainClass.py consolegui/application/MainClass.py
index f61b542..de8d824 100755
--- consolegui/application/MainClass.py
+++ consolegui/application/MainClass.py
@@ -48,7 +48,7 @@ class ApiClient(QtGui.QWidget):
 
         # other variable 
         self.default_host = 'localhost'
-        self.default_port = '8888'
+        self.default_port = self.VarsGui.Get('core.cl_core_port')
 
         self.client = None
         self.host_name = None
diff --git consolegui/application/conf_connection.py consolegui/application/conf_connection.py
index fd81844..ff853bd 100755
--- consolegui/application/conf_connection.py
+++ consolegui/application/conf_connection.py
@@ -254,7 +254,7 @@ class FrameConnection(QtGui.QWidget):
             host_name = '127.0.0.1'
 
         if not port:
-            port = 8888
+            port = self.ClientObj.VarsApi.GetInteger('core.cl_core_port')
 
         if hasattr (self.ClientObj._parent, 'find_host'):
             if self.ClientObj._parent.find_host(host_name, port):
diff --git consolegui/application/create_cert.py consolegui/application/create_cert.py
index 505eaca..24c30ef 100755
--- consolegui/application/create_cert.py
+++ consolegui/application/create_cert.py
@@ -59,6 +59,7 @@ class RequestCreate (QtGui.QWidget):
         
         # Add UserName field
         username = ClientObj.VarsApi.Get('ur_fullname')
+        suffixPort = ":%s"%ClientObj.VarsApi.Get('core.cl_core_port')
 
         username = username.decode('utf-8')
         self.mainlayout.addWidget(LabelWordWrap(_('User Name'), self),1,0 \
@@ -75,7 +76,7 @@ class RequestCreate (QtGui.QWidget):
         # Add Network address field
         self.mainlayout.addWidget(LabelWordWrap(_('Network address'), self), \
                                                 3,0,QtCore.Qt.AlignRight)
-        self.netw_adr = QtGui.QLineEdit(host_name+':8888', self)
+        self.netw_adr = QtGui.QLineEdit(host_name+suffixPort, self)
         self.mainlayout.addWidget(self.netw_adr, 3,1)
 
         # Add State name field
