diff --git pym/consolegui/application/helpwidget.py pym/consolegui/application/helpwidget.py
index e4d4a86..35e1b27 100755
--- pym/consolegui/application/helpwidget.py
+++ pym/consolegui/application/helpwidget.py
@@ -17,19 +17,23 @@
 from PySide import QtGui, QtCore
 from more import LabelWordWrap, show_msg
 import datetime
+from calculate.lib.datavars import DataVars
 
 _('The user should not be root')
 class HelpWgt(QtGui.QWidget):
     def __init__(self, parent):
         QtGui.QWidget.__init__(self)
 
+        clVars = DataVars()
+        clVars.flIniFile()
         cur_year = str(datetime.date.today().year)
         copy_right = "&copy;"
         motto = 'Easy Linux from the Source.<br>'
-        help_text = '%s v%s. \n' %(parent.ClientObj.Name, \
-                parent.ClientObj.Version) + \
-                _('Makes part of Calculate Utilities 3.0.0') + '<br>' + \
-                _('Company') + ' Calculate %s 2007-%s' %(copy_right,cur_year)
+        help_text = ('%s v%s. \n' %(parent.ClientObj.Name,
+                parent.ClientObj.Version) + 
+                _('Makes part of Calculate Utilities %s')%clVars.Get('cl_ver')
+                + '<br>' + 
+                _('Company') + ' Calculate %s 2007-%s' %(copy_right,cur_year))
 
         html_help = ("<!DOCTYPE HTML PUBLIC '-//W3C//DTD HTML 4.0//EN' "
                   "'http://www.w3.org/TR/REC-html40/strict.dtd'>"
@@ -38,12 +42,12 @@ class HelpWgt(QtGui.QWidget):
 
         helpLabel = LabelWordWrap(html_help, self)
         helpLabel.setContentsMargins(10,10,10,5)
-        company_name = _('Company website')
+        company_name = _('Developer:') + " " +  _("Calculate company")
         company_site = 'http://www.calculate.ru'
         distr_name = _('Distribution website')
         distr_site = 'http://www.calculate-linux.org'
         linkLabel = LabelWordWrap(company_name + \
-                ": <br><a href='http://www.calculate.ru'>" + \
+                "<br><a href='http://www.calculate.ru'>" + \
                 company_site + "</a><br><br>" + distr_name+ \
                 ": <br><a href='http://www.calculate-linux.org'>" + \
                 distr_site + "</a>", self)
