# Calculate format=diff merge(xfce-base/garcon)>=0.3
diff --git a/garcon-gtk/garcon-gtk-menu.c b/garcon-gtk/garcon-gtk-menu.c
index ed32acb..9922453 100644
--- a/garcon-gtk/garcon-gtk-menu.c
+++ b/garcon-gtk/garcon-gtk-menu.c
@@ -486,6 +486,8 @@ garcon_gtk_menu_reload (GarconGtkMenu *menu)
   if (menu->priv->reload_id == 0
       && menu->priv->is_loaded)
     {
+      GarconMenuItemCache *cache = garcon_menu_item_cache_get_default();
+      garcon_menu_item_cache_invalidate(cache);
       menu->priv->reload_id = g_timeout_add (100, garcon_gtk_menu_reload_idle, menu);
     }
 }
