commit d0868853121fa1110949125e9015caae16325d03
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Jul 26 18:20:55 2011 +0400

    Fix detect filesystem for lvm.

diff --git a/util/grub-install.in b/util/grub-install.in
index 16236ba..7ebbb50 100644
--- a/util/grub-install.in
+++ b/util/grub-install.in
@@ -466,6 +466,10 @@ fi
 # Create the core image. First, auto-detect the filesystem module.
 fs_module="`"$grub_probe" --device-map="${device_map}" --target=fs --device "${grub_device}"`"
 if test "x$fs_module" = x ; then
+    grub_device_fs=$(readlink -f ${grub_device})
+    fs_module="`"$grub_probe" --device-map="${device_map}" --target=fs --device "${grub_device_fs}"`"
+fi
+if test "x$fs_module" = x ; then
     echo "Auto-detection of a filesystem of ${grub_device} failed." 1>&2
     echo "Try with --recheck." 1>&2
     echo "If the problem persists please report this together with the output of \"$grub_probe --device-map=\"${device_map}\" --target=fs -v ${grubdir}\" to <bug-grub@gnu.org>" 1>&2
