commit 379e4d501b726d4e684215986fe10a35cae3eec7
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Mar 18 09:18:19 2011 +0300

    Add all nvidia to binary and unmask list.

diff --git a/pym/cl_assemble.py b/pym/cl_assemble.py
index b663655..25a68fb 100644
--- a/pym/cl_assemble.py
+++ b/pym/cl_assemble.py
@@ -24,7 +24,8 @@ import traceback
 from os import path
 from cl_template import template,iniParser
 from cl_utils import process,pathJoin,getRunCommands,getTupleVersion,isMount,\
-                     isFstabMount,childMounts,_toUNICODE,cmpVersion
+                     isFstabMount,childMounts,_toUNICODE,cmpVersion, \
+                     listDirectory
 from subprocess import STDOUT,PIPE
 from cl_print import color_print
 from cl_datavars import DataVars
@@ -39,6 +40,7 @@ from server.utils import dialogYesNo
 import cl_install
 import cl_overriding
 import random
+from shutil import copyfile
 
 def DEBUG(msg):
     if "DEBUG" in os.environ:
@@ -1520,6 +1522,17 @@ class cl_assemble(color_print):
              map(lambda x:path.join(pkgDir,x),
              os.listdir(pkgDir))))
 
+            packagesDir = 'usr/portage/packages'
+            x11_drivers = 'x11-drivers'
+            nvidiaDir = pathJoin(chrootPath, packagesDir,x11_drivers)
+
+            pkgDirX11Drivers = pathJoin(pkgDir,'x11-drivers')
+            if not path.exists(pkgDirX11Drivers):
+                os.makedirs(pkgDirX11Drivers,mode=0755)
+            map(lambda x:copyfile(pathJoin(nvidiaDir,x),
+                                  pathJoin(pkgDirX11Drivers,x)),
+             listDirectory(nvidiaDir))
+
             self.regenPackages(chrootPath,pkgDir[len(chrootPath):])
         else:
             self.printByResult(skip=True)
@@ -1547,11 +1560,17 @@ class cl_assemble(color_print):
     def createUnmaskList(self,chrootdir,filename):
         """Create package unmask list by chrootdir"""
         pkgdir = path.join(chrootdir,'var/db/pkg')
+        x11_bindir = path.join(chrootdir,'usr/portage/packages/x11-drivers')
         if not path.exists(chrootdir):
             return False
         try:
             packageList = sorted(reduce(lambda x,y:x+map(lambda x:path.join(y,x),
                 os.listdir(path.join(pkgdir,y))), os.listdir(pkgdir),[]))
+            packageList = \
+                sorted(packageList+
+                       map(lambda x:path.join('x11-drivers',x[:-5]),
+                       filter(lambda x:x.endswith('.tbz2'),
+                       listDirectory(x11_bindir))))
             open(filename,'w').writelines(map(lambda x:"=%s\n"%x,packageList))
         except (IOError,OSError),e:
             return False
