commit d7be7be4564baae604bb7e5e36aa0ff9cbe257ff
Author: Mike khiretskiy <mh@calculate.ru>
Date:   Wed Jul 31 16:28:30 2013 +0400

    Исправлена дописывание пакета с видеодрайвером в world

diff --git pym/cl_fill_assemble.py pym/cl_fill_assemble.py
index b59398d..22b57ad 100644
--- pym/cl_fill_assemble.py
+++ pym/cl_fill_assemble.py
@@ -500,6 +500,7 @@ class fillVars(object, glob_attr):
         yield "PORTAGE=/usr/portage"
         yield "OVERLAY=/var/lib/layman/calculate"
         yield "WORLD=/var/lib/portage/world"
+        yield '[[ -z "$(tail -c1 $WORLD)" ]] || echo >> $WORLD'
         for verName,maskName,drvVer,ebuilds in info:
             yield "if [[ $PKG == 'x11-drivers/{verName}' ]] " \
                   "&& [[ $MASK == '{maskName}' ]]"\
