commit 95ebf418d5cb098f70d35b4319a76dc8a353a5ab
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Thu Mar 24 16:09:11 2011 +0300

    Fix removing directory in /usr/portage/distfiles.

diff --git a/pym/cl_assemble.py b/pym/cl_assemble.py
index 803fd08..ae9ec2b 100644
--- a/pym/cl_assemble.py
+++ b/pym/cl_assemble.py
@@ -1452,7 +1452,7 @@ class cl_assemble(color_print):
         removedList = fetchedList - needFiles
         self.printMessageForTest(" " +
             _("Cleaning %s")%distDir)
-        map(lambda x:os.unlink(x),
+        map(lambda x:os.rmdir(x) if path.isdir(x) else os.unlink(x),
          filter(lambda x:path.exists(x),
          map(lambda x:pathJoin(chrootPath,distDir,x),
          removedList)))
