commit e36996d23c5ecfe83807261ebcb87bedb61eda48
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Wed Aug 1 15:50:43 2012 +0400

    Fix IssueID 456

diff --git pym/cl_assemble.py pym/cl_assemble.py
index d8a1f90..4615d58 100644
--- pym/cl_assemble.py
+++ pym/cl_assemble.py
@@ -610,7 +610,7 @@ class cl_assemble(color_print):
         self._migratePortage()
         if not isPkgInstalled('dev-vcs/git',prefix=self.assemblePath):
             self.printMessageForTest(_("Installing %s")%'dev-vcs/git')
-            self.runChroot("USE='-perl -curl' emerge -1 git")
+            self.runChroot("USE='-perl -curl -webdav' emerge -1 git")
 
         emergeSync = "emerge --sync"
         self.printMessageForTest(_("Updating portages"))
