commit 8b8d34568da30821d178c86d821086a6da18a45a
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Thu May 3 17:28:16 2012 +0400

    Add --with-bdpes for emerge world

diff --git a/pym/cl_assemble.py b/pym/cl_assemble.py
index dd8110a..d0455c4 100644
--- a/pym/cl_assemble.py
+++ b/pym/cl_assemble.py
@@ -166,7 +166,7 @@ class cl_assemble(color_print):
              #{'command':'emerge -D --newuse world',
              # 'step':'newuse'},
              # update world
-             {'command':'emerge -uDN  world',
+             {'command':'emerge -uDN --with-bdeps=y world',
               'step':'update'},
              # depclean
              {'command':'emerge --depclean',
