commit b90abbe01990d1a8afc7107f58f047f3828781e3
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Mar 15 14:54:20 2011 +0300

    Fix listdir fault.

diff --git a/pym/cl_fill_assemble.py b/pym/cl_fill_assemble.py
index f759606..2b78443 100644
--- a/pym/cl_fill_assemble.py
+++ b/pym/cl_fill_assemble.py
@@ -24,7 +24,7 @@ from cl_distr import DistributiveRepository
 from cl_template import iniParser
 from cl_fill_install import fillVars as installFillVars
 from cl_vars import Data as libData
-from cl_utils import _toUNICODE,isMount,pathJoin
+from cl_utils import _toUNICODE,isMount,pathJoin,listDirectory
 from datetime import datetime
 
 class fillVars(installFillVars):
@@ -197,7 +197,8 @@ class fillVars(installFillVars):
         devices = self.Get('os_device_hash').keys()
         disks = reduce( lambda x,y: x +
                     map( lambda x: "/dev/%s"%x,
-                    filter(lambda x: y in x,os.listdir('/sys/block/%s'%y))),
+                    filter(lambda x: y in x,
+                        listDirectory('/sys/block/%s'%y))),
                     devices, [] )
         return disks
 
