commit 6b497fb0649dff671391f87319a27fd8a471c6f2
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Tue Aug 21 17:59:18 2012 +0400

    Fix without video

diff --git pym/cl_assemble.py pym/cl_assemble.py
index 620d7f0..88dadda 100644
--- pym/cl_assemble.py
+++ pym/cl_assemble.py
@@ -1657,9 +1657,9 @@ class cl_assemble(color_print):
             curDistdir = curDistdir[0]
         else:
             curDistdir = '/var/calculate/remote/distfiles'
+        driversMatrix = []
         while drivers:
             repeatDrivers = []
-            driversMatrix = []
             for verName,maskName,drvVer in drivers:
                 rePackage = re.compile("^\[ebuild[^\]]+]\s+(\S+)")
                 checkprocess = self.runChroot(
