commit 2db660f23178b780cc4d1c83aa85d085adad11af
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Tue Apr 16 09:51:54 2013 +0400

    Force checkout branches on assemble

diff --git pym/cl_assemble.py pym/cl_assemble.py
index 55cd748..6456d80 100644
--- pym/cl_assemble.py
+++ pym/cl_assemble.py
@@ -567,7 +567,7 @@ class cl_assemble(color_print):
                     gitrep=self.clVars.Get('cl_assemble_sync'),
                     portage=portagePath)
                 self.runCommand(getPortage)
-            updateBranch= "cd {portage};git checkout {branch}".format(
+            updateBranch= "cd {portage};git checkout {branch} -f".format(
                     portage=portagePath,
                     branch=self.clVars.Get('cl_assemble_branch'))
             self.runCommand(updateBranch)
@@ -608,7 +608,7 @@ class cl_assemble(color_print):
             if path.exists(pathOverlay):
                 updateBranch= \
                     "cd {portage};git pull;" \
-                    "git checkout {branch}".format(
+                    "git checkout {branch} -f".format(
                         portage=pathOverlay,
                         branch=self.clVars.Get('cl_assemble_branch'))
                 self.runCommand(updateBranch)
@@ -646,7 +646,7 @@ class cl_assemble(color_print):
             self.runChroot("layman -L")
             self.runChroot("layman -a calculate")
         if self.clVars.Get('cl_assemble_branch') in ('master','update'):
-            updateBranch= "cd {portage};git checkout {branch}".format(
+            updateBranch= "cd {portage};git checkout {branch} -f".format(
                     portage='/var/lib/layman/calculate',
                     branch=self.clVars.Get('cl_assemble_branch'))
             self.runChroot(updateBranch)
