diff --git pym/cl_assemble.py pym/cl_assemble.py
index 4f7c5f8..f4bf767 100644
--- pym/cl_assemble.py
+++ pym/cl_assemble.py
@@ -388,7 +388,7 @@ class cl_assemble(color_print):
                                   ":\n")
                 for system in self.clVars.Get('cl_assemble_distro'):
                     self.printSUCCESS(self.__profileShortname(system))
-            self.removeDistroInfo()
+                self.removeDistroInfo()
             return False
         return True
 
@@ -686,6 +686,7 @@ class cl_assemble(color_print):
             self.sourceDistr = distRep.getDistributiveByFile(distName)
             if not self._askUser(force, _("Continue system assemble")):
                 self.printERROR(_("Assemble interrupted"))
+                self.removeDistroInfo()
                 return False
             self.writeVar('cl_assemble_make','configure')
             # test /mnt
@@ -1504,10 +1505,11 @@ class cl_assemble(color_print):
         assembleEnv = iniParser(self.assembleEnvFile)
         discardEmpty = lambda x : filter(lambda y:y,x)
         busyDevs = \
+            map(lambda x:x.encode('utf-8') if type(x) == unicode else x,
             discardEmpty(
             map(lambda x:assembleEnv.getVar(x,'os_assemble_root_dev'),
-            distros))
-        discardBusy = lambda x : filter(lambda y:not (x in busyDevs 
+            distros)))
+        discardBusy = lambda x : filter(lambda y:not (y in busyDevs 
                                                       or isMount(y)),x)
         availableDev = self.clVars.Get('cl_assemble_available_dev').split(',')
         availableDev = discardEmpty(availableDev)
@@ -1515,6 +1517,7 @@ class cl_assemble(color_print):
         if freeDev:
             rootDev = random.choice(freeDev)
             self.clVars.Set('os_assemble_root_dev',rootDev,True)
+            self.saveVars('os_assemble_root_dev')
             return True
         return False
 
