diff --git a/pym/cl_assemble.py b/pym/cl_assemble.py
index 48582ad..765feba 100644
--- a/pym/cl_assemble.py
+++ b/pym/cl_assemble.py
@@ -1481,7 +1481,8 @@ class cl_assemble(color_print):
                       imap(lambda x:x.partition('="')[2].strip('\n"'),
                       ifilterfalse(lambda x:x.startswith('mask_71') or \
                                             x.startswith('mask_96') or \
-                                            x.startswith('mask_173'),
+                                            x.startswith('mask_173') or \
+                                            x.startswith('mask_304'),
                       ifilter(lambda x:x.startswith('mask_'),
                       open(nvidiaEclass)))))+\
                       [('ati-drivers',"",'x11-drivers/ati-drivers')]
