commit ced695df30d22c304558eb3ef895fd33b97612a4
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Aug 16 18:01:57 2011 +0400

    Fix detect available device for assemble.

diff --git a/pym/cl_fill_assemble.py b/pym/cl_fill_assemble.py
index 1130ccf..b729ca3 100644
--- a/pym/cl_fill_assemble.py
+++ b/pym/cl_fill_assemble.py
@@ -194,7 +194,8 @@ class fillVars(installFillVars):
 
     def get_os_disk_dev(self):
         reSdaPart = re.compile("^/dev/sd([a-z])(\d+)$")
-        devices = self.Get('os_device_hash').keys()
+        devices = map(lambda x:path.basename(x),
+                  self.Get('os_device_hash').keys())
         disks = reduce( lambda x,y: x +
                     map( lambda x: "/dev/%s"%x,
                     filter(lambda x: y in x,
