commit 59187d5a114f5d01d33096cdf114357878a8ad6d
Author: Mike khiretskiy <mh@calculate.ru>
Date:   Thu Jan 9 11:57:36 2014 +0400

    Пересборка модулей ядра осуществляется через emerge @module-rebuild

diff --git pym/cl_assemble.py pym/cl_assemble.py
index da7327e..ded1c47 100644
--- pym/cl_assemble.py
+++ pym/cl_assemble.py
@@ -177,11 +177,9 @@ class cl_assemble(color_print):
              {'command':'emerge --depclean',
               'step':'depclean'},
              # module rebuild
-             {'command':'modulesRebuild()',
+             {'command':'emerge @module-rebuild',
               'step':'module',
-              'condition':'sys-kernel/.*source.*',
-              'message':_("Executing '%s'")%'module-rebuild -X rebuild',
-              'progname':'module-rebuild'},
+              'condition':'sys-kernel/.*source.*'},
              # xorg module rebuild
              {'command':'emerge @x11-module-rebuild',
               'step':'x11module',
