commit c389ce74128d9c01fa65f49a2d082aa881d3c07a
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Wed Apr 4 10:48:25 2012 +0400

    Add downgrade checking.

diff --git a/pym/cl_assemble.py b/pym/cl_assemble.py
index a7521d6..dd8110a 100644
--- a/pym/cl_assemble.py
+++ b/pym/cl_assemble.py
@@ -227,7 +227,11 @@ class cl_assemble(color_print):
              {'command':'eselect news read new',
               'step':'readnews',
               'message':_("Reading news"),
-              'progname':'eselect'}]
+              'progname':'eselect'},
+             {'command':"checkDowngrades()",
+              'message':_("Checking downgrades"),
+              'step':'showdowngrade',
+              'progname':'eix'}]
 
     reVerSplit = re.compile(r"^(.*?)-(([^-]+?)(?:-(r\d+))?)(?:.(tbz2))?$",re.S)
 
@@ -1388,6 +1392,13 @@ class cl_assemble(color_print):
                 _("An error occurred when executing the command")+
                   ":\n    %s"%self._getCommand(cmd.command))
 
+    def checkDowngrades(self):
+        """Run check downgrades"""
+        cmd = self.runChroot("FORMAT='{downgrade}%{FORMAT_ALL}{}' eix -I",
+                             raiseError=False,
+                             showProcess=True)
+        return True
+
     def modulesRebuild(self):
         """Run revdep-rebuild and check result"""
         if not self.assemblePath:
