commit 38d1dc4538badc93b292182366c9a171734d0a9a
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Aug 15 16:19:53 2011 +0400

    Add display dep packages for proproitary drivers.

diff --git a/pym/cl_assemble.py b/pym/cl_assemble.py
index 72f4e57..e6f4fcf 100644
--- a/pym/cl_assemble.py
+++ b/pym/cl_assemble.py
@@ -1443,13 +1443,16 @@ class cl_assemble(color_print):
         while drivers:
             repeatDrivers = []
             for verName,drvVer in drivers:
+                rePackage = re.compile("^\[ebuild[^\]]+]\s+(\S+)")
                 checkprocess = self.runChroot(
-                            "emerge -p %s"%
+                            "emerge -pv %s"%
                             drvVer, raiseError=False,joinErr=True)
-                checkprocess = self.runChroot(
-                            "emerge -vp %s"%
-                            drvVer,raiseError=False,joinErr=True)
-                rePackage = re.compile("^\[ebuild[^\]]+]\s+(\S+)")
+                # get names of dep packages
+                installPkgList = \
+                    map(lambda x:x.group(1),
+                    filter(lambda x:x,
+                    map(rePackage.search,
+                    checkprocess)))
                 # get version from emerging nvidia-driver
                 nvidiaVersions = \
                     map(lambda x:x.groups()[1],
@@ -1459,13 +1462,25 @@ class cl_assemble(color_print):
                     filter(lambda x:x,
                     map(rePackage.search,
                     filter(lambda x:verName in x,
-                    checkprocess)))))))
+                    checkprocess.readlines())))))))
 
                 if nvidiaVersions:
                     nvidiaVersions = nvidiaVersions[0]
                 else:
                     nvidiaVersions = "0"
-                self.printMessageForTest(" " +
+                self.printSUCCESS(_("For installation '%s' will be "
+                                    "installed follow packages")
+                                    %"%s-%s"%(verName,nvidiaVersions)+":")
+                # print dep packages (error message for probably wrong packages)
+                nameRight = "/%s-"%verName.partition("-")[0]
+                for pkg in installPkgList:
+                    if nameRight in pkg:
+                        self.printSUCCESS("  %s"%pkg)
+                    else:
+                        self.printERROR("  %s"%pkg)
+                if installPkgList:
+                    self.printSUCCESS("")
+                self.printMessageForTest(
                     _("Fetching %s")%"%s-%s"%(verName,nvidiaVersions))
                 # run verbose package emergin for get version
                 fetchprocess = self.runChroot(
@@ -1484,6 +1499,7 @@ class cl_assemble(color_print):
                 else:
                     self.printByResult(fetchprocess.success())
                     needFiles += fetchedFiles
+                self.defaultPrint("\n")
             drivers = repeatDrivers
             if drivers:
                 self.printMessageForTest(" " +
