commit 7f3a787f8cadb56540e640f04f9ca679a7f3d328
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue May 10 16:11:37 2011 +0400

    Add preserved-rebuild after revdep.

diff --git a/pym/cl_assemble.py b/pym/cl_assemble.py
index 19583bf..e713b72 100644
--- a/pym/cl_assemble.py
+++ b/pym/cl_assemble.py
@@ -161,6 +161,11 @@ class cl_assemble(color_print):
               'condition':'.',
               'message':_("Execute '%s'")%'revdep-rebuild -i',
               'progname':'revdep-rebuild'},
+             # emerge @preserved-rebuild
+             {'command':'emerge @preserved-rebuild',
+              'step':'presrebuild',
+              'condition':'.'
+             },
              # prelink all packages
              {'command':'prelink -afmR',
               'step':'prelink',
