diff --git data/client data/client
index 64da473..e3285f2 100644
--- data/client
+++ data/client
@@ -43,14 +43,14 @@ start() {
 		local NET_CONF=`cl-core-variables-show --only-value install.os_install_net_conf`
 		if [[ $NET_CONF = "networkmanager" ]]
 		then
-		        for COUNT in $( seq 0 24 )
+			for COUNT in $( seq 0 24 )
 			do
-        			NET_STATE=`qdbus --system org.freedesktop.NetworkManager /org/freedesktop/NetworkManager org.freedesktop.NetworkManager.state`
+				NET_STATE=`qdbus --system org.freedesktop.NetworkManager /org/freedesktop/NetworkManager org.freedesktop.NetworkManager.state`
 				if [[ $NET_STATE = 40 ]]
 				then
-                                    ping -w4 -i0.5 -c3 $SERVERLIVE &>/dev/null && break || sleep 0.5
+					ping -w4 -i0.5 -c3 $SERVERLIVE &>/dev/null && break || sleep 0.5
 				else
-				    break
+					break
 				fi
 			done
 		else
@@ -61,7 +61,8 @@ start() {
 		fi
 		if [[ -z $SERVER ]]
 		then
-			cl-client $SERVERLIVE && cl-setup-system -l remote &>/dev/null
+			cl-client -T overlay,local $SERVERLIVE && res=$? &&
+				cl-setup-system -T remote &>/dev/null
 		else
 			cl-client --mount
 		fi
