diff --git client/cl_client.py client/cl_client.py
index f0f54b1..5b0bd7e 100644
--- client/cl_client.py
+++ client/cl_client.py
@@ -371,8 +371,9 @@ class Client(commandServer, encrypt):
             self.startTask(_("Applying non-domain templates"),progress=True)
         if self.clTempl:
             self.closeClTemplate()
+        self.clVars.Set('cl_autoupdate_set','on',True)
         self.clTempl = ProgressTemplate(self.setProgress,self.clVars,
-                                        cltObj=True,
+                                        cltObj=True,cltFilter=True,
                                         printSUCCESS=self.printSUCCESS,
                                         printERROR=self.printERROR,
                                         askConfirm=self.askConfirm,
