diff --git client/cl_client.py client/cl_client.py
index 3bd7059..e677af8 100644
--- client/cl_client.py
+++ client/cl_client.py
@@ -1137,12 +1137,12 @@ class Client(commandServer, encrypt):
         if not userPwd or userPwd == "XXXXXXXX":
             raise ClientError(_("User password not found"))
         ecryptfsPath = path.join('/home/.ecryptfs',userName)
+        error = ""
         if path.exists(ecryptfsPath):
             for d in (".ecryptfs",".Private"):
                 source,target = path.join(ecryptfsPath,d),path.join(userDir,d)
                 if not path.lexists(target):
                     os.symlink(source,target)
-            error = ""
             try:
                 if mountEcryptfs(userName,userPwd,userDir):
                     return True
