commit d37e8a7aa13ed0de016dc546edd56c565899998f
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Mon Jun 18 08:57:13 2012 +0400

    Fix wait end session and relogin

diff --git a/pym/cl_client.py b/pym/cl_client.py
index c7b26a8..d147ee6 100644
--- a/pym/cl_client.py
+++ b/pym/cl_client.py
@@ -1607,11 +1607,33 @@ class client(share, commandServer, encrypt):
             return False
         return True
 
+    def getRunCommandsWithEnv(self):
+        """List run program"""
+        def getCmd(procNum):
+            cmdLineFile = '/proc/%s/cmdline'%procNum
+            environFile = '/proc/%s/environ'%procNum
+            try:
+                if os.path.exists(cmdLineFile) and \
+                   os.path.exists(environFile):
+                    return (open(cmdLineFile,'r').read().strip(),
+                            open(environFile,'r').read().strip())
+            except:
+                pass
+            return ("","")
+        if not os.access('/proc',os.R_OK):
+            return []
+        return map(getCmd,
+               filter(lambda x:x.isdigit(),
+               os.listdir('/proc')))
+
+
     def clearUserKey(self, userName):
         """Очищает пользовательский ключ ядра"""
-        # Ищем ключ в ядре
-        if getKey(userName):
-            # Если нашли очищаем
+        # Ищем ключ в ядре и проверяем не выполняет ли он повторный вход
+        if getKey(userName) and not \
+            filter(lambda x:"xdm/xdm\x00--login" in x[0] and \
+               ("USER=%s"%userName) in x[1],self.getRunCommandsWithEnv()):
+            # очищаем
             ret = clearKey(userName)
             if ret == 0:
                 return True
