diff --git pym/cl_client.py pym/cl_client.py
index 66b2ac0..973f729 100644
--- pym/cl_client.py
+++ pym/cl_client.py
@@ -344,9 +344,9 @@ class client(share, commandServer, encrypt):
     # объект конвертирования из старого remote env файла
     convObj = None
     # Приватный файлы пользователя
-    privateFiles = ['.kde4/share/apps/kwallet/kdewallet.kwl', configFileServer]
+    privateFiles = [configFileServer]
     # Приватные директории пользователя
-    privateDirs =  [".ssh"]
+    privateDirs =  []
 
 
     def removeVars(self):
@@ -1980,8 +1980,8 @@ class client(share, commandServer, encrypt):
         if remoteHost:
             self.printERROR(_("The computer is already in the domain %s")\
                               %remoteHost)
-            self.printWARNING(_("Before joining the domain, \
-you need to remove it from the previous domain"))
+            self.printWARNING(_("Before joining the domain, "
+                "you need to remove it from the previous domain"))
             self.printWARNING(_("Run 'cl-client -r'"))
             return False
         
@@ -2324,7 +2324,7 @@ you need to remove it from the previous domain"))
             return False
         self.printOK(_("Changed password of user %s")%os.environ["USER"] + \
                      " ...")
-        self.printWARNING(_("Password will be changed when you logout from the \
-X session"))
+        self.printWARNING(_("Password will be changed when you logout from the"
+                    " X session"))
         return True
 
diff --git pym/cl_client_cmd.py pym/cl_client_cmd.py
index f7375a5..fdf510a 100644
--- pym/cl_client_cmd.py
+++ pym/cl_client_cmd.py
@@ -26,8 +26,8 @@ lang().setLanguage(sys.modules[__name__])
 USAGE = _("%prog [options] domain")
 
 # Коментарии к использованию программы
-COMMENT_EXAMPLES = _("Adds settings for connecting to domain \
-server.calculate.ru")
+COMMENT_EXAMPLES = _("Adds settings for connecting to domain "
+        "server.calculate.ru")
 
 # Пример использования программы
 EXAMPLES = _("%prog server.calculate.ru")
diff --git pym/cl_sync_cmd.py pym/cl_sync_cmd.py
index f96c331..53847f2 100644
--- pym/cl_sync_cmd.py
+++ pym/cl_sync_cmd.py
@@ -36,14 +36,14 @@ DESCRIPTION = _("Mounting resources and synchronize the user profile")
 
 # Опции командной строки
 CMD_OPTIONS = [{'longOption':"login",
-                'help':_("mount user resource and synchronize the user \
-profile")},
+                'help':_("mount user resource and synchronize the user "
+                "profile")},
                {'longOption':"logout",
-                 'help':_("synchronize the user profile and umount user \
-resource")},
+                 'help':_("synchronize the user profile and umount user "
+                 "resource")},
                {'longOption':"nosync",
-                 'help':_("not synchronize the user preferences, is used \
-in conjunction with the 'login' or 'logout'")},
+                 'help':_("not synchronize the user preferences, is used "
+                  "in conjunction with the 'login' or 'logout'")},
                {'longOption':"set"},
                {'longOption':"progress",
                 'help':_("show progress bar for xdm startup")}]
