diff --git client/cl_client.py client/cl_client.py
index 2cf6d12..1d1a1d2 100644
--- client/cl_client.py
+++ client/cl_client.py
@@ -1148,7 +1148,7 @@ class Client(commandServer, encrypt):
         if self.domain:
             foundMountRemote = isMount("/var/calculate/remote")
             if not foundMountRemote:
-                self.mountRemote();
+                self.mountRemote(byUserSync=True);
         # check on two session user login
         if self.isTwoSessionsUser(self.userName):
             return False
@@ -2108,7 +2108,7 @@ class Client(commandServer, encrypt):
                 os.rmdir(previousHome)
         return True
 
-    def mountRemote(self):
+    def mountRemote(self,byUserSync=False):
         """
         Mount remote domain resource for domain workstation.
 
@@ -2145,8 +2145,10 @@ class Client(commandServer, encrypt):
                 os.makedirs(pathRemote)
             if self.mountSambaRes(domain,"client",pwdRemote,
                                0,0,"remote",pathRemote) is False:
-                self.printWARNING(_("Failed to mount Samba resource [%s]")%\
-                                    "remote" + " ...")
+                if not byUserSync:
+                    self.printWARNING(
+                        _("Failed to mount Samba resource [%s]")%\
+                                        "remote" + " ...")
                 beforeRemoteAuth = self.clVars.Get('os_remote_auth')
                 # apply templates if current up-to-date
                 if not self.applyRelevanceTemplates():
