diff --git a/client-profiles/20-kde-always/.kde4/share/config/plasma-desktop-appletsrc b/client-profiles/20-kde-always/.kde4/share/config/plasma-desktop-appletsrc
index 50b945d..a3ecef9 100644
--- a/client-profiles/20-kde-always/.kde4/share/config/plasma-desktop-appletsrc
+++ b/client-profiles/20-kde-always/.kde4/share/config/plasma-desktop-appletsrc
@@ -1,4 +1,4 @@
-# Calculate format=plasma cl_remote_host!=&ini(domain)!=cl_remote_host
+# Calculate format=plasma cl_remote_host!=&ini(domain)!=#-cl_remote_host-#
 #------------------------------------------------------------------------------
 #рабочий стол
 #------------------------------------------------------------------------------
diff --git a/client-profiles/30-gnome-always/.calculate_directory b/client-profiles/30-gnome-always/.calculate_directory
index abd22c5..4e6c732 100644
--- a/client-profiles/30-gnome-always/.calculate_directory
+++ b/client-profiles/30-gnome-always/.calculate_directory
@@ -1 +1 @@
-# Calculate ini(main.update)!=off&ini(update.gnome)!=off&pkg(gnome-base/libgnome)>=2.26&ini(domain)!=cl_remote_host
+# Calculate ini(main.update)!=off&ini(update.gnome)!=off&pkg(gnome-base/libgnome)>=2.26&ini(domain)!=#-cl_remote_host-#
diff --git a/client-profiles/40-xfce-always/.calculate_directory b/client-profiles/40-xfce-always/.calculate_directory
index a628783..9ca2624 100644
--- a/client-profiles/40-xfce-always/.calculate_directory
+++ b/client-profiles/40-xfce-always/.calculate_directory
@@ -1 +1 @@
-# Calculate ini(main.update)!=off&ini(update.xfce)!=off&pkg(xfce-base/xfdesktop)>=4.6&ini(domain)!=cl_remote_host
+# Calculate ini(main.update)!=off&ini(update.xfce)!=off&pkg(xfce-base/xfdesktop)>=4.6&ini(domain)!=#-cl_remote_host-#
