diff --git a/pym/cl_client.py b/pym/cl_client.py
index b00c874..a2987ac 100644
--- a/pym/cl_client.py
+++ b/pym/cl_client.py
@@ -124,7 +124,7 @@ class GProgressBar(NoneProgressBar):
     def setMaximum(self,max):
         '''Установить максимальное значения для прогресса'''
         self.max = max
-        self.kmax = max / 100.0
+        self.kmax = (max / 100.0) or 1.0
 
     def setTitle(self,title):
         '''Установить описания прогресса'''
@@ -232,7 +232,7 @@ org.kde.kdialog.ProgressDialog'
             os.system(env + '/usr/bin/qdbus %s %s maximum %d >/dev/null'\
                       %(self.kdialog,self.suffixSet, 100))
         self.max = max
-        self.kmax = max / 100.0
+        self.kmax = (max / 100.0) or 1.0
 
 
     def setTitle(self,title):
@@ -275,8 +275,6 @@ class RsyncProgressBar:
     при необходимости вывода progressbar
     '''
 
-    # получение номера передаваемого файла из инф потока rsync
-    senderre = re.compile("\[sender\] i=(\d+) ", re.S)
     # получение номера получаемого файла из потока rsync
     receiverre = re.compile("recv_generator\(.+,([0-9]+)\)", re.S)
     pipe = None
@@ -327,12 +325,6 @@ class RsyncProgressBar:
             s = self.pipe.stdout.readline()
             if len(s) == 0:
                 break
-            q = self.senderre.search(s)
-            if q:
-                maximum = int(q.groups()[0])
-                if self.maximum < maximum:
-                    self.maximum = maximum
-                continue
             q = self.receiverre.search(s)
             if q:
                 if not self.copyStarting:
@@ -2867,6 +2859,8 @@ or ld_bind_dn or ld_bind_pw")
 --exclude="/.Xauthority" \
 --exclude="/.thumbnails" \
 --exclude="/.mozilla/firefox/*/Cache" \
+--exclude="/.cache" \
+--exclude="/.opera/cache" \
 --filter="P /.googleearth" --filter="P /Home" --filter="P /Disks" \
 --filter="P /.local/share/akonadi/db_data" --filter="P /.VirtualBox" \
 --filter="P /.mozilla/firefox/calculate.default/urlclassifier3.sqlite" \
@@ -2876,7 +2870,9 @@ or ld_bind_dn or ld_bind_pw")
 --filter="P /.Xauthority" \
 --filter="P /.thumbnails" \
 --filter="P /.mozilla/firefox/*/Cache" \
---filter="P /FTP" -a -x -v -v -v -v  %s/ %s/' %(homeProfile,userHome)
+--filter="P /.cache" \
+--filter="P /.opera/cache" \
+--filter="P /FTP" -a -x -v -v -v  %s/ %s/' %(homeProfile,userHome)
         elif sync == "logout":
             if os.path.exists(userHome) and os.listdir(userHome) and\
                 os.path.exists(homeProfile):
@@ -2891,7 +2887,9 @@ or ld_bind_dn or ld_bind_pw")
 --exclude="/.kde4/socket-*" --exclude="/.kde4/share/config/phonondevicesrc" \
 --exclude="/.thumbnails" \
 --exclude="/.mozilla/firefox/*/Cache" \
--a -x -v -v -v -v %s/ %s/'%(userHome,homeProfile)
+--exclude="/.cache" \
+--exclude="/.opera/cache" \
+-a -x -v -v -v  %s/ %s/'%(userHome,homeProfile)
         else:
             self.printERROR(_("Method syncUser: option sync=%s incorrect")\
             %str(sync))
@@ -2921,11 +2919,11 @@ or ld_bind_dn or ld_bind_pw")
                     if numfiles is False:
                         if os.path.exists(configFileName):
                             os.remove(configFileName)
-                        numfiles = 1
+                        numfiles = 0
                     else:
                         numfiles = int(numfiles)
                 except:
-                    numfiles = 1
+                    numfiles = 0
                 rsync.maximum = numfiles
                 if progress:
                     rsync.run()
