diff --git client/cl_client.py client/cl_client.py
index 393f185..4d1282c 100644
--- client/cl_client.py
+++ client/cl_client.py
@@ -460,12 +460,14 @@ class Client(commandServer, encrypt):
         """Mount samba resource"""
         if res in mountUidList:
             # mount by uid
-            mountStr = "mount -t cifs -o user=%s,uid=%s,gid=%s,noperm"\
+            mountStr = \
+                "mount -t cifs -o cache=loose,user=%s,uid=%s,gid=%s,noperm"\
                        %(userName,uid,gid) +\
                        " //%s/%s %s" %(host, res, rpath)
         else:
             # mount by root
-            mountStr = "mount -t cifs -o user=%s"%(userName)+\
+            mountStr = \
+                "mount -t cifs -o cache=loose,user=%s"%(userName)+\
                        " //%s/%s %s" %(host, res, rpath)
         textLine = self.execProg(mountStr, envProg={"PASSWD":userPwd})
         return textLine
