# Calculate format=diff merge(sys-kernel/calculate-sources)==5.2.11
diff --git a/mm/zsmalloc.c b/mm/zsmalloc.c
index 515b00801..7d62ef2da 100644
--- a/mm/zsmalloc.c
+++ b/mm/zsmalloc.c
@@ -2432,7 +2432,9 @@ struct zs_pool *zs_create_pool(const char *name)
 	if (!pool->name)
 		goto err;
 
+#ifdef CONFIG_COMPACTION
 	init_waitqueue_head(&pool->migration_wait);
+#endif
 
 	if (create_cache(pool))
 		goto err;
