# Calculate merge(sys-kernel/dracut)>=44 format=diff
diff --git a/dracut-044-orig/dracut.sh b/dracut-044/dracut.sh
index a2fc496..d3ec592 100755
--- a/dracut-044-orig/dracut.sh
+++ b/dracut-044/dracut.sh
@@ -1554,13 +1554,18 @@ PRELINK_BIN="$(command -v prelink)"
 if [[ $UID = 0 ]] && [[ $PRELINK_BIN ]]; then
     if [[ $DRACUT_FIPS_MODE ]]; then
         dinfo "*** Installing prelink files ***"
-        inst_multiple -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf /etc/prelink.cache
+        inst_multiple -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf /etc/prelink.cache /lib*/libelf*.so* /usr/lib*/libelf*.so*
     elif [[ $do_prelink == yes ]]; then
         dinfo "*** Pre-linking files ***"
-        inst_multiple -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf
+        inst_multiple -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf /lib*/libelf*.so* /usr/lib*/libelf*.so*
+        [[ -d $initdir/proc ]] || mkdir $initdir/proc
+        mount -n -t proc none $initdir/proc
         chroot "$initdir" "$PRELINK_BIN" -a
+        umount -n $initdir/proc
         rm -f -- "$initdir/$PRELINK_BIN"
         rm -fr -- "$initdir"/etc/prelink.*
+        rm -f --  "$initdir"/usr/lib*/libelf*.so*
+        rm -f --  "$initdir"/lib*/libelf*.so*
         dinfo "*** Pre-linking files done ***"
     fi
 fi
