# Calculate format=diff
Патч для загрузки с неполных raid массивов
diff --git modules.d/90mdraid/parse-md.sh modules.d/90mdraid/parse-md.sh
index dd7bda2..3c88b6f 100755
--- modules.d/90mdraid/parse-md.sh
+++ modules.d/90mdraid/parse-md.sh
@@ -8,6 +8,7 @@ if ( ! [ -n "$MD_UUID" ] && ! getargbool 0 rd.auto ) || ! getargbool 1 rd.md -d
     info "rd.md=0: removing MD RAID activation"
     udevproperty rd_NO_MD=1
 else
+    /sbin/initqueue --timeout --name 50-mdraid_start --onetime --unique /sbin/mdraid_start
     # rewrite the md rules to only process the specified raid array
     if [ -n "$MD_UUID" ]; then
         for f in /etc/udev/rules.d/65-md-incremental*.rules; do
