diff -ruN genkernel.orig/arch/x86/modules_load genkernel/arch/x86/modules_load
--- genkernel.orig/arch/x86/modules_load	2009-08-07 16:16:44.000000000 +0400
+++ genkernel/arch/x86/modules_load	2009-08-07 16:19:10.000000000 +0400
@@ -22,5 +22,5 @@
 MODULES_USB="ehci-hcd uhci usb-ohci hid usb-storage uhci-hcd ohci-hcd usbhid sl811-hcd"
 
 # Filesystems
-MODULES_FS="ext2 ext3 reiserfs jfs nfs xfs fuse"
+MODULES_FS="ext2 ext3 ext4 reiserfs jfs nfs xfs fuse"
 
diff -ruN genkernel.orig/arch/x86_64/modules_load genkernel/arch/x86_64/modules_load
--- genkernel.orig/arch/x86_64/modules_load	2009-08-07 16:16:45.000000000 +0400
+++ genkernel/arch/x86_64/modules_load	2009-08-07 16:19:31.000000000 +0400
@@ -22,4 +22,4 @@
 MODULES_USB="ehci-hcd uhci usb-ohci hid usb-storage uhci-hcd ohci-hcd usbhid sl811-hcd"
 
 # Filesystems
-MODULES_FS="ext2 ext3 reiserfs jfs nfs xfs fuse"
+MODULES_FS="ext2 ext3 ext4 reiserfs jfs nfs xfs fuse"
