diff -ruN genkernel.orig/gen_initramfs.sh genkernel/gen_initramfs.sh
--- genkernel.orig/gen_initramfs.sh	2009-08-07 16:16:45.000000000 +0400
+++ genkernel/gen_initramfs.sh	2009-08-07 16:33:36.000000000 +0400
@@ -607,8 +607,17 @@
 		append_data 'overlay'
 	fi
 
-	gzip -9 "${CPIO}"
-	mv -f "${CPIO}.gz" "${CPIO}"
+	# Select arch util
+	if [ "`cat ${KERNEL_DIR}/.config | 
+		grep -Pc "^CONFIG_RD_LZMA=y|^CONFIG_INITRAMFS_COMPRESSION_LZMA=y"`" == "2" ] &&
+		which lzma &>/dev/null;
+	then
+		lzma -9 "${CPIO}"
+		mv -f "${CPIO}.lzma" "${CPIO}"
+	else
+		gzip -9 "${CPIO}"
+		mv -f "${CPIO}.gz" "${CPIO}"
+	fi
 
 	if isTrue "${INTEGRATED_INITRAMFS}"
 	then
