diff -ruN genkernel.orig/defaults/initrd.defaults genkernel/defaults/initrd.defaults
--- genkernel.orig/defaults/initrd.defaults	2009-08-07 16:16:45.000000000 +0400
+++ genkernel/defaults/initrd.defaults	2009-08-07 16:20:03.000000000 +0400
@@ -67,6 +67,7 @@
 fi
 
 REAL_ROOT=''
+SCRATCH='0'
 CDROOT='0'
 CDROOT_DEV=''
 CDROOT_TYPE='auto'
diff -ruN genkernel.orig/defaults/initrd.scripts genkernel/defaults/initrd.scripts
--- genkernel.orig/defaults/initrd.scripts	2009-08-07 16:16:45.000000000 +0400
+++ genkernel/defaults/initrd.scripts	2009-08-07 16:29:44.000000000 +0400
@@ -986,9 +986,61 @@
 		mkdir -p ${UNION}
 #		mkdir -p $CHANGES
 #		mount -t unionfs -o dirs=$CHANGES=rw unionfs ${UNION}
+
+		if [ "${SCRATCH}" -ne 0 ];
+		then
+			good_msg "Loading in scratch mode"
+			BUILDERDIR=${UNION}/mnt/builder
+			SCRATCHDIR=${UNION}/mnt/scratch
+			DELTADIR=${rw_dir}/mnt/delta
+			WORKSPACEDIR=${rw_dir}/mnt/workspace
+			mount -o remount,rw ${rw_dir}/mnt/cdrom >/dev/null 2>&1
+			touch ${rw_dir}/mnt/cdrom/livecd >/dev/null 2>&1
+			if [ "$?" -ne 0 ];
+			then
+				ISRW=0
+			else
+				ISRW=1
+			fi
+
+			mkdir -p ${DELTADIR}
+			mkdir -p ${WORKSPACEDIR}
+			mount -t unionfs -o dirs=${WORKSPACEDIR}=rw:${DELTADIR}=ro:${rw_dir}/mnt/livecd=ro unionfs ${UNION}
+			tmpret=$?;[ "$tmpret" -ne 0 ] && ret=$tmpret
+			mkdir -p ${SCRATCHDIR}/calculate
+			mkdir -p ${SCRATCHDIR}/delta
+			mkdir -p ${SCRATCHDIR}/workspace
+			mount -o bind ${rw_dir}/mnt/delta ${SCRATCHDIR}/delta
+			tmpret=$?;[ "$tmpret" -ne 0 ] && ret=$tmpret
+			mount -o bind ${rw_dir}/mnt/workspace ${SCRATCHDIR}/workspace
+			tmpret=$?;[ "$tmpret" -ne 0 ] && ret=$tmpret
+			mkdir -p ${BUILDERDIR}
+			mount -t unionfs -o dirs=${DELTADIR}=rw:${rw_dir}/mnt/livecd=ro unionfs ${BUILDERDIR}
+			tmpret=$?;[ "$tmpret" -ne 0 ] && ret=$tmpret
+			mount -o bind ${rw_dir}/mnt/livecd ${SCRATCHDIR}/calculate
+			tmpret=$?;[ "$tmpret" -ne 0 ] && ret=$tmpret
+			if [ "$ISRW" -ne 0 ];
+			then
+				FLASHDIR=${UNION}/mnt/flash
+				mkdir -p ${FLASHDIR}
+				mount -o bind ${rw_dir}/mnt/cdrom ${FLASHDIR}
+				tmpret=$?;[ "$tmpret" -ne 0 ] && ret=$tmpret
+			else
+				CDROMDIR=${UNION}/mnt/cdrom
+				mkdir -p ${CDROMDIR}
+				mount -o bind ${rw_dir}/mnt/cdrom ${CDROMDIR}
+				tmpret=$?;[ "$tmpret" -ne 0 ] && ret=$tmpret
+			fi
+		else
+			mount -t unionfs -o dirs=${rw_dir}:${ro_dir}=ro unionfs ${UNION}
+			ret=$?
+		fi
+
+		chmod 1777 ${UNION}/tmp
+
 		good_msg "Creating union mount"
-		unionfs -o allow_other,cow,noinitgroups,suid,dev,default_permissions,use_ino ${rw_dir}=RW:${ro_dir}=RO ${UNION} 2>/dev/null
-		ret=$?
+#		unionfs -o allow_other,cow,noinitgroups,suid,dev,default_permissions,use_ino ${rw_dir}=RW:${ro_dir}=RO ${UNION} 2>/dev/null
+#		ret=$?
 		if [ "${ret}" -ne 0 ]
 		then
 			bad_msg "Can't setup union mount!"
diff -ruN genkernel.orig/defaults/linuxrc genkernel/defaults/linuxrc
--- genkernel.orig/defaults/linuxrc	2009-08-07 16:16:45.000000000 +0400
+++ genkernel/defaults/linuxrc	2009-08-07 16:32:10.000000000 +0400
@@ -62,6 +62,10 @@
 		init_opts\=*)
 			INIT_OPTS=`parse_opt "${x}"`
 		;;
+		# Scratch usbload
+		scratch)
+			SCRATCH=1
+		;;
 		# Livecd options
 		cdroot)
 			CDROOT=1
@@ -464,9 +468,36 @@
 			then
 				break
 			else
-				bad_msg "The filesystem mounted at ${REAL_ROOT} does not appear to be a valid /, try again"
-				got_good_root=0
-				REAL_ROOT=''
+				# check for scratch
+				if [ -e ${NEW_ROOT}/livecd.squashfs -a -d ${NEW_ROOT}/delta -a -d ${NEW_ROOT}/workspace ];
+				then
+					good_msg 'Found scratch data...'
+					modprobe unionfs >/dev/null
+					modprobe squashfs >/dev/null
+					if mount -o remount,rw ${NEW_ROOT};
+					then
+						mkdir -p ${NEW_ROOT}/calculate && mkdir -p /union && mkdir -p ${NEW_ROOT}/workspace/mnt/builder && mkdir -p ${NEW_ROOT}/workspace/mnt/scratch
+						if mount -o loop,ro -t squashfs ${NEW_ROOT}/livecd.squashfs ${NEW_ROOT}/calculate;
+						then
+							if mount -t unionfs -o ro,dirs=${NEW_ROOT}/workspace:${NEW_ROOT}/delta=ro:${NEW_ROOT}/calculate=ro unionfs /union;
+							then
+								mount -t unionfs -o ro,dirs=${NEW_ROOT}/delta:${NEW_ROOT}/calculate=ro unionfs /union/mnt/builder
+								CHROOT=/union
+								break;
+							else
+								bad_msg 'Could not merge scratch layers'
+							fi
+						else
+							bad_msg 'Could not mount scratch squashfs image'
+						fi
+					else
+						bad_msg 'Could not remount ${NEW_ROOT}'
+					fi
+				else
+					bad_msg "The filesystem mounted at ${REAL_ROOT} does not appear to be a valid /, try again"
+					got_good_root=0
+					REAL_ROOT=''
+				fi
 			fi
 		else
 			bad_msg "Could not mount specified ROOT, try again"
