diff --git a/pyuic/uic/uiparser.py b/pyuic/uic/uiparser.py
index 921c273..84feb95 100644
--- a/pyuic/uic/uiparser.py
+++ b/pyuic/uic/uiparser.py
@@ -158,7 +158,7 @@ class UIParser(object):
         # Ignore the parent if it is a container.
         parent = self.stack.topwidget
         if isinstance(parent, (QtGui.QDockWidget, QtGui.QMdiArea,
-                               QtGui.QScrollArea, QtGui.StackedWidget,
+                               QtGui.QScrollArea, QtGui.QStackedWidget,
                                QtGui.QToolBox, QtGui.QTabWidget,
                                QtGui.QWizard)):
             parent = None
