# Calculate format=diff
diff --git a/style/adwaita.h b/style/adwaita.h
index 8aaece0..ec2a0c1 100644
--- a/style/adwaita.h
+++ b/style/adwaita.h
@@ -54,7 +54,7 @@ namespace Adwaita
     {
 
         const bool SingleClick { true };
-        const bool ShowIconsOnPushButtons { true };
+        const bool ShowIconsOnPushButtons { false };
         const int ToolButtonStyle { Qt::ToolButtonTextBesideIcon };
         const bool ShowIconsInMenuItems { true };
 
