diff --git a/gtk/gtkmm/recentinfo.cc b/gtk/gtkmm/recentinfo.cc
index 3d8f7ea..6d0e14b 100644
--- a/gtk/gtkmm/recentinfo.cc
+++ b/gtk/gtkmm/recentinfo.cc
@@ -189,7 +189,7 @@ bool RecentInfo::get_private_hint() const
 
 bool RecentInfo::get_application_info(const Glib::ustring& app_name, Glib::StringArrayHandle& app_exec, guint& count, time_t& time) const
 {
-  return gtk_recent_info_get_application_info(const_cast<GtkRecentInfo*>(gobj()), app_name.c_str(), const_cast<gchar**>((app_exec).data()), &(count), &(time));
+  return gtk_recent_info_get_application_info(const_cast<GtkRecentInfo*>(gobj()), app_name.c_str(), const_cast<const gchar**>((app_exec).data()), &(count), &(time));
 }
 
 Glib::ustring RecentInfo::last_application() const
