diff --git a/desktop/desktop/00-patch-2.2.9/.calculate_directory b/desktop/desktop/00-patch-2.2.9/.calculate_directory
new file mode 100644
index 0000000..ee4dbd5
--- /dev/null
+++ b/desktop/desktop/00-patch-2.2.9/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.8.1
diff --git a/desktop/desktop/00-patch-2.2.9/ini.env b/desktop/desktop/00-patch-2.2.9/ini.env
new file mode 100644
index 0000000..d565ea7
--- /dev/null
+++ b/desktop/desktop/00-patch-2.2.9/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba path=~/.calculate
+
+[ver]
+patch = 2.2.8.1
+
+[update]
+patch = on
diff --git a/desktop/desktop/00-patch-2.2.9/send-attach.desktop b/desktop/desktop/00-patch-2.2.9/send-attach.desktop
new file mode 100644
index 0000000..d08740f
--- /dev/null
+++ b/desktop/desktop/00-patch-2.2.9/send-attach.desktop
@@ -0,0 +1,21 @@
+# Calculate format=kde ini(ver.kde)!= path= ~/.kde4/share/kde4/services/ServiceMenu
+[Desktop Entry]
+Type=Service
+X-KDE-ServiceTypes=KonqPopupMenu/Plugin,all/allfiles
+Actions=openKmailAttachment
+Encoding=UTF-8
+MimeType=application/octet-stream
+ExcludeServiceTypes=application/x-iso,kdedevice/*,inode/directory
+
+[Desktop Action openKmailAttachment]
+Name=Send File As Attachment
+Name[de]=Datei als Anhang versenden
+Name[en]=Send File As Attachment
+Name[fr]=Envoyer le fichier en pièce jointe
+Name[pt]=Enviar por correio-e
+Name[sk]=Poslať e-mailom
+Name[ru]=Отправить в KMail
+
+Icon=kmail
+Exec=kmail --composer --attach %F `echo attachment=\'file://%F\' | sed 's/ \//,file:\/\/\//g'`
+X-Ubuntu-Gettext-Domain=desktop_kdebase
diff --git a/desktop/desktop/20-kde-4.5/.kde4/share/kde4/services/ServiceMenu/send-attach.desktop b/desktop/desktop/20-kde-4.5/.kde4/share/kde4/services/ServiceMenu/send-attach.desktop
new file mode 100644
index 0000000..fa95700
--- /dev/null
+++ b/desktop/desktop/20-kde-4.5/.kde4/share/kde4/services/ServiceMenu/send-attach.desktop
@@ -0,0 +1,20 @@
+[Desktop Entry]
+Type=Service
+X-KDE-ServiceTypes=KonqPopupMenu/Plugin,all/allfiles
+Actions=openKmailAttachment
+Encoding=UTF-8
+MimeType=application/octet-stream
+ExcludeServiceTypes=application/x-iso,kdedevice/*,inode/directory
+
+[Desktop Action openKmailAttachment]
+Name=Send File As Attachment
+Name[de]=Datei als Anhang versenden
+Name[en]=Send File As Attachment
+Name[fr]=Envoyer le fichier en pièce jointe
+Name[pt]=Enviar por correio-e
+Name[sk]=Poslať e-mailom
+Name[ru]=Отправить в KMail
+
+Icon=kmail
+Exec=kmail --composer --attach %F `echo attachment=\'file://%F\' | sed 's/ \//,file:\/\/\//g'`
+X-Ubuntu-Gettext-Domain=desktop_kdebase
