diff --git desktop/desktop/00-patch-2.2.23/.calculate_directory desktop/desktop/00-patch-2.2.23/.calculate_directory
new file mode 100644
index 0000000..dd6d923
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.23
diff --git desktop/desktop/00-patch-2.2.23/audioplayer.desktop desktop/desktop/00-patch-2.2.23/audioplayer.desktop
new file mode 100644
index 0000000..c39d33d
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/audioplayer.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-39/10000000039.desktop)!=&&exists(/usr/share/applications/calculate-audioplayer.desktop)!= link=/usr/share/applications/calculate-audioplayer.desktop path=~/.config/xfce4/panel/launcher-39 name=10000000039.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/browser.desktop desktop/desktop/00-patch-2.2.23/browser.desktop
new file mode 100644
index 0000000..6783f06
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/browser.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-30/10000000030.desktop)!=&&exists(/usr/share/applications/calculate-browser.desktop)!= link=/usr/share/applications/calculate-browser.desktop path=~/.config/xfce4/panel/launcher-30 name=10000000030.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/calc.desktop desktop/desktop/00-patch-2.2.23/calc.desktop
new file mode 100644
index 0000000..dea0ce6
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/calc.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-35/10000000035.desktop)!=&&exists(/usr/share/applications/calculate-calc.desktop)!= link=/usr/share/applications/calculate-calc.desktop path=~/.config/xfce4/panel/launcher-35 name=10000000035.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/calculator.desktop desktop/desktop/00-patch-2.2.23/calculator.desktop
new file mode 100644
index 0000000..d9de472
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/calculator.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-36/10000000036.desktop)!=&&exists(/usr/share/applications/calculate-calculator.desktop)!= link=/usr/share/applications/calculate-calculator.desktop path=~/.config/xfce4/panel/launcher-36 name=10000000036.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/chat.desktop desktop/desktop/00-patch-2.2.23/chat.desktop
new file mode 100644
index 0000000..7aca1c7
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/chat.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-32/10000000032.desktop)!=&&exists(/usr/share/applications/calculate-chat.desktop)!= link=/usr/share/applications/calculate-chat.desktop path=~/.config/xfce4/panel/launcher-32 name=10000000032.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/dict.desktop desktop/desktop/00-patch-2.2.23/dict.desktop
new file mode 100644
index 0000000..bd030d7
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/dict.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-42/10000000042.desktop)!=&&exists(/usr/share/applications/calculate-dict.desktop)!= link=/usr/share/applications/calculate-dict.desktop path=~/.config/xfce4/panel/launcher-42 name=10000000042.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/im.desktop desktop/desktop/00-patch-2.2.23/im.desktop
new file mode 100644
index 0000000..cecf261
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/im.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-33/10000000033.desktop)!=&&exists(/usr/share/applications/calculate-im.desktop)!= link=/usr/share/applications/calculate-im.desktop path=~/.config/xfce4/panel/launcher-33 name=10000000033.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/imageburn.desktop desktop/desktop/00-patch-2.2.23/imageburn.desktop
new file mode 100644
index 0000000..d1fead5
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/imageburn.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-41/10000000041.desktop)!=&&exists(/usr/share/applications/calculate-imageburn.desktop)!= link=/usr/share/applications/calculate-imageburn.desktop path=~/.config/xfce4/panel/launcher-41 name=10000000041.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/imageedit.desktop desktop/desktop/00-patch-2.2.23/imageedit.desktop
new file mode 100644
index 0000000..3d8cf83
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/imageedit.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-37/10000000037.desktop)!=&&exists(/usr/share/applications/calculate-imageedit.desktop)!= link=/usr/share/applications/calculate-imageedit.desktop path=~/.config/xfce4/panel/launcher-37 name=10000000037.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/imageview.desktop desktop/desktop/00-patch-2.2.23/imageview.desktop
new file mode 100644
index 0000000..915366a
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/imageview.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-38/10000000038.desktop)!=&&exists(/usr/share/applications/calculate-imageview.desktop)!= link=/usr/share/applications/calculate-imageview.desktop path=~/.config/xfce4/panel/launcher-38 name=10000000038.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/ini.env desktop/desktop/00-patch-2.2.23/ini.env
new file mode 100644
index 0000000..b93a55a
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba path=~/.calculate
+
+[ver]
+patch = 2.2.23
+
+[update]
+patch = on
diff --git desktop/desktop/00-patch-2.2.23/mail.desktop desktop/desktop/00-patch-2.2.23/mail.desktop
new file mode 100644
index 0000000..2bea389
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/mail.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-31/10000000031.desktop)!=&&exists(/usr/share/applications/calculate-mail.desktop)!= link=/usr/share/applications/calculate-mail.desktop path=~/.config/xfce4/panel/launcher-31 name=10000000031.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/terminal.desktop desktop/desktop/00-patch-2.2.23/terminal.desktop
new file mode 100644
index 0000000..53b9741
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/terminal.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-44/10000000044.desktop)!=&&exists(/usr/share/applications/calculate-terminal.desktop)!= link=/usr/share/applications/calculate-terminal.desktop path=~/.config/xfce4/panel/launcher-44 name=10000000044.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/textedit.desktop desktop/desktop/00-patch-2.2.23/textedit.desktop
new file mode 100644
index 0000000..a86d325
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/textedit.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-43/10000000043.desktop)!=&&exists(/usr/share/applications/calculate-textedit.desktop)!= link=/usr/share/applications/calculate-textedit.desktop path=~/.config/xfce4/panel/launcher-43 name=10000000043.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/videoplayer.desktop desktop/desktop/00-patch-2.2.23/videoplayer.desktop
new file mode 100644
index 0000000..2fc2577
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/videoplayer.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-40/10000000040.desktop)!=&&exists(/usr/share/applications/calculate-videoplayer.desktop)!= link=/usr/share/applications/calculate-videoplayer.desktop path=~/.config/xfce4/panel/launcher-40 name=10000000040.desktop symbolic force
diff --git desktop/desktop/00-patch-2.2.23/writer.desktop desktop/desktop/00-patch-2.2.23/writer.desktop
new file mode 100644
index 0000000..eb9b8cc
--- /dev/null
+++ desktop/desktop/00-patch-2.2.23/writer.desktop
@@ -0,0 +1 @@
+# Calculate exists(~/.config/xfce4/panel/launcher-34/10000000034.desktop)!=&&exists(/usr/share/applications/calculate-writer.desktop)!= link=/usr/share/applications/calculate-writer.desktop path=~/.config/xfce4/panel/launcher-34 name=10000000034.desktop symbolic force
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/kxkbrc desktop/desktop/20-kde-4.6/.kde4/share/config/kxkbrc
index c0d7733..20105e8 100644
--- desktop/desktop/20-kde-4.6/.kde4/share/config/kxkbrc
+++ desktop/desktop/20-kde-4.6/.kde4/share/config/kxkbrc
@@ -12,7 +12,7 @@ Options=terminate:ctrl_alt_bksp
 Options=grp:caps_toggle,grp_led:scroll,terminate:ctrl_alt_bksp
 #os_locale_xkb#
 ResetOldOptions=true
-ShowFlag=true
+ShowFlag=false
 ShowSingle=false
 SwitchMode=WinClass
 Use=true
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-30/browser.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-30/browser.desktop
index 71080ee..cbd7725 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-30/browser.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-30/browser.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-browser.desktop)!= link=/usr/share/applications/calculate-browser.desktop name=10000000030.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-browser.desktop
+# Calculate exists(/usr/share/applications/calculate-browser.desktop)!= link=/usr/share/applications/calculate-browser.desktop name=10000000030.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-31/mail.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-31/mail.desktop
index 4bddcb2..26fcecd 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-31/mail.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-31/mail.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-mail.desktop)!= link=/usr/share/applications/calculate-mail.desktop name=10000000031.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-mail.desktop
+# Calculate exists(/usr/share/applications/calculate-mail.desktop)!= link=/usr/share/applications/calculate-mail.desktop name=10000000031.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-32/chat.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-32/chat.desktop
index 84da5ad..94e8002 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-32/chat.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-32/chat.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-chat.desktop)!= link=/usr/share/applications/calculate-chat.desktop name=10000000032.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-chat.desktop
+# Calculate exists(/usr/share/applications/calculate-chat.desktop)!= link=/usr/share/applications/calculate-chat.desktop name=10000000032.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-33/im.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-33/im.desktop
index 792351f..babdad5 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-33/im.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-33/im.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-im.desktop)!= link=/usr/share/applications/calculate-im.desktop name=10000000033.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-im.desktop
+# Calculate exists(/usr/share/applications/calculate-im.desktop)!= link=/usr/share/applications/calculate-im.desktop name=10000000033.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-34/writer.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-34/writer.desktop
index 3844f4c..0f8ee73 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-34/writer.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-34/writer.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-writer.desktop)!= link=/usr/share/applications/calculate-writer.desktop name=10000000034.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-writer.desktop
+# Calculate exists(/usr/share/applications/calculate-writer.desktop)!= link=/usr/share/applications/calculate-writer.desktop name=10000000034.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-35/calc.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-35/calc.desktop
index c8b098f..2eaf1dd 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-35/calc.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-35/calc.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-calc.desktop)!= link=/usr/share/applications/calculate-calc.desktop name=10000000035.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-calc.desktop
+# Calculate exists(/usr/share/applications/calculate-calc.desktop)!= link=/usr/share/applications/calculate-calc.desktop name=10000000035.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-36/calculator.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-36/calculator.desktop
index e3db59c..306f0d8 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-36/calculator.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-36/calculator.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-calculator.desktop)!= link=/usr/share/applications/calculate-calculator.desktop name=10000000036.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-calculator.desktop
+# Calculate exists(/usr/share/applications/calculate-calculator.desktop)!= link=/usr/share/applications/calculate-calculator.desktop name=10000000036.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-37/imageedit.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-37/imageedit.desktop
index 088648c..a42cc67 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-37/imageedit.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-37/imageedit.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-imageedit.desktop)!= link=/usr/share/applications/calculate-imageedit.desktop name=10000000037.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-imageedit.desktop
+# Calculate exists(/usr/share/applications/calculate-imageedit.desktop)!= link=/usr/share/applications/calculate-imageedit.desktop name=10000000037.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-38/imageview.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-38/imageview.desktop
index cb488a8..6a8f5e7 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-38/imageview.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-38/imageview.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-imageview.desktop)!= link=/usr/share/applications/calculate-imageview.desktop name=10000000038.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-imageview.desktop
+# Calculate exists(/usr/share/applications/calculate-imageview.desktop)!= link=/usr/share/applications/calculate-imageview.desktop name=10000000038.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-39/audioplayer.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-39/audioplayer.desktop
index db61067..1e65abb 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-39/audioplayer.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-39/audioplayer.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-audioplayer.desktop)!= link=/usr/share/applications/calculate-audioplayer.desktop name=10000000039.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-audioplayer.desktop
+# Calculate exists(/usr/share/applications/calculate-audioplayer.desktop)!= link=/usr/share/applications/calculate-audioplayer.desktop name=10000000039.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-40/videoplayer.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-40/videoplayer.desktop
index 00614a7..c5e5084 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-40/videoplayer.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-40/videoplayer.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-videoplayer.desktop)!= link=/usr/share/applications/calculate-videoplayer.desktop name=10000000040.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-videoplayer.desktop
+# Calculate exists(/usr/share/applications/calculate-videoplayer.desktop)!= link=/usr/share/applications/calculate-videoplayer.desktop name=10000000040.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-41/imageburn.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-41/imageburn.desktop
index 16fac96..5f386c4 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-41/imageburn.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-41/imageburn.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-imageburn.desktop)!= link=/usr/share/applications/calculate-imageburn.desktop name=10000000041.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-imageburn.desktop
+# Calculate exists(/usr/share/applications/calculate-imageburn.desktop)!= link=/usr/share/applications/calculate-imageburn.desktop name=10000000041.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-42/dict.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-42/dict.desktop
index 07fe94c..c983499 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-42/dict.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-42/dict.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-dict.desktop)!= link=/usr/share/applications/calculate-dict.desktop name=10000000042.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-dict.desktop
+# Calculate exists(/usr/share/applications/calculate-dict.desktop)!= link=/usr/share/applications/calculate-dict.desktop name=10000000042.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-43/textedit.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-43/textedit.desktop
index e8fa798..26ae5c1 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-43/textedit.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-43/textedit.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-textedit.desktop)!= link=/usr/share/applications/calculate-textedit.desktop name=10000000043.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-textedit.desktop
+# Calculate exists(/usr/share/applications/calculate-textedit.desktop)!= link=/usr/share/applications/calculate-textedit.desktop name=10000000043.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-44/terminal.desktop desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-44/terminal.desktop
index 132307d..422b1a1 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-44/terminal.desktop
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/panel/launcher-44/terminal.desktop
@@ -1,3 +1 @@
-# Calculate format=kde exists(/usr/share/applications/calculate-terminal.desktop)!= link=/usr/share/applications/calculate-terminal.desktop name=10000000044.desktop
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/calculate-terminal.desktop
+# Calculate exists(/usr/share/applications/calculate-terminal.desktop)!= link=/usr/share/applications/calculate-terminal.desktop name=10000000044.desktop symbolic force
diff --git desktop/desktop/40-xfce-4.8/.config/xfce4/xfconf/xfce-perchannel-xml/xfce4-panel.xml desktop/desktop/40-xfce-4.8/.config/xfce4/xfconf/xfce-perchannel-xml/xfce4-panel.xml
index 6cda23a..a1311f2 100644
--- desktop/desktop/40-xfce-4.8/.config/xfce4/xfconf/xfce-perchannel-xml/xfce4-panel.xml
+++ desktop/desktop/40-xfce-4.8/.config/xfce4/xfconf/xfce-perchannel-xml/xfce4-panel.xml
@@ -202,5 +202,10 @@
         <value type="string" value="10000000044.desktop"/>
       </property>
     </property>
+    <property name="clipman" type="empty">
+      <property name="settings" type="empty">
+        <property name="add-primary-clipboard" type="bool" value="true"/>
+      </property>
+    </property>
   </property>
 </channel>
diff --git desktop/desktop/60-xbmc-10.0/.calculate/ini.env desktop/desktop/60-xbmc-10.0/.calculate/ini.env
new file mode 100644
index 0000000..764e4ad
--- /dev/null
+++ desktop/desktop/60-xbmc-10.0/.calculate/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba
+
+[ver]
+xbmc = 10.1
+
+[update]
+xbmc = on
diff --git desktop/desktop/60-xbmc-10.0/.calculate_directory desktop/desktop/60-xbmc-10.0/.calculate_directory
new file mode 100644
index 0000000..dcb6016
--- /dev/null
+++ desktop/desktop/60-xbmc-10.0/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(xbmc)!=&&ini(update.xbmc)!=off&&ini(ver.xbmc)==&&pkg(media-tv/xbmc)>=10.0
diff --git desktop/desktop/60-xbmc-10.0/guisettings.xml desktop/desktop/60-xbmc-10.0/guisettings.xml
new file mode 100644
index 0000000..094edba
--- /dev/null
+++ desktop/desktop/60-xbmc-10.0/guisettings.xml
@@ -0,0 +1,10 @@
+# Calculate path=~/.xbmc/userdata
+<settings>
+    <locale>
+        <charset>CP1251</charset>
+        <country>USA</country>
+        <language>Russian</language>
+        <timezone>Europe/Moscow</timezone>
+        <timezonecountry>Russia</timezonecountry>
+    </locale>
+</settings>
diff --git desktop/desktop/60-xbmc-10.0/sources.xml desktop/desktop/60-xbmc-10.0/sources.xml
new file mode 100644
index 0000000..ca2bedf
--- /dev/null
+++ desktop/desktop/60-xbmc-10.0/sources.xml
@@ -0,0 +1,30 @@
+# Calculate path=~/.xbmc/userdata
+<sources>
+    <programs>
+        <default pathversion="1"></default>
+    </programs>
+    <video>
+        <default pathversion="1"></default>
+        <source>
+            <name>My Video</name>
+            <path pathversion="1">#-ini(resource.videos)-#</path>
+        </source>
+    </video>
+    <music>
+        <default pathversion="1"></default>
+        <source>
+            <name>My Music</name>
+            <path pathversion="1">#-ini(resource.music)-#</path>
+        </source>
+    </music>
+    <pictures>
+        <default pathversion="1"></default>
+        <source>
+            <name>My Pictures</name>
+            <path pathversion="1">#-ini(resource.pictures)-#</path>
+        </source>
+    </pictures>
+    <files>
+        <default pathversion="1"></default>
+    </files>
+</sources>
diff --git install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
index a0bc2a5..ff49e6f 100644
--- install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
+++ install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/share/applications/audacious2.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=2.4&&pkg(media-sound/audacious)<3.0
+# Calculate format=kde link=/usr/share/applications/audacious2.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=2.4&&pkg(media-sound/audacious)<2.5
 
 # Update info
 #-ini(desktop.audioplayer, audacious)-#
diff --git install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.5.desktop install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.5.desktop
new file mode 100644
index 0000000..d24772b
--- /dev/null
+++ install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.5.desktop
@@ -0,0 +1,9 @@
+# Calculate format=kde link=/usr/share/applications/audacious.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=2.5
+
+# Update info
+#-ini(desktop.audioplayer, audacious)-#
+
+# Hide icon
+[Desktop Entry]
+OnlyShowIn=X-NeverShowThis;
+
diff --git install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop
deleted file mode 100644
index 6387c9e..0000000
--- install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop
+++ /dev/null
@@ -1,9 +0,0 @@
-# Calculate format=kde link=/usr/share/applications/audacious.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=3.0
-
-# Update info
-#-ini(desktop.audioplayer, audacious)-#
-
-# Hide icon
-[Desktop Entry]
-OnlyShowIn=X-NeverShowThis;
-
diff --git install/1merge/xorg-server/X11/xorg.conf install/1merge/xorg-server/X11/xorg.conf
index 210d3a1..204c197 100644
--- install/1merge/xorg-server/X11/xorg.conf
+++ install/1merge/xorg-server/X11/xorg.conf
@@ -1,4 +1,4 @@
-# Calculate comment=# os_install_video_drv!=other
+# Calculate comment=# os_install_x11_video_drv!=other
 
 Section "ServerLayout"
   Identifier "Xorg Configured"
diff --git install/1merge/xorg-server/X11/xorg.conf.other install/1merge/xorg-server/X11/xorg.conf.other
index 9342c02..72ce36b 100644
--- install/1merge/xorg-server/X11/xorg.conf.other
+++ install/1merge/xorg-server/X11/xorg.conf.other
@@ -1,4 +1,4 @@
-# Calculate comment=# os_install_video_drv==other name=xorg.conf
+# Calculate comment=# os_install_x11_video_drv==other name=xorg.conf
 
 Section "ServerLayout"
  Identifier "Xorg Configured"
