diff --git desktop/desktop/60-skype-2.2/Logs/.calculate_directory desktop/desktop/60-skype-2.2/Logs/.calculate_directory
new file mode 100644
index 0000000..afec7e6
--- /dev/null
+++ desktop/desktop/60-skype-2.2/Logs/.calculate_directory
@@ -0,0 +1 @@
+# Calculate
diff --git install/1live/calculate-utilities/fixOpenrc install/1live/calculate-utilities/fixOpenrc
new file mode 100644
index 0000000..0af0038
--- /dev/null
+++ install/1live/calculate-utilities/fixOpenrc
@@ -0,0 +1,8 @@
+# Calculate path=/tmp name=fixOpenrc exec=/bin/bash ac_install_system!=up&&cl_belong_pkg!=&&pkg(sys-apps/openrc)!=
+
+if [[ -z `find /etc/conf.d -name "*cfg*modules" -exec grep "^modules_3" {} \;` ]]
+then
+    find /etc/conf.d -name "*cfg*modules" -delete
+    rc-update add calculate-builder boot
+fi
+true
diff --git install/1live/openrc/conf.d/modules install/1live/openrc/conf.d/modules
new file mode 100644
index 0000000..5d6b9bd
--- /dev/null
+++ install/1live/openrc/conf.d/modules
@@ -0,0 +1,2 @@
+# Calculate format=openrc ac_install_system!=up&&cl_belong_pkg!=
+modules_3="#-os_install_kernel_cpufreq-#"
diff --git install/1live/portage/make.conf install/1live/portage/make.conf
index 3118ec0..602546d 100644
--- install/1live/portage/make.conf
+++ install/1live/portage/make.conf
@@ -46,16 +46,16 @@ ACCEPT_LICENSE="*"
 
 #?os_install_locale_language!=ru&&os_install_locale_language!=bg#
 # Alternative mirror updates repository. In case of problems with access
-# at the beginning try to update the overlay by running `layman-s calculate`.
+# at the beginning try to update the overlay by running `eix-sync`.
 #os_install_locale_language#
 #?os_install_locale_language==ru#
 # Альтернативные зеркала обновленого репозитория. В случае проблем с доступом
-# попробуйте в начале обновить оверлей, выполнив `layman -s calculate`.
+# попробуйте в начале обновить оверлей, выполнив `eix-sync`.
 #os_install_locale_language#
 #?os_install_locale_language==bg#
 # Алтернативни огледални сървъри за обновяване на локалните хранилища.
 # В случай че има проблем с достъпа пробвайте отначало да обновите
-# хранилищата `layman -s calculate`.
+# хранилищата `eix-sync`.
 #os_install_locale_language#
 #?os_install_linux_shortname==CLD||os_install_linux_shortname==CLDG||os_install_linux_shortname==CLDX||os_install_linux_shortname==CDS||os_install_linux_shortname==CSS#
 #PORTAGE_BINHOST="http://77.88.19.68/calculate/#-os_install_linux_shortname-#/grp/#-os_install_arch_machine-#"
diff --git install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop
index c2f95e7..dd212d6 100644
--- install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop
+++ install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2
+# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2&&os_install_arch_machine==i686
 
 # Update info
 #-ini(desktop.writer, libreoffice)-#
diff --git install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop.amd64 install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop.amd64
new file mode 100644
index 0000000..3acfbe9
--- /dev/null
+++ install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop.amd64
@@ -0,0 +1,8 @@
+# Calculate format=kde link=/usr/lib64/libreoffice/share/xdg/writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2&&os_install_arch_machine==x86_64
+
+# Update info
+#-ini(desktop.writer, libreoffice)-#
+
+# Hide icon
+[Desktop Entry]
+NoDisplay=true
diff --git install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop
index 8ec5dcb..972c545 100644
--- install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop
+++ install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2
+# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2&&os_install_arch_machine==i686
 
 # Update info
 #-ini(desktop.calc, libreoffice)-#
diff --git install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop.amd64 install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop.amd64
new file mode 100644
index 0000000..f9fa2d7
--- /dev/null
+++ install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop.amd64
@@ -0,0 +1,8 @@
+# Calculate format=kde link=/usr/lib64/libreoffice/share/xdg/calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2&&os_install_arch_machine==x86_64
+
+# Update info
+#-ini(desktop.calc, libreoffice)-#
+
+# Hide icon
+[Desktop Entry]
+NoDisplay=true
diff --git install/1merge/alsa-utils/init.d/alsasound install/1merge/alsa-utils/init.d/alsasound
index f6cb83a..3a8cf94 100644
--- install/1merge/alsa-utils/init.d/alsasound
+++ install/1merge/alsa-utils/init.d/alsasound
@@ -16,7 +16,7 @@
 		if ! grep -q "PCM Playback Volume" ${alsastatedir}/asound.state
 		then
 			sed -ri "s/^\}/\
-	control.39 \{\\n\
+	control.99 \{\\n\
 		iface MIXER\\n\
 		name 'PCM Playback Volume'\\n\
 		value.0 255\\n\
diff --git install/1merge/libreoffice-3.4.4.2/.calculate_directory install/1merge/libreoffice-3.4.4.2/.calculate_directory
new file mode 100644
index 0000000..c2a5655
--- /dev/null
+++ install/1merge/libreoffice-3.4.4.2/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong(libreoffice)!=&&pkg(app-office/libreoffice)>3.4.4&&pkg(app-office/libreoffice)<3.6&&pkg(kde-base/kdelibs)!= path=/etc name=env.d
diff --git install/1merge/libreoffice-3.4.4.2/99libreoffice install/1merge/libreoffice-3.4.4.2/99libreoffice
new file mode 100644
index 0000000..6b500a8
--- /dev/null
+++ install/1merge/libreoffice-3.4.4.2/99libreoffice
@@ -0,0 +1,2 @@
+# Calculate pkg(app-office/libreoffice)>3.4.4&&pkg(app-office/libreoffice)<3.5&&pkg(kde-base/kdelibs)!=
+OOO_FORCE_DESKTOP=kde
diff --git install/1merge/libreoffice-3.4.4.2/99libreoffice.del install/1merge/libreoffice-3.4.4.2/99libreoffice.del
new file mode 100644
index 0000000..38a79f5
--- /dev/null
+++ install/1merge/libreoffice-3.4.4.2/99libreoffice.del
@@ -0,0 +1 @@
+# Calculate append=remove name=99libreoffice pkg(app-office/libreoffice)>=3.5&&pkg(kde-base/kdelibs)!=
