diff --git builder/squash/etc/calculate/calculate2.env builder/squash/etc/calculate/calculate2.env
index 2f499e2..e6e2c4d 100644
--- builder/squash/etc/calculate/calculate2.env
+++ builder/squash/etc/calculate/calculate2.env
@@ -10,3 +10,6 @@ os_linux_build = #-os_builder_linux_build-#
 #?os_builder_linux_build==||pkg(sys-apps/calculate-lib)<=2.2.2#
 !os_linux_build = empty
 #os_builder_linux_build#
+
+[install]
+!os_install_net_settings = empty
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/kwalletrc desktop/desktop/20-kde-4.6/.kde4/share/config/kwalletrc
index ea56be2..02ee787 100644
--- desktop/desktop/20-kde-4.6/.kde4/share/config/kwalletrc
+++ desktop/desktop/20-kde-4.6/.kde4/share/config/kwalletrc
@@ -1,7 +1,7 @@
 # Calculate format=kde
-#приложения, которым доступ разрешен
+#applications that access is allowed
 [Auto Allow]
-kdewallet=Kopete,KMail,Konqueror,Chrome
+kdewallet=Kopete,KMail,Konqueror,Chrome,Network Management,KDE Daemon,KDE-Dienst,Servicio de KDE,Démon de KDE,Demone KDE,Servidor do KDE,Serviço do KDE,Demon KDE,Служба KDE,Фонова служба KDE
 
 [Wallet]
 Close When Idle=false
diff --git install/1merge/baselayout/motd.en install/1merge/baselayout/motd.en
index eef29e7..0eace80 100644
--- install/1merge/baselayout/motd.en
+++ install/1merge/baselayout/motd.en
@@ -3,11 +3,15 @@ Welcome to the #-os_install_linux_name-# Installation CD!
 
 If any ethernet adapters were detected at boot, they should be auto-configured
 if DHCP is available on your network.
+To specify network settings, use utility "cl_install".
+Example:"cl-install --live --ip 192.168.1.100/24 \
+          --route default:192.168.1.1 \
+          --dns 192.168.1.1"
 #?os_install_linux_system==server||os_install_linux_shortname==CLS#
-Type "net-setup eth0" to specify eth0 IP address settings by hand.
+And restart net.eth0 by command "/etc/init.d/net.eth0 restart".
 #os_install_linux_system#
 #?os_install_linux_system==desktop&&os_install_linux_shortname!=CLS#
-Type "wicd-curses" to specify eth0 IP address settings by hand.
+And restart NetworkManager by command "/etc/init.d/NetworkManager restart".
 #os_install_linux_system#
 
 The latest version of the Handbook is always available from the Calculate Linux
diff --git install/1merge/baselayout/motd.ru install/1merge/baselayout/motd.ru
index 639a782..c9fbad8 100644
--- install/1merge/baselayout/motd.ru
+++ install/1merge/baselayout/motd.ru
@@ -3,6 +3,16 @@
 
 Если какой-либо из сетевых адаπтеров был обнаружен πри загрузке, он будет
 автоматически сконфигурирован, если в сети достуπен DHCP сервер.
+Для указания сетевых настроек используйте утилиту "cl-install".
+Пример:"cl-install --live --ip 192.168.1.100/24 \
+        --route default:192.168.1.1 \
+        --dns 192.168.1.1"
+#?os_install_linux_system==server||os_install_linux_shortname==CLS#
+И перезапустите net.eth0 командой "/etc/init.d/net.eth0 restart".
+#os_install_linux_system#
+#?os_install_linux_system==desktop&&os_install_linux_shortname!=CLS#
+И перезапустите NetworkManager командой "/etc/init.d/NetworkManager restart".
+#os_install_linux_system#
 #?os_install_linux_system==server||os_install_linux_shortname==CLS#
 Выπолните "net-setup eth0" для указания сетевой карты и IP адреса вручную.
 #os_install_linux_system#
diff --git install/1merge/grub2/default.grub install/1merge/grub2/default.grub
index f719541..8b859a8 100644
--- install/1merge/grub2/default.grub
+++ install/1merge/grub2/default.grub
@@ -1,4 +1,4 @@
-# Calculate format=openrc path=/etc/default name=grub
+# Calculate format=openrc path=/etc/default name=grub os_install_root_type!=usb-hdd
 GRUB_DISABLE_RECOVERY=true
 GRUB_DEFAULT=saved
 GRUB_TIMEOUT=3
diff --git install/1merge/grub2/default.grub.usbhdd install/1merge/grub2/default.grub.usbhdd
new file mode 100644
index 0000000..3764977
--- /dev/null
+++ install/1merge/grub2/default.grub.usbhdd
@@ -0,0 +1,29 @@
+# Calculate format=openrc path=/etc/default name=grub os_install_root_type==usb-hdd
+GRUB_DISABLE_RECOVERY=true
+GRUB_DEFAULT=saved
+GRUB_TIMEOUT=3
+#?pkg(media-gfx/cld-themes)!=||pkg(media-gfx/cldx-themes)!=||pkg(media-gfx/cldg-themes)!=||pkg(media-gfx/cls-themes)!=||pkg(media-gfx/cds-themes)!=||pkg(media-gfx/cmc-themes)!=#
+GRUB_CMDLINE_LINUX="video=uvesafb:ywrap,#-os_install_fb_resolution-#,mtrr:3,splash=silent,theme:tty1 #-os_install_kernel_resume-# elevator=#-os_install_kernel_scheduler-# doscsi#-os_install_kernel_attr-# console=tty1 udev quiet"
+#pkg#
+#?pkg(media-gfx/cld-themes)==&&pkg(media-gfx/cldx-themes)==&&pkg(media-gfx/cldg-themes)==&&pkg(media-gfx/cls-themes)==&&pkg(media-gfx/cds-themes)==&&pkg(media-gfx/cmc-themes)==#
+GRUB_CMDLINE_LINUX="video=uvesafb:ywrap,#-os_install_fb_resolution-#,mtrr:3,splash=off,theme:tty1 #-os_install_kernel_resume-# elevator=#-os_install_kernel_scheduler-# doscsi#-os_install_kernel_attr-# console=tty1 udev quiet"
+#pkg#
+!GRUB_TERMINAL=del
+GRUB_GFXMODE="1024x768"
+GRUB_BACKGROUND="/boot/grub/grub-calculate.png"
+GRUB_COLOR_NORMAL="white/black" 
+GRUB_COLOR_HIGHLIGHT="magenta/black"
+GRUB_VIDEO_BACKEND="vbe"
+#?os_install_linux_subname==&&os_install_scratch==off#
+GRUB_DISTRIBUTOR="#-os_install_linux_name-#"
+#os_install_linux_subname#
+#?os_install_linux_subname!=&&os_install_scratch==off#
+GRUB_DISTRIBUTOR="#-os_install_linux_name-# #-os_install_linux_subname-#"
+#os_install_linux_subname#
+#?os_install_linux_subname==&&os_install_scratch==on#
+GRUB_DISTRIBUTOR="#-os_install_linux_name-# Builder"
+#os_install_linux_subname#
+#?os_install_linux_subname!=&&os_install_scratch==on#
+GRUB_DISTRIBUTOR="#-os_install_linux_name-# #-os_install_linux_subname-# Builder"
+#os_install_linux_subname#
+GRUB_DISABLE_OS_PROBER=true
diff --git install/1merge/grub2/update-grub install/1merge/grub2/update-grub
index 561679f..9ffe0a3 100644
--- install/1merge/grub2/update-grub
+++ install/1merge/grub2/update-grub
@@ -1,3 +1,5 @@
 # Calculate path=/tmp exec=/bin/bash os_install_root_type!=livecd&&exists(/sbin/grub-mkconfig,root)!=
 
-CHROOT_PATH=#-cl_chroot_grub-# /sbin/grub-mkconfig >#-cl_chroot_path-#/boot/grub/grub.cfg 2>/dev/null
+/sbin/grub-mkdevicemap
+/sbin/grub-mkdevicemap -m #-cl_chroot_path-#/boot/grub/device.map
+CHROOT_PATH=#-cl_chroot_grub-# /sbin/grub-mkconfig -o #-cl_chroot_path-#/boot/grub/grub.cfg 2>/dev/null
diff --git install/1system/grub/device.map install/1system/grub/device.map
index 730cb1e..9d3e84f 100644
--- install/1system/grub/device.map
+++ install/1system/grub/device.map
@@ -1 +1,2 @@
+# Calculate os_grub2_path==
 #-os_install_grub_devicemap_conf-#
