diff --git builder/iso/calcboot/isolinux/isolinux.cfg.desktop.oldkernel builder/iso/calcboot/isolinux/isolinux.cfg.desktop.oldkernel
index ead6f23..6cc2f39 100644
--- builder/iso/calcboot/isolinux/isolinux.cfg.desktop.oldkernel
+++ builder/iso/calcboot/isolinux/isolinux.cfg.desktop.oldkernel
@@ -1,4 +1,4 @@
-# Calculate name=isolinux.cfg os_builder_linux_system==desktop&&pkg(sys-kernel/calculate-sources)<2.6.32.33||os_builder_linux_system==desktop&&pkg(sys-kernel/calculate-sources)>2.6.33&&pkg(sys-kernel/calculate-sources)<2.6.37.4#
+# Calculate name=isolinux.cfg os_builder_linux_system==desktop&&pkg(sys-kernel/calculate-sources)<2.6.32.33||os_builder_linux_system==desktop&&pkg(sys-kernel/calculate-sources)>2.6.33&&pkg(sys-kernel/calculate-sources)<2.6.37.4
 DEFAULT calcmenu.c32
 INCLUDE common.cfg
 
diff --git builder/iso/calcboot/isolinux/isolinux.cfg.server.oldkernel builder/iso/calcboot/isolinux/isolinux.cfg.server.oldkernel
index 69245ee..b6ba85c 100644
--- builder/iso/calcboot/isolinux/isolinux.cfg.server.oldkernel
+++ builder/iso/calcboot/isolinux/isolinux.cfg.server.oldkernel
@@ -1,4 +1,4 @@
-# Calculate name=isolinux.cfg os_builder_linux_system==server&&pkg(sys-kernel/calculate-sources)<2.6.32.33||os_builder_linux_system==server&&pkg(sys-kernel/calculate-sources)>2.6.33&&pkg(sys-kernel/calculate-sources)<2.6.37.4#
+# Calculate name=isolinux.cfg os_builder_linux_system==server&&pkg(sys-kernel/calculate-sources)<2.6.32.33||os_builder_linux_system==server&&pkg(sys-kernel/calculate-sources)>2.6.33&&pkg(sys-kernel/calculate-sources)<2.6.37.4
 DEFAULT calcmenu.c32
 INCLUDE common.cfg
 
diff --git install/1merge/baselayout/gentoo-release install/1merge/baselayout/gentoo-release
new file mode 100644
index 0000000..739cc33
--- /dev/null
+++ install/1merge/baselayout/gentoo-release
@@ -0,0 +1,6 @@
+#?os_install_linux_subname!=#
+#-os_install_linux_name-# #-os_install_linux_ver-# #-os_install_linux_subname-#
+#os_install_linux_subname#
+#?os_install_linux_subname==#
+#-os_install_linux_name-# #-os_install_linux_ver-#
+#os_install_linux_subname#
diff --git install/1merge/portage/make.conf install/1merge/portage/make.conf
index a539fe6..6928e79 100644
--- install/1merge/portage/make.conf
+++ install/1merge/portage/make.conf
@@ -60,6 +60,7 @@ ACCEPT_LICENSE="*"
 #?os_install_linux_shortname==CLD||os_install_linux_shortname==CLDG||os_install_linux_shortname==CLDX||os_install_linux_shortname==CDS#
 #PORTAGE_BINHOST="http://77.88.19.68/calculate/#-os_install_linux_shortname-#/grp/#-os_install_arch_machine-#"
 #PORTAGE_BINHOST="ftp://ftp.linux.kiev.ua/pub/Linux/Calculate/#-os_install_linux_shortname-#/grp/#-os_install_arch_machine-#"
+#PORTAGE_BINHOST="ftp://opennix.tula.su/calculate/#-os_install_linux_shortname-#/grp/#-os_install_arch_machine-#"
 #os_install_linux_shortname#
 
 #?os_install_locale_language!=ru&&os_install_locale_language!=bg#
