diff --git a/desktop/desktop/20-kde-4.6/.kde4/share/config/ksplashrc b/desktop/desktop/20-kde-4.6/.kde4/share/config/ksplashrc
index e323c6c..a69927b 100644
--- a/desktop/desktop/20-kde-4.6/.kde4/share/config/ksplashrc
+++ b/desktop/desktop/20-kde-4.6/.kde4/share/config/ksplashrc
@@ -2,9 +2,9 @@
 #тема загрузки рабочего стола
 [KSplash]
 Engine=KSplashX
-#?os_install_locale_lang!=pt_BR#
+#?os_locale_lang!=pt_BR#
 Theme=CalculateSplash#-case(capitalize,os_locale_language)-#
-#os_install_locale_lang#
-#?os_install_locale_lang==pt_BR#
+#os_locale_lang#
+#?os_locale_lang==pt_BR#
 Theme=CalculateSplashPt
-#os_install_locale_lang#
+#os_locale_lang#
diff --git a/install/1merge/glibc/locale.gen b/install/1merge/glibc/locale.gen
index cf85a94..15cff6b 100644
--- a/install/1merge/glibc/locale.gen
+++ b/install/1merge/glibc/locale.gen
@@ -26,7 +26,6 @@ nn_NO.UTF-8 UTF-8
 pl_PL.UTF-8 UTF-8
 #Portuguese
 pt_PT.UTF-8 UTF-8
-pt_PT.UTF-8@euro UTF-8
 #Russian
 ru_RU.UTF-8 UTF-8
 #Spanish
diff --git a/install/1merge/grub2/default.grub b/install/1merge/grub2/default.grub
index a3f5f9f..67ddb16 100644
--- a/install/1merge/grub2/default.grub
+++ b/install/1merge/grub2/default.grub
@@ -2,17 +2,27 @@
 GRUB_DISABLE_RECOVERY=true
 GRUB_DEFAULT=saved
 GRUB_TIMEOUT=3
-GRUB_CMDLINE_LINUX_DEFAULT="console=tty1 udev quiet doscsi"
-GRUB_CMDLINE_LINUX="video=uvesafb:ywrap,#-os_install_fb_resolution-#,mtrr:3,splash=silent,theme:tty1 #-os_install_kernel_resume-# elevator=#-os_install_kernel_scheduler-# doscsi#-os_install_kernel_attr-# calculate=video:#-os_install_x11_video_drv-#"
+#?pkg(media-gfx/cld-themes)!=||pkg(media-gfx/cldx-themes)!=||pkg(media-gfx/cldg-themes)!=||pkg(media-gfx/cls-themes)!=||pkg(media-gfx/cds-themes)!=||pkg(media-gfx/cmc-themes)!=#
+GRUB_CMDLINE_LINUX="video=uvesafb:ywrap,#-os_install_fb_resolution-#,mtrr:3,splash=silent,theme:tty1 #-os_install_kernel_resume-# elevator=#-os_install_kernel_scheduler-# doscsi#-os_install_kernel_attr-# calculate=video:#-os_install_x11_video_drv-# console=tty1 udev quiet"
+#pkg#
+#?pkg(media-gfx/cld-themes)==&&pkg(media-gfx/cldx-themes)==&&pkg(media-gfx/cldg-themes)==&&pkg(media-gfx/cls-themes)==&&pkg(media-gfx/cds-themes)==&&pkg(media-gfx/cmc-themes)==#
+GRUB_CMDLINE_LINUX="video=uvesafb:ywrap,#-os_install_fb_resolution-#,mtrr:3,splash=off,theme:tty1 #-os_install_kernel_resume-# elevator=#-os_install_kernel_scheduler-# doscsi#-os_install_kernel_attr-# calculate=video:#-os_install_x11_video_drv-# console=tty1 udev quiet"
+#pkg#
 GRUB_TERMINAL="gfxterm"
 GRUB_GFXMODE="auto"
 GRUB_BACKGROUND="/boot/grub/grub-calculate.png"
 GRUB_COLOR_NORMAL="white/black" 
 GRUB_COLOR_HIGHLIGHT="magenta/black"
 GRUB_VIDEO_BACKEND="vbe"
-#?os_install_linux_subname==#
+#?os_install_linux_subname==&&os_install_scratch==off#
 GRUB_DISTRIBUTOR="#-os_install_linux_name-#"
 #os_install_linux_subname#
-#?os_install_linux_subname!=#
+#?os_install_linux_subname!=&&os_install_scratch==off#
 GRUB_DISTRIBUTOR="#-os_install_linux_name-# #-os_install_linux_subname-#"
 #os_install_linux_subname#
+#?os_install_linux_subname==&&os_install_scratch==on#
+GRUB_DISTRIBUTOR="#-os_install_linux_name-# Builder"
+#os_install_linux_subname#
+#?os_install_linux_subname!=&&os_install_scratch==on#
+GRUB_DISTRIBUTOR="#-os_install_linux_name-# #-os_install_linux_subname-# Builder"
+#os_install_linux_subname#
diff --git a/install/1merge/openrc/boot.runlevels/.calculate_directory b/install/1merge/openrc/boot.runlevels/.calculate_directory
new file mode 100644
index 0000000..e5f23f4
--- /dev/null
+++ b/install/1merge/openrc/boot.runlevels/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate path=/etc/runlevels name=boot
+
diff --git a/install/1merge/openrc/boot.runlevels/consolefont b/install/1merge/openrc/boot.runlevels/consolefont
new file mode 100644
index 0000000..fb62863
--- /dev/null
+++ b/install/1merge/openrc/boot.runlevels/consolefont
@@ -0,0 +1 @@
+# Calculate link=/etc/init.d/consolefont symbolic
diff --git a/install/1merge/openrc/default.runlevels/.calculate_directory b/install/1merge/openrc/default.runlevels/.calculate_directory
new file mode 100644
index 0000000..7e25c99
--- /dev/null
+++ b/install/1merge/openrc/default.runlevels/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate path=/etc/runlevels name=default cl_chroot_status==off
+
diff --git a/install/1merge/openrc/default.runlevels/numlock b/install/1merge/openrc/default.runlevels/numlock
new file mode 100644
index 0000000..0bc43c1
--- /dev/null
+++ b/install/1merge/openrc/default.runlevels/numlock
@@ -0,0 +1 @@
+# Calculate link=/etc/init.d/numlock symbolic hr_laptop!=
diff --git a/install/1merge/openrc/defaultrunlevel/.calculate_directory b/install/1merge/openrc/defaultrunlevel/.calculate_directory
deleted file mode 100644
index 7e25c99..0000000
--- a/install/1merge/openrc/defaultrunlevel/.calculate_directory
+++ /dev/null
@@ -1,2 +0,0 @@
-# Calculate path=/etc/runlevels name=default cl_chroot_status==off
-
diff --git a/install/1merge/openrc/defaultrunlevel/wicd b/install/1merge/openrc/defaultrunlevel/wicd
deleted file mode 100644
index 0bc43c1..0000000
--- a/install/1merge/openrc/defaultrunlevel/wicd
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate link=/etc/init.d/numlock symbolic hr_laptop!=
diff --git a/install/1merge/udev/sysinit/.calculate_directory b/install/1merge/udev/sysinit/.calculate_directory
new file mode 100644
index 0000000..0f7257b
--- /dev/null
+++ b/install/1merge/udev/sysinit/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate path=/etc/runlevels
+
diff --git a/install/1merge/udev/sysinit/udev b/install/1merge/udev/sysinit/udev
new file mode 100644
index 0000000..568c5de
--- /dev/null
+++ b/install/1merge/udev/sysinit/udev
@@ -0,0 +1 @@
+# Calculate link=/etc/init.d/udev symbolic
