commit 8d0365508714757b1f6ad3808a6a3a2702b4b49d
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jul 29 11:59:18 2011 +0400

    Fix cpu freq increase and decrase.

diff --git install/1merge/acpid/default.sh install/1merge/acpid/default.sh
index 5dc8186..d3ee684 100644
--- install/1merge/acpid/default.sh
+++ install/1merge/acpid/default.sh
@@ -30,19 +30,54 @@ dec_backlight() {
 		echo $(($cur_backlight - 1)) >$BACKLIGHT_DIR/brightness
 }
 
-FREQLIST=(2530000 1600000)
+# set powersave cpu mode
+powersave_cpu() {
+	for cpunum in $(seq 0 $(( $(ls -d /sys/devices/system/cpu/cpu[0-9] | wc -w)-1)));do
+		cpufreq-set -c$cpunum -g powersave
+	done
+}
+
+# set ondemand cpu mode
+ondemand_cpu() {
+	for cpunum in $(seq 0 $(( $(ls -d /sys/devices/system/cpu/cpu[0-9] | wc -w)-1)));do
+		cpufreq-set -c$cpunum -g ondemand
+	done
+}
+
+# set specified freq for all process
+set_freq() {
+	for cpunum in $(seq 0 $(( $(ls -d /sys/devices/system/cpu/cpu[0-9] | wc -w)-1)));do
+		cpufreq-set -c$cpunum -f$1
+	done
+}
 # increase cpu freq
 inc_freq() {
-	cpufreq-set -c0 -f  ${FREQLIST[0]}
-	cpufreq-set -c1 -f  ${FREQLIST[0]}
-	#cpufreq-set -g performance
+	NEWCPUFREQ=
+	for i in `cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_available_frequencies`;do
+		if [[ $i == `cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq` ]];then
+			if [[ -n $NEWCPUFREQ ]];then
+				set_freq $NEWCPUFREQ
+			else
+				set_freq $i
+			fi
+			break
+		fi
+		S=$NEWCPUFREQ
+	done
 }
 
 # decrease cpu freq
 dec_freq() {
-	cpufreq-set -c0 -f  ${FREQLIST[1]}
-	cpufreq-set -c1 -f  ${FREQLIST[1]}
-	#cpufreq-set -g powersave
+	NEWCPUFREQ=
+	for i in `cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_available_frequencies`;do
+		if [[ -n $NEWCPUFREQ ]];then
+				set_freq $i
+			break
+		fi
+		if [[ $i == `cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq` ]];then
+			NEWCPUFREQ=$i
+		fi
+	done
 }
 
 log_unhandled() {
@@ -75,7 +110,7 @@ case "$group" in
 			# multicore systems, make sure you set powersave mode
 			# for each core!
 			*0)
-				dec_freq
+				powersave_cpu
 				;;
 
 			# Add code here to handle when the system is plugged in
@@ -83,7 +118,7 @@ case "$group" in
 			# multicore systems, make sure you set performance mode
 			# for each core!
 			*1)
-				inc_freq
+				ondemand_cpu
 				;;
 
 			*)	log_unhandled $* ;;
