commit cf463b3383fe2c47ebc21e21ae5214cade177f0e
Author: Alexander Tratsevskiy <at@calculate.ru>
Date:   Fri Jun 24 11:46:40 2011 +0400

    Add ro_RO.UTF-8

diff --git a/install/1merge/glibc/locale.gen b/install/1merge/glibc/locale.gen
index 15cff6b..c087558 100644
--- a/install/1merge/glibc/locale.gen
+++ b/install/1merge/glibc/locale.gen
@@ -26,6 +26,8 @@ nn_NO.UTF-8 UTF-8
 pl_PL.UTF-8 UTF-8
 #Portuguese
 pt_PT.UTF-8 UTF-8
+#Romanian
+ro_RO.UTF-8 UTF-8
 #Russian
 ru_RU.UTF-8 UTF-8
 #Spanish
