diff --git install/1merge/00-calculate-install/calculate2.env install/1merge/00-calculate-install/calculate2.env
index cead32f..6866fd0 100644
--- install/1merge/00-calculate-install/calculate2.env
+++ install/1merge/00-calculate-install/calculate2.env
@@ -1,4 +1,8 @@
-# Calculate format=samba path=/etc/calculate
+# Calculate format=samba path=/etc/calculate cl_chroot_status==off||ac_install_system==up autoupdate
+
+#?ac_install_system!=up&&exists(/etc/calculate/calculate2.env,root)!=#
+#-load(empty,/etc/calculate/calculate2.env,root)-#
+#ac_install_system#
 
 [install]
 os_install_net_settings = #-os_install_net_conf-#
diff --git install/1merge/baselayout/motd.en install/1merge/baselayout/motd.en
index a9b53cb..eef29e7 100644
--- install/1merge/baselayout/motd.en
+++ install/1merge/baselayout/motd.en
@@ -1,4 +1,4 @@
-# Calculate append=replace os_install_locale_language!=ru
+# Calculate append=replace os_install_locale_language!=ru name=motd
 Welcome to the #-os_install_linux_name-# Installation CD!
 
 If any ethernet adapters were detected at boot, they should be auto-configured
diff --git install/1merge/baselayout/motd.ru install/1merge/baselayout/motd.ru
index 2a42829..639a782 100644
--- install/1merge/baselayout/motd.ru
+++ install/1merge/baselayout/motd.ru
@@ -1,4 +1,4 @@
-# Calculate append=replace os_install_locale_language==ru
+# Calculate append=replace os_install_locale_language==ru name=motd
 Добро πожаловать в установочный диск #-os_install_linux_name-#!
 
 Если какой-либо из сетевых адаπтеров был обнаружен πри загрузке, он будет
diff --git install/1merge/networkmanager/system-connections/.calculate_directory install/1merge/networkmanager/system-connections/.calculate_directory
index b3a2ea7..fee6107 100644
--- install/1merge/networkmanager/system-connections/.calculate_directory
+++ install/1merge/networkmanager/system-connections/.calculate_directory
@@ -1 +1 @@
-# Calculate path=/etc/NetworkManager os_install_net_settings!=networkmanager&&cl_chroot_status==off&&os_install_root_type!=livecd||ac_install_system==up
+# Calculate path=/etc/NetworkManager os_install_net_settings!=networkmanager&&cl_chroot_status==off&&os_install_net_conf==networkmanager||ac_install_system==up&&os_install_net_conf==networkmanager||os_install_root_type==livecd&&os_install_net_conf==networkmanager
diff --git install/1merge/openrc/conf.d/net install/1merge/openrc/conf.d/net
index 4d39ad5..863cc52 100644
--- install/1merge/openrc/conf.d/net
+++ install/1merge/openrc/conf.d/net
@@ -1,4 +1,4 @@
-# Calculate comment=# os_install_net_settings!=openrc&&cl_chroot_status==off&&os_install_root_type!=livecd&&os_install_net_conf==openrc||ac_install_system==up&&os_install_net_conf==openrc
+# Calculate comment=# os_install_net_settings!=openrc&&cl_chroot_status==off&&os_install_net_conf==openrc||ac_install_system==up&&os_install_net_conf==openrc||os_install_root_type==livecd&&os_install_net_conf==openrc
 
 #?os_install_linux_system==server#
 modules="!plug"
@@ -6,24 +6,36 @@ modules="!plug"
 #?list(os_install_net_interfaces,0)!=&&list(os_install_net_dhcp_set,0)!=on&&list(os_install_net_ip,0)!=#
 config_#-list(os_install_net_interfaces,0)-#=( "#-list(os_install_net_ip,0)-#/#-list(os_install_net_cidr,0)-#" )
 #list#
+#?list(os_install_net_interfaces,0)!=&&list(os_install_net_dhcp_set,0)==on#
+config_#-list(os_install_net_interfaces,0)-#="dhcp"
+#list#
 #?list(os_install_net_interfaces,0)!=&&list(os_install_net_dhcp_set,0)!=on&&list(os_install_net_ip,0)!=&&list(os_install_net_route,0)!=#
 routes_#-list(os_install_net_interfaces,0)-#="#-list(os_install_net_route,0)-#"
 #list#
 #?list(os_install_net_interfaces,1)!=&&list(os_install_net_dhcp_set,1)!=on&&list(os_install_net_ip,1)!=#
 config_#-list(os_install_net_interfaces,1)-#=( "#-list(os_install_net_ip,1)-#/#-list(os_install_net_cidr,1)-#" )
 #list#
+#?list(os_install_net_interfaces,1)!=&&list(os_install_net_dhcp_set,1)==on#
+config_#-list(os_install_net_interfaces,1)-#="dhcp"
+#list#
 #?list(os_install_net_interfaces,1)!=&&list(os_install_net_dhcp_set,1)!=on&&list(os_install_net_ip,1)!=&&list(os_install_net_route,1)!=#
 routes_#-list(os_install_net_interfaces,1)-#="#-list(os_install_net_route,1)-#"
 #list#
 #?list(os_install_net_interfaces,2)!=&&list(os_install_net_dhcp_set,2)!=on&&list(os_install_net_ip,2)!=#
 config_#-list(os_install_net_interfaces,2)-#=( "#-list(os_install_net_ip,2)-#/#-list(os_install_net_cidr,2)-#" )
 #list#
+#?list(os_install_net_interfaces,2)!=&&list(os_install_net_dhcp_set,2)==on#
+config_#-list(os_install_net_interfaces,2)-#="dhcp"
+#list#
 #?list(os_install_net_interfaces,2)!=&&list(os_install_net_dhcp_set,2)!=on&&list(os_install_net_ip,2)!=&&list(os_install_net_route,2)!=#
 routes_#-list(os_install_net_interfaces,2)-#="#-list(os_install_net_route,2)-#"
 #list#
 #?list(os_install_net_interfaces,3)!=&&list(os_install_net_dhcp_set,3)!=on&&list(os_install_net_ip,3)!=#
 config_#-list(os_install_net_interfaces,3)-#=( "#-list(os_install_net_ip,3)-#/#-list(os_install_net_cidr,3)-#" )
 #list#
+#?list(os_install_net_interfaces,3)!=&&list(os_install_net_dhcp_set,3)==on#
+config_#-list(os_install_net_interfaces,3)-#="dhcp"
+#list#
 #?list(os_install_net_interfaces,3)!=&&list(os_install_net_dhcp_set,3)!=on&&list(os_install_net_ip,3)!=&&list(os_install_net_route,3)!=#
 routes_#-list(os_install_net_interfaces,3)-#="#-list(os_install_net_route,3)-#"
 #list#
