commit c49e2ebf209d2441479b8f3bcfbe82af2a7875db
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Mar 22 14:20:56 2011 +0300

    Update calckernel.

diff --git install/1merge/calckernel/.calculate_directory install/1merge/calckernel/.calculate_directory
new file mode 100644
index 0000000..a63bfbb
--- /dev/null
+++ install/1merge/calckernel/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate belong()!=&&cl_chroot_status==off&&pkg(sys-kernel/calckernel)!= path=/ name=etc
+
diff --git a/install/1merge/calckernel/genkernel.conf b/install/1merge/calckernel/genkernel.conf
new file mode 100644
index 0000000..4834954
--- /dev/null
+++ install/1merge/calckernel/genkernel.conf
@@ -0,0 +1 @@
+# Calculate format=patch autoupdate
