commit 1ef37e5965d35905336ac66be926005f9c48752e
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Apr 4 16:54:10 2011 +0400

    Add panel resizing.

diff --git desktop/desktop/20-kde-always/.kde4/share/config/plasma-desktop-appletsrc.panel-resize desktop/desktop/20-kde-always/.kde4/share/config/plasma-desktop-appletsrc.panel-resize
new file mode 100644
index 0000000..23079d0
--- /dev/null
+++ desktop/desktop/20-kde-always/.kde4/share/config/plasma-desktop-appletsrc.panel-resize
@@ -0,0 +1,19 @@
+# Calculate name=plasma-desktop-appletsrc format=plasma ini(resolution)!=#-os_x11_standart-#
+#------------------------------------------------------------------------------
+#рабочий стол
+#------------------------------------------------------------------------------
+
+[Containments][#-sum(cont,1)-#]
+geometry=0,0,#-os_x11_width-#,#-os_x11_height-#
+
+#------------------------------------------------------------------------------
+#верхняя панель
+#------------------------------------------------------------------------------
+#-sum(panel,,31)-#
+
+[Containments][#-sum(cont,cont+1)-#]
+geometry=0,0,#-os_x11_width-#,#-sum(panel,panel)-#
+
+[Containments][#-sum(cont,cont)-#][Configuration]
+maximumSize=#-os_x11_width-#,#-sum(panel,panel)-#
+minimumSize=#-os_x11_width-#,#-sum(panel,panel)-#
diff --git desktop/desktop/99-final/.calculate/ini.env desktop/desktop/99-final/.calculate/ini.env
index 4fdbb28..b382f58 100644
--- desktop/desktop/99-final/.calculate/ini.env
+++ desktop/desktop/99-final/.calculate/ini.env
@@ -6,4 +6,5 @@ domain = #-env(client.os_remote_auth)-#
 #?env(client.os_remote_auth)==||ur_domain_set==off#
 !domain = 
 #env#
+resolution = #-os_x11_standart-#
 xsession = #-cl_desktop_xsession-#
