diff --git desktop/desktop/60-chromium-11.0/.calculate/ini.env desktop/desktop/60-chromium-11.0/.calculate/ini.env
new file mode 100644
index 0000000..cf4fce5
--- /dev/null
+++ desktop/desktop/60-chromium-11.0/.calculate/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba
+
+[ver]
+chromium = 11.0
+
+[update]
+chromium = on
diff --git desktop/desktop/60-chromium-11.0/.calculate_directory desktop/desktop/60-chromium-11.0/.calculate_directory
new file mode 100644
index 0000000..ced47a5
--- /dev/null
+++ desktop/desktop/60-chromium-11.0/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(chromium)!=&&ini(update.chromium)!=off&&ini(ver.chromium)<11.0&&pkg(www-client/chromium)>=11.0
diff --git desktop/desktop/60-chromium-11.0/chromium-chromium.desktop desktop/desktop/60-chromium-11.0/chromium-chromium.desktop
new file mode 100644
index 0000000..29b8f6d
--- /dev/null
+++ desktop/desktop/60-chromium-11.0/chromium-chromium.desktop
@@ -0,0 +1,4 @@
+# Calculate format=kde link=/usr/share/applications/chromium-browser-chromium.desktop mirror path=~/.local/share/applications
+
+[Desktop Entry]
+OnlyShowIn=X-NeverShowThis;
