diff --git a/install/1merge/xorg-server/X11/xorg.conf b/install/1merge/xorg-server/X11/xorg.conf
index 5dc4940..0f1851e 100644
--- a/install/1merge/xorg-server/X11/xorg.conf
+++ b/install/1merge/xorg-server/X11/xorg.conf
@@ -89,7 +89,7 @@ Section "DRI"
 EndSection
 #os_install_x11_video_drv#
 
-#?os_install_x11_video_drv!=vesa&&os_install_x11_video_drv!=vmware&&os_install_x11_video_drv!=vboxvideo#
+#?os_install_x11_video_drv!=vmware&&os_install_x11_video_drv!=vboxvideo#
 Section "Extensions"
 #os_install_x11_video_drv#
 #?os_install_x11_video_drv==fglrx#
@@ -102,6 +102,9 @@ Section "Extensions"
 #?os_install_x11_composite!=on&&os_install_x11_video_drv!=vesa&&os_install_x11_video_drv!=vmware&&os_install_x11_video_drv!=vboxvideo#
   Option "Composite" "Off"
 #os_install_x11_composite#
-#?os_install_x11_video_drv!=vesa&&os_install_x11_video_drv!=vmware&&os_install_x11_video_drv!=vboxvideo#
+#?os_install_x11_video_drv==vesa#
+  Option "Composite" "Off"
+#os_install_x11_video_drv#
+#?os_install_x11_video_drv!=vmware&&os_install_x11_video_drv!=vboxvideo#
 EndSection
 #os_install_x11_video_drv#
