diff --git assemble/prepare/etc/python.config assemble/prepare/etc/python.config
index 6422815..f810ae2 100644
--- assemble/prepare/etc/python.config
+++ assemble/prepare/etc/python.config
@@ -1,2 +1,2 @@
 # Calculate path=/etc/env.d/python name=config
-python2.6
+python2.7
diff --git builder/iso/calcboot/isolinux/isolinux.cfg.desktop.oldkernel builder/iso/calcboot/isolinux/isolinux.cfg.desktop.oldkernel
index ead6f23..6cc2f39 100644
--- builder/iso/calcboot/isolinux/isolinux.cfg.desktop.oldkernel
+++ builder/iso/calcboot/isolinux/isolinux.cfg.desktop.oldkernel
@@ -1,4 +1,4 @@
-# Calculate name=isolinux.cfg os_builder_linux_system==desktop&&pkg(sys-kernel/calculate-sources)<2.6.32.33||os_builder_linux_system==desktop&&pkg(sys-kernel/calculate-sources)>2.6.33&&pkg(sys-kernel/calculate-sources)<2.6.37.4#
+# Calculate name=isolinux.cfg os_builder_linux_system==desktop&&pkg(sys-kernel/calculate-sources)<2.6.32.33||os_builder_linux_system==desktop&&pkg(sys-kernel/calculate-sources)>2.6.33&&pkg(sys-kernel/calculate-sources)<2.6.37.4
 DEFAULT calcmenu.c32
 INCLUDE common.cfg
 
diff --git builder/iso/calcboot/isolinux/isolinux.cfg.server.oldkernel builder/iso/calcboot/isolinux/isolinux.cfg.server.oldkernel
index 69245ee..b6ba85c 100644
--- builder/iso/calcboot/isolinux/isolinux.cfg.server.oldkernel
+++ builder/iso/calcboot/isolinux/isolinux.cfg.server.oldkernel
@@ -1,4 +1,4 @@
-# Calculate name=isolinux.cfg os_builder_linux_system==server&&pkg(sys-kernel/calculate-sources)<2.6.32.33||os_builder_linux_system==server&&pkg(sys-kernel/calculate-sources)>2.6.33&&pkg(sys-kernel/calculate-sources)<2.6.37.4#
+# Calculate name=isolinux.cfg os_builder_linux_system==server&&pkg(sys-kernel/calculate-sources)<2.6.32.33||os_builder_linux_system==server&&pkg(sys-kernel/calculate-sources)>2.6.33&&pkg(sys-kernel/calculate-sources)<2.6.37.4
 DEFAULT calcmenu.c32
 INCLUDE common.cfg
 
diff --git install/1merge/baselayout/gentoo-release install/1merge/baselayout/gentoo-release
new file mode 100644
index 0000000..739cc33
--- /dev/null
+++ install/1merge/baselayout/gentoo-release
@@ -0,0 +1,6 @@
+#?os_install_linux_subname!=#
+#-os_install_linux_name-# #-os_install_linux_ver-# #-os_install_linux_subname-#
+#os_install_linux_subname#
+#?os_install_linux_subname==#
+#-os_install_linux_name-# #-os_install_linux_ver-#
+#os_install_linux_subname#
diff --git install/1merge/gdm/custom.conf install/1merge/gdm/custom.conf
index 2b4bf50..d242939 100644
--- install/1merge/gdm/custom.conf
+++ install/1merge/gdm/custom.conf
@@ -12,7 +12,7 @@ GraphicalThemedColor=#170b0a
 #pkg#
 
 [server-Standard]
-command=/usr/bin/X -audit 0 vt7 -dpi 96
+command=/usr/bin/X -audit 0  -dpi 96
 
 [daemon]
 #?os_install_root_type!=livecd#
diff --git install/1merge/kdm/kdmrc install/1merge/kdm/kdmrc
index a6f145c..fbc04b2 100644
--- install/1merge/kdm/kdmrc
+++ install/1merge/kdm/kdmrc
@@ -73,7 +73,7 @@ AllowNullPasswd=true
 AllowShutdown=All
 NoPassEnable=false
 NoPassUsers=
-ServerArgsLocal=-nolisten tcp vt7 -dpi 96
+ServerArgsLocal=-nolisten tcp -dpi 96
 ServerCmd=/usr/bin/X -br -novtswitch
 #?os_install_x11_video_drv==fglrx#
 # System freezes after logout with GDM or KDM ATI
diff --git install/1merge/openrc/rc.conf install/1merge/openrc/rc.conf
index 1616719..71a8d1d 100644
--- install/1merge/openrc/rc.conf
+++ install/1merge/openrc/rc.conf
@@ -5,3 +5,5 @@ rc_interactive="YES"
 no_umounts="/newroot.*:/lib/splash/cache:/lib64/splash/cache"
 #use all net.* for network atapters (need for CDS)
 rc_hotplug="*"
+#nothing special type
+rc_sys=""
diff --git install/1merge/portage/make.conf install/1merge/portage/make.conf
index a539fe6..270567f 100644
--- install/1merge/portage/make.conf
+++ install/1merge/portage/make.conf
@@ -60,6 +60,8 @@ ACCEPT_LICENSE="*"
 #?os_install_linux_shortname==CLD||os_install_linux_shortname==CLDG||os_install_linux_shortname==CLDX||os_install_linux_shortname==CDS#
 #PORTAGE_BINHOST="http://77.88.19.68/calculate/#-os_install_linux_shortname-#/grp/#-os_install_arch_machine-#"
 #PORTAGE_BINHOST="ftp://ftp.linux.kiev.ua/pub/Linux/Calculate/#-os_install_linux_shortname-#/grp/#-os_install_arch_machine-#"
+#PORTAGE_BINHOST="ftp://opennix.tula.su/calculate/#-os_install_linux_shortname-#/grp/#-os_install_arch_machine-#"
+#PORTAGE_BINHOST="http://mirror.cnet.kz/calculate/#-os_install_linux_shortname-#/grp/#-os_install_arch_machine-#"
 #os_install_linux_shortname#
 
 #?os_install_locale_language!=ru&&os_install_locale_language!=bg#
diff --git install/1merge/slim/slim.conf install/1merge/slim/slim.conf
index 45ba731..7aa80e1 100644
--- install/1merge/slim/slim.conf
+++ install/1merge/slim/slim.conf
@@ -1,5 +1,5 @@
 # Calculate format=proftpd
-xserver_arguments	-nolisten tcp -br -deferglyphs 16 vt7 -dpi 96
+xserver_arguments	-nolisten tcp -br -deferglyphs 16 -dpi 96
 
 #?os_install_root_type==livecd#
 auto_login	yes
