diff --git a/install/1merge/cds-themes/tty1/1024x768.cfg b/install/1merge/cds-themes/tty1/1024x768.cfg
index 7b2a760..654f264 100644
--- a/install/1merge/cds-themes/tty1/1024x768.cfg
+++ b/install/1merge/cds-themes/tty1/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cds-themes/tty1/800x600.cfg b/install/1merge/cds-themes/tty1/800x600.cfg
index cb63e9a..728abc0 100644
--- a/install/1merge/cds-themes/tty1/800x600.cfg
+++ b/install/1merge/cds-themes/tty1/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cds-themes/tty2/1024x768.cfg b/install/1merge/cds-themes/tty2/1024x768.cfg
index db78bd4..3cb0c55 100644
--- a/install/1merge/cds-themes/tty2/1024x768.cfg
+++ b/install/1merge/cds-themes/tty2/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cds-themes/tty2/800x600.cfg b/install/1merge/cds-themes/tty2/800x600.cfg
index fdb4ca9..047462b 100644
--- a/install/1merge/cds-themes/tty2/800x600.cfg
+++ b/install/1merge/cds-themes/tty2/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cds-themes/tty3/1024x768.cfg b/install/1merge/cds-themes/tty3/1024x768.cfg
index db3a84e..185734b 100644
--- a/install/1merge/cds-themes/tty3/1024x768.cfg
+++ b/install/1merge/cds-themes/tty3/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cds-themes/tty3/800x600.cfg b/install/1merge/cds-themes/tty3/800x600.cfg
index 56ec395..1ef5b80 100644
--- a/install/1merge/cds-themes/tty3/800x600.cfg
+++ b/install/1merge/cds-themes/tty3/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cds-themes/tty4/1024x768.cfg b/install/1merge/cds-themes/tty4/1024x768.cfg
index edadc69..0f47adb 100644
--- a/install/1merge/cds-themes/tty4/1024x768.cfg
+++ b/install/1merge/cds-themes/tty4/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cds-themes/tty4/800x600.cfg b/install/1merge/cds-themes/tty4/800x600.cfg
index d49b5be..7a78c5a 100644
--- a/install/1merge/cds-themes/tty4/800x600.cfg
+++ b/install/1merge/cds-themes/tty4/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cds-themes/tty5/1024x768.cfg b/install/1merge/cds-themes/tty5/1024x768.cfg
index f0792d2..37b2b81 100644
--- a/install/1merge/cds-themes/tty5/1024x768.cfg
+++ b/install/1merge/cds-themes/tty5/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cds-themes/tty5/800x600.cfg b/install/1merge/cds-themes/tty5/800x600.cfg
index e7f4837..25411ba 100644
--- a/install/1merge/cds-themes/tty5/800x600.cfg
+++ b/install/1merge/cds-themes/tty5/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cds-themes/tty6/1024x768.cfg b/install/1merge/cds-themes/tty6/1024x768.cfg
index d248b2f..624f517 100644
--- a/install/1merge/cds-themes/tty6/1024x768.cfg
+++ b/install/1merge/cds-themes/tty6/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cds-themes/tty6/800x600.cfg b/install/1merge/cds-themes/tty6/800x600.cfg
index 4a49e55..6c211c9 100644
--- a/install/1merge/cds-themes/tty6/800x600.cfg
+++ b/install/1merge/cds-themes/tty6/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cld-themes/tty1/1024x768.cfg b/install/1merge/cld-themes/tty1/1024x768.cfg
index 785793a..f91add3 100644
--- a/install/1merge/cld-themes/tty1/1024x768.cfg
+++ b/install/1merge/cld-themes/tty1/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cld-themes/tty1/800x600.cfg b/install/1merge/cld-themes/tty1/800x600.cfg
index 36018a4..d9a17bb 100644
--- a/install/1merge/cld-themes/tty1/800x600.cfg
+++ b/install/1merge/cld-themes/tty1/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cld-themes/tty2/1024x768.cfg b/install/1merge/cld-themes/tty2/1024x768.cfg
index 2af3f4c..3cf9d67 100644
--- a/install/1merge/cld-themes/tty2/1024x768.cfg
+++ b/install/1merge/cld-themes/tty2/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cld-themes/tty2/800x600.cfg b/install/1merge/cld-themes/tty2/800x600.cfg
index d836b48..2bb14df 100644
--- a/install/1merge/cld-themes/tty2/800x600.cfg
+++ b/install/1merge/cld-themes/tty2/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cld-themes/tty3/1024x768.cfg b/install/1merge/cld-themes/tty3/1024x768.cfg
index edfd14f..0af6c2b 100644
--- a/install/1merge/cld-themes/tty3/1024x768.cfg
+++ b/install/1merge/cld-themes/tty3/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cld-themes/tty3/800x600.cfg b/install/1merge/cld-themes/tty3/800x600.cfg
index 97d4060..2111257 100644
--- a/install/1merge/cld-themes/tty3/800x600.cfg
+++ b/install/1merge/cld-themes/tty3/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cld-themes/tty4/1024x768.cfg b/install/1merge/cld-themes/tty4/1024x768.cfg
index e4112ac..913ec3b 100644
--- a/install/1merge/cld-themes/tty4/1024x768.cfg
+++ b/install/1merge/cld-themes/tty4/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cld-themes/tty4/800x600.cfg b/install/1merge/cld-themes/tty4/800x600.cfg
index 70de266..1db4103 100644
--- a/install/1merge/cld-themes/tty4/800x600.cfg
+++ b/install/1merge/cld-themes/tty4/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cld-themes/tty5/1024x768.cfg b/install/1merge/cld-themes/tty5/1024x768.cfg
index b788de2..b8c04a6 100644
--- a/install/1merge/cld-themes/tty5/1024x768.cfg
+++ b/install/1merge/cld-themes/tty5/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cld-themes/tty5/800x600.cfg b/install/1merge/cld-themes/tty5/800x600.cfg
index 5715b06..17714f3 100644
--- a/install/1merge/cld-themes/tty5/800x600.cfg
+++ b/install/1merge/cld-themes/tty5/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cld-themes/tty6/1024x768.cfg b/install/1merge/cld-themes/tty6/1024x768.cfg
index 00d9942..556d726 100644
--- a/install/1merge/cld-themes/tty6/1024x768.cfg
+++ b/install/1merge/cld-themes/tty6/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cld-themes/tty6/800x600.cfg b/install/1merge/cld-themes/tty6/800x600.cfg
index 440d49f..0aa9dd4 100644
--- a/install/1merge/cld-themes/tty6/800x600.cfg
+++ b/install/1merge/cld-themes/tty6/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldg-themes/tty1/1024x768.cfg b/install/1merge/cldg-themes/tty1/1024x768.cfg
index 00e4881..48ed0c0 100644
--- a/install/1merge/cldg-themes/tty1/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty1/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldg-themes/tty1/800x600.cfg b/install/1merge/cldg-themes/tty1/800x600.cfg
index a157aa1..993a822 100644
--- a/install/1merge/cldg-themes/tty1/800x600.cfg
+++ b/install/1merge/cldg-themes/tty1/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldg-themes/tty2/1024x768.cfg b/install/1merge/cldg-themes/tty2/1024x768.cfg
index d3f458d..1399296 100644
--- a/install/1merge/cldg-themes/tty2/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty2/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldg-themes/tty2/800x600.cfg b/install/1merge/cldg-themes/tty2/800x600.cfg
index 0d8995e..accf2a6 100644
--- a/install/1merge/cldg-themes/tty2/800x600.cfg
+++ b/install/1merge/cldg-themes/tty2/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldg-themes/tty3/1024x768.cfg b/install/1merge/cldg-themes/tty3/1024x768.cfg
index 1384ea8..40d3e98 100644
--- a/install/1merge/cldg-themes/tty3/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty3/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldg-themes/tty3/800x600.cfg b/install/1merge/cldg-themes/tty3/800x600.cfg
index 8952c54..0449b7d 100644
--- a/install/1merge/cldg-themes/tty3/800x600.cfg
+++ b/install/1merge/cldg-themes/tty3/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldg-themes/tty4/1024x768.cfg b/install/1merge/cldg-themes/tty4/1024x768.cfg
index 1d83d52..3e8abe5 100644
--- a/install/1merge/cldg-themes/tty4/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty4/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldg-themes/tty4/800x600.cfg b/install/1merge/cldg-themes/tty4/800x600.cfg
index 17f30e0..18c5a07 100644
--- a/install/1merge/cldg-themes/tty4/800x600.cfg
+++ b/install/1merge/cldg-themes/tty4/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldg-themes/tty5/1024x768.cfg b/install/1merge/cldg-themes/tty5/1024x768.cfg
index 8eb74b0..0c942ed 100644
--- a/install/1merge/cldg-themes/tty5/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty5/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldg-themes/tty5/800x600.cfg b/install/1merge/cldg-themes/tty5/800x600.cfg
index 062354c..5fdddb3 100644
--- a/install/1merge/cldg-themes/tty5/800x600.cfg
+++ b/install/1merge/cldg-themes/tty5/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldg-themes/tty6/1024x768.cfg b/install/1merge/cldg-themes/tty6/1024x768.cfg
index 697f082..b41d7eb 100644
--- a/install/1merge/cldg-themes/tty6/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty6/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldg-themes/tty6/800x600.cfg b/install/1merge/cldg-themes/tty6/800x600.cfg
index aee51d7..9fffa64 100644
--- a/install/1merge/cldg-themes/tty6/800x600.cfg
+++ b/install/1merge/cldg-themes/tty6/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldx-themes/tty1/1024x768.cfg b/install/1merge/cldx-themes/tty1/1024x768.cfg
index e0b5a2c..f7e94f3 100644
--- a/install/1merge/cldx-themes/tty1/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty1/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldx-themes/tty1/800x600.cfg b/install/1merge/cldx-themes/tty1/800x600.cfg
index 3a8f5ec..8b3e280 100644
--- a/install/1merge/cldx-themes/tty1/800x600.cfg
+++ b/install/1merge/cldx-themes/tty1/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldx-themes/tty2/1024x768.cfg b/install/1merge/cldx-themes/tty2/1024x768.cfg
index 1dbd469..9cfce2c 100644
--- a/install/1merge/cldx-themes/tty2/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty2/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldx-themes/tty2/800x600.cfg b/install/1merge/cldx-themes/tty2/800x600.cfg
index 4505708..67a81d9 100644
--- a/install/1merge/cldx-themes/tty2/800x600.cfg
+++ b/install/1merge/cldx-themes/tty2/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldx-themes/tty3/1024x768.cfg b/install/1merge/cldx-themes/tty3/1024x768.cfg
index 905e44d..0d842ba 100644
--- a/install/1merge/cldx-themes/tty3/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty3/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldx-themes/tty3/800x600.cfg b/install/1merge/cldx-themes/tty3/800x600.cfg
index 32e3c31..d6ec07e 100644
--- a/install/1merge/cldx-themes/tty3/800x600.cfg
+++ b/install/1merge/cldx-themes/tty3/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldx-themes/tty4/1024x768.cfg b/install/1merge/cldx-themes/tty4/1024x768.cfg
index 67594b5..09917ea 100644
--- a/install/1merge/cldx-themes/tty4/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty4/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldx-themes/tty4/800x600.cfg b/install/1merge/cldx-themes/tty4/800x600.cfg
index 640a795..1d1e1a4 100644
--- a/install/1merge/cldx-themes/tty4/800x600.cfg
+++ b/install/1merge/cldx-themes/tty4/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldx-themes/tty5/1024x768.cfg b/install/1merge/cldx-themes/tty5/1024x768.cfg
index dd8ec79..4ea78d6 100644
--- a/install/1merge/cldx-themes/tty5/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty5/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldx-themes/tty5/800x600.cfg b/install/1merge/cldx-themes/tty5/800x600.cfg
index cd23762..520d883 100644
--- a/install/1merge/cldx-themes/tty5/800x600.cfg
+++ b/install/1merge/cldx-themes/tty5/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cldx-themes/tty6/1024x768.cfg b/install/1merge/cldx-themes/tty6/1024x768.cfg
index a7cfb5d..bcd2ee9 100644
--- a/install/1merge/cldx-themes/tty6/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty6/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cldx-themes/tty6/800x600.cfg b/install/1merge/cldx-themes/tty6/800x600.cfg
index 808feac..9240cf0 100644
--- a/install/1merge/cldx-themes/tty6/800x600.cfg
+++ b/install/1merge/cldx-themes/tty6/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cls-themes/tty1/1024x768.cfg b/install/1merge/cls-themes/tty1/1024x768.cfg
index 1b2479b..ca656bd 100644
--- a/install/1merge/cls-themes/tty1/1024x768.cfg
+++ b/install/1merge/cls-themes/tty1/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cls-themes/tty1/800x600.cfg b/install/1merge/cls-themes/tty1/800x600.cfg
index 8b8fdd2..fe2777c 100644
--- a/install/1merge/cls-themes/tty1/800x600.cfg
+++ b/install/1merge/cls-themes/tty1/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cls-themes/tty2/1024x768.cfg b/install/1merge/cls-themes/tty2/1024x768.cfg
index 0c41cbf..7ca81d2 100644
--- a/install/1merge/cls-themes/tty2/1024x768.cfg
+++ b/install/1merge/cls-themes/tty2/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cls-themes/tty2/800x600.cfg b/install/1merge/cls-themes/tty2/800x600.cfg
index e0340df..bbea78e 100644
--- a/install/1merge/cls-themes/tty2/800x600.cfg
+++ b/install/1merge/cls-themes/tty2/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cls-themes/tty3/1024x768.cfg b/install/1merge/cls-themes/tty3/1024x768.cfg
index d900b58..74a6812 100644
--- a/install/1merge/cls-themes/tty3/1024x768.cfg
+++ b/install/1merge/cls-themes/tty3/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cls-themes/tty3/800x600.cfg b/install/1merge/cls-themes/tty3/800x600.cfg
index 9fa9cd0..61d25f2 100644
--- a/install/1merge/cls-themes/tty3/800x600.cfg
+++ b/install/1merge/cls-themes/tty3/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cls-themes/tty4/1024x768.cfg b/install/1merge/cls-themes/tty4/1024x768.cfg
index 4575404..b0ccf13 100644
--- a/install/1merge/cls-themes/tty4/1024x768.cfg
+++ b/install/1merge/cls-themes/tty4/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cls-themes/tty4/800x600.cfg b/install/1merge/cls-themes/tty4/800x600.cfg
index 87f4312..58829b9 100644
--- a/install/1merge/cls-themes/tty4/800x600.cfg
+++ b/install/1merge/cls-themes/tty4/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cls-themes/tty5/1024x768.cfg b/install/1merge/cls-themes/tty5/1024x768.cfg
index 9655309..41abda0 100644
--- a/install/1merge/cls-themes/tty5/1024x768.cfg
+++ b/install/1merge/cls-themes/tty5/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cls-themes/tty5/800x600.cfg b/install/1merge/cls-themes/tty5/800x600.cfg
index 018292e..1fa0264 100644
--- a/install/1merge/cls-themes/tty5/800x600.cfg
+++ b/install/1merge/cls-themes/tty5/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1merge/cls-themes/tty6/1024x768.cfg b/install/1merge/cls-themes/tty6/1024x768.cfg
index c22cabe..afd4d0a 100644
--- a/install/1merge/cls-themes/tty6/1024x768.cfg
+++ b/install/1merge/cls-themes/tty6/1024x768.cfg
@@ -46,9 +46,9 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
 box     19 19  19 679 #313234
diff --git a/install/1merge/cls-themes/tty6/800x600.cfg b/install/1merge/cls-themes/tty6/800x600.cfg
index 56ece74..f32ee5e 100644
--- a/install/1merge/cls-themes/tty6/800x600.cfg
+++ b/install/1merge/cls-themes/tty6/800x600.cfg
@@ -46,9 +46,9 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
 box     14 14  14 530 #313234
diff --git a/install/1system/openrc/conf.d/net b/install/1system/openrc/conf.d/net
new file mode 100644
index 0000000..9f72552
--- /dev/null
+++ b/install/1system/openrc/conf.d/net
@@ -0,0 +1,2 @@
+# Calculate comment=# os_install_linux_system==server
+modules="!plug"
