commit d64738c7f81fc5ca2a04619446bde4280e9e2435
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Jan 30 17:06:42 2012 +0400

    Switch off nepomuk indexer.

diff --git desktop/desktop/00-patch-2.2.27/.calculate_directory desktop/desktop/00-patch-2.2.27/.calculate_directory
new file mode 100644
index 0000000..cf46608
--- /dev/null
+++ desktop/desktop/00-patch-2.2.27/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(calculate-utilities)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.27
diff --git desktop/desktop/00-patch-2.2.27/ini.env desktop/desktop/00-patch-2.2.27/ini.env
new file mode 100644
index 0000000..c268287
--- /dev/null
+++ desktop/desktop/00-patch-2.2.27/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba path=~/.calculate
+
+[ver]
+patch = 2.2.27
+
+[update]
+patch = on
diff --git desktop/desktop/00-patch-2.2.27/nepomukserverrc desktop/desktop/00-patch-2.2.27/nepomukserverrc
new file mode 100644
index 0000000..828eabb
--- /dev/null
+++ desktop/desktop/00-patch-2.2.27/nepomukserverrc
@@ -0,0 +1,10 @@
+# Calculate format=kde path=~/.kde4/share/config
+
+[Basic Settings]
+Start Nepomuk=true
+
+[Service-nepomukfileindexer]
+autostart=false
+
+[Service-nepomukstrigiservice]
+autostart=false
diff --git desktop/desktop/20-kde-4.7/.kde4/share/config/nepomukserverrc desktop/desktop/20-kde-4.7/.kde4/share/config/nepomukserverrc
index bfe2b98..24ee2fd 100644
--- desktop/desktop/20-kde-4.7/.kde4/share/config/nepomukserverrc
+++ desktop/desktop/20-kde-4.7/.kde4/share/config/nepomukserverrc
@@ -6,3 +6,5 @@ Start Nepomuk=true
 [Service-nepomukstrigiservice]
 autostart=false
 
+[Service-nepomukfileindexer]
+autostart=false
