diff --git install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop
index c2f95e7..dd212d6 100644
--- install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop
+++ install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2
+# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2&&os_install_arch_machine==i686
 
 # Update info
 #-ini(desktop.writer, libreoffice)-#
diff --git install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop.amd64 install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop.amd64
new file mode 100644
index 0000000..3acfbe9
--- /dev/null
+++ install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop.amd64
@@ -0,0 +1,8 @@
+# Calculate format=kde link=/usr/lib64/libreoffice/share/xdg/writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2&&os_install_arch_machine==x86_64
+
+# Update info
+#-ini(desktop.writer, libreoffice)-#
+
+# Hide icon
+[Desktop Entry]
+NoDisplay=true
diff --git install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop
index 8ec5dcb..972c545 100644
--- install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop
+++ install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2
+# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2&&os_install_arch_machine==i686
 
 # Update info
 #-ini(desktop.calc, libreoffice)-#
diff --git install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop.amd64 install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop.amd64
new file mode 100644
index 0000000..f9fa2d7
--- /dev/null
+++ install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop.amd64
@@ -0,0 +1,8 @@
+# Calculate format=kde link=/usr/lib64/libreoffice/share/xdg/calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2&&os_install_arch_machine==x86_64
+
+# Update info
+#-ini(desktop.calc, libreoffice)-#
+
+# Hide icon
+[Desktop Entry]
+NoDisplay=true
diff --git install/1merge/libreoffice-3.4.4.2/.calculate_directory install/1merge/libreoffice-3.4.4.2/.calculate_directory
new file mode 100644
index 0000000..82eef0c
--- /dev/null
+++ install/1merge/libreoffice-3.4.4.2/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong(libreoffice)!=&&pkg(app-office/libreoffice)>3.4.4&&pkg(app-office/libreoffice)<3.5&&pkg(kde-base/kdelibs)!= path=/etc name=env.d
diff --git install/1merge/libreoffice-3.4.4.2/99libreoffice install/1merge/libreoffice-3.4.4.2/99libreoffice
new file mode 100644
index 0000000..d01b3db
--- /dev/null
+++ install/1merge/libreoffice-3.4.4.2/99libreoffice
@@ -0,0 +1 @@
+OOO_FORCE_DESKTOP=kde
