diff --git builder/squash/etc/portage/package.keywords/custom builder/squash/etc/portage/package.keywords/custom
index e69de29..bce5bd6 100644
--- builder/squash/etc/portage/package.keywords/custom
+++ builder/squash/etc/portage/package.keywords/custom
@@ -0,0 +1 @@
+# Calculate exists(/etc/portage/package.keywords/custom)==
diff --git builder/squash/etc/portage/package.mask/custom builder/squash/etc/portage/package.mask/custom
index e69de29..d00a01b 100644
--- builder/squash/etc/portage/package.mask/custom
+++ builder/squash/etc/portage/package.mask/custom
@@ -0,0 +1 @@
+# Calculate exists(/etc/portage/package.mask/custom)==
diff --git builder/squash/etc/portage/package.unmask/custom builder/squash/etc/portage/package.unmask/custom
index e69de29..32c589d 100644
--- builder/squash/etc/portage/package.unmask/custom
+++ builder/squash/etc/portage/package.unmask/custom
@@ -0,0 +1 @@
+# Calculate exists(/etc/portage/package.unmask/custom)==
diff --git builder/squash/etc/portage/package.use/custom builder/squash/etc/portage/package.use/custom
index e69de29..31a0700 100644
--- builder/squash/etc/portage/package.use/custom
+++ builder/squash/etc/portage/package.use/custom
@@ -0,0 +1 @@
+# Calculate exists(/etc/portage/package.use/custom)==
diff --git builder/squash/etc/portage/profile/package.provided builder/squash/etc/portage/profile/package.provided
index 2bcb5ea..04758ba 100644
--- builder/squash/etc/portage/profile/package.provided
+++ builder/squash/etc/portage/profile/package.provided
@@ -1,4 +1,4 @@
+# Calculate exists(/etc/portage/profile/package.provided)==
 # Packages that you do not want to update (see: man portage)
 # Example:
 # net-misc/wicd-1.7.0
-
diff --git desktop/desktop/40-xfce-4.6/.local/share/applications/gparted.desktop desktop/desktop/40-xfce-4.6/.local/share/applications/gparted.desktop
index c8a8ff6..12db8d9 100644
--- desktop/desktop/40-xfce-4.6/.local/share/applications/gparted.desktop
+++ desktop/desktop/40-xfce-4.6/.local/share/applications/gparted.desktop
@@ -4,6 +4,6 @@
 Exec=sudo /usr/sbin/gparted %f
 #os_root_type#
 #?os_root_type!=livecd#
-Exec=ktsuss /usr/sbin/gparted %f
+Exec=gksu /usr/sbin/gparted %f
 #os_root_type#
 !OnlyShowIn=
diff --git desktop/desktop/40-xfce-4.8/.local/share/applications/gparted.desktop desktop/desktop/40-xfce-4.8/.local/share/applications/gparted.desktop
index c8a8ff6..12db8d9 100644
--- desktop/desktop/40-xfce-4.8/.local/share/applications/gparted.desktop
+++ desktop/desktop/40-xfce-4.8/.local/share/applications/gparted.desktop
@@ -4,6 +4,6 @@
 Exec=sudo /usr/sbin/gparted %f
 #os_root_type#
 #?os_root_type!=livecd#
-Exec=ktsuss /usr/sbin/gparted %f
+Exec=gksu /usr/sbin/gparted %f
 #os_root_type#
 !OnlyShowIn=
