diff --git a/install/1merge/cds-themes/tty1/1024x768.cfg b/install/1merge/cds-themes/tty1/1024x768.cfg
index 7b2a760..3895bc6 100644
--- a/install/1merge/cds-themes/tty1/1024x768.cfg
+++ b/install/1merge/cds-themes/tty1/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty1/800x600.cfg b/install/1merge/cds-themes/tty1/800x600.cfg
index cb63e9a..28e7a72 100644
--- a/install/1merge/cds-themes/tty1/800x600.cfg
+++ b/install/1merge/cds-themes/tty1/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty2/1024x768.cfg b/install/1merge/cds-themes/tty2/1024x768.cfg
index db78bd4..1dd57cf 100644
--- a/install/1merge/cds-themes/tty2/1024x768.cfg
+++ b/install/1merge/cds-themes/tty2/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty2/800x600.cfg b/install/1merge/cds-themes/tty2/800x600.cfg
index fdb4ca9..23eee54 100644
--- a/install/1merge/cds-themes/tty2/800x600.cfg
+++ b/install/1merge/cds-themes/tty2/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty3/1024x768.cfg b/install/1merge/cds-themes/tty3/1024x768.cfg
index db3a84e..2eefd26 100644
--- a/install/1merge/cds-themes/tty3/1024x768.cfg
+++ b/install/1merge/cds-themes/tty3/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty3/800x600.cfg b/install/1merge/cds-themes/tty3/800x600.cfg
index 56ec395..8bc61cd 100644
--- a/install/1merge/cds-themes/tty3/800x600.cfg
+++ b/install/1merge/cds-themes/tty3/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty4/1024x768.cfg b/install/1merge/cds-themes/tty4/1024x768.cfg
index edadc69..03bd5e8 100644
--- a/install/1merge/cds-themes/tty4/1024x768.cfg
+++ b/install/1merge/cds-themes/tty4/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty4/800x600.cfg b/install/1merge/cds-themes/tty4/800x600.cfg
index d49b5be..035200f 100644
--- a/install/1merge/cds-themes/tty4/800x600.cfg
+++ b/install/1merge/cds-themes/tty4/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty5/1024x768.cfg b/install/1merge/cds-themes/tty5/1024x768.cfg
index f0792d2..2938667 100644
--- a/install/1merge/cds-themes/tty5/1024x768.cfg
+++ b/install/1merge/cds-themes/tty5/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty5/800x600.cfg b/install/1merge/cds-themes/tty5/800x600.cfg
index e7f4837..ad4fc09 100644
--- a/install/1merge/cds-themes/tty5/800x600.cfg
+++ b/install/1merge/cds-themes/tty5/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty6/1024x768.cfg b/install/1merge/cds-themes/tty6/1024x768.cfg
index d248b2f..27366e6 100644
--- a/install/1merge/cds-themes/tty6/1024x768.cfg
+++ b/install/1merge/cds-themes/tty6/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cds-themes/tty6/800x600.cfg b/install/1merge/cds-themes/tty6/800x600.cfg
index 4a49e55..78dd248 100644
--- a/install/1merge/cds-themes/tty6/800x600.cfg
+++ b/install/1merge/cds-themes/tty6/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cds-themes)<10.9#
+#?pkg(media-gfx/cds-themes)<10.9||pkg(media-gfx/cds-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty1/1024x768.cfg b/install/1merge/cld-themes/tty1/1024x768.cfg
index 785793a..65b8bd0 100644
--- a/install/1merge/cld-themes/tty1/1024x768.cfg
+++ b/install/1merge/cld-themes/tty1/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty1/800x600.cfg b/install/1merge/cld-themes/tty1/800x600.cfg
index 36018a4..89fa2bd 100644
--- a/install/1merge/cld-themes/tty1/800x600.cfg
+++ b/install/1merge/cld-themes/tty1/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty2/1024x768.cfg b/install/1merge/cld-themes/tty2/1024x768.cfg
index 2af3f4c..b0a7e56 100644
--- a/install/1merge/cld-themes/tty2/1024x768.cfg
+++ b/install/1merge/cld-themes/tty2/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty2/800x600.cfg b/install/1merge/cld-themes/tty2/800x600.cfg
index d836b48..98f4369 100644
--- a/install/1merge/cld-themes/tty2/800x600.cfg
+++ b/install/1merge/cld-themes/tty2/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty3/1024x768.cfg b/install/1merge/cld-themes/tty3/1024x768.cfg
index edfd14f..78dd7b0 100644
--- a/install/1merge/cld-themes/tty3/1024x768.cfg
+++ b/install/1merge/cld-themes/tty3/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty3/800x600.cfg b/install/1merge/cld-themes/tty3/800x600.cfg
index 97d4060..865212b 100644
--- a/install/1merge/cld-themes/tty3/800x600.cfg
+++ b/install/1merge/cld-themes/tty3/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty4/1024x768.cfg b/install/1merge/cld-themes/tty4/1024x768.cfg
index e4112ac..6765443 100644
--- a/install/1merge/cld-themes/tty4/1024x768.cfg
+++ b/install/1merge/cld-themes/tty4/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty4/800x600.cfg b/install/1merge/cld-themes/tty4/800x600.cfg
index 70de266..c89c245 100644
--- a/install/1merge/cld-themes/tty4/800x600.cfg
+++ b/install/1merge/cld-themes/tty4/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty5/1024x768.cfg b/install/1merge/cld-themes/tty5/1024x768.cfg
index b788de2..a88daf0 100644
--- a/install/1merge/cld-themes/tty5/1024x768.cfg
+++ b/install/1merge/cld-themes/tty5/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty5/800x600.cfg b/install/1merge/cld-themes/tty5/800x600.cfg
index 5715b06..bcd9135 100644
--- a/install/1merge/cld-themes/tty5/800x600.cfg
+++ b/install/1merge/cld-themes/tty5/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty6/1024x768.cfg b/install/1merge/cld-themes/tty6/1024x768.cfg
index 00d9942..f506674 100644
--- a/install/1merge/cld-themes/tty6/1024x768.cfg
+++ b/install/1merge/cld-themes/tty6/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cld-themes/tty6/800x600.cfg b/install/1merge/cld-themes/tty6/800x600.cfg
index 440d49f..84538b9 100644
--- a/install/1merge/cld-themes/tty6/800x600.cfg
+++ b/install/1merge/cld-themes/tty6/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cld-themes)<10.9#
+#?pkg(media-gfx/cld-themes)<10.9||pkg(media-gfx/cld-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty1/1024x768.cfg b/install/1merge/cldg-themes/tty1/1024x768.cfg
index 00e4881..650dc72 100644
--- a/install/1merge/cldg-themes/tty1/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty1/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty1/800x600.cfg b/install/1merge/cldg-themes/tty1/800x600.cfg
index a157aa1..0124b11 100644
--- a/install/1merge/cldg-themes/tty1/800x600.cfg
+++ b/install/1merge/cldg-themes/tty1/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty2/1024x768.cfg b/install/1merge/cldg-themes/tty2/1024x768.cfg
index d3f458d..30d47a2 100644
--- a/install/1merge/cldg-themes/tty2/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty2/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty2/800x600.cfg b/install/1merge/cldg-themes/tty2/800x600.cfg
index 0d8995e..b3fed2c 100644
--- a/install/1merge/cldg-themes/tty2/800x600.cfg
+++ b/install/1merge/cldg-themes/tty2/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty3/1024x768.cfg b/install/1merge/cldg-themes/tty3/1024x768.cfg
index 1384ea8..1820cd9 100644
--- a/install/1merge/cldg-themes/tty3/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty3/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty3/800x600.cfg b/install/1merge/cldg-themes/tty3/800x600.cfg
index 8952c54..f93ad0f 100644
--- a/install/1merge/cldg-themes/tty3/800x600.cfg
+++ b/install/1merge/cldg-themes/tty3/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty4/1024x768.cfg b/install/1merge/cldg-themes/tty4/1024x768.cfg
index 1d83d52..f8e539b 100644
--- a/install/1merge/cldg-themes/tty4/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty4/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty4/800x600.cfg b/install/1merge/cldg-themes/tty4/800x600.cfg
index 17f30e0..6796082 100644
--- a/install/1merge/cldg-themes/tty4/800x600.cfg
+++ b/install/1merge/cldg-themes/tty4/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty5/1024x768.cfg b/install/1merge/cldg-themes/tty5/1024x768.cfg
index 8eb74b0..b2c514d 100644
--- a/install/1merge/cldg-themes/tty5/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty5/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty5/800x600.cfg b/install/1merge/cldg-themes/tty5/800x600.cfg
index 062354c..d3c0b4f 100644
--- a/install/1merge/cldg-themes/tty5/800x600.cfg
+++ b/install/1merge/cldg-themes/tty5/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty6/1024x768.cfg b/install/1merge/cldg-themes/tty6/1024x768.cfg
index 697f082..b8db7d1 100644
--- a/install/1merge/cldg-themes/tty6/1024x768.cfg
+++ b/install/1merge/cldg-themes/tty6/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldg-themes/tty6/800x600.cfg b/install/1merge/cldg-themes/tty6/800x600.cfg
index aee51d7..eb4f80c 100644
--- a/install/1merge/cldg-themes/tty6/800x600.cfg
+++ b/install/1merge/cldg-themes/tty6/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldg-themes)<10.9#
+#?pkg(media-gfx/cldg-themes)<10.9||pkg(media-gfx/cldg-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty1/1024x768.cfg b/install/1merge/cldx-themes/tty1/1024x768.cfg
index e0b5a2c..4085913 100644
--- a/install/1merge/cldx-themes/tty1/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty1/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty1/800x600.cfg b/install/1merge/cldx-themes/tty1/800x600.cfg
index 3a8f5ec..5dd4ca0 100644
--- a/install/1merge/cldx-themes/tty1/800x600.cfg
+++ b/install/1merge/cldx-themes/tty1/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty2/1024x768.cfg b/install/1merge/cldx-themes/tty2/1024x768.cfg
index 1dbd469..94e7cae 100644
--- a/install/1merge/cldx-themes/tty2/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty2/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty2/800x600.cfg b/install/1merge/cldx-themes/tty2/800x600.cfg
index 4505708..76d0bed 100644
--- a/install/1merge/cldx-themes/tty2/800x600.cfg
+++ b/install/1merge/cldx-themes/tty2/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty3/1024x768.cfg b/install/1merge/cldx-themes/tty3/1024x768.cfg
index 905e44d..a922702 100644
--- a/install/1merge/cldx-themes/tty3/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty3/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty3/800x600.cfg b/install/1merge/cldx-themes/tty3/800x600.cfg
index 32e3c31..d24e126 100644
--- a/install/1merge/cldx-themes/tty3/800x600.cfg
+++ b/install/1merge/cldx-themes/tty3/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty4/1024x768.cfg b/install/1merge/cldx-themes/tty4/1024x768.cfg
index 67594b5..bcd814a 100644
--- a/install/1merge/cldx-themes/tty4/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty4/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty4/800x600.cfg b/install/1merge/cldx-themes/tty4/800x600.cfg
index 640a795..06f33fd 100644
--- a/install/1merge/cldx-themes/tty4/800x600.cfg
+++ b/install/1merge/cldx-themes/tty4/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty5/1024x768.cfg b/install/1merge/cldx-themes/tty5/1024x768.cfg
index dd8ec79..852bf25 100644
--- a/install/1merge/cldx-themes/tty5/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty5/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty5/800x600.cfg b/install/1merge/cldx-themes/tty5/800x600.cfg
index cd23762..7f14bbf 100644
--- a/install/1merge/cldx-themes/tty5/800x600.cfg
+++ b/install/1merge/cldx-themes/tty5/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty6/1024x768.cfg b/install/1merge/cldx-themes/tty6/1024x768.cfg
index a7cfb5d..b22070f 100644
--- a/install/1merge/cldx-themes/tty6/1024x768.cfg
+++ b/install/1merge/cldx-themes/tty6/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cldx-themes/tty6/800x600.cfg b/install/1merge/cldx-themes/tty6/800x600.cfg
index 808feac..2e2642d 100644
--- a/install/1merge/cldx-themes/tty6/800x600.cfg
+++ b/install/1merge/cldx-themes/tty6/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cldx-themes)<10.9#
+#?pkg(media-gfx/cldx-themes)<10.9||pkg(media-gfx/cldx-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty1/1024x768.cfg b/install/1merge/cls-themes/tty1/1024x768.cfg
index 1b2479b..94da396 100644
--- a/install/1merge/cls-themes/tty1/1024x768.cfg
+++ b/install/1merge/cls-themes/tty1/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty1/800x600.cfg b/install/1merge/cls-themes/tty1/800x600.cfg
index 8b8fdd2..464df4b 100644
--- a/install/1merge/cls-themes/tty1/800x600.cfg
+++ b/install/1merge/cls-themes/tty1/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty2/1024x768.cfg b/install/1merge/cls-themes/tty2/1024x768.cfg
index 0c41cbf..a49f6d6 100644
--- a/install/1merge/cls-themes/tty2/1024x768.cfg
+++ b/install/1merge/cls-themes/tty2/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty2/800x600.cfg b/install/1merge/cls-themes/tty2/800x600.cfg
index e0340df..84df01f 100644
--- a/install/1merge/cls-themes/tty2/800x600.cfg
+++ b/install/1merge/cls-themes/tty2/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty3/1024x768.cfg b/install/1merge/cls-themes/tty3/1024x768.cfg
index d900b58..2ddef20 100644
--- a/install/1merge/cls-themes/tty3/1024x768.cfg
+++ b/install/1merge/cls-themes/tty3/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty3/800x600.cfg b/install/1merge/cls-themes/tty3/800x600.cfg
index 9fa9cd0..e7173bf 100644
--- a/install/1merge/cls-themes/tty3/800x600.cfg
+++ b/install/1merge/cls-themes/tty3/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty4/1024x768.cfg b/install/1merge/cls-themes/tty4/1024x768.cfg
index 4575404..7332445 100644
--- a/install/1merge/cls-themes/tty4/1024x768.cfg
+++ b/install/1merge/cls-themes/tty4/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty4/800x600.cfg b/install/1merge/cls-themes/tty4/800x600.cfg
index 87f4312..48e6607 100644
--- a/install/1merge/cls-themes/tty4/800x600.cfg
+++ b/install/1merge/cls-themes/tty4/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty5/1024x768.cfg b/install/1merge/cls-themes/tty5/1024x768.cfg
index 9655309..b5001f6 100644
--- a/install/1merge/cls-themes/tty5/1024x768.cfg
+++ b/install/1merge/cls-themes/tty5/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty5/800x600.cfg b/install/1merge/cls-themes/tty5/800x600.cfg
index 018292e..d903d2c 100644
--- a/install/1merge/cls-themes/tty5/800x600.cfg
+++ b/install/1merge/cls-themes/tty5/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty6/1024x768.cfg b/install/1merge/cls-themes/tty6/1024x768.cfg
index c22cabe..322c4ff 100644
--- a/install/1merge/cls-themes/tty6/1024x768.cfg
+++ b/install/1merge/cls-themes/tty6/1024x768.cfg
@@ -46,13 +46,13 @@ box silent		360 414 664 414 #eef4ff
 box silent		360 408 360 414 #313234
 box silent		664 408 664 414 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 20 20 1004 678 #1B2D3A
+#box noover 20 20 1004 678 #12233A
 
 #rectangle around text window in verbose mode
-box     19 19  19 679 #313234
-box     19 19 1005 19 #313234
+box     19 19  19 679 #8C9096
+box     19 19 1005 19 #8C9096
 box     1005 19 1005 679 #eef4ff
 box     19 679 1005 679 #eef4ff
 #pkg#
diff --git a/install/1merge/cls-themes/tty6/800x600.cfg b/install/1merge/cls-themes/tty6/800x600.cfg
index 56ece74..6dc1f8d 100644
--- a/install/1merge/cls-themes/tty6/800x600.cfg
+++ b/install/1merge/cls-themes/tty6/800x600.cfg
@@ -46,13 +46,13 @@ box silent		281 323 518 323 #eef4ff
 box silent		281 318 281 323 #313234
 box silent		518 318 518 323 #eef4ff
 
-#?pkg(media-gfx/cls-themes)<10.9#
+#?pkg(media-gfx/cls-themes)<10.9||pkg(media-gfx/cls-themes)>=11.0##
 #background color for text window in verbose mode
-box noover 15 15 784 529 #1B2D3A
+#box noover 15 15 784 529 #12233A
 
 #rectangle around text window in verbose mode
-box     14 14  14 530 #313234
-box     14 14 785 14 #313234
+box     14 14  14 530 #8C9096
+box     14 14 785 14 #8C9096
 box     785 14 785 530 #eef4ff
 box     14 530 785 530 #eef4ff
 #pkg#
diff --git a/install/1system/openrc/conf.d/net b/install/1system/openrc/conf.d/net
new file mode 100644
index 0000000..9f72552
--- /dev/null
+++ b/install/1system/openrc/conf.d/net
@@ -0,0 +1,2 @@
+# Calculate comment=# os_install_linux_system==server
+modules="!plug"
