diff --git desktop/00-desktop-init/empty_dir/0ini.env desktop/00-desktop-init/empty_dir/0ini.env
index fd0937a..6560917 100644
--- desktop/00-desktop-init/empty_dir/0ini.env
+++ desktop/00-desktop-init/empty_dir/0ini.env
@@ -4,7 +4,7 @@
 update = on
 
 [ver]
-patch = 2.2.13
+patch = 2.2.24
 
 [update]
 patch = on
diff --git desktop/desktop/00-patch-2.2.10/.calculate_directory desktop/desktop/00-patch-2.2.10/.calculate_directory
index 9ea3814..7b5186f 100644
--- desktop/desktop/00-patch-2.2.10/.calculate_directory
+++ desktop/desktop/00-patch-2.2.10/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.10
+# Calculate append=skip belong(calculate-utilities)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.10
diff --git desktop/desktop/00-patch-2.2.12/.calculate_directory desktop/desktop/00-patch-2.2.12/.calculate_directory
index 3f135e2..3bad719 100644
--- desktop/desktop/00-patch-2.2.12/.calculate_directory
+++ desktop/desktop/00-patch-2.2.12/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.12
+# Calculate append=skip belong(calculate-utilities)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.12
diff --git desktop/desktop/00-patch-2.2.13/.calculate_directory desktop/desktop/00-patch-2.2.13/.calculate_directory
index 81943f4..3932c91 100644
--- desktop/desktop/00-patch-2.2.13/.calculate_directory
+++ desktop/desktop/00-patch-2.2.13/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.13
+# Calculate append=skip belong(calculate-utilities)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.13
diff --git desktop/desktop/00-patch-2.2.22/.calculate_directory desktop/desktop/00-patch-2.2.22/.calculate_directory
index 9259f63..0ed217d 100644
--- desktop/desktop/00-patch-2.2.22/.calculate_directory
+++ desktop/desktop/00-patch-2.2.22/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.22
+# Calculate append=skip belong(calculate-utilities)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.22
diff --git desktop/desktop/00-patch-2.2.23/.calculate_directory desktop/desktop/00-patch-2.2.23/.calculate_directory
index dd6d923..3c18666 100644
--- desktop/desktop/00-patch-2.2.23/.calculate_directory
+++ desktop/desktop/00-patch-2.2.23/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.23
+# Calculate append=skip belong(calculate-utilities)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.23
diff --git desktop/desktop/00-patch-2.2.24/.calculate_directory desktop/desktop/00-patch-2.2.24/.calculate_directory
new file mode 100644
index 0000000..4d30994
--- /dev/null
+++ desktop/desktop/00-patch-2.2.24/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(calculate-utilities)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.24
diff --git desktop/desktop/00-patch-2.2.24/adblock-fix.sh desktop/desktop/00-patch-2.2.24/adblock-fix.sh
new file mode 100644
index 0000000..acf4f7f
--- /dev/null
+++ desktop/desktop/00-patch-2.2.24/adblock-fix.sh
@@ -0,0 +1,7 @@
+# Calculate path=/tmp pkg(www-apps/chromium-adblock)!= exec=/bin/bash name=adblock-fix-#-ur_login-#.sh
+
+if [[ -d #-ur_home_path-#/.config/chromium/Default/Extensions/gighmmpiobklfepjocnamgkkbiglidom ]]
+then
+    chown #-ur_login-#:#-ur_group-# -R #-ur_home_path-#/.config/chromium/Default/Extensions/gighmmpiobklfepjocnamgkkbiglidom &>/dev/null
+fi
+exit 0
diff --git desktop/desktop/00-patch-2.2.24/ini.env desktop/desktop/00-patch-2.2.24/ini.env
new file mode 100644
index 0000000..8d9b683
--- /dev/null
+++ desktop/desktop/00-patch-2.2.24/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba path=~/.calculate
+
+[ver]
+patch = 2.2.24
+
+[update]
+patch = on
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/adblock-install.sh desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/adblock-install.sh
index 1e6fad0..04633df 100644
--- desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/adblock-install.sh
+++ desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/adblock-install.sh
@@ -1,4 +1,5 @@
 # Calculate path=/tmp pkg(www-apps/chromium-adblock)!= exec=/bin/bash name=adblock-install-#-ur_login-#.sh
 
 cp -r /usr/share/chromium-adblock/gighmmpiobklfepjocnamgkkbiglidom #-ur_home_path-#/.config/chromium/Default/Extensions/
+chown #-ur_login-#:#-ur_group-# -R #-ur_home_path-#/.config/chromium/Default/Extensions/gighmmpiobklfepjocnamgkkbiglidom &>/dev/null
 exit 0
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/First Run desktop/desktop/60-chromium-6.0/.config/chromium/First Run
index 8b3bbff..e69de29 100644
--- desktop/desktop/60-chromium-6.0/.config/chromium/First Run	
+++ desktop/desktop/60-chromium-6.0/.config/chromium/First Run	
@@ -1,2 +0,0 @@
-# Calculate exists(~/.mozilla)==
-# will not migrate if Firefox had not run
\ No newline at end of file
