commit 0b04de81481615520fe913005d8a79397c4427c3
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jul 22 10:17:36 2011 +0400

    Add scan delay for live image boot.

diff --git a/builder/iso/calcboot/isolinux/isolinux.cfg.desktop b/builder/iso/calcboot/isolinux/isolinux.cfg.desktop
index 433c1f6..0681bc1 100644
--- a/builder/iso/calcboot/isolinux/isolinux.cfg.desktop
+++ b/builder/iso/calcboot/isolinux/isolinux.cfg.desktop
@@ -22,7 +22,7 @@ LABEL cld-docache
 	MENU LABEL #-os_builder_linux_name-# #-os_builder_linux_ver-# Live#-cl_builder_cdname-# (RAM)
 #os_builder_linux_subname#
 	KERNEL /boot/vmlinuz
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume doscsi docache
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume doscsi docache scandelay=2
 
 LABEL cld-nox
 #?os_builder_linux_subname!=#
@@ -32,7 +32,7 @@ LABEL cld-nox
 	MENU LABEL #-os_builder_linux_name-# #-os_builder_linux_ver-# Live#-cl_builder_cdname-# (No-X)
 #os_builder_linux_subname#
 	KERNEL /boot/vmlinuz
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume doscsi nox
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume doscsi nox scandelay=2
 
 LABEL cld-builder
 #?os_builder_linux_subname!=#
@@ -42,7 +42,7 @@ LABEL cld-builder
 	MENU LABEL #-os_builder_linux_name-# #-os_builder_linux_ver-# Live#-cl_builder_cdname-# Builder
 #os_builder_linux_subname#
 	KERNEL /boot/vmlinuz
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume doscsi scratch
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume doscsi scratch scandelay=2
 
 LABEL memtest
 	MENU LABEL Memtest86+
diff --git a/builder/iso/calcboot/isolinux/isolinux.cfg.server b/builder/iso/calcboot/isolinux/isolinux.cfg.server
index 0aff1e9..ca72883 100644
--- a/builder/iso/calcboot/isolinux/isolinux.cfg.server
+++ b/builder/iso/calcboot/isolinux/isolinux.cfg.server
@@ -8,37 +8,37 @@ LABEL cds
 	MENU LABEL #-os_builder_linux_name-# #-os_builder_linux_ver-# Live#-cl_builder_cdname-#
 	KERNEL /boot/vmlinuz
 #?os_builder_linux_shortname!=CSS#
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume elevator=deadline
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume elevator=deadline scandelay=2
 #os_builder_linux_shortname#
 #?os_builder_linux_shortname==CSS#
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot console=tty1 udev quiet noresume elevator=deadline splash=off
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot console=tty1 udev quiet noresume elevator=deadline splash=off scandelay=2
 #os_builder_linux_shortname#
 
 LABEL cds-docache
 	MENU LABEL #-os_builder_linux_name-# #-os_builder_linux_ver-# Live#-cl_builder_cdname-# (RAM)
 	KERNEL /boot/vmlinuz
 #?os_builder_linux_shortname!=CSS#
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume elevator=deadline docache
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume elevator=deadline docache scandelay=2
 #os_builder_linux_shortname#
 #?os_builder_linux_shortname==CSS#
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot console=tty1 udev quiet noresume elevator=deadline splash=off docache
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot console=tty1 udev quiet noresume elevator=deadline splash=off docache scandelay=2
 #os_builder_linux_shortname#
 
 #?os_builder_linux_shortname!=CSS#
 LABEL cds-nofb
 	MENU LABEL #-os_builder_linux_name-# #-os_builder_linux_ver-# Live#-cl_builder_cdname-# (No-FB)
 	KERNEL /boot/vmlinuz
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot console=tty1 udev quiet noresume elevator=deadline splash=off
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot console=tty1 udev quiet noresume elevator=deadline splash=off scandelay=2
 #os_builder_linux_shortname#
 
 LABEL cds-builder
 	MENU LABEL #-os_builder_linux_name-# #-os_builder_linux_ver-# Live#-cl_builder_cdname-# Builder
 	KERNEL /boot/vmlinuz
 #?os_builder_linux_shortname!=CSS#
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume elevator=deadline scratch
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot video=uvesafb:ywrap,1024x768-32@60,mtrr:3,splash=silent,theme:tty1 console=tty1 udev quiet noresume elevator=deadline scratch scandelay=2
 #os_builder_linux_shortname#
 #?os_builder_linux_shortname==CSS#
-	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot console=tty1 udev quiet noresume elevator=deadline splash=off scratch
+	APPEND root=/dev/ram0 initrd=/boot/initrd init=/linuxrc looptype=squashfs unionfs doload=squashfs,isofs,unionfs loop=/#-cl_builder_current_squash-# nodevfs cdroot console=tty1 udev quiet noresume elevator=deadline splash=off scratch scandelay=2
 #os_builder_linux_shortname#
 
 #?os_builder_linux_shortname!=CSS#
