diff --git builder/iso/calcboot/isolinux/keymap.cfg builder/iso/calcboot/isolinux/keymap.cfg
index d4abb5e..9da49d6 100644
--- builder/iso/calcboot/isolinux/keymap.cfg
+++ builder/iso/calcboot/isolinux/keymap.cfg
@@ -80,6 +80,11 @@ LABEL keymap:pt_PT
 	KERNEL calcmenu.c32
 	APPEND isolinux.cfg calculate=,keymap:pt_PT
 
+LABEL keymap:ro_RO
+	MENU LABEL Romanian
+	KERNEL calcmenu.c32
+	APPEND isolinux.cfg calculate=,keymap:ro_RO
+
 LABEL keymap:ru_RU
 	MENU LABEL Russian
 	KERNEL calcmenu.c32
diff --git builder/iso/calcboot/isolinux/lang.cfg builder/iso/calcboot/isolinux/lang.cfg
index aefb539..c008495 100644
--- builder/iso/calcboot/isolinux/lang.cfg
+++ builder/iso/calcboot/isolinux/lang.cfg
@@ -50,6 +50,11 @@ LABEL lang:ru_RU
         KERNEL calcmenu.c32
         APPEND isolinux.cfg calculate=lang:ru_RU,keymap:ru_RU,timezone:Europe/Moscow
 
+LABEL lang:ro_RO
+        MENU LABEL Romanian
+        KERNEL calcmenu.c32
+        APPEND isolinux.cfg calculate=lang:ro_RO,keymap:ro_RO,timezone:Europe/Istanbul
+
 LABEL lang:es_ES
         MENU LABEL Espanol
         KERNEL calcmenu.c32
diff --git desktop/00-desktop-init/empty_dir/2ini_res.env desktop/00-desktop-init/empty_dir/2ini_res.env
index ba4c0a6..13bac65 100644
--- desktop/00-desktop-init/empty_dir/2ini_res.env
+++ desktop/00-desktop-init/empty_dir/2ini_res.env
@@ -118,6 +118,19 @@ videos = #-ini(resource.path)-#/Видео
 desktop = /home/#-ur_login-#/Рабочий стол
 #os_locale_lang#
 
+#?os_locale_lang==ro_RO#
+documents = #-ini(resource.path)-#/Documente
+downloads = #-ini(resource.path)-#/Descărcări
+music = #-ini(resource.path)-#/Muzică
+pictures = #-ini(resource.path)-#/Poze
+public = #-ini(resource.path)-#/Public
+templates = #-ini(resource.path)-#/Templates
+videos = #-ini(resource.path)-#/Video
+#os_locale_lang#
+#?os_locale_lang==ro_RO&&ur_domain_set==off#
+desktop = /home/#-ur_login-#/Desktop
+#os_locale_lang#
+
 #?os_locale_lang==uk_UA#
 documents = #-ini(resource.path)-#/Документи
 downloads = #-ini(resource.path)-#/Завантаження
diff --git desktop/desktop/00-patch-2.2.0.16/2ini_res.env desktop/desktop/00-patch-2.2.0.16/2ini_res.env
index ba4c0a6..ec7ae86 100644
--- desktop/desktop/00-patch-2.2.0.16/2ini_res.env
+++ desktop/desktop/00-patch-2.2.0.16/2ini_res.env
@@ -105,6 +105,19 @@ videos = #-ini(resource.path)-#/Vídeos
 desktop = /home/#-ur_login-#/Desktop
 #os_locale_lang#
 
+#?os_locale_lang==ro_RO#
+documents = #-ini(resource.path)-#/Documente
+downloads = #-ini(resource.path)-#/Descărcări
+music = #-ini(resource.path)-#/Muzică
+pictures = #-ini(resource.path)-#/Poze
+public = #-ini(resource.path)-#/Public
+templates = #-ini(resource.path)-#/Templates
+videos = #-ini(resource.path)-#/Video
+#os_locale_lang#
+#?os_locale_lang==ro_RO&&ur_domain_set==off#
+desktop = /home/#-ur_login-#/Desktop
+#os_locale_lang#
+
 #?os_locale_lang==ru_RU#
 documents = #-ini(resource.path)-#/Документы
 downloads = #-ini(resource.path)-#/Загрузки
diff --git desktop/desktop/30-gnome-2.26/.gconf/desktop/gnome/peripherals/keyboard/kbd/%gconf.xml.ro desktop/desktop/30-gnome-2.26/.gconf/desktop/gnome/peripherals/keyboard/kbd/%gconf.xml.ro
new file mode 100644
index 0000000..a40c1af
--- /dev/null
+++ desktop/desktop/30-gnome-2.26/.gconf/desktop/gnome/peripherals/keyboard/kbd/%gconf.xml.ro
@@ -0,0 +1,12 @@
+# Calculate name=%gconf.xml format=xml_gconf append=join os_locale_lang==ro_RO
+<?xml version="1.0"?>
+<gconf>
+	<entry name="layouts" type="list" ltype="string">
+		<li type="string">
+			<stringvalue>ro</stringvalue>
+		</li>
+		<li type="string">
+			<stringvalue>us</stringvalue>
+		</li>
+	</entry>
+</gconf>
diff --git desktop/desktop/30-gnome-2.30/.gconf/desktop/gnome/peripherals/keyboard/kbd/%gconf.xml.ro desktop/desktop/30-gnome-2.30/.gconf/desktop/gnome/peripherals/keyboard/kbd/%gconf.xml.ro
new file mode 100644
index 0000000..a40c1af
--- /dev/null
+++ desktop/desktop/30-gnome-2.30/.gconf/desktop/gnome/peripherals/keyboard/kbd/%gconf.xml.ro
@@ -0,0 +1,12 @@
+# Calculate name=%gconf.xml format=xml_gconf append=join os_locale_lang==ro_RO
+<?xml version="1.0"?>
+<gconf>
+	<entry name="layouts" type="list" ltype="string">
+		<li type="string">
+			<stringvalue>ro</stringvalue>
+		</li>
+		<li type="string">
+			<stringvalue>us</stringvalue>
+		</li>
+	</entry>
+</gconf>
diff --git desktop/desktop/60-firefox-3.5/.mozilla/firefox/calculate.default/prefs.js desktop/desktop/60-firefox-3.5/.mozilla/firefox/calculate.default/prefs.js
index 846e584..3d9432e 100644
--- desktop/desktop/60-firefox-3.5/.mozilla/firefox/calculate.default/prefs.js
+++ desktop/desktop/60-firefox-3.5/.mozilla/firefox/calculate.default/prefs.js
@@ -71,6 +71,9 @@ user_pref("spellchecker.dictionary", "pt");
 #?pkg(www-plugins/spell-mozilla-plugin)!=&&os_locale_language==ru#
 user_pref("spellchecker.dictionary", "ru");
 #pkg#
+#?pkg(www-plugins/spell-mozilla-plugin)!=&&os_locale_language==ro#
+user_pref("spellchecker.dictionary", "ro");
+#pkg#
 #?pkg(www-plugins/spell-mozilla-plugin)!=&&os_locale_language==uk#
 user_pref("spellchecker.dictionary", "uk-UA");
 #pkg#
diff --git install/1merge/00-calculate-install/calculate2.env install/1merge/00-calculate-install/calculate2.env
new file mode 100644
index 0000000..cead32f
--- /dev/null
+++ install/1merge/00-calculate-install/calculate2.env
@@ -0,0 +1,4 @@
+# Calculate format=samba path=/etc/calculate
+
+[install]
+os_install_net_settings = #-os_install_net_conf-#
diff --git install/1merge/glibc/locale.gen install/1merge/glibc/locale.gen
index 15cff6b..c087558 100644
--- install/1merge/glibc/locale.gen
+++ install/1merge/glibc/locale.gen
@@ -26,6 +26,8 @@ nn_NO.UTF-8 UTF-8
 pl_PL.UTF-8 UTF-8
 #Portuguese
 pt_PT.UTF-8 UTF-8
+#Romanian
+ro_RO.UTF-8 UTF-8
 #Russian
 ru_RU.UTF-8 UTF-8
 #Spanish
diff --git install/1merge/grub2/default.grub install/1merge/grub2/default.grub
index 67ddb16..f719541 100644
--- install/1merge/grub2/default.grub
+++ install/1merge/grub2/default.grub
@@ -8,8 +8,8 @@ GRUB_CMDLINE_LINUX="video=uvesafb:ywrap,#-os_install_fb_resolution-#,mtrr:3,spla
 #?pkg(media-gfx/cld-themes)==&&pkg(media-gfx/cldx-themes)==&&pkg(media-gfx/cldg-themes)==&&pkg(media-gfx/cls-themes)==&&pkg(media-gfx/cds-themes)==&&pkg(media-gfx/cmc-themes)==#
 GRUB_CMDLINE_LINUX="video=uvesafb:ywrap,#-os_install_fb_resolution-#,mtrr:3,splash=off,theme:tty1 #-os_install_kernel_resume-# elevator=#-os_install_kernel_scheduler-# doscsi#-os_install_kernel_attr-# calculate=video:#-os_install_x11_video_drv-# console=tty1 udev quiet"
 #pkg#
-GRUB_TERMINAL="gfxterm"
-GRUB_GFXMODE="auto"
+!GRUB_TERMINAL=del
+GRUB_GFXMODE="1024x768"
 GRUB_BACKGROUND="/boot/grub/grub-calculate.png"
 GRUB_COLOR_NORMAL="white/black" 
 GRUB_COLOR_HIGHLIGHT="magenta/black"
diff --git install/1merge/networkmanager/plasma-applet-networkmanagement.desktop install/1merge/networkmanager/plasma-applet-networkmanagement.desktop
new file mode 100644
index 0000000..2b3b677
--- /dev/null
+++ install/1merge/networkmanager/plasma-applet-networkmanagement.desktop
@@ -0,0 +1,8 @@
+# Calculate path=/usr/share/kde4/services pkg(kde-base/kdelibs)!= format=kde
+[Desktop Entry]
+#?os_install_net_conf==networkmanager#
+X-KDE-Library=plasma_applet_networkmanagement
+#os_install_net_conf#
+#?os_install_net_conf!=networkmanager#
+X-KDE-Library=
+#os_install_net_conf#
diff --git install/1merge/networkmanager/system-connections/.calculate_directory install/1merge/networkmanager/system-connections/.calculate_directory
index 4401bf6..b3a2ea7 100644
--- install/1merge/networkmanager/system-connections/.calculate_directory
+++ install/1merge/networkmanager/system-connections/.calculate_directory
@@ -1 +1 @@
-# Calculate path=/etc/NetworkManager exists(/etc/NetworkManager/system-connections,root)==&&cl_chroot_status==off&&os_install_root_type!=livecd||ac_install_system==up
+# Calculate path=/etc/NetworkManager os_install_net_settings!=networkmanager&&cl_chroot_status==off&&os_install_root_type!=livecd||ac_install_system==up
diff --git install/1merge/nm-applet/.calculate_directory install/1merge/nm-applet/.calculate_directory
new file mode 100644
index 0000000..6f79f0c
--- /dev/null
+++ install/1merge/nm-applet/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate belong()!=&&cl_chroot_status==off&&pkg(gnome-extra/nm-applet)!= path=/etc/xdg name=autostart
+
diff --git install/1merge/nm-applet/nm-applet.desktop install/1merge/nm-applet/nm-applet.desktop
new file mode 100644
index 0000000..d32af0f
--- /dev/null
+++ install/1merge/nm-applet/nm-applet.desktop
@@ -0,0 +1,8 @@
+# Calculate format=kde
+[Desktop Entry]
+#?os_install_net_conf==networkmanager#
+X-GNOME-Autostart-enabled=true
+#os_install_net_conf#
+#?os_install_net_conf!=networkmanager#
+X-GNOME-Autostart-enabled=false
+#os_install_net_conf#
diff --git install/1merge/openrc/conf.d/net install/1merge/openrc/conf.d/net
index a40ea98..4d39ad5 100644
--- install/1merge/openrc/conf.d/net
+++ install/1merge/openrc/conf.d/net
@@ -1,22 +1,30 @@
-# Calculate comment=# exists(/etc/conf.d/net,root)==&&cl_chroot_status==off&&os_install_root_type!=livecd&&os_install_net_conf==openrc||ac_install_system==up&&os_install_net_conf==openrc
+# Calculate comment=# os_install_net_settings!=openrc&&cl_chroot_status==off&&os_install_root_type!=livecd&&os_install_net_conf==openrc||ac_install_system==up&&os_install_net_conf==openrc
 
 #?os_install_linux_system==server#
 modules="!plug"
 #os_install_linux_system#
 #?list(os_install_net_interfaces,0)!=&&list(os_install_net_dhcp_set,0)!=on&&list(os_install_net_ip,0)!=#
 config_#-list(os_install_net_interfaces,0)-#=( "#-list(os_install_net_ip,0)-#/#-list(os_install_net_cidr,0)-#" )
+#list#
+#?list(os_install_net_interfaces,0)!=&&list(os_install_net_dhcp_set,0)!=on&&list(os_install_net_ip,0)!=&&list(os_install_net_route,0)!=#
 routes_#-list(os_install_net_interfaces,0)-#="#-list(os_install_net_route,0)-#"
 #list#
 #?list(os_install_net_interfaces,1)!=&&list(os_install_net_dhcp_set,1)!=on&&list(os_install_net_ip,1)!=#
 config_#-list(os_install_net_interfaces,1)-#=( "#-list(os_install_net_ip,1)-#/#-list(os_install_net_cidr,1)-#" )
+#list#
+#?list(os_install_net_interfaces,1)!=&&list(os_install_net_dhcp_set,1)!=on&&list(os_install_net_ip,1)!=&&list(os_install_net_route,1)!=#
 routes_#-list(os_install_net_interfaces,1)-#="#-list(os_install_net_route,1)-#"
 #list#
 #?list(os_install_net_interfaces,2)!=&&list(os_install_net_dhcp_set,2)!=on&&list(os_install_net_ip,2)!=#
 config_#-list(os_install_net_interfaces,2)-#=( "#-list(os_install_net_ip,2)-#/#-list(os_install_net_cidr,2)-#" )
+#list#
+#?list(os_install_net_interfaces,2)!=&&list(os_install_net_dhcp_set,2)!=on&&list(os_install_net_ip,2)!=&&list(os_install_net_route,2)!=#
 routes_#-list(os_install_net_interfaces,2)-#="#-list(os_install_net_route,2)-#"
 #list#
 #?list(os_install_net_interfaces,3)!=&&list(os_install_net_dhcp_set,3)!=on&&list(os_install_net_ip,3)!=#
 config_#-list(os_install_net_interfaces,3)-#=( "#-list(os_install_net_ip,3)-#/#-list(os_install_net_cidr,3)-#" )
+#list#
+#?list(os_install_net_interfaces,3)!=&&list(os_install_net_dhcp_set,3)!=on&&list(os_install_net_ip,3)!=&&list(os_install_net_route,3)!=#
 routes_#-list(os_install_net_interfaces,3)-#="#-list(os_install_net_route,3)-#"
 #list#
 #?os_install_net_dns!=#
diff --git install/1merge/sudo/sudoers install/1merge/sudo/sudoers
index 961a5bf..25c4cde 100644
--- install/1merge/sudo/sudoers
+++ install/1merge/sudo/sudoers
@@ -1,4 +1,4 @@
-# Calculate comment=#
+# Calculate comment=# chmod=0400
 
 Defaults	env_reset
 #?os_install_root_type==livecd#
diff --git install/1system/calculate-install/calculate2.env install/1system/calculate-install/calculate2.env
index d83a0f6..27df9f5 100644
--- install/1system/calculate-install/calculate2.env
+++ install/1system/calculate-install/calculate2.env
@@ -6,6 +6,10 @@ os_install_dev_from = #-os_root_dev-#
 # NTP is a protocol designed to synchronize the clocks of computers over a network.
 os_install_ntp = #-os_install_ntp-#
 #os_root_type#
+#pkg(sys-apps/calculate-install)>2.2.17-r1&&os_root_type=hdd#
+# name of net manager which is used
+os_install_net_settings = #-os_install_net_conf-#
+#pkg#
 
 #?os_install_linux_ver>10.8.81#
 [main]
diff --git install/2laptop/lenovo/sudo/sudoers install/2laptop/lenovo/sudo/sudoers
index dc81e5e..f252dae 100644
--- install/2laptop/lenovo/sudo/sudoers
+++ install/2laptop/lenovo/sudo/sudoers
@@ -1,4 +1,4 @@
-# Calculate format=patch path=/etc
+# Calculate format=patch path=/etc chmod=0400
 <reg>(# BEGIN laptop configuration)
 .*(# END laptop configuration)</reg>
 <text>\1
diff --git install/2laptop/samsung/sudo/sudoers install/2laptop/samsung/sudo/sudoers
index ee88674..4b5b0f7 100644
--- install/2laptop/samsung/sudo/sudoers
+++ install/2laptop/samsung/sudo/sudoers
@@ -1,4 +1,4 @@
-# Calculate format=patch path=/etc
+# Calculate format=patch path=/etc chmod=0400
 <reg>(# BEGIN laptop configuration)
 .*(# END laptop configuration)</reg>
 <text>\1
