diff --git install/1merge/desktopcouch/.calculate_directory install/1merge/desktopcouch/.calculate_directory
new file mode 100644
index 0000000..b56d4cd
--- /dev/null
+++ install/1merge/desktopcouch/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate belong()!=&&cl_chroot_status==off&&pkg(dev-db/desktopcouch)!= path=/usr/share name=applications
+
diff --git install/1merge/desktopcouch/desktopcouch-pair.desktop install/1merge/desktopcouch/desktopcouch-pair.desktop
new file mode 100644
index 0000000..bb065d8
--- /dev/null
+++ install/1merge/desktopcouch/desktopcouch-pair.desktop
@@ -0,0 +1,3 @@
+# Calculate format=kde
+[Desktop Entry]
+NoDisplay=true
