diff --git builder/squash/etc/init.d/calculate builder/squash/etc/init.d/calculate
new file mode 100644
index 0000000..35c0c98
--- /dev/null
+++ builder/squash/etc/init.d/calculate
@@ -0,0 +1,3 @@
+# Calculate format=patch cl_builder_live_set==off
+<reg>LANG=C /usr/bin/cl-install --startup --live</reg>
+<text>LANG=C /usr/bin/cl-install --startup</text>
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
index 150bd84..0a2655a 100644
--- desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
+++ desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
@@ -5,7 +5,7 @@ emoticonsTheme=KMess-Cartoon
 
 #Fonts
 [General]
-BrowserApplication[$e]=file:///usr/share/applications/calculate-browser.desktop
+BrowserApplication[$e]=calculate-browser.desktop
 XftHintStyle=hintmedium
 #?pkg(media-fonts/droid)!=#
 desktopFont=Droid Sans,10,-1,5,50,0,0,0,0,0
diff --git desktop/desktop/40-xfce-4.8-always/.config/xfce4/desktop/icons.screen0.rc desktop/desktop/40-xfce-4.8-always/.config/xfce4/desktop/icons.screen0.rc
index be2a74e..0108a26 100644
--- desktop/desktop/40-xfce-4.8-always/.config/xfce4/desktop/icons.screen0.rc
+++ desktop/desktop/40-xfce-4.8-always/.config/xfce4/desktop/icons.screen0.rc
@@ -26,15 +26,15 @@ col=1
 
 #env#
 #?env(client.os_remote_auth)!=&&ur_domain_set==on#
-[Home.desktop]
+[Home]
 row=0
 col=0
 
-[Share.desktop]
+[Share]
 row=0
 col=1
 
-[FTP.desktop]
+[FTP]
 row=0
 col=2
 #env#
diff --git desktop/desktop/60-chromium-5.0/.calculate/ini.env desktop/desktop/60-chromium-5.0/.calculate/ini.env
deleted file mode 100644
index 5908f2a..0000000
--- desktop/desktop/60-chromium-5.0/.calculate/ini.env
+++ /dev/null
@@ -1,7 +0,0 @@
-# Calculate format=samba
-
-[ver]
-chromium = 5.0
-
-[update]
-chromium = on
diff --git desktop/desktop/60-chromium-5.0/.calculate_directory desktop/desktop/60-chromium-5.0/.calculate_directory
deleted file mode 100644
index 8af9047..0000000
--- desktop/desktop/60-chromium-5.0/.calculate_directory
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate append=skip belong(chromium)!=&&exists(~/.config/chromium)==&&ini(update.chromium)!=off&&ini(ver.chromium)==&&pkg(www-client/chromium)>=5.0
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Bookmarks.en desktop/desktop/60-chromium-5.0/.config/chromium/Default/Bookmarks.en
deleted file mode 100644
index ea21e90..0000000
--- desktop/desktop/60-chromium-5.0/.config/chromium/Default/Bookmarks.en
+++ /dev/null
@@ -1,68 +0,0 @@
-# Calculate os_locale_language!=ru name=Bookmarks
-{
-   "roots": {
-      "bookmark_bar": {
-         "children": [{
-            "name": "Home",
-            "type": "url",
-            "url": "http://www.calculate-linux.org"
-         },
-         {
-            "name": "Docs",
-            "type": "url",
-            "url": "http://www.calculate-linux.org/main/en"
-         },
-         {
-            "name": "Forums",
-            "type": "url",
-            "url": "http://www.calculate-linux.org/projects/en/boards"
-         },
-         {
-            "name": "Lists",
-            "type": "url",
-            "url": "http://www.calculate-linux.org/main/en/maillist"
-         },
-         {
-            "name": "Bugs",
-            "type": "url",
-            "url": "http://www.calculate-linux.org/projects/calculatelinux/issues"
-         },
-         {
-            "name": "Blogs",
-            "type": "url",
-            "url": "http://www.calculate-linux.org/blogs/en"
-         }],
-         "type": "folder"
-      },
-      "other": {
-         "children": [
-         {
-            "name": "Calculate Linux | Facebook",
-            "type": "url",
-            "url": "http://www.facebook.com/group.php?gid=97309000613"
-         },
-         {
-            "name": "Calculate - LinuxQuestions.org",
-            "type": "url",
-            "url": "http://www.linuxquestions.org/questions/calculate-89/"
-         },
-         {
-            "name": "DistroWatch.com",
-            "type": "url",
-            "url": "http://distrowatch.com/"
-         },
-         {
-            "name": "Gentoo Linux",
-            "type": "url",
-            "url": "http://www.gentoo.org/"
-         },
-         {
-            "name": "LinuxTracker",
-            "type": "url",
-            "url": "http://linuxtracker.org/"
-         }],
-         "type": "folder"
-      }
-   },
-   "version": 1
-}
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Bookmarks.ru desktop/desktop/60-chromium-5.0/.config/chromium/Default/Bookmarks.ru
deleted file mode 100644
index 9b94ce2..0000000
--- desktop/desktop/60-chromium-5.0/.config/chromium/Default/Bookmarks.ru
+++ /dev/null
@@ -1,63 +0,0 @@
-# Calculate os_locale_language==ru name=Bookmarks
-{
-   "roots": {
-      "bookmark_bar": {
-         "children": [{
-            "name": "Домашняя",
-            "type": "url",
-            "url": "http://www.calculate-linux.ru"
-         },
-         {
-            "name": "Документация",
-            "type": "url",
-            "url": "http://www.calculate-linux.ru/main/ru"
-         },
-         {
-            "name": "Форумы",
-            "type": "url",
-            "url": "http://www.calculate-linux.ru/projects/ru/boards"
-         },
-         {
-            "name": "Рассылка",
-            "type": "url",
-            "url": "http://www.calculate-linux.ru/main/ru/maillist"
-         },
-         {
-            "name": "Ошибки",
-            "type": "url",
-            "url": "http://www.calculate-linux.ru/projects/calculatelinux/issues"
-         },
-         {
-            "name": "Блоги",
-            "type": "url",
-            "url": "http://www.calculate-linux.ru/blogs/ru"
-         }],
-         "type": "folder"
-      },
-      "other": {
-         "children": [
-         {
-            "name": "DistroWatch.com",
-            "type": "url",
-            "url": "http://distrowatch.com/"
-         },
-         {
-            "name": "Gentoo Linux",
-            "type": "url",
-            "url": "http://www.gentoo.ru/"
-         },
-         {
-            "name": "Lunuxoid",
-            "type": "url",
-            "url": "http://tux.in.ua"
-         },
-         {
-            "name": "В Контакте | Calculate Linux",
-            "type": "url",
-            "url": "http://vkontakte.ru/club10221243"
-         }],
-         "type": "folder"
-      }
-   },
-   "version": 1
-}
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/Cached Theme.pak desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/Cached Theme.pak
deleted file mode 100644
index 179d05f..0000000
Binary files desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/Cached Theme.pak and /dev/null differ
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA
deleted file mode 100644
index e58e2d2..0000000
Binary files desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA and /dev/null differ
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA
deleted file mode 100644
index c0e1abc..0000000
Binary files desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA and /dev/null differ
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA
deleted file mode 100644
index 74e359a..0000000
Binary files desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA and /dev/null differ
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA
deleted file mode 100644
index 7b191b6..0000000
Binary files desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA and /dev/null differ
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/manifest.json desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/manifest.json
deleted file mode 100644
index be93954..0000000
--- desktop/desktop/60-chromium-5.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/manifest.json
+++ /dev/null
@@ -1,32 +0,0 @@
-{
-   "key": "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCndMu10pCnXVJNSCvmc8K0CXEZZ8DU2DNgt77nkTQ2flcGgmf4yCY5QAKHMv2K9mDos38iVIcUSxuUruIXdOuQIY5JZy7Bitb5WzHQ6UfWWqoc/tEpVYi/k/+8TvORflKfgbmNX2kxrbI7gxkaTssMLYrgbrlG3KlwI7k890T2KQIDAQAB",
-   "name": "Minimal",
-   "theme": {
-      "colors": {
-         "bookmark_text": [ 149, 149, 149 ],
-         "frame": [ 255, 255, 255 ],
-         "ntp_background": [ 255, 255, 255 ],
-         "ntp_link": [ 170, 170, 170 ],
-         "ntp_section": [ 235, 235, 235, 1 ],
-         "ntp_section_link": [ 149, 149, 149 ],
-         "ntp_section_text": [ 149, 149, 149 ],
-         "ntp_text": [ 149, 149, 149 ],
-         "tab_background_text": [ 149, 149, 149 ],
-         "tab_text": [ 149, 149, 149 ],
-         "toolbar": [ 255, 255, 255 ]
-      },
-      "images": {
-         "theme_button_background": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA",
-         "theme_frame": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA",
-         "theme_tab_background": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA",
-         "theme_toolbar": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA"
-      },
-      "properties": {
-
-      },
-      "tints": {
-         "buttons": [ 0.001, 0, 0.706 ]
-      }
-   },
-   "version": "1.0"
-}
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/History desktop/desktop/60-chromium-5.0/.config/chromium/Default/History
deleted file mode 100644
index b667cdd..0000000
Binary files desktop/desktop/60-chromium-5.0/.config/chromium/Default/History and /dev/null differ
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Preferences desktop/desktop/60-chromium-5.0/.config/chromium/Default/Preferences
deleted file mode 100644
index 0b0ee65..0000000
--- desktop/desktop/60-chromium-5.0/.config/chromium/Default/Preferences
+++ /dev/null
@@ -1,162 +0,0 @@
-{
-   "alternate_error_pages": {
-      "enabled": true
-   },
-   "autofill": {
-      "auxiliary_profiles_enabled": false,
-      "default_profile": "",
-      "enabled": true,
-      "infobar_shown": true
-   },
-   "bookmark_bar": {
-      "show_on_all_tabs": true
-   },
-   "browser": {
-      "check_default_browser": false,
-      "clear_data": {
-         "browsing_history": true,
-         "cache": true,
-         "cookies": true,
-         "download_history": true,
-         "form_data": true,
-         "passwords": false,
-         "time_period": 0
-      },
-      "custom_chrome_frame": false,
-      "enable_spellchecking": true,
-      "ntp": {
-         "promo_image_remaining": -79,
-         "promo_line_remaining": -79
-      },
-      "show_home_button": true
-   },
-   "content_settings_window": {
-      "last_tab_index": 4
-   },
-   "countryid_at_install": 21077,
-   "default_search_provider": {
-      "id": "2",
-      "name": "Google",
-      "prepopulate_id": "1",
-      "search_url": "{google:baseURL}search?{google:RLZ}{google:acceptedSuggestion}{google:originalQueryForSuggestion}sourceid=chrome&ie={inputEncoding}&q={searchTerms}",
-      "suggest_url": "{google:baseSuggestURL}search?client=chrome&hl={language}&q={searchTerms}"
-   },
-   "devtools": {
-      "open_docked": false
-   },
-   "dns_prefetching": {
-      "enabled": true
-   },
-   "download": {
-      "directory_upgrade": true,
-      "extensions_to_open": "",
-      "prompt_for_download": false
-   },
-   "extensions": {
-      "autoupdate": {
-         "last_check": "12918142274298271",
-         "next_check": "12918215676716661"
-      },
-      "blacklistupdate": {
-         "lastpingday": "12918121200665425"
-      },
-      "chrome_url_overrides": {
-         "bookmarks": [ "chrome-extension://eemcgdkfndhakfknompkggombfjjjeno/main.html" ]
-      },
-      "settings": {
-         "hnfhcmjkebafbfikmbkhdpbmfpfjgiog": {
-            "lastpingday": "12918121199767875",
-            "location": 1,
-            "manifest": {
-               "key": "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCndMu10pCnXVJNSCvmc8K0CXEZZ8DU2DNgt77nkTQ2flcGgmf4yCY5QAKHMv2K9mDos38iVIcUSxuUruIXdOuQIY5JZy7Bitb5WzHQ6UfWWqoc/tEpVYi/k/+8TvORflKfgbmNX2kxrbI7gxkaTssMLYrgbrlG3KlwI7k890T2KQIDAQAB",
-               "name": "Minimal",
-               "theme": {
-                  "colors": {
-                     "bookmark_text": [ 149, 149, 149 ],
-                     "frame": [ 255, 255, 255 ],
-                     "ntp_background": [ 255, 255, 255 ],
-                     "ntp_link": [ 170, 170, 170 ],
-                     "ntp_section": [ 235, 235, 235, 1 ],
-                     "ntp_section_link": [ 149, 149, 149 ],
-                     "ntp_section_text": [ 149, 149, 149 ],
-                     "ntp_text": [ 149, 149, 149 ],
-                     "tab_background_text": [ 149, 149, 149 ],
-                     "tab_text": [ 149, 149, 149 ],
-                     "toolbar": [ 255, 255, 255 ]
-                  },
-                  "images": {
-                     "theme_button_background": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA",
-                     "theme_frame": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA",
-                     "theme_tab_background": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA",
-                     "theme_toolbar": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA"
-                  },
-                  "tints": {
-                     "buttons": [ 0.001, 0, 0.706 ]
-                  }
-               },
-               "version": "1.0"
-            },
-            "path": "hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0",
-            "state": 1
-         }
-      },
-      "theme": {
-         "id": "hnfhcmjkebafbfikmbkhdpbmfpfjgiog",
-         "pack": "/home/#-ur_login-#/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/Cached Theme.pak",
-         "use_system": false
-      },
-      "toolbarsize": -1,
-      "ui": {
-         "developer_mode": true
-      }
-   },
-   "geolocation": {
-      "default_content_setting": 3
-   },
-#?os_locale_language==ru#
-   "homepage": "http://www.calculate-linux.ru/",
-#os_locale_language#
-#?os_locale_language!=ru#
-   "homepage": "http://www.calculate-linux.org/",
-#os_locale_language#
-   "homepage_is_newtabpage": false,
-   "profile": {
-      "block_third_party_cookies": false,
-      "clear_site_data_on_exit": false,
-      "content_settings": {
-         "pref_version": 1
-      },
-      "exited_cleanly": true,
-      "password_manager_enabled": true,
-      "per_host_zoom_levels": {
-         "www.computerra.ru": 1
-      }
-   },
-   "safebrowsing": {
-      "enabled": true
-   },
-   "search": {
-      "suggest_enabled": true
-   },
-   "session": {
-      "restore_on_startup": 0
-   },
-   "sync": {
-      "autofill": true,
-      "bookmarks": true,
-      "has_setup_completed": true,
-      "last_synced_time": "12918205902976887",
-      "preferences": true,
-      "themes": true
-   },
-   "translate": {
-      "enabled": true
-   },
-   "translate_language_blacklist": [ "en" ],
-   "webkit": {
-      "webprefs": {
-         "inspector_settings": "lastActivePanel:string:scripts\n",
-         "uses_universal_detector": false
-      }
-   }
-}
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/Default/Thumbnails desktop/desktop/60-chromium-5.0/.config/chromium/Default/Thumbnails
deleted file mode 100644
index 2cb2b4a..0000000
Binary files desktop/desktop/60-chromium-5.0/.config/chromium/Default/Thumbnails and /dev/null differ
diff --git desktop/desktop/60-chromium-5.0/.config/chromium/First Run desktop/desktop/60-chromium-5.0/.config/chromium/First Run
deleted file mode 100644
index 8b3bbff..0000000
--- desktop/desktop/60-chromium-5.0/.config/chromium/First Run	
+++ /dev/null
@@ -1,2 +0,0 @@
-# Calculate exists(~/.mozilla)==
-# will not migrate if Firefox had not run
\ No newline at end of file
diff --git desktop/desktop/60-chromium-6.0/.calculate/ini.env desktop/desktop/60-chromium-6.0/.calculate/ini.env
new file mode 100644
index 0000000..3a94e61
--- /dev/null
+++ desktop/desktop/60-chromium-6.0/.calculate/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba
+
+[ver]
+chromium = 6.0
+
+[update]
+chromium = on
diff --git desktop/desktop/60-chromium-6.0/.calculate_directory desktop/desktop/60-chromium-6.0/.calculate_directory
new file mode 100644
index 0000000..45adab3
--- /dev/null
+++ desktop/desktop/60-chromium-6.0/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(chromium)!=&&exists(~/.config/chromium)==&&ini(update.chromium)!=off&&ini(ver.chromium)==&&pkg(www-client/chromium)>=6.0
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Bookmarks.en desktop/desktop/60-chromium-6.0/.config/chromium/Default/Bookmarks.en
new file mode 100644
index 0000000..ea21e90
--- /dev/null
+++ desktop/desktop/60-chromium-6.0/.config/chromium/Default/Bookmarks.en
@@ -0,0 +1,68 @@
+# Calculate os_locale_language!=ru name=Bookmarks
+{
+   "roots": {
+      "bookmark_bar": {
+         "children": [{
+            "name": "Home",
+            "type": "url",
+            "url": "http://www.calculate-linux.org"
+         },
+         {
+            "name": "Docs",
+            "type": "url",
+            "url": "http://www.calculate-linux.org/main/en"
+         },
+         {
+            "name": "Forums",
+            "type": "url",
+            "url": "http://www.calculate-linux.org/projects/en/boards"
+         },
+         {
+            "name": "Lists",
+            "type": "url",
+            "url": "http://www.calculate-linux.org/main/en/maillist"
+         },
+         {
+            "name": "Bugs",
+            "type": "url",
+            "url": "http://www.calculate-linux.org/projects/calculatelinux/issues"
+         },
+         {
+            "name": "Blogs",
+            "type": "url",
+            "url": "http://www.calculate-linux.org/blogs/en"
+         }],
+         "type": "folder"
+      },
+      "other": {
+         "children": [
+         {
+            "name": "Calculate Linux | Facebook",
+            "type": "url",
+            "url": "http://www.facebook.com/group.php?gid=97309000613"
+         },
+         {
+            "name": "Calculate - LinuxQuestions.org",
+            "type": "url",
+            "url": "http://www.linuxquestions.org/questions/calculate-89/"
+         },
+         {
+            "name": "DistroWatch.com",
+            "type": "url",
+            "url": "http://distrowatch.com/"
+         },
+         {
+            "name": "Gentoo Linux",
+            "type": "url",
+            "url": "http://www.gentoo.org/"
+         },
+         {
+            "name": "LinuxTracker",
+            "type": "url",
+            "url": "http://linuxtracker.org/"
+         }],
+         "type": "folder"
+      }
+   },
+   "version": 1
+}
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Bookmarks.ru desktop/desktop/60-chromium-6.0/.config/chromium/Default/Bookmarks.ru
new file mode 100644
index 0000000..9b94ce2
--- /dev/null
+++ desktop/desktop/60-chromium-6.0/.config/chromium/Default/Bookmarks.ru
@@ -0,0 +1,63 @@
+# Calculate os_locale_language==ru name=Bookmarks
+{
+   "roots": {
+      "bookmark_bar": {
+         "children": [{
+            "name": "Домашняя",
+            "type": "url",
+            "url": "http://www.calculate-linux.ru"
+         },
+         {
+            "name": "Документация",
+            "type": "url",
+            "url": "http://www.calculate-linux.ru/main/ru"
+         },
+         {
+            "name": "Форумы",
+            "type": "url",
+            "url": "http://www.calculate-linux.ru/projects/ru/boards"
+         },
+         {
+            "name": "Рассылка",
+            "type": "url",
+            "url": "http://www.calculate-linux.ru/main/ru/maillist"
+         },
+         {
+            "name": "Ошибки",
+            "type": "url",
+            "url": "http://www.calculate-linux.ru/projects/calculatelinux/issues"
+         },
+         {
+            "name": "Блоги",
+            "type": "url",
+            "url": "http://www.calculate-linux.ru/blogs/ru"
+         }],
+         "type": "folder"
+      },
+      "other": {
+         "children": [
+         {
+            "name": "DistroWatch.com",
+            "type": "url",
+            "url": "http://distrowatch.com/"
+         },
+         {
+            "name": "Gentoo Linux",
+            "type": "url",
+            "url": "http://www.gentoo.ru/"
+         },
+         {
+            "name": "Lunuxoid",
+            "type": "url",
+            "url": "http://tux.in.ua"
+         },
+         {
+            "name": "В Контакте | Calculate Linux",
+            "type": "url",
+            "url": "http://vkontakte.ru/club10221243"
+         }],
+         "type": "folder"
+      }
+   },
+   "version": 1
+}
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/adblock-install.sh desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/adblock-install.sh
new file mode 100644
index 0000000..077edd2
--- /dev/null
+++ desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/adblock-install.sh
@@ -0,0 +1,4 @@
+# Calculate path=/tmp pkg(www-apps/chromium-adblock)!= exec=/bin/bash
+
+cp -r /usr/share/chromium-adblock/gighmmpiobklfepjocnamgkkbiglidom #-ur_home_path-#/.config/chromium/Default/Extensions/
+exit 0
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/Cached Theme.pak desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/Cached Theme.pak
new file mode 100644
index 0000000..179d05f
Binary files /dev/null and desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/Cached Theme.pak differ
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA
new file mode 100644
index 0000000..e58e2d2
Binary files /dev/null and desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA differ
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA
new file mode 100644
index 0000000..c0e1abc
Binary files /dev/null and desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA differ
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA
new file mode 100644
index 0000000..74e359a
Binary files /dev/null and desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA differ
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA
new file mode 100644
index 0000000..7b191b6
Binary files /dev/null and desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA differ
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/manifest.json desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/manifest.json
new file mode 100644
index 0000000..be93954
--- /dev/null
+++ desktop/desktop/60-chromium-6.0/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/manifest.json
@@ -0,0 +1,32 @@
+{
+   "key": "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCndMu10pCnXVJNSCvmc8K0CXEZZ8DU2DNgt77nkTQ2flcGgmf4yCY5QAKHMv2K9mDos38iVIcUSxuUruIXdOuQIY5JZy7Bitb5WzHQ6UfWWqoc/tEpVYi/k/+8TvORflKfgbmNX2kxrbI7gxkaTssMLYrgbrlG3KlwI7k890T2KQIDAQAB",
+   "name": "Minimal",
+   "theme": {
+      "colors": {
+         "bookmark_text": [ 149, 149, 149 ],
+         "frame": [ 255, 255, 255 ],
+         "ntp_background": [ 255, 255, 255 ],
+         "ntp_link": [ 170, 170, 170 ],
+         "ntp_section": [ 235, 235, 235, 1 ],
+         "ntp_section_link": [ 149, 149, 149 ],
+         "ntp_section_text": [ 149, 149, 149 ],
+         "ntp_text": [ 149, 149, 149 ],
+         "tab_background_text": [ 149, 149, 149 ],
+         "tab_text": [ 149, 149, 149 ],
+         "toolbar": [ 255, 255, 255 ]
+      },
+      "images": {
+         "theme_button_background": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA",
+         "theme_frame": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA",
+         "theme_tab_background": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA",
+         "theme_toolbar": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA"
+      },
+      "properties": {
+
+      },
+      "tints": {
+         "buttons": [ 0.001, 0, 0.706 ]
+      }
+   },
+   "version": "1.0"
+}
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/History desktop/desktop/60-chromium-6.0/.config/chromium/Default/History
new file mode 100644
index 0000000..b667cdd
Binary files /dev/null and desktop/desktop/60-chromium-6.0/.config/chromium/Default/History differ
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Local Storage/chrome-extension_gighmmpiobklfepjocnamgkkbiglidom_0.localstorage desktop/desktop/60-chromium-6.0/.config/chromium/Default/Local Storage/chrome-extension_gighmmpiobklfepjocnamgkkbiglidom_0.localstorage
new file mode 100644
index 0000000..944f58f
Binary files /dev/null and desktop/desktop/60-chromium-6.0/.config/chromium/Default/Local Storage/chrome-extension_gighmmpiobklfepjocnamgkkbiglidom_0.localstorage differ
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Preferences desktop/desktop/60-chromium-6.0/.config/chromium/Default/Preferences
new file mode 100644
index 0000000..f018d5d
--- /dev/null
+++ desktop/desktop/60-chromium-6.0/.config/chromium/Default/Preferences
@@ -0,0 +1,123 @@
+{
+   "bookmark_bar": {
+      "show_on_all_tabs": true
+   },
+   "browser": {
+      "check_default_browser": false,
+      "enable_spellchecking": true,
+      "show_home_button": true
+   },
+   "extensions": {
+      "chrome_url_overrides": {
+         "bookmarks": [ "chrome-extension://eemcgdkfndhakfknompkggombfjjjeno/main.html" ]
+      },
+      "settings": {
+         "hnfhcmjkebafbfikmbkhdpbmfpfjgiog": {
+            "location": 1,
+            "manifest": {
+               "key": "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCndMu10pCnXVJNSCvmc8K0CXEZZ8DU2DNgt77nkTQ2flcGgmf4yCY5QAKHMv2K9mDos38iVIcUSxuUruIXdOuQIY5JZy7Bitb5WzHQ6UfWWqoc/tEpVYi/k/+8TvORflKfgbmNX2kxrbI7gxkaTssMLYrgbrlG3KlwI7k890T2KQIDAQAB",
+               "name": "Minimal",
+               "theme": {
+                  "colors": {
+                     "bookmark_text": [ 149, 149, 149 ],
+                     "frame": [ 255, 255, 255 ],
+                     "ntp_background": [ 255, 255, 255 ],
+                     "ntp_link": [ 170, 170, 170 ],
+                     "ntp_section": [ 235, 235, 235, 1 ],
+                     "ntp_section_link": [ 149, 149, 149 ],
+                     "ntp_section_text": [ 149, 149, 149 ],
+                     "ntp_text": [ 149, 149, 149 ],
+                     "tab_background_text": [ 149, 149, 149 ],
+                     "tab_text": [ 149, 149, 149 ],
+                     "toolbar": [ 255, 255, 255 ]
+                  },
+                  "images": {
+                     "theme_button_background": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYktsBDA",
+                     "theme_frame": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYoaQBDA",
+                     "theme_tab_background": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYmtMBDA",
+                     "theme_toolbar": "i/agxjaHJvbWV0aGVtZXNyDAsSBEZpbGUYltMBDA"
+                  },
+                  "tints": {
+                     "buttons": [ 0.001, 0, 0.706 ]
+                  }
+               },
+               "version": "1.0"
+            },
+            "path": "hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0",
+            "state": 1
+#?pkg(www-apps/chromium-adblock)!=#
+         },
+         "gighmmpiobklfepjocnamgkkbiglidom": {
+            "allowFileAccess": true,
+            "app_launcher_index": 0,
+            "granted_permissions": {
+               "api": [ "contextMenus", "tabs" ],
+               "full": false,
+               "host": [ "*://*.hk-pub.com/forum/thread-*", "*://*.mail.live.com/*", "*://*.youtube.com/*", "http://*/*", "https://*/*" ]
+            },
+	    "install_time": "12959953182540259",
+            "location": 1,
+            "manifest": {
+               "background_page": "background.html",
+               "browser_action": {
+                  "default_icon": "img/icon19.png",
+                  "default_title": "\u041F\u0430\u0440\u0430\u043C\u0435\u0442\u0440\u044B AdBlock",
+                  "popup": "button/popup.html"
+               },
+               "content_scripts": [ {
+                  "all_frames": true,
+                  "js": [ "port.js", "jquery/jquery.min.js", "functions.js", "filtering/filteroptions.js", "filtering/filtertypes.js", "filtering/filterset.js", "adblock_start.js" ],
+                  "matches": [ "http://*/*", "https://*/*" ],
+                  "run_at": "document_start"
+               }, {
+                  "all_frames": true,
+                  "js": [ "bandaids.js" ],
+                  "matches": [ "*://*.mail.live.com/*", "*://*.hk-pub.com/forum/thread-*", "*://*.youtube.com/*" ],
+                  "run_at": "document_end"
+               }, {
+                  "all_frames": true,
+                  "js": [ "adblock.js" ],
+                  "matches": [ "http://*/*", "https://*/*" ],
+                  "run_at": "document_end"
+               }, {
+                  "all_frames": false,
+                  "js": [ "uiscripts/blacklisting/rightclick_hook.js" ],
+                  "matches": [ "http://*/*", "https://*/*" ],
+                  "run_at": "document_end"
+               } ],
+               "current_locale": "ru",
+               "default_locale": "en",
+               "description": "\u041D\u0430\u0438\u0431\u043E\u043B\u0435\u0435 \u043F\u043E\u043F\u0443\u043B\u044F\u0440\u043D\u043E\u0435 \u0440\u0430\u0441\u0448\u0438\u0440\u0435\u043D\u0438\u0435 Chrome \u0441 \u0431\u043E\u043B\u0435\u0435 \u0447\u0435\u043C 2 \u043C\u0438\u043B\u043B\u0438\u043E\u043D\u0430\u043C\u0438 \u043F\u043E\u043B\u044C\u0437\u043E\u0432\u0430\u0442\u0435\u043B\u0435\u0439! \u0411\u043B\u043E\u043A\u0438\u0440\u0443\u0435\u0442 \u0440\u0435\u043A\u043B\u0430\u043C\u0443 \u043F\u043E \u0432\u0441\u0435\u043C\u0443 \u0418\u043D\u0442\u0435\u0440\u043D\u0435\u0442\u0443.",
+               "icons": {
+                  "128": "img/icon128.png",
+                  "16": "img/icon16.png",
+                  "48": "img/icon48.png"
+               },
+               "incognito": "spanning",
+               "key": "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDZWWNkerYNFB0SIg87WwcrcniMYkCqxLz4OB3CLNoogUUsHN+Rk9vRB7o3Jjc4shlz2LBvAbtPbrGkU+NK9Hc3ubVydzpnoRDEl1RKz3GhqKFkECvTyAxrsKainXH+4Ni2+K7bg1U0tLoQMPJ9f3ieYLIwXeHh9+lYXQi6soq0pwIDAQAB",
+               "minimum_chrome_version": "6",
+               "name": "AdBlock",
+               "options_page": "options/index.html",
+               "permissions": [ "http://*/*", "https://*/*", "contextMenus", "tabs" ],
+               "update_url": "http://clients2.google.com/service/update2/crx",
+               "version": "2.4.27"
+            },
+            "path": "gighmmpiobklfepjocnamgkkbiglidom/2.4.27_0",
+            "state": 1
+#pkg#
+         }
+      },
+      "theme": {
+         "id": "hnfhcmjkebafbfikmbkhdpbmfpfjgiog",
+         "pack": "/home/#-ur_login-#/.config/chromium/Default/Extensions/hnfhcmjkebafbfikmbkhdpbmfpfjgiog/1.0/Cached Theme.pak",
+         "use_system": false
+      }
+   },
+#?os_locale_language==ru#
+   "homepage": "http://www.calculate-linux.ru/",
+#os_locale_language#
+#?os_locale_language!=ru#
+   "homepage": "http://www.calculate-linux.org/",
+#os_locale_language#
+   "homepage_is_newtabpage": false
+}
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/Default/Thumbnails desktop/desktop/60-chromium-6.0/.config/chromium/Default/Thumbnails
new file mode 100644
index 0000000..2cb2b4a
Binary files /dev/null and desktop/desktop/60-chromium-6.0/.config/chromium/Default/Thumbnails differ
diff --git desktop/desktop/60-chromium-6.0/.config/chromium/First Run desktop/desktop/60-chromium-6.0/.config/chromium/First Run
new file mode 100644
index 0000000..8b3bbff
--- /dev/null
+++ desktop/desktop/60-chromium-6.0/.config/chromium/First Run	
@@ -0,0 +1,2 @@
+# Calculate exists(~/.mozilla)==
+# will not migrate if Firefox had not run
\ No newline at end of file
diff --git desktop/desktop/99-final/.dmrc desktop/desktop/99-final/.dmrc
index 1406119..0894b2b 100644
--- desktop/desktop/99-final/.dmrc
+++ desktop/desktop/99-final/.dmrc
@@ -1,2 +1,2 @@
-# Calculate path=~ name=.dmrc append=remove
+# Calculate path=~ name=.dmrc append=remove belong(calculate-desktop)!=
 # To delete a file necessary when changing the system
diff --git install/1merge/Desktop/300-ImageEdit/gimp/calculate/zzz-gimp-2.6.11.desktop install/1merge/Desktop/300-ImageEdit/gimp/calculate/zzz-gimp-2.6.11.desktop
new file mode 100644
index 0000000..71984f2
--- /dev/null
+++ install/1merge/Desktop/300-ImageEdit/gimp/calculate/zzz-gimp-2.6.11.desktop
@@ -0,0 +1,9 @@
+# Calculate format=kde link=/usr/share/applications/zzz-gimp.desktop name=calculate-imageedit.desktop mirror pkg(media-gfx/gimp)>=2.6.11&&exists(/usr/share/applications/zzz-gimp.desktop)!=
+
+# Update info
+#-ini(desktop.imageedit, gimp)-#
+
+# Hide icon
+[Desktop Entry]
+OnlyShowIn=X-NeverShowThis;
+
diff --git install/1merge/gimp/.calculate_directory install/1merge/gimp/.calculate_directory
new file mode 100644
index 0000000..d2e799d
--- /dev/null
+++ install/1merge/gimp/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate belong()!=&&pkg(media-gfx/gimp)!= path=/etc name=env.d
+
diff --git install/1merge/gimp/99gimp install/1merge/gimp/99gimp
new file mode 100644
index 0000000..615128e
--- /dev/null
+++ install/1merge/gimp/99gimp
@@ -0,0 +1,3 @@
+# discard /usr/lib64/gimp/2.0/plug-ins/hot due to an segfault prelink
+# prelink (20110106-20110511), gimp (2.6.11) (CLD-64)
+PRELINK_PATH_MASK=/usr/lib64/gimp/2.0/plug-ins/hot
