diff --git install/1merge/00-calculate-install/calculate2.env install/1merge/00-calculate-install/calculate2.env
index cead32f..6866fd0 100644
--- install/1merge/00-calculate-install/calculate2.env
+++ install/1merge/00-calculate-install/calculate2.env
@@ -1,4 +1,8 @@
-# Calculate format=samba path=/etc/calculate
+# Calculate format=samba path=/etc/calculate cl_chroot_status==off||ac_install_system==up autoupdate
+
+#?ac_install_system!=up&&exists(/etc/calculate/calculate2.env,root)!=#
+#-load(empty,/etc/calculate/calculate2.env,root)-#
+#ac_install_system#
 
 [install]
 os_install_net_settings = #-os_install_net_conf-#
