diff --git install/1merge/grub/.calculate_directory install/1merge/grub/.calculate_directory
new file mode 100644
index 0000000..74c47ca
--- /dev/null
+++ install/1merge/grub/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate belong()!=&&cl_chroot_status==off&&pkg(sys-boot/grub)!= path=/boot
+
diff --git install/1merge/grub/grub.conf install/1merge/grub/grub.conf
new file mode 100644
index 0000000..e87d8a5
--- /dev/null
+++ install/1merge/grub/grub.conf
@@ -0,0 +1,5 @@
+# Calculate format=patch os_install_x11_video_drv!=os_x11_video_drv&&cl_belong_pkg==&&ac_install_system!=up&&pkg(sys-kernel/calckernel)>=3.4.15-r5
+<reg>(kernel /boot/vmlinuz-#-cl_kernel_uid-#.*)(:? calculate=\S+)</reg>
+<text>\1</text>
+<reg>(kernel /boot/vmlinuz-#-cl_kernel_uid-#.*)</reg>
+<text>\1 calculate=video:#-os_install_x11_video_drv-#</text>
