commit bc65810e87b9ad982476d4ca05743ba035a39ef5
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jun 10 17:59:53 2011 +0400

    Fix variable bug.

diff --git a/desktop/desktop/20-kde-4.6/.kde4/share/config/ksplashrc b/desktop/desktop/20-kde-4.6/.kde4/share/config/ksplashrc
index e323c6c..a69927b 100644
--- a/desktop/desktop/20-kde-4.6/.kde4/share/config/ksplashrc
+++ b/desktop/desktop/20-kde-4.6/.kde4/share/config/ksplashrc
@@ -2,9 +2,9 @@
 #тема загрузки рабочего стола
 [KSplash]
 Engine=KSplashX
-#?os_install_locale_lang!=pt_BR#
+#?os_locale_lang!=pt_BR#
 Theme=CalculateSplash#-case(capitalize,os_locale_language)-#
-#os_install_locale_lang#
-#?os_install_locale_lang==pt_BR#
+#os_locale_lang#
+#?os_locale_lang==pt_BR#
 Theme=CalculateSplashPt
-#os_install_locale_lang#
+#os_locale_lang#
