diff --git desktop/desktop/00-patch-2.2.0.16/.calculate_directory desktop/desktop/00-patch-2.2.0.16/.calculate_directory
deleted file mode 100644
index f6d18e2..0000000
--- desktop/desktop/00-patch-2.2.0.16/.calculate_directory
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.0.16
diff --git desktop/desktop/00-patch-2.2.0.16/1ini_res_def.env desktop/desktop/00-patch-2.2.0.16/1ini_res_def.env
deleted file mode 100644
index 023f00c..0000000
--- desktop/desktop/00-patch-2.2.0.16/1ini_res_def.env
+++ /dev/null
@@ -1,12 +0,0 @@
-# Calculate format=samba path=~/.calculate name=ini.env
-
-[resource]
-desktop = /home/#-ur_login-#/Desktop
-documents = #-ini(resource.path)-#/Documents
-downloads = #-ini(resource.path)-#/Downloads
-music = #-ini(resource.path)-#/Music
-pictures = #-ini(resource.path)-#/Pictures
-public = #-ini(resource.path)-#/Public
-templates = #-ini(resource.path)-#/Templates
-videos = #-ini(resource.path)-#/Videos
-
diff --git desktop/desktop/00-patch-2.2.0.16/2ini_res.env desktop/desktop/00-patch-2.2.0.16/2ini_res.env
deleted file mode 100644
index ec7ae86..0000000
--- desktop/desktop/00-patch-2.2.0.16/2ini_res.env
+++ /dev/null
@@ -1,146 +0,0 @@
-# Calculate format=samba path=~/.calculate name=ini.env
-
-[resource]
-#?os_locale_lang==bg_BG#
-documents = #-ini(resource.path)-#/Документи
-downloads = #-ini(resource.path)-#/Свалени
-music = #-ini(resource.path)-#/Музика
-pictures = #-ini(resource.path)-#/Изображения
-public = #-ini(resource.path)-#/Публични
-templates = #-ini(resource.path)-#/Шаблони
-videos = #-ini(resource.path)-#/Видео
-#os_locale_lang#
-#?os_locale_lang==bg_BG&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Работен плот
-#os_locale_lang#
-
-#?os_locale_lang==de_DE#
-documents = #-ini(resource.path)-#/Dokumente
-downloads = #-ini(resource.path)-#/Downloads
-music = #-ini(resource.path)-#/Musik
-pictures = #-ini(resource.path)-#/Bilder
-public = #-ini(resource.path)-#/Öffentlich
-templates = #-ini(resource.path)-#/Vorlagen
-videos = #-ini(resource.path)-#/Videos
-#os_locale_lang#
-#?os_locale_lang==de_DE&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Arbeitsfläche
-#os_locale_lang#
-
-#?os_locale_lang==es_ES#
-documents = #-ini(resource.path)-#/Documentos
-downloads = #-ini(resource.path)-#/Descargas
-music = #-ini(resource.path)-#/Música
-pictures = #-ini(resource.path)-#/Imágenes
-public = #-ini(resource.path)-#/Público
-templates = #-ini(resource.path)-#/Plantillas
-videos = #-ini(resource.path)-#/Vídeos
-#os_locale_lang#
-#?os_locale_lang==es_ES&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Escritorio
-#os_locale_lang#
-
-#?os_locale_lang==fr_FR#
-documents = #-ini(resource.path)-#/Documents
-downloads = #-ini(resource.path)-#/Téléchargements
-music = #-ini(resource.path)-#/Musique
-pictures = #-ini(resource.path)-#/Images
-public = #-ini(resource.path)-#/Public
-templates = #-ini(resource.path)-#/Modèles
-videos = #-ini(resource.path)-#/Vidéos
-#os_locale_lang#
-#?os_locale_lang==fr_FR&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Bureau
-#os_locale_lang#
-
-#?os_locale_lang==it_IT#
-documents = #-ini(resource.path)-#/Documenti
-downloads = #-ini(resource.path)-#/Scaricati
-music = #-ini(resource.path)-#/Musica
-pictures = #-ini(resource.path)-#/Immagini
-public = #-ini(resource.path)-#/Pubblici
-templates = #-ini(resource.path)-#/Modelli
-videos = #-ini(resource.path)-#/Video
-#os_locale_lang#
-#?os_locale_lang==it_IT&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Ambiente grafico
-#os_locale_lang#
-
-#?os_locale_lang==pl_PL#
-documents = #-ini(resource.path)-#/Dokumenty
-downloads = #-ini(resource.path)-#/Pobrane
-music = #-ini(resource.path)-#/Muzyka
-pictures = #-ini(resource.path)-#/Obrazy
-public = #-ini(resource.path)-#/Publiczny
-templates = #-ini(resource.path)-#/Szablony
-videos = #-ini(resource.path)-#/Wideo
-#os_locale_lang#
-#?os_locale_lang==pl_PL&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Pulpit
-#os_locale_lang#
-
-#?os_locale_lang==pt_BR#
-documents = #-ini(resource.path)-#/Documentos
-downloads = #-ini(resource.path)-#/Download
-music = #-ini(resource.path)-#/Música
-pictures = #-ini(resource.path)-#/Imagens
-public = #-ini(resource.path)-#/Público
-templates = #-ini(resource.path)-#/Modelos
-videos = #-ini(resource.path)-#/Vídeos
-#os_locale_lang#
-#?os_locale_lang==pt_BR&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Área de trabalho
-#os_locale_lang#
-
-#?os_locale_lang==pt_PT#
-documents = #-ini(resource.path)-#/Documentos
-downloads = #-ini(resource.path)-#/Transferências
-music = #-ini(resource.path)-#/Música
-pictures = #-ini(resource.path)-#/Imagens
-public = #-ini(resource.path)-#/Público
-templates = #-ini(resource.path)-#/Modelos
-videos = #-ini(resource.path)-#/Vídeos
-#os_locale_lang#
-#?os_locale_lang==pt_PT&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Desktop
-#os_locale_lang#
-
-#?os_locale_lang==ro_RO#
-documents = #-ini(resource.path)-#/Documente
-downloads = #-ini(resource.path)-#/Descărcări
-music = #-ini(resource.path)-#/Muzică
-pictures = #-ini(resource.path)-#/Poze
-public = #-ini(resource.path)-#/Public
-templates = #-ini(resource.path)-#/Templates
-videos = #-ini(resource.path)-#/Video
-#os_locale_lang#
-#?os_locale_lang==ro_RO&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Desktop
-#os_locale_lang#
-
-#?os_locale_lang==ru_RU#
-documents = #-ini(resource.path)-#/Документы
-downloads = #-ini(resource.path)-#/Загрузки
-music = #-ini(resource.path)-#/Музыка
-pictures = #-ini(resource.path)-#/Картинки
-public = #-ini(resource.path)-#/Общедоступные
-templates = #-ini(resource.path)-#/Шаблоны
-videos = #-ini(resource.path)-#/Видео
-#os_locale_lang#
-#?os_locale_lang==ru_RU&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Рабочий стол
-#os_locale_lang#
-
-#?os_locale_lang==uk_UA#
-documents = #-ini(resource.path)-#/Документи
-downloads = #-ini(resource.path)-#/Завантаження
-music = #-ini(resource.path)-#/Музика
-pictures = #-ini(resource.path)-#/Картинки
-public = #-ini(resource.path)-#/Загальнодоступні
-templates = #-ini(resource.path)-#/Шаблони
-videos = #-ini(resource.path)-#/Відео
-#os_locale_lang#
-#?os_locale_lang==uk_UA&&ur_domain_set==off#
-desktop = /home/#-ur_login-#/Робочий стіл
-#os_locale_lang#
-
diff --git desktop/desktop/00-patch-2.2.0.16/ini.env desktop/desktop/00-patch-2.2.0.16/ini.env
deleted file mode 100644
index cccccca..0000000
--- desktop/desktop/00-patch-2.2.0.16/ini.env
+++ /dev/null
@@ -1,7 +0,0 @@
-# Calculate format=samba path=~/.calculate
-
-[ver]
-patch = 2.2.0.16
-
-[update]
-patch = on
diff --git desktop/desktop/00-patch-2.2.10/.calculate_directory desktop/desktop/00-patch-2.2.10/.calculate_directory
index d44b259..9ea3814 100644
--- desktop/desktop/00-patch-2.2.10/.calculate_directory
+++ desktop/desktop/00-patch-2.2.10/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.10
+# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.10
diff --git desktop/desktop/00-patch-2.2.12/.calculate_directory desktop/desktop/00-patch-2.2.12/.calculate_directory
index 92513a1..3f135e2 100644
--- desktop/desktop/00-patch-2.2.12/.calculate_directory
+++ desktop/desktop/00-patch-2.2.12/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.12
+# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.12
diff --git desktop/desktop/00-patch-2.2.13/.calculate_directory desktop/desktop/00-patch-2.2.13/.calculate_directory
index ea391de..81943f4 100644
--- desktop/desktop/00-patch-2.2.13/.calculate_directory
+++ desktop/desktop/00-patch-2.2.13/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.13
+# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.13
diff --git desktop/desktop/00-patch-2.2.22/.calculate_directory desktop/desktop/00-patch-2.2.22/.calculate_directory
new file mode 100644
index 0000000..9259f63
--- /dev/null
+++ desktop/desktop/00-patch-2.2.22/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(calculate-templates)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.22
diff --git desktop/desktop/00-patch-2.2.22/ini-patch.env desktop/desktop/00-patch-2.2.22/ini-patch.env
new file mode 100644
index 0000000..6b18c0f
--- /dev/null
+++ desktop/desktop/00-patch-2.2.22/ini-patch.env
@@ -0,0 +1,11 @@
+# Calculate format=samba path=~/.calculate name=ini.env ini(ver.kde)!=
+
+[ver]
+kopete = 4.6
+kwallet = 4.6
+plasma = 4.6
+
+[update]
+kopete = on
+kwallet = on
+plasma = on
diff --git desktop/desktop/00-patch-2.2.22/ini.env desktop/desktop/00-patch-2.2.22/ini.env
new file mode 100644
index 0000000..7f8174a
--- /dev/null
+++ desktop/desktop/00-patch-2.2.22/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba path=~/.calculate
+
+[ver]
+patch = 2.2.22
+
+[update]
+patch = on
diff --git desktop/desktop/00-patch-2.2.6/.calculate_directory desktop/desktop/00-patch-2.2.6/.calculate_directory
deleted file mode 100644
index f1b5891..0000000
--- desktop/desktop/00-patch-2.2.6/.calculate_directory
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.6
diff --git desktop/desktop/00-patch-2.2.6/audacious2-gtkui.desktop desktop/desktop/00-patch-2.2.6/audacious2-gtkui.desktop
deleted file mode 100644
index 241f475..0000000
--- desktop/desktop/00-patch-2.2.6/audacious2-gtkui.desktop
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate append=remove path=~/.local/share/applications
diff --git desktop/desktop/00-patch-2.2.6/audacious2.desktop desktop/desktop/00-patch-2.2.6/audacious2.desktop
deleted file mode 100644
index 241f475..0000000
--- desktop/desktop/00-patch-2.2.6/audacious2.desktop
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate append=remove path=~/.local/share/applications
diff --git desktop/desktop/00-patch-2.2.6/ini.env desktop/desktop/00-patch-2.2.6/ini.env
deleted file mode 100644
index c166855..0000000
--- desktop/desktop/00-patch-2.2.6/ini.env
+++ /dev/null
@@ -1,7 +0,0 @@
-# Calculate format=samba path=~/.calculate
-
-[ver]
-patch = 2.2.6
-
-[update]
-patch = on
diff --git desktop/desktop/00-patch-2.2.7/.calculate_directory desktop/desktop/00-patch-2.2.7/.calculate_directory
deleted file mode 100644
index f06ef76..0000000
--- desktop/desktop/00-patch-2.2.7/.calculate_directory
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.7
diff --git desktop/desktop/00-patch-2.2.7/Xcursor.xrdb desktop/desktop/00-patch-2.2.7/Xcursor.xrdb
deleted file mode 100644
index b3c8ab0..0000000
--- desktop/desktop/00-patch-2.2.7/Xcursor.xrdb
+++ /dev/null
@@ -1,2 +0,0 @@
-# Calculate append=replace path=~/.config/xfce4 ini(ver.xfce)!=
-Xcursor.theme: neutral
diff --git desktop/desktop/00-patch-2.2.7/ini.env desktop/desktop/00-patch-2.2.7/ini.env
deleted file mode 100644
index 5cf6df3..0000000
--- desktop/desktop/00-patch-2.2.7/ini.env
+++ /dev/null
@@ -1,7 +0,0 @@
-# Calculate format=samba path=~/.calculate
-
-[ver]
-patch = 2.2.7
-
-[update]
-patch = on
diff --git desktop/desktop/00-patch-2.2.7/xsettings.xml desktop/desktop/00-patch-2.2.7/xsettings.xml
deleted file mode 100644
index 54ae4a0..0000000
--- desktop/desktop/00-patch-2.2.7/xsettings.xml
+++ /dev/null
@@ -1,8 +0,0 @@
-# Calculate path=~/.config/xfce4/xfconf/xfce-perchannel-xml format=xml_xfce ini(ver.xfce)!=
-<?xml version="1.0" encoding="UTF-8"?>
-
-<channel name="xsettings" version="1.0">
-  <property name="Gtk" type="empty">
-    <property name="CursorThemeName" type="string" value="neutral"/>
-  </property>
-</channel>
diff --git desktop/desktop/00-patch-2.2.8/.calculate_directory desktop/desktop/00-patch-2.2.8/.calculate_directory
deleted file mode 100644
index ef8044f..0000000
--- desktop/desktop/00-patch-2.2.8/.calculate_directory
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.8
diff --git desktop/desktop/00-patch-2.2.8/10000000043.desktop desktop/desktop/00-patch-2.2.8/10000000043.desktop
deleted file mode 100644
index 4e116f5..0000000
--- desktop/desktop/00-patch-2.2.8/10000000043.desktop
+++ /dev/null
@@ -1,3 +0,0 @@
-# Calculate format=kde ini(ver.xfce)!= link=/usr/share/applications/leafpad.desktop path=~/.config/xfce4/panel/launcher-43
-[Desktop Entry]
-X-XFCE-Source=file:///usr/share/applications/leafpad.desktop
diff --git desktop/desktop/00-patch-2.2.8/ini.env desktop/desktop/00-patch-2.2.8/ini.env
deleted file mode 100644
index 35b0f29..0000000
--- desktop/desktop/00-patch-2.2.8/ini.env
+++ /dev/null
@@ -1,7 +0,0 @@
-# Calculate format=samba path=~/.calculate
-
-[ver]
-patch = 2.2.8
-
-[update]
-patch = on
diff --git desktop/desktop/00-patch-2.2.8/kcminputrc desktop/desktop/00-patch-2.2.8/kcminputrc
deleted file mode 100644
index 059b0d8..0000000
--- desktop/desktop/00-patch-2.2.8/kcminputrc
+++ /dev/null
@@ -1,8 +0,0 @@
-# Calculate format=kde ini(ver.kde)!= path=~/.kde4/share/config
-[$Version]
-#для отмены сброса настройки cursorTheme
-update_info=mouse_cursor_theme.upd:kde3.4.99
-
-[Mouse]
-#тема курсоров мыши
-cursorTheme=neutral
diff --git desktop/desktop/00-patch-2.2.8/kcminputrc-bad desktop/desktop/00-patch-2.2.8/kcminputrc-bad
deleted file mode 100644
index b9e0ea2..0000000
--- desktop/desktop/00-patch-2.2.8/kcminputrc-bad
+++ /dev/null
@@ -1,2 +0,0 @@
-# Calculate ini(ver.kde)!= append=remove name=kcminputrc
-
diff --git desktop/desktop/00-patch-2.2.8/leafpadrc desktop/desktop/00-patch-2.2.8/leafpadrc
deleted file mode 100644
index 62d9e78..0000000
--- desktop/desktop/00-patch-2.2.8/leafpadrc
+++ /dev/null
@@ -1,8 +0,0 @@
-# Calculate ini(ver.xfce)!= path=~/.config/leafpad
-0.8.17
-600
-400
-Droid Sans Mono 10
-1
-0
-0
diff --git desktop/desktop/00-patch-2.2.8/xfce4-panel.xml desktop/desktop/00-patch-2.2.8/xfce4-panel.xml
deleted file mode 100644
index 8eeac10..0000000
--- desktop/desktop/00-patch-2.2.8/xfce4-panel.xml
+++ /dev/null
@@ -1,12 +0,0 @@
-# Calculate format=xml_xfce ini(ver.xfce)!= path=~/.config/xfce4/xfconf/xfce-perchannel-xml
-<?xml version="1.0" encoding="UTF-8"?>
-
-<channel name="xfce4-panel" version="1.0">
-  <property name="plugins" type="empty">
-    <property name="plugin-43" type="string" value="launcher">
-      <property name="items" type="array">
-        <value type="string" value="10000000043.desktop"/>
-      </property>
-    </property>
-  </property>
-</channel>
diff --git desktop/desktop/00-patch-2.2.9/.calculate_directory desktop/desktop/00-patch-2.2.9/.calculate_directory
deleted file mode 100644
index 4a1be37..0000000
--- desktop/desktop/00-patch-2.2.9/.calculate_directory
+++ /dev/null
@@ -1 +0,0 @@
-# Calculate append=skip belong(calculate-desktop)!=&&ini(update.patch)!=off&&ini(ver.patch)<2.2.9
diff --git desktop/desktop/00-patch-2.2.9/dolphinrc desktop/desktop/00-patch-2.2.9/dolphinrc
deleted file mode 100644
index 875c51e..0000000
--- desktop/desktop/00-patch-2.2.9/dolphinrc
+++ /dev/null
@@ -1,4 +0,0 @@
-# Calculate format=kde ini(ver.kde)!= path=~/.kde4/share/config
-[MainWindow]
-ToolBarsMovable=Disabled
-
diff --git desktop/desktop/00-patch-2.2.9/ini.env desktop/desktop/00-patch-2.2.9/ini.env
deleted file mode 100644
index 3f00b77..0000000
--- desktop/desktop/00-patch-2.2.9/ini.env
+++ /dev/null
@@ -1,7 +0,0 @@
-# Calculate format=samba path=~/.calculate
-
-[ver]
-patch = 2.2.9
-
-[update]
-patch = on
diff --git desktop/desktop/00-patch-2.2.9/plasmarc desktop/desktop/00-patch-2.2.9/plasmarc
deleted file mode 100644
index b9a1272..0000000
--- desktop/desktop/00-patch-2.2.9/plasmarc
+++ /dev/null
@@ -1,4 +0,0 @@
-# Calculate format=plasma ini(ver.kde)!= path=~/.kde4/share/config
-[Theme]
-name=slim-glow
-
diff --git desktop/desktop/00-patch-2.2.9/send-attach.desktop desktop/desktop/00-patch-2.2.9/send-attach.desktop
deleted file mode 100644
index 9590ed5..0000000
--- desktop/desktop/00-patch-2.2.9/send-attach.desktop
+++ /dev/null
@@ -1,21 +0,0 @@
-# Calculate format=kde ini(ver.kde)!= path=~/.kde4/share/kde4/services/ServiceMenu
-[Desktop Entry]
-Type=Service
-X-KDE-ServiceTypes=KonqPopupMenu/Plugin,all/allfiles
-Actions=openKmailAttachment
-Encoding=UTF-8
-MimeType=application/octet-stream
-ExcludeServiceTypes=application/x-iso,kdedevice/*,inode/directory
-
-[Desktop Action openKmailAttachment]
-Name=Send File As Attachment
-Name[de]=Datei als Anhang versenden
-Name[en]=Send File As Attachment
-Name[fr]=Envoyer le fichier en pièce jointe
-Name[pt]=Enviar por correio-e
-Name[sk]=Poslať e-mailom
-Name[ru]=Отправить в KMail
-
-Icon=kmail
-Exec=kmail --composer --attach %F `echo attachment=\'file://%F\' | sed 's/ \//,file:\/\/\//g'`
-X-Ubuntu-Gettext-Domain=desktop_kdebase
diff --git desktop/desktop/20-kde-4.6/.kde4/share/apps/kopete/kopetechatwindow.rc desktop/desktop/20-kde-4.6/.kde4/share/apps/kopete/kopetechatwindow.rc
deleted file mode 100644
index 4708080..0000000
--- desktop/desktop/20-kde-4.6/.kde4/share/apps/kopete/kopetechatwindow.rc
+++ /dev/null
@@ -1,7 +0,0 @@
-<!DOCTYPE kpartgui>
-<kpartgui version="34" name="kopetechatwindow" >
- <ActionProperties>
-  <Action shortcut="Ctrl+Return" name="chat_send" />
-  <Action shortcut="Esc" name="file_quit" />
- </ActionProperties>
-</kpartgui>
diff --git desktop/desktop/20-kde-4.6/.kde4/share/apps/kopete/kopeteemailwindow.rc desktop/desktop/20-kde-4.6/.kde4/share/apps/kopete/kopeteemailwindow.rc
deleted file mode 100644
index 5f7c52b..0000000
--- desktop/desktop/20-kde-4.6/.kde4/share/apps/kopete/kopeteemailwindow.rc
+++ /dev/null
@@ -1,7 +0,0 @@
-<!DOCTYPE kpartgui>
-<kpartgui version="16" name="kopeteemailwindow" >
- <ActionProperties>
-  <Action shortcut="Ctrl+Return" name="chat_send" />
-  <Action shortcut="Esc" name="file_quit" />
- </ActionProperties>
-</kpartgui>
diff --git desktop/desktop/20-kde-4.6/.kde4/share/apps/kwallet/kdewallet.kwl desktop/desktop/20-kde-4.6/.kde4/share/apps/kwallet/kdewallet.kwl
deleted file mode 100644
index 3e8774c..0000000
Binary files desktop/desktop/20-kde-4.6/.kde4/share/apps/kwallet/kdewallet.kwl and /dev/null differ
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
index 4015931..150bd84 100644
--- desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
+++ desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
@@ -5,15 +5,7 @@ emoticonsTheme=KMess-Cartoon
 
 #Fonts
 [General]
-#?pkg(www-client/chromium)!=#
-BrowserApplication[$e]=file:///usr/share/applications/chromium-chromium.desktop
-#pkg#
-#?pkg(www-client/firefox)!=&&pkg(www-client/chromium)==#
-BrowserApplication[$e]=file:///home/#-ur_login-#/.local/share/applications/mozilla-firefox.desktop
-#pkg#
-#?pkg(www-client/firefox)==&&pkg(www-client/chromium)==#
-BrowserApplication[$e]=file:///usr/share/applications/kde4/konqbrowser.desktop
-#pkg#
+BrowserApplication[$e]=file:///usr/share/applications/calculate-browser.desktop
 XftHintStyle=hintmedium
 #?pkg(media-fonts/droid)!=#
 desktopFont=Droid Sans,10,-1,5,50,0,0,0,0,0
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/kopete.notifyrc desktop/desktop/20-kde-4.6/.kde4/share/config/kopete.notifyrc
deleted file mode 100644
index 0e256cb..0000000
--- desktop/desktop/20-kde-4.6/.kde4/share/config/kopete.notifyrc
+++ /dev/null
@@ -1,9 +0,0 @@
-# Calculate format=kde
-[Event/kopete_contact_incoming]
-#реакция на входящее сообщение
-Action=Sound|Popup|Taskbar
-
-[Event/kopete_contact_online]
-#реакция когда пользователь вошел в сеть
-Action=
-
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/kopeterc desktop/desktop/20-kde-4.6/.kde4/share/config/kopeterc
deleted file mode 100644
index cb33211..0000000
--- desktop/desktop/20-kde-4.6/.kde4/share/config/kopeterc
+++ /dev/null
@@ -1,89 +0,0 @@
-# Calculate format=kde append=replace
-#?env(client.os_remote_auth)!=&&ur_domain_set==on&&server(jabber.host)!=#
-[Account_JabberProtocol_#-ur_jid-#]
-AccountId=#-ur_jid-#
-AllowPlainTextPassword=true
-CustomServer=true
-Port=#-server(jabber.port)-#
-Protocol=JabberProtocol
-RememberPassword=true
-Server=#-server(jabber.host)-#
-UseSSL=true
-#env#
-
-[Behavior]
-#подключаться автоматически
-#?pkg(kde-base/kopete)<4.2#
-autoConnect=true
-#pkg#
-#?pkg(kde-base/kopete)>=4.2#
-initialStatus=Online
-#pkg#
-
-[ContactList]
-#показывать фотографии пользователей (если есть)
-contactListIconMode=IconPhoto
-#показывать пользователей не в сети
-showOfflineUsers=false
-
-[General Options]
-#размер окна
-Geometry=230,600
-#фиксация панели инструментов
-#ToolBarsMovable=Disabled
-#отображение строки меню
-MenuBar=Disabled
-#статусная строка свернута
-ShownIdentityId=
-
-#[KopeteChatWindowIndividualMode][Toolbar mainToolBar]
-#отображение иконок в окне диалога
-#ToolButtonStyle=IconOnly
-
-#размер окна чата
-[KopeteChatWindowIndividualMode]
-#положение разделителя истории
-ChatViewSplitter=AAAA/wAAAAAAAAACAAAAlQAAAFMAAAAAAwEAAAAC
-Height 768=400
-Height 800=400
-Height 900=400
-Height 1024=400
-Height 1050=400
-Height 1200=400
-Width 1024=500
-Width 1280=500
-Width 1440=500
-Width 1600=500
-Width 1680=500
-Width 1920=500
-
-[MainWindow]
-Height 768=400
-Height 800=400
-Height 900=400
-Height 1024=400
-Height 1050=400
-Height 1200=400
-Width 1024=500
-Width 1280=500
-Width 1440=500
-Width 1600=500
-Width 1680=500
-Width 1920=500
-
-[Notification Messages]
-#?env(client.os_remote_auth)!=&&ur_domain_set==on&&server(jabber.host)!=#
-#выдавать предупреждение при проверке сертификата защищенного соединения
-KopeteTLSWarning#-ur_jid_host-#InvalidCertSelfSigned=false
-#env#
-#предупреждать что программа остается запущенной при закрытии главного окна
-hideOnCloseInfo=false
-
-[QuickSearchBar Settings]
-#скрытие панели поиска
-Hidden=true
-
-[ToolBar Settings]
-#отображение иконок в окне приложения
-ToolButtonStyle=IconOnly
-
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/kwalletrc desktop/desktop/20-kde-4.6/.kde4/share/config/kwalletrc
deleted file mode 100644
index 02ee787..0000000
--- desktop/desktop/20-kde-4.6/.kde4/share/config/kwalletrc
+++ /dev/null
@@ -1,10 +0,0 @@
-# Calculate format=kde
-#applications that access is allowed
-[Auto Allow]
-kdewallet=Kopete,KMail,Konqueror,Chrome,Network Management,KDE Daemon,KDE-Dienst,Servicio de KDE,Démon de KDE,Demone KDE,Servidor do KDE,Serviço do KDE,Demon KDE,Служба KDE,Фонова служба KDE
-
-[Wallet]
-Close When Idle=false
-Enabled=true
-First Use=false
-Use One Wallet=true
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/plasma-desktop-appletsrc desktop/desktop/20-kde-4.6/.kde4/share/config/plasma-desktop-appletsrc
deleted file mode 100644
index 672f36e..0000000
--- desktop/desktop/20-kde-4.6/.kde4/share/config/plasma-desktop-appletsrc
+++ /dev/null
@@ -1,358 +0,0 @@
-# Calculate format=plasma append=replace
-#------------------------------------------------------------------------------
-#рабочий стол
-#------------------------------------------------------------------------------
-[AppletGlobals][plasma_applet_pager]
-rows=1
-
-[Containments][#-sum(cont,1)-#]
-activity=
-alignToGrid=true
-customIconSize=48
-desktop=0
-filter=0
-filterFiles=*
-formfactor=0
-geometry=0,0,#-os_x11_width-#,#-os_x11_height-#
-iconsLocked=false
-immutability=1
-location=0
-mimeFilter=
-plugin=folderview
-screen=0
-url=desktop:/
-wallpaperplugin=image
-wallpaperpluginmode=SingleImage
-#-sum(iconx,,10)-#
-#-sum(icony,,10)-#
-#-sum(iconcx,,112)-#
-#-sum(iconcy,,100)-#
-#-sum(iconcount,,2)-#
-savedPositions=1,7,Home.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,Share.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,FTP.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,#-sum(iconx,,10)-##-sum(icony,,icony+iconcy)-#calculate-install.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,partitionmanager.desktop,#-sum(iconx,iconx,10)-#,#-sum(icony,icony,icony+iconcy)-#,README.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,calculate-community.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#
-zvalue=0
-sortColumn=-1
-
-[Containments][#-sum(cont,cont)-#][Wallpaper][image]
-slideTimer=600
-slidepaths=/usr/share/wallpapers/
-userswallpapers=
-wallpaper=/usr/share/wallpapers/Calculate_Linux
-wallpapercolor=56,111,150
-wallpaperposition=0
-
-#------------------------------------------------------------------------------
-#верхняя панель
-#------------------------------------------------------------------------------
-#-sum(panel,,31)-#
-#-sum(apporder,,0)-#
-
-[Containments][-#-sum(cont,cont+1)-#]
-desktop=-1
-formfactor=2
-geometry=0,0,#-os_x11_width-#,#-sum(panel,panel)-#
-immutability=1
-location=3
-plugin=panel
-screen=0
-zvalue=150
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,cont*100)-#]
-immutability=1
-plugin=simplelauncher
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-icon=kde
-#?env(client.os_remote_auth)==||ur_domain_set==off#
-views=Applications,RunCommand,Leave
-#env#
-#?env(client.os_remote_auth)!=&&ur_domain_set==on#
-views=Applications,RunCommand,SwitchUser,LockScreen,Logout
-#env#
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder)-#
-
-#?env(client.os_remote_auth)==||ur_domain_set==off#
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///home/#-ur_login-#
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-#env#
-
-#?env(client.os_remote_auth)!=&&ur_domain_set==on#
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///home/#-ur_login-#/Home
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///home/#-ur_login-#/Share
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-#env#
-
-#?env(client.os_remote_auth)!=&&ur_domain_set==on&&server(ftp.host)!=#
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///home/#-ur_login-#/FTP
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-#env#
-#?server(ftp.host)==#
-[Containments][#-sum(cont,cont)-#][Applets][!#-sum(app,app+1)-#]
-#-sum(apporder,,apporder+1)-#
-#server#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=pager
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-#показывать номер рабочего стола - 0
-displayedText=0
-#отображать значки
-showWindowIcons=true
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=tasks
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-#группирование
-groupingStrategy=0
-#максимум строк
-maxRows=1
-#показывать задачи только с текущего рабочего стола
-showOnlyCurrentDesktop=true
-#сортировка
-sortingStrategy=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=trash
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][!#-sum(app,app+1)-#]
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=systemtray
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-hidden=kwalletmanager
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=digital-clock
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][999]
-immutability=1
-plugin=lockout
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][999][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Configuration]
-maximumSize=#-os_x11_width-#,#-sum(panel,panel)-#
-minimumSize=#-os_x11_width-#,#-sum(panel,panel)-#
-
-#------------------------------------------------------------------------------
-#нижняя панель
-#------------------------------------------------------------------------------
-
-#-sum(height,,44)-#
-#?hr_cdrom_set==on#
-#-sum(width,,height*14)-#
-#hr_cdrom_set#
-#?hr_cdrom_set==off#
-#-sum(width,,height*13)-#
-#hr_cdrom_set#
-#-sum(apporder,,0)-#
-
-#?pkg(net-irc/konversation)!=#
-#-sum(width,,width+height)-#
-#pkg#
-
-[Containments][-#-sum(cont,cont+1)-#]
-desktop=-1
-formfactor=2
-geometry=0,0,#-sum(width,width)-#,#-sum(height,height)-#
-immutability=1
-location=4
-plugin=panel
-screen=0
-zvalue=150
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,cont*100)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-browser.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-mail.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-#?pkg(net-irc/konversation)!=#
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-chat.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-#pkg#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-im.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-writer.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-calc.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-calculator.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-imageedit.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-imageview.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-audioplayer.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-videoplayer.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-#?hr_cdrom_set==on#
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-imageburn.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-#hr_cdrom_set#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-dict.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-textedit.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
-immutability=1
-plugin=icon
-zvalue=0
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
-Url=file:///usr/share/applications/calculate-terminal.desktop
-[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
-Order=#-sum(apporder,apporder+1)-#
-
-[General]
-immutability=2
-
-[Containments][#-sum(cont,cont)-#][Configuration]
-maximumSize=#-sum(width,width)-#,#-sum(height,height)-#
-minimumSize=#-sum(width,width)-#,#-sum(height,height)-#
-
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/plasma-desktoprc desktop/desktop/20-kde-4.6/.kde4/share/config/plasma-desktoprc
deleted file mode 100644
index da6d215..0000000
--- desktop/desktop/20-kde-4.6/.kde4/share/config/plasma-desktoprc
+++ /dev/null
@@ -1,21 +0,0 @@
-# Calculate format=plasma
-#расположение панелей
-[PlasmaViews][1]
-Alignment=132
-Offset=0
-panelVisibility=1
-
-[PlasmaViews][2]
-Alignment=132
-Offset=0
-panelVisibility=1
-
-[PlasmaViews][3]
-Offset=0
-panelVisibility=0
-
-[ViewIds]
-1=1
-2=3
-3=2
-
diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/plasmarc desktop/desktop/20-kde-4.6/.kde4/share/config/plasmarc
deleted file mode 100644
index 133c3ab..0000000
--- desktop/desktop/20-kde-4.6/.kde4/share/config/plasmarc
+++ /dev/null
@@ -1,5 +0,0 @@
-# Calculate format=plasma
-#оформление
-[Theme]
-name=slim-glow
-
diff --git desktop/desktop/20-kopete-4.6/.calculate_directory desktop/desktop/20-kopete-4.6/.calculate_directory
new file mode 100644
index 0000000..03e969f
--- /dev/null
+++ desktop/desktop/20-kopete-4.6/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(kopete)!=&&ini(update.kopete)!=off&&ini(ver.kopete)==&&pkg(kde-base/kopete)>=4.6
diff --git desktop/desktop/20-kopete-4.6/ini.env desktop/desktop/20-kopete-4.6/ini.env
new file mode 100644
index 0000000..f4970fe
--- /dev/null
+++ desktop/desktop/20-kopete-4.6/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba path=~/.calculate
+
+[ver]
+kopete = 4.6
+
+[update]
+kopete = on
diff --git desktop/desktop/20-kopete-4.6/kopete.notifyrc desktop/desktop/20-kopete-4.6/kopete.notifyrc
new file mode 100644
index 0000000..21911c8
--- /dev/null
+++ desktop/desktop/20-kopete-4.6/kopete.notifyrc
@@ -0,0 +1,10 @@
+# Calculate format=kde path=~/.kde4/share/config
+
+[Event/kopete_contact_incoming]
+# response to an incoming message
+Action=Sound|Popup|Taskbar
+
+[Event/kopete_contact_online]
+# reaction to the birth of a user in the online
+Action=
+
diff --git desktop/desktop/20-kopete-4.6/kopetechatwindow.rc desktop/desktop/20-kopete-4.6/kopetechatwindow.rc
new file mode 100644
index 0000000..0ad4d8d
--- /dev/null
+++ desktop/desktop/20-kopete-4.6/kopetechatwindow.rc
@@ -0,0 +1,8 @@
+# Calculate path=~/.kde4/share/apps/kopete
+<!DOCTYPE kpartgui>
+<kpartgui version="34" name="kopetechatwindow" >
+ <ActionProperties>
+  <Action shortcut="Ctrl+Return" name="chat_send" />
+  <Action shortcut="Esc" name="file_quit" />
+ </ActionProperties>
+</kpartgui>
diff --git desktop/desktop/20-kopete-4.6/kopeteemailwindow.rc desktop/desktop/20-kopete-4.6/kopeteemailwindow.rc
new file mode 100644
index 0000000..c7dfbff
--- /dev/null
+++ desktop/desktop/20-kopete-4.6/kopeteemailwindow.rc
@@ -0,0 +1,8 @@
+# Calculate path=~/.kde4/share/apps/kopete
+<!DOCTYPE kpartgui>
+<kpartgui version="16" name="kopeteemailwindow" >
+ <ActionProperties>
+  <Action shortcut="Ctrl+Return" name="chat_send" />
+  <Action shortcut="Esc" name="file_quit" />
+ </ActionProperties>
+</kpartgui>
diff --git desktop/desktop/20-kopete-4.6/kopeterc desktop/desktop/20-kopete-4.6/kopeterc
new file mode 100644
index 0000000..d888cfa
--- /dev/null
+++ desktop/desktop/20-kopete-4.6/kopeterc
@@ -0,0 +1,86 @@
+# Calculate format=kde append=replace path=~/.kde4/share/config
+
+#?env(client.os_remote_auth)!=&&ur_domain_set==on&&server(jabber.host)!=#
+[Account_JabberProtocol_#-ur_jid-#]
+AccountId=#-ur_jid-#
+AllowPlainTextPassword=true
+CustomServer=true
+Port=#-server(jabber.port)-#
+Protocol=JabberProtocol
+RememberPassword=true
+Server=#-server(jabber.host)-#
+UseSSL=true
+#env#
+
+[Behavior]
+# automatically connect
+#?pkg(kde-base/kopete)<4.2#
+autoConnect=true
+#pkg#
+#?pkg(kde-base/kopete)>=4.2#
+initialStatus=Online
+#pkg#
+
+[ContactList]
+contactListIconMode=IconPhoto
+showOfflineUsers=false
+
+[General Options]
+Geometry=230,600
+# fixation toolbar
+#ToolBarsMovable=Disabled
+MenuBar=Disabled
+# the status bar is minimized
+ShownIdentityId=
+
+#[KopeteChatWindowIndividualMode][Toolbar mainToolBar]
+# map icon in the dialog box
+#ToolButtonStyle=IconOnly
+
+# size of the chat window
+[KopeteChatWindowIndividualMode]
+# position of the splitter stories
+ChatViewSplitter=AAAA/wAAAAAAAAACAAAAlQAAAFMAAAAAAwEAAAAC
+Height 768=400
+Height 800=400
+Height 900=400
+Height 1024=400
+Height 1050=400
+Height 1200=400
+Width 1024=500
+Width 1280=500
+Width 1440=500
+Width 1600=500
+Width 1680=500
+Width 1920=500
+
+[MainWindow]
+Height 768=400
+Height 800=400
+Height 900=400
+Height 1024=400
+Height 1050=400
+Height 1200=400
+Width 1024=500
+Width 1280=500
+Width 1440=500
+Width 1600=500
+Width 1680=500
+Width 1920=500
+
+[Notification Messages]
+#?env(client.os_remote_auth)!=&&ur_domain_set==on&&server(jabber.host)!=#
+# issue a warning when verifying a certificate protected connection
+KopeteTLSWarning#-ur_jid_host-#InvalidCertSelfSigned=false
+#env#
+# warn that the program is running when you close the main window
+hideOnCloseInfo=false
+
+[QuickSearchBar Settings]
+# hide search bar
+Hidden=true
+
+[ToolBar Settings]
+# display icons in the application
+ToolButtonStyle=IconOnly
+
diff --git desktop/desktop/20-kwallet-4.6/.calculate_directory desktop/desktop/20-kwallet-4.6/.calculate_directory
new file mode 100644
index 0000000..b2d6a13
--- /dev/null
+++ desktop/desktop/20-kwallet-4.6/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(kwallet)!=&&ini(update.kwallet)!=off&&ini(ver.kwallet)==&&pkg(kde-base/kwallet)>=4.6
diff --git desktop/desktop/20-kwallet-4.6/ini.env desktop/desktop/20-kwallet-4.6/ini.env
new file mode 100644
index 0000000..713626c
--- /dev/null
+++ desktop/desktop/20-kwallet-4.6/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba path=~/.calculate
+
+[ver]
+kwallet = 4.6
+
+[update]
+kwallet = on
diff --git desktop/desktop/20-kwallet-4.6/kwallet/.calculate_directory desktop/desktop/20-kwallet-4.6/kwallet/.calculate_directory
new file mode 100644
index 0000000..ab3ea29
--- /dev/null
+++ desktop/desktop/20-kwallet-4.6/kwallet/.calculate_directory
@@ -0,0 +1 @@
+# Calculate path=~/.kde4/share/apps
diff --git desktop/desktop/20-kwallet-4.6/kwallet/kdewallet.kwl desktop/desktop/20-kwallet-4.6/kwallet/kdewallet.kwl
new file mode 100644
index 0000000..3e8774c
Binary files /dev/null and desktop/desktop/20-kwallet-4.6/kwallet/kdewallet.kwl differ
diff --git desktop/desktop/20-kwallet-4.6/kwalletrc desktop/desktop/20-kwallet-4.6/kwalletrc
new file mode 100644
index 0000000..5243adc
--- /dev/null
+++ desktop/desktop/20-kwallet-4.6/kwalletrc
@@ -0,0 +1,11 @@
+# Calculate format=kde path=~/.kde4/share/config
+
+# Applications that access is allowed
+[Auto Allow]
+kdewallet=Kopete,KMail,Konqueror,Chrome,Network Management,KDE Daemon,KDE-Dienst,Servicio de KDE,Démon de KDE,Demone KDE,Servidor do KDE,Serviço do KDE,Demon KDE,Служба KDE,Фонова служба KDE
+
+[Wallet]
+Close When Idle=false
+Enabled=true
+First Use=false
+Use One Wallet=true
diff --git desktop/desktop/20-plasma-4.6/.calculate_directory desktop/desktop/20-plasma-4.6/.calculate_directory
new file mode 100644
index 0000000..1eed3c1
--- /dev/null
+++ desktop/desktop/20-plasma-4.6/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(plasma-workspace)!=&&ini(update.plasma)!=off&&ini(ver.plasma)==&&pkg(kde-base/plasma-workspace)>=4.6
diff --git desktop/desktop/20-plasma-4.6/ini.env desktop/desktop/20-plasma-4.6/ini.env
new file mode 100644
index 0000000..73be5bf
--- /dev/null
+++ desktop/desktop/20-plasma-4.6/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba path=~/.calculate
+
+[ver]
+plasma = 4.6
+
+[update]
+plasma = on
diff --git desktop/desktop/20-plasma-4.6/plasma-desktop-appletsrc desktop/desktop/20-plasma-4.6/plasma-desktop-appletsrc
new file mode 100644
index 0000000..d4799d6
--- /dev/null
+++ desktop/desktop/20-plasma-4.6/plasma-desktop-appletsrc
@@ -0,0 +1,358 @@
+# Calculate format=plasma append=replace path=~/.kde4/share/config
+#------------------------------------------------------------------------------
+#рабочий стол
+#------------------------------------------------------------------------------
+[AppletGlobals][plasma_applet_pager]
+rows=1
+
+[Containments][#-sum(cont,1)-#]
+activity=
+alignToGrid=true
+customIconSize=48
+desktop=0
+filter=0
+filterFiles=*
+formfactor=0
+geometry=0,0,#-os_x11_width-#,#-os_x11_height-#
+iconsLocked=false
+immutability=1
+location=0
+mimeFilter=
+plugin=folderview
+screen=0
+url=desktop:/
+wallpaperplugin=image
+wallpaperpluginmode=SingleImage
+#-sum(iconx,,10)-#
+#-sum(icony,,10)-#
+#-sum(iconcx,,112)-#
+#-sum(iconcy,,100)-#
+#-sum(iconcount,,2)-#
+savedPositions=1,7,Home.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,Share.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,FTP.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,#-sum(iconx,,10)-##-sum(icony,,icony+iconcy)-#calculate-install.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,partitionmanager.desktop,#-sum(iconx,iconx,10)-#,#-sum(icony,icony,icony+iconcy)-#,README.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#,calculate-community.desktop,#-sum(iconx,iconx,iconx+iconcx)-#,#-sum(icony,icony)-#
+zvalue=0
+sortColumn=-1
+
+[Containments][#-sum(cont,cont)-#][Wallpaper][image]
+slideTimer=600
+slidepaths=/usr/share/wallpapers/
+userswallpapers=
+wallpaper=/usr/share/wallpapers/Calculate_Linux
+wallpapercolor=56,111,150
+wallpaperposition=0
+
+#------------------------------------------------------------------------------
+#верхняя панель
+#------------------------------------------------------------------------------
+#-sum(panel,,31)-#
+#-sum(apporder,,0)-#
+
+[Containments][-#-sum(cont,cont+1)-#]
+desktop=-1
+formfactor=2
+geometry=0,0,#-os_x11_width-#,#-sum(panel,panel)-#
+immutability=1
+location=3
+plugin=panel
+screen=0
+zvalue=150
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,cont*100)-#]
+immutability=1
+plugin=simplelauncher
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+icon=kde
+#?env(client.os_remote_auth)==||ur_domain_set==off#
+views=Applications,RunCommand,Leave
+#env#
+#?env(client.os_remote_auth)!=&&ur_domain_set==on#
+views=Applications,RunCommand,SwitchUser,LockScreen,Logout
+#env#
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder)-#
+
+#?env(client.os_remote_auth)==||ur_domain_set==off#
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///home/#-ur_login-#
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+#env#
+
+#?env(client.os_remote_auth)!=&&ur_domain_set==on#
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///home/#-ur_login-#/Home
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///home/#-ur_login-#/Share
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+#env#
+
+#?env(client.os_remote_auth)!=&&ur_domain_set==on&&server(ftp.host)!=#
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///home/#-ur_login-#/FTP
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+#env#
+#?server(ftp.host)==#
+[Containments][#-sum(cont,cont)-#][Applets][!#-sum(app,app+1)-#]
+#-sum(apporder,,apporder+1)-#
+#server#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=pager
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+#показывать номер рабочего стола - 0
+displayedText=0
+#отображать значки
+showWindowIcons=true
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=tasks
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+#группирование
+groupingStrategy=0
+#максимум строк
+maxRows=1
+#показывать задачи только с текущего рабочего стола
+showOnlyCurrentDesktop=true
+#сортировка
+sortingStrategy=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=trash
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][!#-sum(app,app+1)-#]
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=systemtray
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+hidden=kwalletmanager
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=digital-clock
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][999]
+immutability=1
+plugin=lockout
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][999][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Configuration]
+maximumSize=#-os_x11_width-#,#-sum(panel,panel)-#
+minimumSize=#-os_x11_width-#,#-sum(panel,panel)-#
+
+#------------------------------------------------------------------------------
+#нижняя панель
+#------------------------------------------------------------------------------
+
+#-sum(height,,44)-#
+#?hr_cdrom_set==on#
+#-sum(width,,height*14)-#
+#hr_cdrom_set#
+#?hr_cdrom_set==off#
+#-sum(width,,height*13)-#
+#hr_cdrom_set#
+#-sum(apporder,,0)-#
+
+#?pkg(net-irc/konversation)!=#
+#-sum(width,,width+height)-#
+#pkg#
+
+[Containments][-#-sum(cont,cont+1)-#]
+desktop=-1
+formfactor=2
+geometry=0,0,#-sum(width,width)-#,#-sum(height,height)-#
+immutability=1
+location=4
+plugin=panel
+screen=0
+zvalue=150
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,cont*100)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-browser.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-mail.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+#?pkg(net-irc/konversation)!=#
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-chat.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+#pkg#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-im.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-writer.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-calc.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-calculator.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-imageedit.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-imageview.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-audioplayer.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-videoplayer.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+#?hr_cdrom_set==on#
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-imageburn.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+#hr_cdrom_set#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-dict.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-textedit.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app+1)-#]
+immutability=1
+plugin=icon
+zvalue=0
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][Configuration]
+Url=file:///usr/share/applications/calculate-terminal.desktop
+[Containments][#-sum(cont,cont)-#][Applets][#-sum(app,app)-#][LayoutInformation]
+Order=#-sum(apporder,apporder+1)-#
+
+[General]
+immutability=2
+
+[Containments][#-sum(cont,cont)-#][Configuration]
+maximumSize=#-sum(width,width)-#,#-sum(height,height)-#
+minimumSize=#-sum(width,width)-#,#-sum(height,height)-#
+
diff --git desktop/desktop/20-plasma-4.6/plasma-desktoprc desktop/desktop/20-plasma-4.6/plasma-desktoprc
new file mode 100644
index 0000000..72fa120
--- /dev/null
+++ desktop/desktop/20-plasma-4.6/plasma-desktoprc
@@ -0,0 +1,22 @@
+# Calculate format=plasma path=~/.kde4/share/config
+
+# location of panels
+[PlasmaViews][1]
+Alignment=132
+Offset=0
+panelVisibility=1
+
+[PlasmaViews][2]
+Alignment=132
+Offset=0
+panelVisibility=1
+
+[PlasmaViews][3]
+Offset=0
+panelVisibility=0
+
+[ViewIds]
+1=1
+2=3
+3=2
+
diff --git desktop/desktop/20-plasma-4.6/plasmarc desktop/desktop/20-plasma-4.6/plasmarc
new file mode 100644
index 0000000..d28b6dc
--- /dev/null
+++ desktop/desktop/20-plasma-4.6/plasmarc
@@ -0,0 +1,5 @@
+# Calculate format=plasma path=~/.kde4/share/config
+#оформление
+[Theme]
+name=slim-glow
+
diff --git install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
index 5f21119..a0bc2a5 100644
--- install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
+++ install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/share/applications/audacious2.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=2.4
+# Calculate format=kde link=/usr/share/applications/audacious2.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=2.4&&pkg(media-sound/audacious)<3.0
 
 # Update info
 #-ini(desktop.audioplayer, audacious)-#
diff --git install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop
new file mode 100644
index 0000000..6387c9e
--- /dev/null
+++ install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop
@@ -0,0 +1,9 @@
+# Calculate format=kde link=/usr/share/applications/audacious.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=3.0
+
+# Update info
+#-ini(desktop.audioplayer, audacious)-#
+
+# Hide icon
+[Desktop Entry]
+OnlyShowIn=X-NeverShowThis;
+
diff --git install/1merge/calculate-desktop/.calculate_directory install/1merge/calculate-desktop/.calculate_directory
new file mode 100644
index 0000000..437bd6a
--- /dev/null
+++ install/1merge/calculate-desktop/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate belong()!=&&cl_chroot_status==off&&pkg(sys-apps/calculate-desktop)!= name=etc path=/usr name=bin
+
diff --git install/1merge/calculate-desktop/xautologin install/1merge/calculate-desktop/xautologin
new file mode 100644
index 0000000..1173e71
--- /dev/null
+++ install/1merge/calculate-desktop/xautologin
@@ -0,0 +1,15 @@
+# Calculate chmod=0755
+#!/bin/bash
+echo $BASHPID >/var/run/xinit.pid
+while true;
+do
+#?list(cl_migrate_user,1)!=#
+        env USER=#-list(cl_migrate_user,1)-# /usr/share/calculate/xdm/xdm --login
+        su - #-list(cl_migrate_user,1)-# -c 'source /etc/profile;/bin/bash /etc/X11/xinit/xinitrc'
+        env USER=#-list(cl_migrate_user,1)-# /usr/share/calculate/xdm/xdm --logout
+#list#
+#?list(cl_migrate_user,1)==#
+        :
+#list#
+done
+exit 0
diff --git install/1merge/xorg-server/conf.d/xdm install/1merge/xorg-server/conf.d/xdm
index eb21521..e7fa676 100644
--- install/1merge/xorg-server/conf.d/xdm
+++ install/1merge/xorg-server/conf.d/xdm
@@ -1,12 +1,19 @@
 # Calculate comment=#
 
 CHECKVT=7
-#?pkg(kde-base/kdm)!=#
+#?os_install_root_type==livecd||os_install_linux_shortname==CMC#
+DISPLAYMANAGER="xinit"
+START_STOP_ARGS="--background -- /usr/bin/xautologin -- vt$CHECKVT"
+#os_install_root_type#
+#?os_install_root_type!=livecd&&os_install_linux_shortname!=CMC&&pkg(kde-base/kdm)!=#
 DISPLAYMANAGER="kdm"
-#pkg#
-#?pkg(gnome-base/gdm)!=&&pkg(kde-base/kdm)==#
+!START_STOP_ARGS=empty
+#os_install_root_type#
+#?os_install_root_type!=livecd&&os_install_linux_shortname!=CMC&&pkg(gnome-base/gdm)!=&&pkg(kde-base/kdm)==#
 DISPLAYMANAGER="gdm"
-#pkg#
-#?pkg(x11-misc/slim)!=&&pkg(kde-base/kdm)==&&pkg(gnome-base/gdm)==#
+!START_STOP_ARGS=empty
+#os_install_root_type#
+#?os_install_root_type!=livecd&&os_install_linux_shortname!=CMC&&pkg(x11-misc/slim)!=&&pkg(kde-base/kdm)==&&pkg(gnome-base/gdm)==#
 DISPLAYMANAGER="slim"
-#pkg#
+!START_STOP_ARGS=empty
+#os_install_root_type#
