commit c937d04f1a227bc260ff9d9146a13851235c658e
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Apr 13 19:21:26 2011 +0400

    Fix grub.conf for CONSOLE.

diff --git install/1merge/calckernel/grub.conf.fix install/1merge/calckernel/grub.conf.fix
new file mode 100644
index 0000000..4a771de
--- /dev/null
+++ install/1merge/calckernel/grub.conf.fix
@@ -0,0 +1,3 @@
+# Calculate exec=/bin/bash path=/tmp
+
+sed -ri 's|(vmlinuz-#-cl_kernel_uid-#.*)CONSOLE=/dev/tty1|\1console=tty1|' /boot/grub/grub.conf 
