commit 81ff96d00c831bd724db169231c7717a8047f560
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Fri Dec 24 11:18:11 2010 +0300

    Fix cmc and cds themes config.

diff --git a/install/1merge/cds-themes/tty2/1024x768.cfg b/install/1merge/cds-themes/tty2/1024x768.cfg
index 16cb3da..88b4152 100644
--- a/install/1merge/cds-themes/tty2/1024x768.cfg
+++ b/install/1merge/cds-themes/tty2/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CDS-tty1.jpg</reg>
-<text>verbose-CDS-tty2.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1.jpg</reg>
+<text>tty2/images/verbose-CDS-tty2.jpg</text>
diff --git a/install/1merge/cds-themes/tty2/800x600.cfg b/install/1merge/cds-themes/tty2/800x600.cfg
index 5ab6160..2846ffa 100644
--- a/install/1merge/cds-themes/tty2/800x600.cfg
+++ b/install/1merge/cds-themes/tty2/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CDS-tty1-800x600.jpg</reg>
-<text>verbose-CDS-tty2-800x600.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1-800x600.jpg</reg>
+<text>tty2/images/verbose-CDS-tty2-800x600.jpg</text>
diff --git a/install/1merge/cds-themes/tty3/1024x768.cfg b/install/1merge/cds-themes/tty3/1024x768.cfg
index f64b691..eb8813b 100644
--- a/install/1merge/cds-themes/tty3/1024x768.cfg
+++ b/install/1merge/cds-themes/tty3/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CDS-tty1.jpg</reg>
-<text>verbose-CDS-tty3.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1.jpg</reg>
+<text>tty3/images/verbose-CDS-tty3.jpg</text>
diff --git a/install/1merge/cds-themes/tty3/800x600.cfg b/install/1merge/cds-themes/tty3/800x600.cfg
index 9a5cc61..f81d9e5 100644
--- a/install/1merge/cds-themes/tty3/800x600.cfg
+++ b/install/1merge/cds-themes/tty3/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CDS-tty1-800x600.jpg</reg>
-<text>verbose-CDS-tty3-800x600.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1-800x600.jpg</reg>
+<text>tty3/images/verbose-CDS-tty3-800x600.jpg</text>
diff --git a/install/1merge/cds-themes/tty4/1024x768.cfg b/install/1merge/cds-themes/tty4/1024x768.cfg
index a725ba5..9b5dd37 100644
--- a/install/1merge/cds-themes/tty4/1024x768.cfg
+++ b/install/1merge/cds-themes/tty4/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CDS-tty1.jpg</reg>
-<text>verbose-CDS-tty4.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1.jpg</reg>
+<text>tty4/images/verbose-CDS-tty4.jpg</text>
diff --git a/install/1merge/cds-themes/tty4/800x600.cfg b/install/1merge/cds-themes/tty4/800x600.cfg
index 426b014..d6cccbb 100644
--- a/install/1merge/cds-themes/tty4/800x600.cfg
+++ b/install/1merge/cds-themes/tty4/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CDS-tty1-800x600.jpg</reg>
-<text>verbose-CDS-tty4-800x600.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1-800x600.jpg</reg>
+<text>tty4/images/verbose-CDS-tty4-800x600.jpg</text>
diff --git a/install/1merge/cds-themes/tty5/1024x768.cfg b/install/1merge/cds-themes/tty5/1024x768.cfg
index e07bae4..9387aa2 100644
--- a/install/1merge/cds-themes/tty5/1024x768.cfg
+++ b/install/1merge/cds-themes/tty5/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CDS-tty1.jpg</reg>
-<text>verbose-CDS-tty5.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1.jpg</reg>
+<text>tty5/images/verbose-CDS-tty5.jpg</text>
diff --git a/install/1merge/cds-themes/tty5/800x600.cfg b/install/1merge/cds-themes/tty5/800x600.cfg
index 483aba6..86d8860 100644
--- a/install/1merge/cds-themes/tty5/800x600.cfg
+++ b/install/1merge/cds-themes/tty5/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CDS-tty1-800x600.jpg</reg>
-<text>verbose-CDS-tty5-800x600.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1-800x600.jpg</reg>
+<text>tty5/images/verbose-CDS-tty5-800x600.jpg</text>
diff --git a/install/1merge/cds-themes/tty6/1024x768.cfg b/install/1merge/cds-themes/tty6/1024x768.cfg
index 2fd478e..747f724 100644
--- a/install/1merge/cds-themes/tty6/1024x768.cfg
+++ b/install/1merge/cds-themes/tty6/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CDS-tty1.jpg</reg>
-<text>verbose-CDS-tty6.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1.jpg</reg>
+<text>tty6/images/verbose-CDS-tty6.jpg</text>
diff --git a/install/1merge/cds-themes/tty6/800x600.cfg b/install/1merge/cds-themes/tty6/800x600.cfg
index b654b2c..dec6a84 100644
--- a/install/1merge/cds-themes/tty6/800x600.cfg
+++ b/install/1merge/cds-themes/tty6/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CDS-tty1-800x600.jpg</reg>
-<text>verbose-CDS-tty6-800x600.jpg</text>
+<reg>tty1/images/verbose-CDS-tty1-800x600.jpg</reg>
+<text>tty6/images/verbose-CDS-tty6-800x600.jpg</text>
diff --git a/install/1merge/cmc-themes/tty2/1024x768.cfg b/install/1merge/cmc-themes/tty2/1024x768.cfg
index fe621d8..389b948 100644
--- a/install/1merge/cmc-themes/tty2/1024x768.cfg
+++ b/install/1merge/cmc-themes/tty2/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CMC-tty1.jpg</reg>
-<text>verbose-CMC-tty2.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1.jpg</reg>
+<text>tty2/images/verbose-CMC-tty2.jpg</text>
diff --git a/install/1merge/cmc-themes/tty2/800x600.cfg b/install/1merge/cmc-themes/tty2/800x600.cfg
index 7aea025..bd30448 100644
--- a/install/1merge/cmc-themes/tty2/800x600.cfg
+++ b/install/1merge/cmc-themes/tty2/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CMC-tty1-800x600.jpg</reg>
-<text>verbose-CMC-tty2-800x600.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1-800x600.jpg</reg>
+<text>tty2/images/verbose-CMC-tty2-800x600.jpg</text>
diff --git a/install/1merge/cmc-themes/tty3/1024x768.cfg b/install/1merge/cmc-themes/tty3/1024x768.cfg
index 4844bbb..bf85c21 100644
--- a/install/1merge/cmc-themes/tty3/1024x768.cfg
+++ b/install/1merge/cmc-themes/tty3/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CMC-tty1.jpg</reg>
-<text>verbose-CMC-tty3.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1.jpg</reg>
+<text>tty3/images/verbose-CMC-tty3.jpg</text>
diff --git a/install/1merge/cmc-themes/tty3/800x600.cfg b/install/1merge/cmc-themes/tty3/800x600.cfg
index 86caff5..64ec8ef 100644
--- a/install/1merge/cmc-themes/tty3/800x600.cfg
+++ b/install/1merge/cmc-themes/tty3/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CMC-tty1-800x600.jpg</reg>
-<text>verbose-CMC-tty3-800x600.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1-800x600.jpg</reg>
+<text>tty3/images/verbose-CMC-tty3-800x600.jpg</text>
diff --git a/install/1merge/cmc-themes/tty4/1024x768.cfg b/install/1merge/cmc-themes/tty4/1024x768.cfg
index 4b26438..ef39f63 100644
--- a/install/1merge/cmc-themes/tty4/1024x768.cfg
+++ b/install/1merge/cmc-themes/tty4/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CMC-tty1.jpg</reg>
-<text>verbose-CMC-tty4.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1.jpg</reg>
+<text>tty4/images/verbose-CMC-tty4.jpg</text>
diff --git a/install/1merge/cmc-themes/tty4/800x600.cfg b/install/1merge/cmc-themes/tty4/800x600.cfg
index fd515c9..0da109f 100644
--- a/install/1merge/cmc-themes/tty4/800x600.cfg
+++ b/install/1merge/cmc-themes/tty4/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CMC-tty1-800x600.jpg</reg>
-<text>verbose-CMC-tty4-800x600.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1-800x600.jpg</reg>
+<text>tty4/images/verbose-CMC-tty4-800x600.jpg</text>
diff --git a/install/1merge/cmc-themes/tty5/1024x768.cfg b/install/1merge/cmc-themes/tty5/1024x768.cfg
index 02c5078..ca9cddf 100644
--- a/install/1merge/cmc-themes/tty5/1024x768.cfg
+++ b/install/1merge/cmc-themes/tty5/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CMC-tty1.jpg</reg>
-<text>verbose-CMC-tty5.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1.jpg</reg>
+<text>tty5/images/verbose-CMC-tty5.jpg</text>
diff --git a/install/1merge/cmc-themes/tty5/800x600.cfg b/install/1merge/cmc-themes/tty5/800x600.cfg
index 0c48515..85197c1 100644
--- a/install/1merge/cmc-themes/tty5/800x600.cfg
+++ b/install/1merge/cmc-themes/tty5/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CMC-tty1-800x600.jpg</reg>
-<text>verbose-CMC-tty5-800x600.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1-800x600.jpg</reg>
+<text>tty5/images/verbose-CMC-tty5-800x600.jpg</text>
diff --git a/install/1merge/cmc-themes/tty6/1024x768.cfg b/install/1merge/cmc-themes/tty6/1024x768.cfg
index c9b8dd5..ada8777 100644
--- a/install/1merge/cmc-themes/tty6/1024x768.cfg
+++ b/install/1merge/cmc-themes/tty6/1024x768.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/1024x768.cfg
-<reg>verbose-CMC-tty1.jpg</reg>
-<text>verbose-CMC-tty6.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1.jpg</reg>
+<text>tty6/images/verbose-CMC-tty6.jpg</text>
diff --git a/install/1merge/cmc-themes/tty6/800x600.cfg b/install/1merge/cmc-themes/tty6/800x600.cfg
index 14ee764..eecea9b 100644
--- a/install/1merge/cmc-themes/tty6/800x600.cfg
+++ b/install/1merge/cmc-themes/tty6/800x600.cfg
@@ -1,3 +1,3 @@
 # Calculate format=patch link=/etc/splash/tty1/800x600.cfg
-<reg>verbose-CMC-tty1-800x600.jpg</reg>
-<text>verbose-CMC-tty6-800x600.jpg</text>
+<reg>tty1/images/verbose-CMC-tty1-800x600.jpg</reg>
+<text>tty6/images/verbose-CMC-tty6-800x600.jpg</text>
