diff --git desktop/desktop/60-opera-10.6/.calculate_directory desktop/desktop/60-opera-10.6/.calculate_directory
index 78a774b..e9e9630 100644
--- desktop/desktop/60-opera-10.6/.calculate_directory
+++ desktop/desktop/60-opera-10.6/.calculate_directory
@@ -1 +1 @@
-# Calculate append=skip belong(opera)!=&&exists(~/.opera)==&&ini(update.opera)!=off&&ini(ver.opera)==&&pkg(www-client/opera)>=10.6
+# Calculate append=skip belong(opera)!=&&exists(~/.opera)==&&ini(update.opera)!=off&&ini(ver.opera)==&&pkg(www-client/opera)>=10.6&&pkg(www-client/opera)<11.0
diff --git desktop/desktop/60-opera-11.5/.calculate/ini.env desktop/desktop/60-opera-11.5/.calculate/ini.env
new file mode 100644
index 0000000..7532d69
--- /dev/null
+++ desktop/desktop/60-opera-11.5/.calculate/ini.env
@@ -0,0 +1,7 @@
+# Calculate format=samba
+
+[ver]
+opera = 11.5
+
+[update]
+opera = on
diff --git desktop/desktop/60-opera-11.5/.calculate_directory desktop/desktop/60-opera-11.5/.calculate_directory
new file mode 100644
index 0000000..d699447
--- /dev/null
+++ desktop/desktop/60-opera-11.5/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong(opera)!=&&exists(~/.opera)==&&ini(update.opera)!=off&&ini(ver.opera)==&&pkg(www-client/opera)>=11.5
diff --git desktop/desktop/60-opera-11.5/.opera/operaprefs.ini desktop/desktop/60-opera-11.5/.opera/operaprefs.ini
new file mode 100644
index 0000000..e904e3e
--- /dev/null
+++ desktop/desktop/60-opera-11.5/.opera/operaprefs.ini
@@ -0,0 +1,45 @@
+﻿# Calculate format=kde
+
+#?pkg(media-fonts/droid)!=#
+[CSS Generic Font Family]
+Fantasy=Droid Sans
+Serif=Droid Serif
+Sans-Serif=Droid Sans
+Cursive=Droid Serif
+Monospace=Droid Sans Mono
+
+[Fonts]
+Normal=14,4,0,0,0,0,Droid Sans
+PRE=14,4,0,0,0,0,Droid Sans Mono
+Form.Text=12,4,0,0,0,0,Droid Sans
+EmailCompose=12,4,0,0,0,0,Droid Sans
+Email=13,4,0,0,0,0,Droid Sans
+HTMLCompose=12,4,0,0,0,0,Droid Sans
+Form.Input=12,4,0,0,0,0,Droid Sans
+Form.Button=12,4,0,0,0,0,Droid Sans
+H1=32,7,0,0,0,0,Droid Sans
+H2=24,7,0,0,0,0,Droid Sans
+H3=18,7,0,0,0,0,Droid Sans
+H4=16,7,0,0,0,0,Droid Sans
+H5=13,7,0,0,0,0,Droid Sans
+H6=12,7,0,0,0,0,Droid Sans
+#pkg#
+
+[Install]
+Newest Used Version=11.52.1100
+
+[Mail]
+Handler=3
+
+[State]
+Run=0
+
+[User Prefs]
+#?os_locale_language!=ru#
+Home URL=http://www.calculate-linux.ru
+#os_locale_language#
+#?os_locale_language==ru#
+Home URL=http://www.calculate-linux.ru
+#os_locale_language#
+Startup Type=2
+
diff --git desktop/desktop/60-opera-11.5/.opera/speeddial.ini-en desktop/desktop/60-opera-11.5/.opera/speeddial.ini-en
new file mode 100644
index 0000000..05faeef
--- /dev/null
+++ desktop/desktop/60-opera-11.5/.opera/speeddial.ini-en
@@ -0,0 +1,20 @@
+﻿# Calculate name=speeddial.ini os_locale_language!=ru
+
+[Speed Dial 1]
+Url=http://www.calculate-linux.org/
+
+[Speed Dial 2]
+Url=http://pastebin.calculate-linux.org/
+
+[Speed Dial 3]
+http://www.linuxquestions.org/questions/calculate-89/
+
+[Speed Dial 4]
+Url=http://www.facebook.com/group.php?gid=97309000613
+
+[Speed Dial 5]
+Url=http://twitter.com/#!/calculatelinux
+
+[Speed Dial 6]
+Url=http://www.distrowatch.com/calculate
+
diff --git desktop/desktop/60-opera-11.5/.opera/speeddial.ini-ru desktop/desktop/60-opera-11.5/.opera/speeddial.ini-ru
new file mode 100644
index 0000000..e1643e9
--- /dev/null
+++ desktop/desktop/60-opera-11.5/.opera/speeddial.ini-ru
@@ -0,0 +1,25 @@
+﻿# Calculate name=speeddial.ini os_locale_language==ru
+
+[Speed Dial 1]
+Url=http://www.calculate-linux.ru/
+
+[Speed Dial 2]
+Url=http://pastebin.calculate-linux.ru/
+
+[Speed Dial 3]
+Url=http://vkontakte.ru/calculatelinux
+
+[Speed Dial 4]
+Url=http://www.facebook.com/group.php?gid=101281876595263
+
+[Speed Dial 5]
+Url=http://twitter.com/#!/calculatelinux
+
+[Speed Dial 6]
+Url=http://www.distrowatch.com/calculate
+
+[Speed Dial 7]
+Url=http://www.gentoo.ru/
+
+[Speed Dial 8]
+Url=http://www.tux.in.ua/
diff --git install/1live/calculate-utilities/.calculate_directory install/1live/calculate-utilities/.calculate_directory
new file mode 100644
index 0000000..0478f8c
--- /dev/null
+++ install/1live/calculate-utilities/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip belong()!=&&pkg(sys-apps/calculate-utilities)!=
diff --git install/1live/calculate-utilities/reconfigurePackages install/1live/calculate-utilities/reconfigurePackages
new file mode 100644
index 0000000..c127bc5
--- /dev/null
+++ install/1live/calculate-utilities/reconfigurePackages
@@ -0,0 +1,4 @@
+# Calculate path=/tmp name=reconfigureLibre exec=/bin/bash ac_install_system!=up&&cl_belong_pkg!=&&pkg(app-office/libreoffice)!=
+
+cl-template -l all --merge app-office/libreoffice &>/dev/null
+true
diff --git install/1live/portage/make.conf install/1live/portage/make.conf
index 3630cbf..3118ec0 100644
--- install/1live/portage/make.conf
+++ install/1live/portage/make.conf
@@ -131,5 +131,5 @@ MAKEOPTS="#-os_install_makeopts-#"
 #?os_install_locale_language==bg#
 # Количество пакети за едновременно компилиране.
 #os_install_locale_language#
-#EMERGE_DEFAULT_OPTS="--jobs=4"
+#EMERGE_DEFAULT_OPTS="${EMERGE_DEFAULT_OPTS} --jobs=4"
 
diff --git install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop
new file mode 100644
index 0000000..c2f95e7
--- /dev/null
+++ install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.3.2.desktop
@@ -0,0 +1,8 @@
+# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2
+
+# Update info
+#-ini(desktop.writer, libreoffice)-#
+
+# Hide icon
+[Desktop Entry]
+NoDisplay=true
diff --git install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.desktop install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.desktop
index c8fea53..ffa463b 100644
--- install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.desktop
+++ install/1merge/Desktop/200-Writer/libreoffice/calculate/writer-3.4.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/share/applications/libreoffice-writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4
+# Calculate format=kde link=/usr/share/applications/libreoffice-writer.desktop name=calculate-writer.desktop mirror pkg(app-office/libreoffice)>=3.4&&pkg(app-office/libreoffice)<3.4.3.2
 
 # Update info
 #-ini(desktop.writer, libreoffice)-#
diff --git install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop
new file mode 100644
index 0000000..8ec5dcb
--- /dev/null
+++ install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.3.2.desktop
@@ -0,0 +1,8 @@
+# Calculate format=kde link=/usr/lib/libreoffice/share/xdg/calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4.3.2
+
+# Update info
+#-ini(desktop.calc, libreoffice)-#
+
+# Hide icon
+[Desktop Entry]
+NoDisplay=true
diff --git install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.desktop install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.desktop
index 6c917ac..59e8fe4 100644
--- install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.desktop
+++ install/1merge/Desktop/210-Calc/libreoffice/calculate/calc-3.4.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/share/applications/libreoffice-calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4
+# Calculate format=kde link=/usr/share/applications/libreoffice-calc.desktop name=calculate-calc.desktop mirror pkg(app-office/libreoffice)>=3.4&&pkg(app-office/libreoffice)<3.4.3.2
 
 # Update info
 #-ini(desktop.calc, libreoffice)-#
