commit d225f74f1a82a1380433dcac2e2ec06c2baee9af
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Apr 13 11:51:10 2011 +0400

    Fix intel bad name in isolinux.cfg.

diff --git a/builder/iso/calcboot/isolinux/video.cfg b/builder/iso/calcboot/isolinux/video.cfg
index 256ca23..0466340 100644
--- a/builder/iso/calcboot/isolinux/video.cfg
+++ b/builder/iso/calcboot/isolinux/video.cfg
@@ -15,12 +15,12 @@ ENDTEXT
 LABEL intel
         MENU LABEL Intel (intel) with composite
         KERNEL calcmenu.c32
-        APPEND vga.cfg calculate=,,,,intel,composite
+        APPEND vga.cfg calculate=,,,,i915,composite
 
 LABEL intel
         MENU LABEL Intel (intel)
         KERNEL calcmenu.c32
-        APPEND vga.cfg calculate=,,,,intel,nocomposite
+        APPEND vga.cfg calculate=,,,,i915,nocomposite
 
 LABEL radeon
         MENU LABEL ATI Radeon Open Source (radeon)
