diff --git desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
index 4015931..150bd84 100644
--- desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
+++ desktop/desktop/20-kde-4.6/.kde4/share/config/kdeglobals
@@ -5,15 +5,7 @@ emoticonsTheme=KMess-Cartoon
 
 #Fonts
 [General]
-#?pkg(www-client/chromium)!=#
-BrowserApplication[$e]=file:///usr/share/applications/chromium-chromium.desktop
-#pkg#
-#?pkg(www-client/firefox)!=&&pkg(www-client/chromium)==#
-BrowserApplication[$e]=file:///home/#-ur_login-#/.local/share/applications/mozilla-firefox.desktop
-#pkg#
-#?pkg(www-client/firefox)==&&pkg(www-client/chromium)==#
-BrowserApplication[$e]=file:///usr/share/applications/kde4/konqbrowser.desktop
-#pkg#
+BrowserApplication[$e]=file:///usr/share/applications/calculate-browser.desktop
 XftHintStyle=hintmedium
 #?pkg(media-fonts/droid)!=#
 desktopFont=Droid Sans,10,-1,5,50,0,0,0,0,0
diff --git install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
index 5f21119..a0bc2a5 100644
--- install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
+++ install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-2.4.desktop
@@ -1,4 +1,4 @@
-# Calculate format=kde link=/usr/share/applications/audacious2.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=2.4
+# Calculate format=kde link=/usr/share/applications/audacious2.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=2.4&&pkg(media-sound/audacious)<3.0
 
 # Update info
 #-ini(desktop.audioplayer, audacious)-#
diff --git install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop
new file mode 100644
index 0000000..6387c9e
--- /dev/null
+++ install/1merge/Desktop/400-AudioPlayer/audacious/calculate/audacious-3.0.desktop
@@ -0,0 +1,9 @@
+# Calculate format=kde link=/usr/share/applications/audacious.desktop name=calculate-audioplayer.desktop mirror pkg(media-sound/audacious)>=3.0
+
+# Update info
+#-ini(desktop.audioplayer, audacious)-#
+
+# Hide icon
+[Desktop Entry]
+OnlyShowIn=X-NeverShowThis;
+
diff --git install/1merge/calculate-desktop/.calculate_directory install/1merge/calculate-desktop/.calculate_directory
new file mode 100644
index 0000000..437bd6a
--- /dev/null
+++ install/1merge/calculate-desktop/.calculate_directory
@@ -0,0 +1,2 @@
+# Calculate belong()!=&&cl_chroot_status==off&&pkg(sys-apps/calculate-desktop)!= name=etc path=/usr name=bin
+
diff --git install/1merge/calculate-desktop/xautologin install/1merge/calculate-desktop/xautologin
new file mode 100644
index 0000000..5387eb1
--- /dev/null
+++ install/1merge/calculate-desktop/xautologin
@@ -0,0 +1,8 @@
+# Calculate chmod=0755
+#!/bin/bash
+#?list(cl_migrate_user,1)!=#
+export USER=#-list(cl_migrate_user,1)-#
+/usr/share/calculate/xdm/xdm --login
+su - #-list(cl_migrate_user,1)-# -c "/bin/bash -lc startx"
+/usr/share/calculate/xdm/xdm --logout
+#list#
diff --git install/1merge/xorg-server/conf.d/xdm install/1merge/xorg-server/conf.d/xdm
index eb21521..d6f5cdf 100644
--- install/1merge/xorg-server/conf.d/xdm
+++ install/1merge/xorg-server/conf.d/xdm
@@ -1,12 +1,15 @@
 # Calculate comment=#
 
 CHECKVT=7
-#?pkg(kde-base/kdm)!=#
+#?os_install_root_type==livecd||os_install_linux_shortname==CMC#
+DISPLAYMANAGER="xautologin"
+#os_install_root_type#
+#?os_install_root_type!=livecd&&os_install_linux_shortname!=CMC&&pkg(kde-base/kdm)!=#
 DISPLAYMANAGER="kdm"
-#pkg#
-#?pkg(gnome-base/gdm)!=&&pkg(kde-base/kdm)==#
+#os_install_root_type#
+#?os_install_root_type!=livecd&&os_install_linux_shortname!=CMC&&pkg(gnome-base/gdm)!=&&pkg(kde-base/kdm)==#
 DISPLAYMANAGER="gdm"
-#pkg#
-#?pkg(x11-misc/slim)!=&&pkg(kde-base/kdm)==&&pkg(gnome-base/gdm)==#
+#os_install_root_type#
+#?os_install_root_type!=livecd&&os_install_linux_shortname!=CMC&&pkg(x11-misc/slim)!=&&pkg(kde-base/kdm)==&&pkg(gnome-base/gdm)==#
 DISPLAYMANAGER="slim"
-#pkg#
+#os_install_root_type#
