commit 37495f855ac8858bae3c4ed4b6a4a6af75fef66f
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Aug 26 22:40:24 2011 +0400

    Fix templates for xorg.conf.

diff --git install/1merge/xorg-server/X11/xorg.conf install/1merge/xorg-server/X11/xorg.conf
index 210d3a1..204c197 100644
--- install/1merge/xorg-server/X11/xorg.conf
+++ install/1merge/xorg-server/X11/xorg.conf
@@ -1,4 +1,4 @@
-# Calculate comment=# os_install_video_drv!=other
+# Calculate comment=# os_install_x11_video_drv!=other
 
 Section "ServerLayout"
   Identifier "Xorg Configured"
diff --git install/1merge/xorg-server/X11/xorg.conf.other install/1merge/xorg-server/X11/xorg.conf.other
index 9342c02..72ce36b 100644
--- install/1merge/xorg-server/X11/xorg.conf.other
+++ install/1merge/xorg-server/X11/xorg.conf.other
@@ -1,4 +1,4 @@
-# Calculate comment=# os_install_video_drv==other name=xorg.conf
+# Calculate comment=# os_install_x11_video_drv==other name=xorg.conf
 
 Section "ServerLayout"
  Identifier "Xorg Configured"
