commit 5b238f93c93edd570af5de95ed020cf1981d3e98
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue Dec 27 09:18:22 2011 +0400

    Add fix init.d for some packages.

diff --git install/1merge/00-patch/.calculate_directory install/1merge/00-patch/.calculate_directory
new file mode 100644
index 0000000..720d52e
--- /dev/null
+++ install/1merge/00-patch/.calculate_directory
@@ -0,0 +1 @@
+# Calculate append=skip
diff --git install/1merge/00-patch/fail2ban/.calculate_directory install/1merge/00-patch/fail2ban/.calculate_directory
new file mode 100644
index 0000000..dc9e02a
--- /dev/null
+++ install/1merge/00-patch/fail2ban/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(net-analyzer/fail2ban)>=0.8.4 name=init.d path=/etc
diff --git install/1merge/00-patch/fail2ban/fail2ban install/1merge/00-patch/fail2ban/fail2ban
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/fail2ban/fail2ban
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/iptables/.calculate_directory install/1merge/00-patch/iptables/.calculate_directory
new file mode 100644
index 0000000..415681b
--- /dev/null
+++ install/1merge/00-patch/iptables/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(net-firewall/iptables)>=1.4.12 name=init.d path=/etc
diff --git install/1merge/00-patch/iptables/ip6tables install/1merge/00-patch/iptables/ip6tables
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/iptables/ip6tables
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/iptables/iptables install/1merge/00-patch/iptables/iptables
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/iptables/iptables
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/net-snmp/.calculate_directory install/1merge/00-patch/net-snmp/.calculate_directory
new file mode 100644
index 0000000..ee6d7af
--- /dev/null
+++ install/1merge/00-patch/net-snmp/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(net-analyzer/net-snmp)>=5.4.2.1 name=init.d path=/etc
diff --git install/1merge/00-patch/net-snmp/snmpd install/1merge/00-patch/net-snmp/snmpd
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/net-snmp/snmpd
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/nfs-utils/.calculate_directory install/1merge/00-patch/nfs-utils/.calculate_directory
new file mode 100644
index 0000000..9a5bf1f
--- /dev/null
+++ install/1merge/00-patch/nfs-utils/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(net-fs/nfs-utils)>=1.2.3 name=init.d path=/etc
diff --git install/1merge/00-patch/nfs-utils/nfs install/1merge/00-patch/nfs-utils/nfs
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/nfs-utils/nfs
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/postfix/.calculate_directory install/1merge/00-patch/postfix/.calculate_directory
new file mode 100644
index 0000000..46ee030
--- /dev/null
+++ install/1merge/00-patch/postfix/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(mail-mta/postfix)>=2.8.4 name=init.d path=/etc
diff --git install/1merge/00-patch/postfix/postfix install/1merge/00-patch/postfix/postfix
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/postfix/postfix
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/postgresql-server/.calculate_directory install/1merge/00-patch/postgresql-server/.calculate_directory
new file mode 100644
index 0000000..8a04d57
--- /dev/null
+++ install/1merge/00-patch/postgresql-server/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(dev-db/postgresql-server)>=9.1&&pkg(dev-db/postgresql-server)<9.2 name=init.d path=/etc
diff --git install/1merge/00-patch/postgresql-server/postgresql-9.1 install/1merge/00-patch/postgresql-server/postgresql-9.1
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/postgresql-server/postgresql-9.1
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/postgrey/.calculate_directory install/1merge/00-patch/postgrey/.calculate_directory
new file mode 100644
index 0000000..c207d24
--- /dev/null
+++ install/1merge/00-patch/postgrey/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(mail-filter/postgrey)>=1.34 name=init.d path=/etc
diff --git install/1merge/00-patch/postgrey/postgrey install/1merge/00-patch/postgrey/postgrey
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/postgrey/postgrey
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/powernowd/.calculate_directory install/1merge/00-patch/powernowd/.calculate_directory
new file mode 100644
index 0000000..06264a7
--- /dev/null
+++ install/1merge/00-patch/powernowd/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(sys-power/powernowd)>=1.00 name=init.d path=/etc
diff --git install/1merge/00-patch/powernowd/powernowd install/1merge/00-patch/powernowd/powernowd
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/powernowd/powernowd
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/samba/.calculate_directory install/1merge/00-patch/samba/.calculate_directory
new file mode 100644
index 0000000..f74390f
--- /dev/null
+++ install/1merge/00-patch/samba/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(net-fs/samba)>=3.5.11 name=init.d path=/etc
diff --git install/1merge/00-patch/samba/samba install/1merge/00-patch/samba/samba
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/samba/samba
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/squid/.calculate_directory install/1merge/00-patch/squid/.calculate_directory
new file mode 100644
index 0000000..1535831
--- /dev/null
+++ install/1merge/00-patch/squid/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(net-proxy/squid)>=3.1.16 name=init.d path=/etc
diff --git install/1merge/00-patch/squid/squid install/1merge/00-patch/squid/squid
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/squid/squid
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
diff --git install/1merge/00-patch/vnstat/.calculate_directory install/1merge/00-patch/vnstat/.calculate_directory
new file mode 100644
index 0000000..ebf76b0
--- /dev/null
+++ install/1merge/00-patch/vnstat/.calculate_directory
@@ -0,0 +1 @@
+# Calculate belong()!=&&pkg(net-analyzer/vnstat)>=1.10 name=init.d path=/etc
diff --git install/1merge/00-patch/vnstat/vnstat install/1merge/00-patch/vnstat/vnstat
new file mode 100644
index 0000000..efb66eb
--- /dev/null
+++ install/1merge/00-patch/vnstat/vnstat
@@ -0,0 +1,5 @@
+# Calculate format=patch
+<reg>
+opts="(\$\{opts\} )?([^"]+)"</reg>
+<text>
+extra_commands="\2"</text>
