diff --git a/install/2laptop/samsung/calculate-install/bin/backlight b/install/2laptop/samsung/calculate-install/bin/backlight
index 740b596..1d9c619 100644
--- a/install/2laptop/samsung/calculate-install/bin/backlight
+++ b/install/2laptop/samsung/calculate-install/bin/backlight
@@ -1,4 +1,4 @@
-# Calculate chmod=0755 comment=# load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011
+# Calculate chmod=0755 comment=# load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011
 #!/bin/bash
 # increase/decrease/set/get the backlight brightness (range 0-255)
 #
diff --git a/install/2laptop/samsung/calculate-install/init.d/laptopd b/install/2laptop/samsung/calculate-install/init.d/laptopd
index b0bbc11..6d780f2 100644
--- a/install/2laptop/samsung/calculate-install/init.d/laptopd
+++ b/install/2laptop/samsung/calculate-install/init.d/laptopd
@@ -14,11 +14,11 @@ start() {
 	setkeycodes e003 236   # Fn+F2 maps to battery
 	setkeycodes e004 148   # Fn+F5 maps to prog1
 	setkeycodes e006 238   # Fn+F9 maps to wlan
-#?load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)!=0x8086||load(/sys/bus/pci/devices/0000:00:02.0/device,root)!=0xa011#
+#?load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)!=0x8086||load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)!=0xa011#
 	setkeycodes e008 225   # Fn+Up maps to brightnessup
 	setkeycodes e009 224   # Fn+Dn maps to brightnessdown
 #load#
-#?load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
+#?load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
 	setkeycodes e008 230   # Fn+Up maps to brightnessup
 	setkeycodes e009 229   # Fn+Dn maps to brightnessdown
 	/usr/bin/backlight set aa
diff --git a/install/2laptop/samsung/gconf/%gconf-tree.xml b/install/2laptop/samsung/gconf/%gconf-tree.xml
index ea92328..ba0b1dc 100644
--- a/install/2laptop/samsung/gconf/%gconf-tree.xml
+++ b/install/2laptop/samsung/gconf/%gconf-tree.xml
@@ -4,7 +4,7 @@
 	<dir name="desktop">
 		<dir name="gnome">
 			<dir name="keybindings">
-#?load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
+#?load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
 				<dir name="custom_brn_up">
 					<entry name="action" schema="/schemas/desktop/gnome/keybindings/custom_brn_up/action"/>
 					<entry name="name" schema="/schemas/desktop/gnome/keybindings/custom_brn_up/name"/>
@@ -33,7 +33,7 @@
 		<dir name="desktop">
 			<dir name="gnome">
 				<dir name="keybindings">
-#?load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
+#?load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
 					<dir name="custom_brn_up">
 						<entry name="action" type="schema" stype="string" owner="calculate">
 							<local_schema locale="C" short_desc="Increase backlight">
diff --git a/install/2laptop/samsung/khotkeys/brightdown.khotkeys b/install/2laptop/samsung/khotkeys/brightdown.khotkeys
index fa100f5..94beb05 100644
--- a/install/2laptop/samsung/khotkeys/brightdown.khotkeys
+++ b/install/2laptop/samsung/khotkeys/brightdown.khotkeys
@@ -1,4 +1,4 @@
-# Calculate load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011
+# Calculate load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011
 [Data]
 DataCount=1
 
diff --git a/install/2laptop/samsung/khotkeys/brightup.khotkeys b/install/2laptop/samsung/khotkeys/brightup.khotkeys
index 824e05e..35b49ab 100644
--- a/install/2laptop/samsung/khotkeys/brightup.khotkeys
+++ b/install/2laptop/samsung/khotkeys/brightup.khotkeys
@@ -1,4 +1,4 @@
-# Calculate load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011
+# Calculate load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011
 [Data]
 DataCount=1
 
diff --git a/install/2laptop/samsung/khotkeys/togglebacklight.khotkeys b/install/2laptop/samsung/khotkeys/togglebacklight.khotkeys
index 55653a7..5442406 100644
--- a/install/2laptop/samsung/khotkeys/togglebacklight.khotkeys
+++ b/install/2laptop/samsung/khotkeys/togglebacklight.khotkeys
@@ -1,4 +1,4 @@
-# Calculate load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011
+# Calculate load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011
 [Data]
 DataCount=1
 
diff --git a/install/2laptop/samsung/libxfcegui4/xfce4-keyboard-shortcuts.xml b/install/2laptop/samsung/libxfcegui4/xfce4-keyboard-shortcuts.xml
index 825ea23..98baf82 100644
--- a/install/2laptop/samsung/libxfcegui4/xfce4-keyboard-shortcuts.xml
+++ b/install/2laptop/samsung/libxfcegui4/xfce4-keyboard-shortcuts.xml
@@ -4,7 +4,7 @@
 <channel name="xfce4-keyboard-shortcuts" version="1.0">
   <property name="commands" type="empty">
     <property name="default" type="empty">
-#?load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
+#?load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
       <property name="XF86KbdBrightnessUp" type="string" value="sudo /usr/bin/backlight up"/>
       <property name="XF86KbdBrightnessDown" type="string" value="sudo /usr/bin/backlight down"/>
       <property name="XF86Launch1" type="string" value="sudo /usr/bin/backlight toggle"/>
diff --git a/install/2laptop/samsung/sudo/sudoers b/install/2laptop/samsung/sudo/sudoers
index 78e10bd..615c2b6 100644
--- a/install/2laptop/samsung/sudo/sudoers
+++ b/install/2laptop/samsung/sudo/sudoers
@@ -2,10 +2,10 @@
 <reg>(# BEGIN laptop configuration)
 .*(# END laptop configuration)</reg>
 <text>\1
-#?load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)!=0x8086||load(/sys/bus/pci/devices/0000:00:02.0/device,root)!=0xa011#
+#?load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)!=0x8086||load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)!=0xa011#
 Cmnd_Alias NETBOOK_CMDS = /usr/bin/cpufreq_toggle_osd, /usr/bin/cpufreq_toggle
 #load#
-#?load(/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
+#?load(char,/sys/bus/pci/devices/0000:00:02.0/vendor,root)==0x8086&&load(char,/sys/bus/pci/devices/0000:00:02.0/device,root)==0xa011#
 Cmnd_Alias NETBOOK_CMDS = /usr/bin/backlight, /usr/bin/cpufreq_toggle_osd, /usr/bin/cpufreq_toggle
 #load#
 %users	ALL=(ALL) NOPASSWD: NETBOOK_CMDS 
