diff --git a/install/2laptop/samsung/calculate-install/bin/backlight b/install/2laptop/samsung/calculate-install/bin/backlight
index eb8e2d0..8ccef8e 100644
--- a/install/2laptop/samsung/calculate-install/bin/backlight
+++ b/install/2laptop/samsung/calculate-install/bin/backlight
@@ -21,13 +21,23 @@ var1d=$((0x$var1))
 case "$1" in
        up)
                #calculate new brightness
-               var2=`echo "ibase=10; obase=16; a=($var1d+$AMOUNT);if (a<255) print a else print 255" | bc`
+               var2=$(($var1d + $AMOUNT))
+               if [[ $var2 -gt 255 ]]
+               then
+                      var2=255
+               fi
+               var2=`printf "%x" $var2`
                echo "$0: increasing brightness from 0x$var1 to 0x$var2"
                setpci -s $DEVICE F4.B=$var2
                ;;
        down)
                #calculate new brightness
-               var2=`echo "ibase=10; obase=16; a=($var1d-$AMOUNT);if (a>$MIN) print a else print $MIN" | bc`
+               var2=$(($var1d - $AMOUNT))
+               if [[ $var2 -lt $MIN ]]
+               then
+                      var2=$MIN
+               fi
+               var2=`printf "%x" $var2`
                echo "$0: decreasing brightness from 0x$var1 to 0x$var2"
                setpci -s $DEVICE F4.B=$var2
                ;;
