diff --git a/core/efuse/rtw_efuse.c b/core/efuse/rtw_efuse.c
index a357253..667f5f8 100755
--- a/core/efuse/rtw_efuse.c
+++ b/core/efuse/rtw_efuse.c
@@ -3204,7 +3204,9 @@ int storeAdaptorInfoFile(char *path, u8 *efuse_data)
 int retriveAdaptorInfoFile(char *path, u8 *efuse_data)
 {
 	int ret = _SUCCESS;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 	mm_segment_t oldfs;
+#endif
 	struct file *fp;
 
 	if (path && efuse_data) {
diff --git a/core/rtw_ieee80211.c b/core/rtw_ieee80211.c
index 846df87..1d15677 100755
--- a/core/rtw_ieee80211.c
+++ b/core/rtw_ieee80211.c
@@ -1522,7 +1522,9 @@ int rtw_get_mac_addr_intel(unsigned char *buf)
 	int ret = 0;
 	int i;
 	struct file *fp = NULL;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0))
 	mm_segment_t oldfs;
+#endif
 	unsigned char c_mac[MAC_ADDRESS_LEN];
 	char fname[] = "/config/wifi/mac.txt";
