diff --git a/cl-lxc b/cl-lxc
index 8096f72..8103d2b 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -426,6 +426,7 @@ create_lxc() {
 	rm -rf $path_lxc/rootfs/var/calculate/distfiles;mkdir $path_lxc/rootfs/var/calculate/distfiles
 	eend
 
+	local calculate_mount= calculate_dir=
 	if [ $prepare_ == 0 ]
 	then
 		ebegin $"Moving /var/calculate to a separate mount point"
@@ -436,19 +437,34 @@ create_lxc() {
 		calculate_dir="${path_lxc}/calculate"
 		eend
 	else
-		calculate_mount=""
 		calculate_dir="${path_lxc}/rootfs/var/calculate"
 	fi
 
-	local container_mount=""
-	if ! [ -d /var/db/repos/container ]
+	local portage_mount= gentoo_from= gentoo_to=
+	if [ -d /var/db/repos/gentoo ]
 	then
-		ewarn $"Skipping mounting Container overlay"
-		container_mount="#lxc.mount.entry = /var/db/repos/container var/db/repos/container none ro,bind 0 0
-"
+		gentoo_from='/var/db/repos/gentoo'
 	else
+		gentoo_from='/usr/portage'
+	fi
+	if [ -d $path_lxc/rootfs/var/db/repos/gentoo ]
+	then
+		gentoo_to='var/db/repos/gentoo'
+	else
+		gentoo_to='usr/portage'
+	fi
+	rm -rf ${path_lxc}/rootfs/${gentoo_to}; mkdir ${path_lxc}/rootfs/${gentoo_to}
+	portage_mount="lxc.mount.entry = ${gentoo_from} ${gentoo_to} none ro,bind 0 0"
+
+	local container_mount=
+	if [ -d /var/db/repos/container ]
+	then
 		rm -rf $path_lxc/rootfs/var/db/repos/container;	mkdir $path_lxc/rootfs/var/db/repos/container
 		container_mount="lxc.mount.entry = /var/db/repos/container var/db/repos/container none ro,bind 0 0
+"
+	else
+		ewarn $"Skipping mounting Container overlay"
+		container_mount="#lxc.mount.entry = /var/db/repos/container var/db/repos/container none ro,bind 0 0
 "
 	fi
 
@@ -479,7 +495,7 @@ lxc.net.0.veth.pair = lxc-${name_lxc}"
 	cat << EOL >> ${path_lxc}/config
 ${network_conf}
 
-${calculate_mount}lxc.mount.entry = /usr/portage usr/portage none ro,bind 0 0
+${calculate_mount}${portage_mount}
 lxc.mount.entry = /var/db/repos/calculate var/db/repos/calculate none ro,bind 0 0
 ${container_mount}lxc.mount.entry = /var/cache/edb/binhost var/cache/edb/binhost none ro,bind 0 0
 lxc.mount.entry = /var/calculate/packages var/calculate/packages none rw,bind 0 0
