diff --git a/cl-lxc b/cl-lxc
index 2914fe9..3f69f91 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -550,6 +550,9 @@ update_prepare() {
 	einfo $"Launching package update"
 	lxc-attach $name_lxc -- cl-update -f
 	rm -rf "$path_lxc/rootfs/var/calculate/{tmp/*,templates/*}"
+	cd $path_lxc/rootfs/etc/runlevels/boot
+	ln -s /etc/init.d/calculate .
+	cd - &>/dev/null
 
 	ebegin $"Stopping container"
 	lxc-stop $name_lxc
@@ -566,7 +569,7 @@ update_prepare() {
 	ebegin $"Removing temporary container directory $name_lxc"
 	if [ $type_fs = 'btrfs' ]
 	then
-		btrfs subvolume delete $path_lxc >&/dev/null
+		btrfs subvolume delete $path_lxc &>/dev/null
 		if [ -n "$path_change" ]
 		then
 			rm $path_def/$name_lxc # удалим символическую ссылку
