diff --git a/cl-lxc b/cl-lxc
index cd411c0..77ee670 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -490,7 +490,7 @@ create_lxc() {
 		net_conf=none
 		network_conf="lxc.net.0.type = none"
 		cp /etc/resolv.conf $path_lxc/rootfs/etc
-		echo -e "\nconfig_eth0=\"noop\"" >> $path_lxc/rootfs/etc/conf.d/net
+		echo -e "\nrc_provide=\"net\"" >> $path_lxc/rootfs/etc/conf.d/net
 	else
 		net_conf=veth
 		local random_mac=$(printf '06:%02X:%02X:%02X:%02X:%02X\n' \
