diff --git a/cl-lxc b/cl-lxc
index 099a0ce..2cee054 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -477,12 +477,13 @@ create_lxc() {
 		net_conf=none
 		network_conf="lxc.net.0.type = none"
 		cp /etc/resolv.conf $path_lxc/rootfs/etc
+		echo -e "\nconfig_eth0=\"noop\"" >> $path_lxc/rootfs/etc/conf.d/net
 	else
 		net_conf=veth
 		local random_mac=$(printf '06:%02X:%02X:%02X:%02X:%02X\n' \
 			$[RANDOM%256] $[RANDOM%256] $[RANDOM%256] $[RANDOM%256] $[RANDOM%256])
 		local veth_pair="lxc.net.0.veth.pair = lxc-${name_lxc::11}"
-		if grep -q "$veth_pair" -r ${path_def}/*/config
+		if grep -q "$veth_pair" -r ${path_def}/*/config &>/dev/null
 		then
 			# исключим настройку сетевого имени, если такое уже используется
 			veth_pair="#${veth_pair}"
