diff --git a/cl-lxc b/cl-lxc
index 5d42270..2fd324a 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -37,6 +37,7 @@ print_help(){
 ${0##*/} installation, configuration and update of an LXC container with Calculate Linux on board
 
 Parameters:
+  -P, --profile		select profile flavour ('list' to view all that applies)
   -c, --create		create new container
   --path		path to container
   -p, --prepare		get rootfs ready for update
@@ -46,7 +47,6 @@ Parameters:
   -m, --mirror		mirror address
   -h, --help		show this help page
 "
-# -P, --profile		select profile flavour ('list' to view all that applies)
 	else
 		echo $usage
 	fi
@@ -147,15 +147,32 @@ get_args(){
 }
 
 print_profile(){
+	local profile_dir
+	if [[ -e /var/db/repos/container/profiles/profiles.desc ]]
+	then
+		profile_dir=/var/db/repos/container
+	else
+		profile_dir=/var/calculate/tmp/update/container
+		if [[ ! -d $profile_dir ]]
+		then
+			git clone https://github.com/calculatelinux/container $profile_dir &>/dev/null
+		else
+			git -C $profile_dir pull &>/dev/null
+		fi
+	fi
+
+	# Отобразим список доступных профилей или проверим наличие запрошенного
 	if (( $# == 0 ))
 	then
-		echo $"Profile:
-  [directory]	LDAP server for centralized storage of user accounts, applications and access rights
-  [scratch]	Calculate Linux scratch (by default)
-  [steam]	Video game digital distribution service and storefront by Valve
-  [taiga]	Project management tool for cross-functional agile teams to work effectively"
-  		return
-  	fi
+		echo $"Profile:"
+		grep "^amd64" ${profile_dir}/profiles/profiles.desc | awk '{print " * " $2, $3}' OFS=' (' ORS=')\n'
+		return
+	else
+		if ! grep -q "^amd64\s*${1}/" ${profile_dir}/profiles/profiles.desc
+		then
+			return 1
+		fi
+	fi
 }
 
 set_vars(){
@@ -235,21 +252,12 @@ set_vars(){
 		num_next=
 	fi
 
-	case "$profile_" in
-	directory)
-		lxc_profile=directory
-		;;
-	scratch)
-		lxc_profile=scratch
-		;;
-	steam)
-		lxc_profile=steam
-		;;
-	taiga)
-		lxc_profile=taiga
-		;;
-	'')
-		if [ $prepare_ == 1 ] || [ $upgrade_ == 1 ]
+	# taiga/amd64/20 -> taiga
+	lxc_profile=${profile_%%/*}
+
+	if [[ $lxc_profile == "" ]]
+	then
+		if [[ $prepare_ == 1 || $upgrade_ == 1 ]]
 		then
 			lxc_profile=$(file $path_work/$lxc_/rootfs$num_cur/etc/portage/make.profile | \
 				awk -F '/profiles/' '{ print $2 }' | awk -F '/' '{ print $1 }')
@@ -265,13 +273,12 @@ set_vars(){
 		else
 			lxc_profile=scratch
 		fi
-		;;
-	*)
-		echo $"Wrong --profile parameter, use 'list' to view accepted values." >&2
-		exit 1
-		;;
-	esac
-
+	else
+		print_profile $lxc_profile || {
+			echo $"Wrong --profile parameter, use 'list' to view accepted values." >&2
+			exit 1
+		}
+	fi
 	true
 }
 
@@ -303,6 +310,10 @@ show(){
 	echo ---------------------------------------------
 }
 
+launch_message(){
+	printf $"Your container is ready. To start it, please run 'lxc-start %s'.\n" $name_lxc
+}
+
 check_vars(){
 	program_name=${0##*/}
 	if [ "$path_" != '' ] && ! [ -d "$path_" ]
@@ -598,7 +609,7 @@ create_lxc(){
 
 	if [[ -z "$name_upgrading" && $lxc_profile == 'scratch' ]]
 	then
-		printf $"Your container is ready. To start it, please run 'lxc-start %s'.\n" $name_lxc
+		launch_message
 	fi
 }
 
@@ -627,6 +638,8 @@ change_profile(){
 	einfo $"Launching package update"
 	lxc-attach $name_lxc -- cl-update -f
 	eend
+
+	launch_message
 }
 
 update_prepare(){
