diff --git a/cl-lxc b/cl-lxc
index 70a965e..9072aed 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -479,7 +479,7 @@ create_lxc() {
 		cp /etc/resolv.conf $path_lxc/rootfs/etc
 	else
 		net_conf=veth
-		local random_mac=$(echo -n '02:'; hexdump -n5 -e '/1 ":%02X"' /dev/random | sed s/^://g)
+		local random_mac=$(printf '06:%02X:%02X:%02X:%02X:%02X\n' $[RANDOM%256] $[RANDOM%256] $[RANDOM%256] $[RANDOM%256] $[RANDOM%256])
 		network_conf="lxc.net.0.type = veth
 lxc.net.0.flags = up
 lxc.net.0.name = eth0
