diff --git a/cl-lxc b/cl-lxc
index 3f69f91..20b2985 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -146,6 +146,7 @@ print_distro(){
 	then
 		echo $"Distributive:
   [CCS]		Calculate Container Scratch (by default)
+  [CDS]         Calculate Directory Container
   [CCG]		Calculate Container Games"
   		return
   	fi
@@ -154,6 +155,9 @@ print_distro(){
 	CCS)
 		echo "Calculate Container Scratch"
 		;;
+	CDC)
+		echo "Calculate Directory Container"
+		;;
 	CCG)
 		echo "Calculate Container Games"
 		;;
@@ -239,6 +243,9 @@ set_vars(){
 	CCS|ccs)
 		lxc_distro=CCS
 		;;
+	CDC|cdc)
+		lxc_distro=CDC
+		;;
 	CCG|ccg)
 		lxc_distro=CCG
 		;;
