diff --git a/usr/sbin/cl-lxc b/cl-lxc
index 4fc3ac1..0a32014 100755
--- a/usr/sbin/cl-lxc
+++ b/cl-lxc
@@ -128,7 +128,7 @@ get_args(){
 	elif (( $# == 1 ))
 	then
 		lxc_=$1
-		if [[ $lxc_ =~ [^A-Za-z0-9_\.] ]]
+		if [[ $lxc_ =~ [^A-Za-z0-9_\._\-] ]]
 		then
 			echo $"Container name contains wrong characters." >&2
 			exit 1
@@ -146,6 +146,7 @@ print_distro(){
 	then
 		echo $"Distributive:
   [CCS]		Calculate Container Scratch (by default)
+  [CDC]		Calculate Directory Container
   [CCG]		Calculate Container Games"
   		return
   	fi
@@ -154,6 +155,9 @@ print_distro(){
 	CCS)
 		echo "Calculate Container Scratch"
 		;;
+	CDC)
+		echo "Calculate Directory Container"
+		;;
 	CCG)
 		echo "Calculate Container Games"
 		;;
