diff --git a/cl-lxc b/cl-lxc
index 8c29bde..4fc3ac1 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -616,7 +616,7 @@ upgrade_lxc() {
 
 	ebegin $"Copying ssh keys"
 	cp $path_work/$name_upgrading/rootfs$num_cur/etc/ssh/*_key* \
-	   $path_work/$name_upgrading/rootfs$num_next/etc/ssh
+	   $path_work/$name_upgrading/rootfs$num_next/etc/ssh 2>/dev/null | true
 	eend
 
 	ebegin $"Preparing first boot setup script"
