diff --git a/cl-lxc b/cl-lxc
index 8103d2b..9a53c2d 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -419,11 +419,20 @@ create_lxc() {
 	fi
 
 	ebegin $"Preparing base mount paths"
-	rm -rf $path_lxc/rootfs/usr/portage;		mkdir $path_lxc/rootfs/usr/portage
-	rm -rf $path_lxc/rootfs/var/db/repos/calculate;	mkdir $path_lxc/rootfs/var/db/repos/calculate
-	rm -rf $path_lxc/rootfs/var/cache/edb/binhost; 	mkdir $path_lxc/rootfs/var/cache/edb/binhost
-	rm -rf $path_lxc/rootfs/var/calculate/packages;	mkdir $path_lxc/rootfs/var/calculate/packages
-	rm -rf $path_lxc/rootfs/var/calculate/distfiles;mkdir $path_lxc/rootfs/var/calculate/distfiles
+	for path_res in	usr/portage \
+			var/cache/edb/binhost \
+			var/calculate/distfiles \
+			var/calculate/packages \
+			var/db/repos/calculate \
+			var/db/repos/gentoo
+	do
+		path_res=${path_lxc}/rootfs/${path_res}
+		if [ -e $path_res ]
+		then
+			rm -rf $path_res
+			mkdir $path_res
+		fi
+	done
 	eend
 
 	local calculate_mount= calculate_dir=
@@ -453,13 +462,13 @@ create_lxc() {
 	else
 		gentoo_to='usr/portage'
 	fi
-	rm -rf ${path_lxc}/rootfs/${gentoo_to}; mkdir ${path_lxc}/rootfs/${gentoo_to}
 	portage_mount="lxc.mount.entry = ${gentoo_from} ${gentoo_to} none ro,bind 0 0"
 
 	local container_mount=
 	if [ -d /var/db/repos/container ]
 	then
-		rm -rf $path_lxc/rootfs/var/db/repos/container;	mkdir $path_lxc/rootfs/var/db/repos/container
+		rm -rf $path_lxc/rootfs/var/db/repos/container
+		mkdir $path_lxc/rootfs/var/db/repos/container
 		container_mount="lxc.mount.entry = /var/db/repos/container var/db/repos/container none ro,bind 0 0
 "
 	else
@@ -543,15 +552,14 @@ update_prepare() {
 	printf $"Package update for container %s\n" $name_lxc
 
 	ebegin $"Moving installed packages, flags and masks from main container"
-	cp $path_work/$name_upgrading/rootfs$num_cur/etc/portage/sets/custom $path_lxc/rootfs/etc/portage/sets/custom &>/dev/null || true
-	cp $path_work/$name_upgrading/rootfs$num_cur/etc/portage/package.accept_keywords/custom \
-		$path_lxc/rootfs/etc/portage/package.accept_keywords/custom &>/dev/null || true
-	cp $path_work/$name_upgrading/rootfs$num_cur/etc/portage/package.use/custom \
-		$path_lxc/rootfs/etc/portage/package.use/custom &>/dev/null || true
-	cp $path_work/$name_upgrading/rootfs$num_cur/etc/portage/package.mask/custom \
-		$path_lxc/rootfs/etc/portage/package.mask/custom &>/dev/null || true
-	cp $path_work/$name_upgrading/rootfs$num_cur/etc/portage/package.unmask/custom \
-		$path_lxc/rootfs/etc/portage/package.unmask/custom &>/dev/null || true
+	for migrate_file in	/etc/portage/package.accept_keywords/custom \
+				/etc/portage/package.mask/custom \
+				/etc/portage/package.unmask/custom \
+				/etc/portage/package.use/custom \
+				/etc/portage/sets/custom
+	do
+		cp ${path_work}/${name_upgrading}/rootfs${num_cur}${migrate_file} ${path_lxc}/rootfs${migrate_file} &>/dev/null || true
+	done
 	eend
 
 	ebegin $"Creating mount points"
