diff --git a/cl-lxc b/cl-lxc
index d18b29d..f93ee15 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -590,7 +590,7 @@ change_profile(){
 	echo $"Change the profile to $lxc_profile"
 	start_container
 
-	cl-update -s
+	cl-update -s -o
 
 	lxc-attach $name_lxc -- cl-update-profile -f $lxc_profile
 
@@ -654,7 +654,7 @@ update_prepare(){
 
 	start_container
 
-	cl-update -s
+	cl-update -s -o
 
 	if [[ $(file $path_lxc/rootfs/etc/portage/make.profile | \
 		awk -F '/profiles/' '{ print $2 }' | awk -F '/' '{ print $1 }') != $lxc_profile ]]
