diff --git a/cl-lxc b/cl-lxc
index 5dc5f1a..cd411c0 100755
--- a/cl-lxc
+++ b/cl-lxc
@@ -37,7 +37,6 @@ print_help(){
 ${0##*/} installation, configuration and update of an LXC container with Calculate Linux on board
 
 Parameters:
-  -D, --distro		select distribution flavour ('list' to view all that applies)
   -c, --create		create new container
   --path		path to container
   -p, --prepare		get rootfs ready for update
@@ -47,6 +46,7 @@ Parameters:
   -m, --mirror		mirror address
   -h, --help		show this help page
 "
+#  -D, --distro		select distribution flavour ('list' to view all that applies)
 	else
 		echo $usage
 	fi
@@ -255,7 +255,7 @@ set_vars(){
 			lxc_distro=$(file $path_work/$lxc_/rootfs$num_cur/etc/portage/make.profile | \
 				awk -F '/profiles/' '{ print $2 }' | awk -F '/' '{ print $1 }')
 		else
-			lxc_distro=CCS
+			lxc_distro=scratch
 		fi
 		;;
 	*)
@@ -368,8 +368,10 @@ create_base() {
 			mkdir $path_cache
 		fi
 	fi
+
 	lxc-create -n $lxc_distro -t download -- --server $mirror_ \
-			--arch x86_64 --dist $lxc_distro --release $lxc_release &>/dev/null || {
+			--arch x86_64 --dist $lxc_distro --release live \
+			2>>/var/log/calculate/cl-lxc-error.log >/dev/null || {
 		eend $? || true
 		eerror $"Failed to run lxc-create" >&2
 		rm -rf $path_cache
