# Calculate format=diff
diff --git a/sc/source/ui/view/cellsh4.cxx b/sc/source/ui/view/cellsh4.cxx
index dc682c5..226e090 100644
--- a/sc/source/ui/view/cellsh4.cxx
+++ b/sc/source/ui/view/cellsh4.cxx
@@ -128,9 +128,6 @@ void ScCellShell::ExecuteCursor( SfxRequest& rReq )
     // If ScrollLock key is active, cell cursor stays on the current cell while
     // scrolling the grid.
     bool bScrollLock = false;
-    KeyIndicatorState eState = GetFrameWin()->GetIndicatorState();
-    if (eState & KeyIndicatorState::SCROLLLOCK)
-        bScrollLock = true;
 
     //OS: once for all should do, however!
     pTabViewShell->ExecuteInputDirect();
