diff --git a/profile/samba/etc/samba/smb.conf b/profile/samba/etc/samba/smb.conf
index d11542c..4b1a877 100644
--- a/profile/samba/etc/samba/smb.conf
+++ b/profile/samba/etc/samba/smb.conf
@@ -86,7 +86,7 @@ chown=root:root
 	hide unreadable = yes
 	hide dot files = yes
 	socket options = TCP_NODELAY SO_RCVBUF=8192 SO_SNDBUF=8192
-
+	server min protocol = NT1
  
 	# Поддержка wins
 	# --------------------------------------------------------------------
