commit 29425ac4e201ff5702d8d7adea55028ce3d2749b
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Fri Mar 30 11:09:57 2012 +0400

    Получение MAC и IP адресов без утилиты ifconfig.

diff --git a/pym/cl_ldap.py b/pym/cl_ldap.py
index 385ccdf..e286eda 100644
--- a/pym/cl_ldap.py
+++ b/pym/cl_ldap.py
@@ -21486,30 +21486,11 @@ network %s")%net)
 
     def getMacInIp(self, ip):
         """Находит mac адрес по ip адресу"""
-        netInterfaces=cl_utils.getdirlist("/sys/class/net/")
-        flagFound = False
-        for i in netInterfaces:
-            execStr = "/sbin/ifconfig %s"%i
-            res = self.execProg(execStr, None, False)
-            if not res:
-                break
-            mac = ""
-            for line in res:
-                sMac = re.search('HWaddr\s+(([0-9a-f]{2}:){5}[0-9a-f]{2})',\
-                        line)
-                if sMac:
-                    # mac адрес
-                    mac = sMac.groups()[0]
-                    continue
-                sIP = re.search('addr:([0-9\.]+).+Bcast:', line)
-                if sIP:
-                    Ip = sIP.groups()[0]
-                    if ip == Ip and mac:
-                        flagFound = True
-                        break
-        if not flagFound:
-            mac = ""
-        return mac
+        for iface in cl_utils.getInterfaces():
+            mac,iface_ip = cl_utils.getMac(iface),cl_utils.getIp(iface)
+            if iface_ip == ip and mac:
+                return mac.lower()
+        return ""
 
     def setupDhcpServer(self, options):
         """Начальная настройка DHCP сервиса"""
