diff --git a/pym/cl_ldap.py b/pym/cl_ldap.py
--- pym/cl_ldap.py
+++ pym/cl_ldap.py
@@ -6449,6 +6449,7 @@ Samba and Unix services") %", ".join(exclGroup)
         if options.has_key('c'):
             gecos = options['c']
             modAttrs.append((ldap.MOD_REPLACE, 'description', gecos))
+            modAttrs.append((ldap.MOD_REPLACE, 'displayName', gecos))
         if modAttrs:
             groupDN = self.addDN("cn=" + modGroupName, self.relGroupsDN)
             res = self.modAttrsDN(groupDN, modAttrs)
