diff --git a/profile/proxy/etc/squid/squid.conf b/profile/proxy/etc/squid/squid.conf
index a0bfc20..9a3fb48 100644
--- a/profile/proxy/etc/squid/squid.conf
+++ b/profile/proxy/etc/squid/squid.conf
@@ -24,7 +24,13 @@ htcp_access allow localnet
 htcp_access deny all
 http_port #-sr_proxy_port-#
 hierarchy_stoplist cgi-bin ?
+#?pkg(net-proxy/squid)<3.1.6#
 logformat squid %ts.%03tu %6tr %>a %Ss/%03Hs %<st %rm %ru %un %ul %ui %Sh/%<A %mt %et
+#pkg#
+#?pkg(net-proxy/squid)>=3.1.6#
+logformat squid %ts.%03tu %6tr %>a %Ss/%03>Hs %<st %rm %ru %un %ul %ui %Sh/%<A %mt %et
+#pkg#
+access_log /var/log/squid/access.log squid
 access_log /var/log/squid/access.log squid
 refresh_pattern ^ftp:           1440    20%     10080
 refresh_pattern ^gopher:        1440    0%      1440
