commit acb67f1857a42aa649b8952c29b3ab1bf5dc8935
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Mon Mar 15 17:08:57 2021 +0300

    Remove min protocol option

diff --git a/profile/samba/etc/samba/smb.conf b/profile/samba/etc/samba/smb.conf
index 4eeb2d4..03fe2f8 100644
--- a/profile/samba/etc/samba/smb.conf
+++ b/profile/samba/etc/samba/smb.conf
@@ -117,7 +117,7 @@ chown=root:root
 	hide unreadable = yes
 	hide dot files = yes
 	socket options = TCP_NODELAY SO_RCVBUF=8192 SO_SNDBUF=8192
-	server min protocol = NT1
+	!server min protocol =
  
 	# Поддержка wins
 	# --------------------------------------------------------------------
