diff --git a/profile/mail/etc/postfix/main.cf b/profile/mail/etc/postfix/main.cf
index f34b032..aca83b7 100644
--- a/profile/mail/etc/postfix/main.cf
+++ b/profile/mail/etc/postfix/main.cf
@@ -16,7 +16,12 @@ command_directory = /usr/sbin
 # daemon programs (i.e. programs listed in the master.cf file). This
 # directory must be owned by root.
 #
+#pkg(mail-mta/postfix)<2.9.3#
 daemon_directory = /usr/lib/postfix
+#pkg#
+#pkg(mail-mta/postfix)>=2.9.3#
+daemon_directory = /usr/libexec/postfix
+#pkg#
 
 #Определяет местоположение всех программ системы postfix.
 mail_owner = postfix
