commit 2787a17c985af58b6d4c7b9b76acdea70e0a03ca
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Tue Dec 11 11:44:56 2012 +0400

    Add starttls

diff --git profile/jabber/etc/jabber/ejabberd.cfg profile/jabber/etc/jabber/ejabberd.cfg
index 747c77c..5459465 100644
--- profile/jabber/etc/jabber/ejabberd.cfg
+++ profile/jabber/etc/jabber/ejabberd.cfg
@@ -13,6 +13,12 @@
   			{certfile, "/etc/jabber/ssl.pem"}, tls,
   			{max_stanza_size, 65536}
   		       ]},
+  {#-sr_jabber_port_starttls-#, ejabberd_c2s, [
+  			{access, c2s},
+  			{shaper, c2s_shaper},
+  			{certfile, "/etc/jabber/ssl.pem"}, starttls,starttls_required,
+  			{max_stanza_size, 65536}
+  		       ]},
 
   {5269, ejabberd_s2s_in, [
 			   {shaper, s2s_shaper},
@@ -125,4 +131,4 @@
 	    ]}
   ]},
   {mod_version,  []}
- ]}.
\ No newline at end of file
+ ]}.
diff --git pym/cl_vars_server.py pym/cl_vars_server.py
index 71561a1..35ecafc 100644
--- pym/cl_vars_server.py
+++ pym/cl_vars_server.py
@@ -175,6 +175,9 @@ class Data:
     # порт jabber сервиса
     sr_jabber_port = {'mode':"w",'value':'5223'}
 
+    # порт jabber сервиса
+    sr_jabber_port_starttls = {'mode':"w",'value':'5222'}
+
     # включение хранения истории сообщений jabber сервиса
     # по умолчанию выключено
     sr_jabber_history = {'mode':"w",
