# Calculate format=diff
diff --git a/pcmanfm/mainwindow.cpp b/pcmanfm/mainwindow.cpp
index 15aa4e2..43f0e5a 100644
--- a/pcmanfm/mainwindow.cpp
+++ b/pcmanfm/mainwindow.cpp
@@ -164,10 +164,22 @@ MainWindow::MainWindow(Fm::FilePath path):
     loadBookmarksMenu();
 
     // set generic icons for view actions
-    ui.actionIconView->setIcon(style()->standardIcon(QStyle::SP_FileDialogContentsView));
-    ui.actionThumbnailView->setIcon(style()->standardIcon(QStyle::SP_FileDialogInfoView));
-    ui.actionCompactView->setIcon(style()->standardIcon(QStyle::SP_FileDialogListView));
-    ui.actionDetailedList->setIcon(style()->standardIcon(QStyle::SP_FileDialogDetailedView));
+    ui.actionIconView->setIcon(QIcon::fromTheme("view-list-icons"));
+    if(ui.actionIconView->icon().isNull()) {
+	    ui.actionIconView->setIcon(style()->standardIcon(QStyle::SP_FileDialogContentsView));
+    }
+    ui.actionThumbnailView->setIcon(QIcon::fromTheme("view-list-images"));
+    if(ui.actionThumbnailView->icon().isNull()) {
+	    ui.actionThumbnailView->setIcon(style()->standardIcon(QStyle::SP_FileDialogInfoView));
+    }
+    ui.actionCompactView->setIcon(QIcon::fromTheme("view-list-text"));
+    if(ui.actionCompactView->icon().isNull()) {
+	    ui.actionCompactView->setIcon(style()->standardIcon(QStyle::SP_FileDialogListView));
+    }
+    ui.actionDetailedList->setIcon(QIcon::fromTheme("view-list-details"));
+    if(ui.actionDetailedList->icon().isNull()) {
+	    ui.actionDetailedList->setIcon(style()->standardIcon(QStyle::SP_FileDialogDetailedView));
+    }
 
     // Fix the menu groups which is not done by Qt designer
     // To my suprise, this was supported in Qt designer 3 :-(
