# Calculate format=diff
diff --git a/src/xfpm-manager.c b/src/xfpm-manager.c
index e9acaa0..a08f42c 100644
--- a/src/xfpm-manager.c
+++ b/src/xfpm-manager.c
@@ -441,12 +441,9 @@ xfpm_manager_lid_changed_cb (XfpmPower *power, gboolean lid_is_closed, XfpmManag
   {
     XFPM_DEBUG_ENUM (action, XFPM_TYPE_LID_TRIGGER_ACTION, "LID close event");
 
-    if ( action == LID_TRIGGER_NOTHING )
-    {
-      if ( !xfpm_is_multihead_connected () )
-        xfpm_dpms_force_level (manager->priv->dpms, DPMSModeOff);
-    }
-    else if ( action == LID_TRIGGER_LOCK_SCREEN )
+    if ( action != LID_TRIGGER_NOTHING && !xfpm_is_multihead_connected () )
+      xfpm_dpms_force_level (manager->priv->dpms, DPMSModeOff);
+    if ( action == LID_TRIGGER_LOCK_SCREEN )
     {
       if ( !xfpm_is_multihead_connected () )
       {
@@ -472,7 +469,7 @@ xfpm_manager_lid_changed_cb (XfpmPower *power, gboolean lid_is_closed, XfpmManag
   else
   {
     XFPM_DEBUG_ENUM (action, XFPM_TYPE_LID_TRIGGER_ACTION, "LID opened");
-
+    g_usleep (1000000);
     xfpm_dpms_force_level (manager->priv->dpms, DPMSModeOn);
   }
 }
