# Calculate format=diff
diff --git a/src/qlippersystray.cpp b/src/qlippersystray.cpp
index 8e3a2f3..ec6b823 100644
--- a/src/qlippersystray.cpp
+++ b/src/qlippersystray.cpp
@@ -35,7 +35,7 @@ QlipperSystray::QlipperSystray(QObject *parent)
       , m_shortcutMenu(0)
 #endif
 {
-    setIcon(QIcon(":/icons/qlipper.png"));
+    setIcon(QIcon::fromTheme("qlipper"));
 
     m_model = new QlipperModel(this);
 
