diff -ruN spellcheck-2.0.orig/applet/SpellCheck.cpp spellcheck-2.0/applet/SpellCheck.cpp
--- spellcheck-2.0.orig/applet/SpellCheck.cpp	2009-04-30 23:06:41.000000000 +0400
+++ spellcheck-2.0/applet/SpellCheck.cpp	2009-05-10 14:25:24.930256063 +0400
@@ -182,7 +182,7 @@
 
         if (pasteText)
         {
-            m_textEdit->setText(QApplication::clipboard()->text());
+            m_textEdit->setText(QApplication::clipboard()->text(QClipboard::Selection));
         }
 
         m_textEdit->setFocus();
@@ -201,10 +201,12 @@
 {
     if (!m_textEdit->textCursor().selectedText().isEmpty())
     {
+        QApplication::clipboard()->setText(m_textEdit->textCursor().selectedText(),QClipboard::Selection);
         QApplication::clipboard()->setText(m_textEdit->textCursor().selectedText());
     }
     else
     {
+        QApplication::clipboard()->setText(m_textEdit->toPlainText(),QClipboard::Selection);
         QApplication::clipboard()->setText(m_textEdit->toPlainText());
     }
 }
