diff --git a/lxqt-leave/CMakeLists.txt b/lxqt-leave/CMakeLists.txt
index d54cdcd..0f8038f 100644
--- a/lxqt-leave/CMakeLists.txt
+++ b/lxqt-leave/CMakeLists.txt
@@ -35,6 +35,16 @@ lxqt_translate_ts(lxqt-leave_QM_FILES
         ${H_FILES}
     INSTALL_DIR
         "${LXQT_TRANSLATIONS_DIR}/${PROJECT_NAME}"
+    PULL_TRANSLATIONS
+        ${PULL_TRANSLATIONS}
+    CLEAN_TRANSLATIONS
+        ${CLEAN_TRANSLATIONS}
+    TRANSLATIONS_REPO
+        ${TRANSLATIONS_REPO}
+    TRANSLATIONS_REFSPEC
+        ${TRANSLATIONS_REFSPEC}
+    REPO_SUBDIR
+        "lxqt-session/${PROJECT_NAME}"
 )
 
 lxqt_app_translation_loader(lxqt-leave_QM_LOADER ${PROJECT_NAME})
diff --git a/lxqt-leave/leavedialog.cpp b/lxqt-leave/leavedialog.cpp
index 879908b..351ce14 100644
--- a/lxqt-leave/leavedialog.cpp
+++ b/lxqt-leave/leavedialog.cpp
@@ -28,24 +28,57 @@
 #include "leavedialog.h"
 
 LeaveDialog::LeaveDialog(QWidget* parent)
-    : QDialog(parent),
+    : QDialog(parent, Qt::Dialog | Qt::WindowStaysOnTopHint | Qt::CustomizeWindowHint),
     ui(new Ui::LeaveDialog),
     mPower(new LXQt::Power(this)),
+    mPowerManager(new LXQt::PowerManager(this)),
     mScreensaver(new LXQt::ScreenSaver(this))
 {
     ui->setupUi(this);
 
+    /* This is a special dialog. We want to make it hard to ignore.
+       We make it:
+           * Undraggable
+           * Frameless
+           * Stays on top of all other windows
+           * Present in all desktops
+    */
+    setWindowFlags((Qt::CustomizeWindowHint | Qt::FramelessWindowHint |
+                    Qt::WindowStaysOnTopHint | Qt::X11BypassWindowManagerHint));
+
     ui->logoutButton->setEnabled(mPower->canAction(LXQt::Power::PowerLogout));
     ui->rebootButton->setEnabled(mPower->canAction(LXQt::Power::PowerReboot));
     ui->shutdownButton->setEnabled(mPower->canAction(LXQt::Power::PowerShutdown));
     ui->suspendButton->setEnabled(mPower->canAction(LXQt::Power::PowerSuspend));
     ui->hibernateButton->setEnabled(mPower->canAction(LXQt::Power::PowerHibernate));
 
-    connect(ui->logoutButton,       &QPushButton::clicked, [&] { close(); mPower->logout(); });
-    connect(ui->rebootButton,       &QPushButton::clicked, [&] { close(); mPower->reboot(); });
-    connect(ui->shutdownButton,     &QPushButton::clicked, [&] { close(); mPower->shutdown(); });
-    connect(ui->suspendButton,      &QPushButton::clicked, [&] { close(); mPower->suspend(); });
-    connect(ui->hibernateButton,    &QPushButton::clicked, [&] { close(); mPower->hibernate(); });
+    /*
+     * Make all the buttons have equal widths
+     */
+    QVector<QToolButton*> buttons(6);
+    buttons[0] = ui->logoutButton;
+    buttons[1] = ui->lockscreenButton;
+    buttons[2] = ui->suspendButton;
+    buttons[3] = ui->hibernateButton;
+    buttons[4] = ui->rebootButton;
+    buttons[5] = ui->shutdownButton;
+
+    int maxWidth = 0;
+    const int N = buttons.size();
+    for (int i = 0; i < N; ++i) {
+        // Make sure that the button size is adjusted to the text width
+        buttons.at(i)->adjustSize();
+        maxWidth = qMax(maxWidth, buttons.at(i)->width());
+    }
+    for (int i = 0; i < N; ++i)
+        buttons.at(i)->setMinimumWidth(maxWidth);
+
+    connect(ui->logoutButton,       &QPushButton::clicked, [&] { close(); mPowerManager->logout();    });
+    connect(ui->rebootButton,       &QPushButton::clicked, [&] { close(); mPowerManager->reboot();    });
+    connect(ui->shutdownButton,     &QPushButton::clicked, [&] { close(); mPowerManager->shutdown();  });
+    connect(ui->suspendButton,      &QPushButton::clicked, [&] { close(); mPowerManager->suspend();   });
+    connect(ui->hibernateButton,    &QPushButton::clicked, [&] { close(); mPowerManager->hibernate(); });
+    connect(ui->cancelButton,       &QPushButton::clicked, [&] { close();                             });
     connect(ui->lockscreenButton,   &QPushButton::clicked, [&] {
         close();
         QEventLoop loop;
@@ -54,10 +87,17 @@ LeaveDialog::LeaveDialog(QWidget* parent)
         loop.exec();
     });
 
-    connect(ui->cancelButton, &QPushButton::clicked, [&] { close(); });
 }
 
 LeaveDialog::~LeaveDialog()
 {
     delete ui;
 }
+
+void LeaveDialog::resizeEvent(QResizeEvent* event)
+{
+    QRect screen = QApplication::desktop()->screenGeometry();
+    move((screen.width()  - this->width()) / 2,
+         (screen.height() - this->height()) / 2);
+
+}
diff --git a/lxqt-leave/leavedialog.h b/lxqt-leave/leavedialog.h
index 58e0535..00bb391 100644
--- a/lxqt-leave/leavedialog.h
+++ b/lxqt-leave/leavedialog.h
@@ -31,7 +31,9 @@
 #include "ui_leavedialog.h"
 
 #include <QDialog>
+#include <QDesktopWidget>
 #include <LXQt/Power>
+#include <LXQt/PowerManager>
 #include <LXQt/ScreenSaver>
 
 namespace Ui {
@@ -46,9 +48,16 @@ public:
     explicit LeaveDialog(QWidget *parent = 0);
     ~LeaveDialog();
 
+protected:
+    virtual void resizeEvent(QResizeEvent* event);
+
 private:
     Ui::LeaveDialog *ui;
+    // LXQt::Power is used to know if the actions are doable, while
+    // LXQt::PowerManager is used to trigger the actions, while
+    // obeying the user option to ask or not for confirmation
     LXQt::Power *mPower;
+    LXQt::PowerManager *mPowerManager;
     LXQt::ScreenSaver *mScreensaver;
 };
 
diff --git a/lxqt-leave/leavedialog.ui b/lxqt-leave/leavedialog.ui
index ea6f62c..f768dcf 100644
--- a/lxqt-leave/leavedialog.ui
+++ b/lxqt-leave/leavedialog.ui
@@ -6,8 +6,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>617</width>
-    <height>135</height>
+    <width>456</width>
+    <height>372</height>
    </rect>
   </property>
   <property name="sizePolicy">
@@ -19,8 +19,38 @@
   <property name="windowTitle">
    <string>Leave</string>
   </property>
-  <layout class="QVBoxLayout" name="verticalLayout">
-   <item>
+  <layout class="QGridLayout" name="gridLayout_2">
+   <item row="2" column="0">
+    <widget class="QWidget" name="widget_2" native="true">
+     <layout class="QHBoxLayout" name="horizontalLayout">
+      <item>
+       <spacer name="horizontalSpacer">
+        <property name="orientation">
+         <enum>Qt::Horizontal</enum>
+        </property>
+        <property name="sizeHint" stdset="0">
+         <size>
+          <width>170</width>
+          <height>20</height>
+         </size>
+        </property>
+       </spacer>
+      </item>
+      <item>
+       <widget class="QPushButton" name="cancelButton">
+        <property name="text">
+         <string>Cancel</string>
+        </property>
+        <property name="icon">
+         <iconset theme="dialog-cancel">
+          <normaloff>.</normaloff>.</iconset>
+        </property>
+       </widget>
+      </item>
+     </layout>
+    </widget>
+   </item>
+   <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
@@ -36,109 +66,173 @@
      </property>
     </widget>
    </item>
-   <item>
+   <item row="1" column="0">
     <widget class="QWidget" name="buttonsWidget" native="true">
-     <layout class="QHBoxLayout" name="horizontalLayout_2">
-      <item>
-       <widget class="QPushButton" name="logoutButton">
-        <property name="text">
-         <string>Logout</string>
-        </property>
-        <property name="icon">
-         <iconset theme="system-log-out">
-          <normaloff/>
-         </iconset>
-        </property>
-       </widget>
-      </item>
-      <item>
-       <widget class="QPushButton" name="rebootButton">
-        <property name="text">
-         <string>Reboot</string>
-        </property>
-        <property name="icon">
-         <iconset theme="system-reboot">
-          <normaloff/>
-         </iconset>
-        </property>
-       </widget>
-      </item>
-      <item>
-       <widget class="QPushButton" name="shutdownButton">
-        <property name="text">
-         <string>Shutdown</string>
-        </property>
-        <property name="icon">
-         <iconset theme="system-shutdown">
-          <normaloff/>
-         </iconset>
+     <layout class="QGridLayout" name="gridLayout">
+      <property name="horizontalSpacing">
+       <number>12</number>
+      </property>
+      <item row="0" column="0">
+       <layout class="QVBoxLayout" name="verticalLayout_2">
+        <property name="spacing">
+         <number>6</number>
         </property>
-       </widget>
+        <item alignment="Qt::AlignHCenter">
+         <widget class="QToolButton" name="logoutButton">
+          <property name="sizePolicy">
+           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
+            <horstretch>0</horstretch>
+            <verstretch>0</verstretch>
+           </sizepolicy>
+          </property>
+          <property name="contextMenuPolicy">
+           <enum>Qt::NoContextMenu</enum>
+          </property>
+          <property name="text">
+           <string>Logout</string>
+          </property>
+          <property name="icon">
+           <iconset theme="system-log-out">
+            <normaloff>.</normaloff>.</iconset>
+          </property>
+          <property name="iconSize">
+           <size>
+            <width>64</width>
+            <height>64</height>
+           </size>
+          </property>
+          <property name="toolButtonStyle">
+           <enum>Qt::ToolButtonTextUnderIcon</enum>
+          </property>
+         </widget>
+        </item>
+        <item alignment="Qt::AlignHCenter">
+         <widget class="QToolButton" name="lockscreenButton">
+          <property name="text">
+           <string>Lock screen</string>
+          </property>
+          <property name="icon">
+           <iconset theme="system-lock-screen">
+            <normaloff>.</normaloff>.</iconset>
+          </property>
+          <property name="iconSize">
+           <size>
+            <width>64</width>
+            <height>64</height>
+           </size>
+          </property>
+          <property name="toolButtonStyle">
+           <enum>Qt::ToolButtonTextUnderIcon</enum>
+          </property>
+         </widget>
+        </item>
+       </layout>
       </item>
-      <item>
-       <widget class="QPushButton" name="suspendButton">
-        <property name="text">
-         <string>Suspend</string>
+      <item row="0" column="4">
+       <layout class="QVBoxLayout" name="verticalLayout_4">
+        <property name="spacing">
+         <number>6</number>
         </property>
-        <property name="icon">
-         <iconset theme="system-suspend">
-          <normaloff/>
-         </iconset>
-        </property>
-       </widget>
+        <item alignment="Qt::AlignHCenter">
+         <widget class="QToolButton" name="suspendButton">
+          <property name="text">
+           <string>Suspend</string>
+          </property>
+          <property name="icon">
+           <iconset theme="system-suspend">
+            <normaloff>.</normaloff>.</iconset>
+          </property>
+          <property name="iconSize">
+           <size>
+            <width>64</width>
+            <height>64</height>
+           </size>
+          </property>
+          <property name="toolButtonStyle">
+           <enum>Qt::ToolButtonTextUnderIcon</enum>
+          </property>
+         </widget>
+        </item>
+        <item alignment="Qt::AlignHCenter">
+         <widget class="QToolButton" name="hibernateButton">
+          <property name="text">
+           <string>Hibernate</string>
+          </property>
+          <property name="icon">
+           <iconset theme="system-suspend-hibernate">
+            <normaloff>.</normaloff>.</iconset>
+          </property>
+          <property name="iconSize">
+           <size>
+            <width>64</width>
+            <height>64</height>
+           </size>
+          </property>
+          <property name="toolButtonStyle">
+           <enum>Qt::ToolButtonTextUnderIcon</enum>
+          </property>
+         </widget>
+        </item>
+       </layout>
       </item>
-      <item>
-       <widget class="QPushButton" name="lockscreenButton">
-        <property name="text">
-         <string>Lock screen</string>
+      <item row="0" column="2">
+       <layout class="QVBoxLayout" name="verticalLayout_3">
+        <property name="spacing">
+         <number>6</number>
         </property>
-        <property name="icon">
-         <iconset theme="system-lock-screen">
-          <normaloff/>
-         </iconset>
-        </property>
-       </widget>
+        <item alignment="Qt::AlignHCenter">
+         <widget class="QToolButton" name="shutdownButton">
+          <property name="text">
+           <string>Shutdown</string>
+          </property>
+          <property name="icon">
+           <iconset theme="system-shutdown">
+            <normaloff>.</normaloff>.</iconset>
+          </property>
+          <property name="iconSize">
+           <size>
+            <width>64</width>
+            <height>64</height>
+           </size>
+          </property>
+          <property name="toolButtonStyle">
+           <enum>Qt::ToolButtonTextUnderIcon</enum>
+          </property>
+         </widget>
+        </item>
+        <item alignment="Qt::AlignHCenter">
+         <widget class="QToolButton" name="rebootButton">
+          <property name="text">
+           <string>Reboot</string>
+          </property>
+          <property name="icon">
+           <iconset theme="system-reboot">
+            <normaloff>.</normaloff>.</iconset>
+          </property>
+          <property name="iconSize">
+           <size>
+            <width>64</width>
+            <height>64</height>
+           </size>
+          </property>
+          <property name="toolButtonStyle">
+           <enum>Qt::ToolButtonTextUnderIcon</enum>
+          </property>
+         </widget>
+        </item>
+       </layout>
       </item>
-      <item>
-       <widget class="QPushButton" name="hibernateButton">
-        <property name="text">
-         <string>Hibernate</string>
-        </property>
-        <property name="icon">
-         <iconset theme="system-suspend-hibernate">
-          <normaloff/>
-         </iconset>
+      <item row="0" column="1">
+       <widget class="Line" name="line">
+        <property name="orientation">
+         <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
-     </layout>
-    </widget>
-   </item>
-   <item>
-    <widget class="QWidget" name="widget_2" native="true">
-     <layout class="QHBoxLayout" name="horizontalLayout">
-      <item>
-       <spacer name="horizontalSpacer">
+      <item row="0" column="3">
+       <widget class="Line" name="line_2">
         <property name="orientation">
-         <enum>Qt::Horizontal</enum>
-        </property>
-        <property name="sizeHint" stdset="0">
-         <size>
-          <width>170</width>
-          <height>20</height>
-         </size>
-        </property>
-       </spacer>
-      </item>
-      <item>
-       <widget class="QPushButton" name="cancelButton">
-        <property name="text">
-         <string>Cancel</string>
-        </property>
-        <property name="icon">
-         <iconset theme="dialog-cancel">
-          <normaloff/>
-         </iconset>
+         <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
@@ -146,7 +240,19 @@
     </widget>
    </item>
   </layout>
+  <zorder>label</zorder>
+  <zorder>buttonsWidget</zorder>
+  <zorder>widget_2</zorder>
  </widget>
+ <tabstops>
+  <tabstop>cancelButton</tabstop>
+  <tabstop>logoutButton</tabstop>
+  <tabstop>lockscreenButton</tabstop>
+  <tabstop>shutdownButton</tabstop>
+  <tabstop>rebootButton</tabstop>
+  <tabstop>suspendButton</tabstop>
+  <tabstop>hibernateButton</tabstop>
+ </tabstops>
  <resources/>
  <connections/>
 </ui>
diff --git a/lxqt-leave/main.cpp b/lxqt-leave/main.cpp
index 972cf42..aad9d4c 100644
--- a/lxqt-leave/main.cpp
+++ b/lxqt-leave/main.cpp
@@ -102,10 +102,10 @@ int main(int argc, char *argv[])
 
     LeaveDialog dialog;
     a.setActivationWindow(&dialog);
+    dialog.setFixedSize(dialog.sizeHint());
     dialog.setGeometry(QStyle::alignedRect(Qt::LeftToRight,
                 Qt::AlignCenter,
-                dialog.size(),
+                dialog.sizeHint(),
                 qApp->desktop()->screenGeometry(QCursor::pos())));
-    dialog.setMaximumSize(dialog.minimumSize());
     return dialog.exec();
 }
diff --git a/lxqt-leave/resources/lxqt-hibernate.desktop.in b/lxqt-leave/resources/lxqt-hibernate.desktop.in
index 19ed383..923b2a2 100644
--- a/lxqt-leave/resources/lxqt-hibernate.desktop.in
+++ b/lxqt-leave/resources/lxqt-hibernate.desktop.in
@@ -5,7 +5,7 @@ GenericName=Hibernate
 Comment=hibernate the machine
 Exec=lxqt-leave --hibernate
 Icon=system-suspend-hibernate
-Categories=System;X-Leave
-OnlyShowIn=LXQt
+Categories=System;X-Leave;
+OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=../translations
diff --git a/lxqt-leave/resources/lxqt-leave.1 b/lxqt-leave/resources/lxqt-leave.1
index 827ebbc..b38d829 100644
--- a/lxqt-leave/resources/lxqt-leave.1
+++ b/lxqt-leave/resources/lxqt-leave.1
@@ -1,10 +1,13 @@
-.TH lxqt-leave 1 "" "" "LXQt\ Helper"
+.TH lxqt-leave 1 "2015-11-05" "LXQt 0.10.0" "LXQt Leave Session Module"
 .SH NAME
-\fBlxqt-leave\fR \- easily logout, reboot, shutdown, suspend, hibernate and lock screen from a dialog
+\fBlxqt-leave\fR \- Dialog to easily select logout, reboot, shutdown,
+suspend, hibernate, and lock screen
 .SH SYNOPSIS
 \fBlxqt-leave\fR [\fIargument\fR]
 .SH DESCRIPTION
-\fBlxqt-leave\fR is a graphical Qt tool for easily triggering leave session actions: logout, reboot, shutdown, hibernate and lock screen. If run with an argument, the dialog is not shown and the action is executed directly.
+\fBlxqt-leave\fR is a graphical Qt tool for easily triggering leave session
+actions: logout, reboot, shutdown, hibernate, and lock screen. If run with
+an argument, the dialog is not shown and the action is executed directly.
 .SH OPTIONS
 .PP
 The options which apply to the \fBlxqt-leave\fR command are:
@@ -31,5 +34,16 @@ The options which apply to the \fBlxqt-leave\fR command are:
 .RE
 .SH "REPORTING BUGS"
 Report bugs to https://github.com/lxde/lxqt/issues
+.SH "SEE ALSO"
+.\" any module must refer to the session application, for module overview and initiation
+\fBstartlxqt.1\fR  LXQt session initialization and launch script (e.g. in \fB.xinitrc\fR)
+.P
+\fBlxqt-session.1\fR  LXQt \fIoverview\fR and complete session environment
+.P
+\fBlxqt-config-session.1\fR  LXQt default and autostart applications settings,
+plus environment settings
+.P
+\fBlxqt-config.1\fR  LXQt settings Configuration Center interface
+.P
 .SH AUTHOR
 This manual page was created by \fBPaulo Lieuthier\fR \fI<paulolieuthier@gmail.com>\fR for the \fBLXQt\fR project.
diff --git a/lxqt-leave/resources/lxqt-leave.desktop.in b/lxqt-leave/resources/lxqt-leave.desktop.in
index 9608a3d..db8a620 100644
--- a/lxqt-leave/resources/lxqt-leave.desktop.in
+++ b/lxqt-leave/resources/lxqt-leave.desktop.in
@@ -5,7 +5,7 @@ GenericName=Leave
 Comment=Leave dialog
 Exec=lxqt-leave
 Icon=system-shutdown
-Categories=System;X-Leave
+Categories=System;X-Leave;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=../translations
diff --git a/lxqt-leave/resources/lxqt-lockscreen.desktop.in b/lxqt-leave/resources/lxqt-lockscreen.desktop.in
index 6866095..81dea85 100644
--- a/lxqt-leave/resources/lxqt-lockscreen.desktop.in
+++ b/lxqt-leave/resources/lxqt-lockscreen.desktop.in
@@ -5,7 +5,7 @@ GenericName=Lock Screen
 Comment=Lock the current session
 Exec=lxqt-leave --lockscreen
 Icon=system-lock-screen
-Categories=System;Screensaver
+Categories=System;Screensaver;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=../translations
diff --git a/lxqt-leave/resources/lxqt-logout.desktop.in b/lxqt-leave/resources/lxqt-logout.desktop.in
index 763f33a..a4da833 100644
--- a/lxqt-leave/resources/lxqt-logout.desktop.in
+++ b/lxqt-leave/resources/lxqt-logout.desktop.in
@@ -5,7 +5,7 @@ GenericName=Logout
 Comment=Logout from the current session
 Exec=lxqt-leave --logout
 Icon=system-log-out
-Categories=System;X-Leave
+Categories=System;X-Leave;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=../translations
diff --git a/lxqt-leave/resources/lxqt-reboot.desktop.in b/lxqt-leave/resources/lxqt-reboot.desktop.in
index f5adde6..bb10c4c 100644
--- a/lxqt-leave/resources/lxqt-reboot.desktop.in
+++ b/lxqt-leave/resources/lxqt-reboot.desktop.in
@@ -5,7 +5,7 @@ GenericName=Reboot
 Comment=reboot the machine
 Exec=lxqt-leave --reboot
 Icon=system-reboot
-Categories=System;X-Leave
+Categories=System;X-Leave;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=../translations
diff --git a/lxqt-leave/resources/lxqt-shutdown.desktop.in b/lxqt-leave/resources/lxqt-shutdown.desktop.in
index bd04c33..a1f069d 100644
--- a/lxqt-leave/resources/lxqt-shutdown.desktop.in
+++ b/lxqt-leave/resources/lxqt-shutdown.desktop.in
@@ -5,7 +5,7 @@ GenericName=Shutdown
 Comment=shutdown the machine
 Exec=lxqt-leave --shutdown
 Icon=system-shutdown
-Categories=System;X-Leave
+Categories=System;X-Leave;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=../translations
diff --git a/lxqt-leave/resources/lxqt-suspend.desktop.in b/lxqt-leave/resources/lxqt-suspend.desktop.in
index 401595d..44a0b87 100644
--- a/lxqt-leave/resources/lxqt-suspend.desktop.in
+++ b/lxqt-leave/resources/lxqt-suspend.desktop.in
@@ -5,7 +5,7 @@ GenericName=Suspend
 Comment=suspend the machine
 Exec=lxqt-leave --suspend
 Icon=system-suspend
-Categories=System;X-Leave
-OnlyShowIn=LXQt
+Categories=System;X-Leave;
+OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=../translations
diff --git a/lxqt-leave/translations/lxqt-leave.ts b/lxqt-leave/translations/lxqt-leave.ts
deleted file mode 100644
index 7c2ba24..0000000
--- a/lxqt-leave/translations/lxqt-leave.ts
+++ /dev/null
@@ -1,85 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS>
-<TS version="2.1">
-<context>
-    <name>LeaveDialog</name>
-    <message>
-        <location filename="../leavedialog.ui" line="20"/>
-        <source>Leave</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="29"/>
-        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;What do you want the computer to do?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="45"/>
-        <source>Logout</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="57"/>
-        <source>Reboot</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="69"/>
-        <source>Shutdown</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="81"/>
-        <source>Suspend</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="93"/>
-        <source>Lock screen</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="105"/>
-        <source>Hibernate</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="136"/>
-        <source>Cancel</source>
-        <translation type="unfinished"></translation>
-    </message>
-</context>
-<context>
-    <name>main</name>
-    <message>
-        <location filename="../main.cpp" line="51"/>
-        <source>Logout.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="54"/>
-        <source>Lockscreen.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="57"/>
-        <source>Suspend.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="60"/>
-        <source>Hibernate.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="63"/>
-        <source>Shutdown.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="66"/>
-        <source>Reboot.</source>
-        <translation type="unfinished"></translation>
-    </message>
-</context>
-</TS>
diff --git a/lxqt-leave/translations/lxqt-leave_de.ts b/lxqt-leave/translations/lxqt-leave_de.ts
deleted file mode 100644
index 2a33baa..0000000
--- a/lxqt-leave/translations/lxqt-leave_de.ts
+++ /dev/null
@@ -1,85 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS>
-<TS version="2.1" language="de">
-<context>
-    <name>LeaveDialog</name>
-    <message>
-        <location filename="../leavedialog.ui" line="20"/>
-        <source>Leave</source>
-        <translation>Verlassen</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="29"/>
-        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;What do you want the computer to do?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Was soll der Computer tun?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="45"/>
-        <source>Logout</source>
-        <translation>Abmelden</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="57"/>
-        <source>Reboot</source>
-        <translation>Neustart</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="69"/>
-        <source>Shutdown</source>
-        <translation>Herunterfahren</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="81"/>
-        <source>Suspend</source>
-        <translation>Bereitschaft</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="93"/>
-        <source>Lock screen</source>
-        <translation>Bildschirm sperren</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="105"/>
-        <source>Hibernate</source>
-        <translation>Tiefschlaf</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="136"/>
-        <source>Cancel</source>
-        <translation>Abbrechen</translation>
-    </message>
-</context>
-<context>
-    <name>main</name>
-    <message>
-        <location filename="../main.cpp" line="51"/>
-        <source>Logout.</source>
-        <translation>Abmelden.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="54"/>
-        <source>Lockscreen.</source>
-        <translation>Sperren.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="57"/>
-        <source>Suspend.</source>
-        <translation>Bereitschaft.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="60"/>
-        <source>Hibernate.</source>
-        <translation>Tiefschlaf.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="63"/>
-        <source>Shutdown.</source>
-        <translation>Herunterfahren.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="66"/>
-        <source>Reboot.</source>
-        <translation>Neustart.</translation>
-    </message>
-</context>
-</TS>
diff --git a/lxqt-leave/translations/lxqt-leave_el.ts b/lxqt-leave/translations/lxqt-leave_el.ts
deleted file mode 100644
index 3172138..0000000
--- a/lxqt-leave/translations/lxqt-leave_el.ts
+++ /dev/null
@@ -1,85 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS>
-<TS version="2.1" language="el">
-<context>
-    <name>LeaveDialog</name>
-    <message>
-        <location filename="../leavedialog.ui" line="20"/>
-        <source>Leave</source>
-        <translation>Έξοδος</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="29"/>
-        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;What do you want the computer to do?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Τι επιθυμείτε να πραγματοποιήσει ο υπολογιστής;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="45"/>
-        <source>Logout</source>
-        <translation>Αποσύνδεση</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="57"/>
-        <source>Reboot</source>
-        <translation>Επανεκκίνηση</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="69"/>
-        <source>Shutdown</source>
-        <translation>Τερματισμός</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="81"/>
-        <source>Suspend</source>
-        <translation>Αναστολή</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="93"/>
-        <source>Lock screen</source>
-        <translation>Κλείδωμα οθόνης</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="105"/>
-        <source>Hibernate</source>
-        <translation>Νάρκη</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="136"/>
-        <source>Cancel</source>
-        <translation>Ακύρωση</translation>
-    </message>
-</context>
-<context>
-    <name>main</name>
-    <message>
-        <location filename="../main.cpp" line="51"/>
-        <source>Logout.</source>
-        <translation>Αποσύνδεση.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="54"/>
-        <source>Lockscreen.</source>
-        <translation>Κλείδωμα οθόνης.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="57"/>
-        <source>Suspend.</source>
-        <translation>Αναστολή.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="60"/>
-        <source>Hibernate.</source>
-        <translation>Νάρκη.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="63"/>
-        <source>Shutdown.</source>
-        <translation>Τερματισμός.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="66"/>
-        <source>Reboot.</source>
-        <translation>Επανεκκίνηση.</translation>
-    </message>
-</context>
-</TS>
diff --git a/lxqt-leave/translations/lxqt-leave_hu.ts b/lxqt-leave/translations/lxqt-leave_hu.ts
deleted file mode 100644
index 0d45a6c..0000000
--- a/lxqt-leave/translations/lxqt-leave_hu.ts
+++ /dev/null
@@ -1,89 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS>
-<TS version="2.1" language="ru">
-<context>
-    <name>LeaveDialog</name>
-    <message>
-        <source>Form</source>
-        <translation type="vanished">Űrlap</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="20"/>
-        <source>Leave</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="29"/>
-        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;What do you want the computer to do?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Mi legyen a számítógéppel?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="45"/>
-        <source>Logout</source>
-        <translation>Kilépés</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="57"/>
-        <source>Reboot</source>
-        <translation>Újraindítás</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="69"/>
-        <source>Shutdown</source>
-        <translation>Kikapcsolás</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="81"/>
-        <source>Suspend</source>
-        <translation>Készenlét</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="93"/>
-        <source>Lock screen</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="105"/>
-        <source>Hibernate</source>
-        <translation>Mélykészenlét</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="136"/>
-        <source>Cancel</source>
-        <translation>Mégse</translation>
-    </message>
-</context>
-<context>
-    <name>main</name>
-    <message>
-        <location filename="../main.cpp" line="51"/>
-        <source>Logout.</source>
-        <translation>Kilépés.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="54"/>
-        <source>Lockscreen.</source>
-        <translation>Képernyőzár.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="57"/>
-        <source>Suspend.</source>
-        <translation>Készenlét.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="60"/>
-        <source>Hibernate.</source>
-        <translation>Mélykészenlét.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="63"/>
-        <source>Shutdown.</source>
-        <translation>Kikapcsolás.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="66"/>
-        <source>Reboot.</source>
-        <translation>Újraindítás.</translation>
-    </message>
-</context>
-</TS>
diff --git a/lxqt-leave/translations/lxqt-leave_it.ts b/lxqt-leave/translations/lxqt-leave_it.ts
deleted file mode 100644
index 9082c1c..0000000
--- a/lxqt-leave/translations/lxqt-leave_it.ts
+++ /dev/null
@@ -1,89 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS>
-<TS version="2.1" language="it">
-<context>
-    <name>LeaveDialog</name>
-    <message>
-        <source>Form</source>
-        <translation type="vanished">Esci</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="20"/>
-        <source>Leave</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="29"/>
-        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;What do you want the computer to do?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Terminare la sessione di LXQt?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="45"/>
-        <source>Logout</source>
-        <translation>Chiudi sessione</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="57"/>
-        <source>Reboot</source>
-        <translation>Riavvia</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="69"/>
-        <source>Shutdown</source>
-        <translation>Spegni</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="81"/>
-        <source>Suspend</source>
-        <translation>Sospendi</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="93"/>
-        <source>Lock screen</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="105"/>
-        <source>Hibernate</source>
-        <translation>Iberna</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="136"/>
-        <source>Cancel</source>
-        <translation>Cancella</translation>
-    </message>
-</context>
-<context>
-    <name>main</name>
-    <message>
-        <location filename="../main.cpp" line="51"/>
-        <source>Logout.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="54"/>
-        <source>Lockscreen.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="57"/>
-        <source>Suspend.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="60"/>
-        <source>Hibernate.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="63"/>
-        <source>Shutdown.</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="66"/>
-        <source>Reboot.</source>
-        <translation type="unfinished"></translation>
-    </message>
-</context>
-</TS>
diff --git a/lxqt-leave/translations/lxqt-leave_ja.ts b/lxqt-leave/translations/lxqt-leave_ja.ts
deleted file mode 100644
index 5003971..0000000
--- a/lxqt-leave/translations/lxqt-leave_ja.ts
+++ /dev/null
@@ -1,85 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS>
-<TS version="2.1" language="ja">
-<context>
-    <name>LeaveDialog</name>
-    <message>
-        <location filename="../leavedialog.ui" line="20"/>
-        <source>Leave</source>
-        <translation>終了</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="29"/>
-        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;What do you want the computer to do?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;どの操作を選択しますか？&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="45"/>
-        <source>Logout</source>
-        <translation>ログアウト</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="57"/>
-        <source>Reboot</source>
-        <translation>再起動</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="69"/>
-        <source>Shutdown</source>
-        <translation>シャットダウン</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="81"/>
-        <source>Suspend</source>
-        <translation>サスペンド</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="93"/>
-        <source>Lock screen</source>
-        <translation>画面のロック</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="105"/>
-        <source>Hibernate</source>
-        <translation>ハイバネート</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="136"/>
-        <source>Cancel</source>
-        <translation>キャンセル</translation>
-    </message>
-</context>
-<context>
-    <name>main</name>
-    <message>
-        <location filename="../main.cpp" line="51"/>
-        <source>Logout.</source>
-        <translation>ログアウト.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="54"/>
-        <source>Lockscreen.</source>
-        <translation>画面のロック.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="57"/>
-        <source>Suspend.</source>
-        <translation>サスペンド.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="60"/>
-        <source>Hibernate.</source>
-        <translation>ハイバネート.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="63"/>
-        <source>Shutdown.</source>
-        <translation>シャットダウン.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="66"/>
-        <source>Reboot.</source>
-        <translation>再起動.</translation>
-    </message>
-</context>
-</TS>
diff --git a/lxqt-leave/translations/lxqt-leave_ru.ts b/lxqt-leave/translations/lxqt-leave_ru.ts
deleted file mode 100644
index 01f8f78..0000000
--- a/lxqt-leave/translations/lxqt-leave_ru.ts
+++ /dev/null
@@ -1,89 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS>
-<TS version="2.1" language="ru">
-<context>
-    <name>LeaveDialog</name>
-    <message>
-        <source>Form</source>
-        <translation type="vanished">Меню выхода</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="20"/>
-        <source>Leave</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="29"/>
-        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;What do you want the computer to do?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Что вы хотите сделать с компьютером?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="45"/>
-        <source>Logout</source>
-        <translation>Выйти</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="57"/>
-        <source>Reboot</source>
-        <translation>Перезагрузить</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="69"/>
-        <source>Shutdown</source>
-        <translation>Выключить</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="81"/>
-        <source>Suspend</source>
-        <translation>Перевести в ждущий режим</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="93"/>
-        <source>Lock screen</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="105"/>
-        <source>Hibernate</source>
-        <translation>Перевести в спящий режим</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="136"/>
-        <source>Cancel</source>
-        <translation>Отмена</translation>
-    </message>
-</context>
-<context>
-    <name>main</name>
-    <message>
-        <location filename="../main.cpp" line="51"/>
-        <source>Logout.</source>
-        <translation>Выйти.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="54"/>
-        <source>Lockscreen.</source>
-        <translation>Блокировать экран.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="57"/>
-        <source>Suspend.</source>
-        <translation>Перевести в ждущий режим.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="60"/>
-        <source>Hibernate.</source>
-        <translation>Перевести в спящий режим.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="63"/>
-        <source>Shutdown.</source>
-        <translation>Выключить.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="66"/>
-        <source>Reboot.</source>
-        <translation>Перезагрузить.</translation>
-    </message>
-</context>
-</TS>
diff --git a/lxqt-leave/translations/lxqt-leave_ru_RU.ts b/lxqt-leave/translations/lxqt-leave_ru_RU.ts
deleted file mode 100644
index 658e38c..0000000
--- a/lxqt-leave/translations/lxqt-leave_ru_RU.ts
+++ /dev/null
@@ -1,89 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS>
-<TS version="2.1" language="ru_RU">
-<context>
-    <name>LeaveDialog</name>
-    <message>
-        <source>Form</source>
-        <translation type="vanished">Меню выхода</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="20"/>
-        <source>Leave</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="29"/>
-        <source>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;What do you want the computer to do?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt;&quot;&gt;Что вы хотите сделать с компьютером?&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="45"/>
-        <source>Logout</source>
-        <translation>Выйти</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="57"/>
-        <source>Reboot</source>
-        <translation>Перезагрузить</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="69"/>
-        <source>Shutdown</source>
-        <translation>Выключить</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="81"/>
-        <source>Suspend</source>
-        <translation>Перевести в ждущий режим</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="93"/>
-        <source>Lock screen</source>
-        <translation type="unfinished"></translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="105"/>
-        <source>Hibernate</source>
-        <translation>Перевести в спящий режим</translation>
-    </message>
-    <message>
-        <location filename="../leavedialog.ui" line="136"/>
-        <source>Cancel</source>
-        <translation>Отмена</translation>
-    </message>
-</context>
-<context>
-    <name>main</name>
-    <message>
-        <location filename="../main.cpp" line="51"/>
-        <source>Logout.</source>
-        <translation>Выйти.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="54"/>
-        <source>Lockscreen.</source>
-        <translation>Блокировать экран.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="57"/>
-        <source>Suspend.</source>
-        <translation>Перевести в ждущий режим.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="60"/>
-        <source>Hibernate.</source>
-        <translation>Перевести в спящий режим.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="63"/>
-        <source>Shutdown.</source>
-        <translation>Выключить.</translation>
-    </message>
-    <message>
-        <location filename="../main.cpp" line="66"/>
-        <source>Reboot.</source>
-        <translation>Перезагрузить.</translation>
-    </message>
-</context>
-</TS>
diff --git a/lxqt-leave/translations/lxqt-lockscreen_ru.desktop b/lxqt-leave/translations/lxqt-lockscreen_ru.desktop
index 2267019..fe5268a 100644
--- a/lxqt-leave/translations/lxqt-lockscreen_ru.desktop
+++ b/lxqt-leave/translations/lxqt-lockscreen_ru.desktop
@@ -1,2 +1,2 @@
-Name[ru]=Блокировать экран
-Comment[ru]=Блокировать текущий сеанс
+Name[ru]=Заблокировать экран
+Comment[ru]=Заблокировать текущий сеанс
