# Calculate format=diff merge(games-action/oolite)=>1.86,1.88
diff --git a/deps/mozilla/js/src/jsapi.cpp b/deps/mozilla/js/src/jsapi.cpp
index ce03cd7..7076519 100644
--- a/deps/mozilla/js/src/jsapi.cpp
+++ b/deps/mozilla/js/src/jsapi.cpp
@@ -3977,7 +3977,7 @@ JS_Enumerate(JSContext *cx, JSObject *obj)
     AutoIdVector props(cx);
     JSIdArray *ida;
     if (!GetPropertyNames(cx, obj, JSITER_OWNONLY, &props) || !VectorToIdArray(cx, props, &ida))
-        return false;
+        return NULL;
     for (size_t n = 0; n < size_t(ida->length); ++n)
         JS_ASSERT(js_CheckForStringIndex(ida->vector[n]) == ida->vector[n]);
     return ida;
diff --git a/deps/mozilla/js/src/jsfun.cpp b/deps/mozilla/js/src/jsfun.cpp
index 71dba0b..f494849 100644
--- a/deps/mozilla/js/src/jsfun.cpp
+++ b/deps/mozilla/js/src/jsfun.cpp
@@ -2052,7 +2052,7 @@ fun_toStringHelper(JSContext *cx, JSObject *obj, uintN indent)
 
     JSString *str = JS_DecompileFunction(cx, fun, indent);
     if (!str)
-        return false;
+        return NULL;
 
     if (!indent)
         cx->compartment->toSourceCache.put(fun, str);
@@ -2658,7 +2658,7 @@ LookupInterpretedFunctionPrototype(JSContext *cx, JSObject *funobj)
     const Shape *shape = funobj->nativeLookup(id);
     if (!shape) {
         if (!ResolveInterpretedFunctionPrototype(cx, funobj))
-            return false;
+            return NULL;
         shape = funobj->nativeLookup(id);
     }
     JS_ASSERT(!shape->configurable());
diff --git a/deps/mozilla/js/src/jsiter.cpp b/deps/mozilla/js/src/jsiter.cpp
index 8d8a62f..5ba46ff 100644
--- a/deps/mozilla/js/src/jsiter.cpp
+++ b/deps/mozilla/js/src/jsiter.cpp
@@ -425,7 +425,7 @@ NewIteratorObject(JSContext *cx, uintN flags)
          */
         JSObject *obj = js_NewGCObject(cx, FINALIZE_OBJECT0);
         if (!obj)
-            return false;
+            return NULL;
         obj->init(cx, &js_IteratorClass, NULL, NULL, NULL, false);
         obj->setMap(cx->compartment->emptyEnumeratorShape);
         return obj;
diff --git a/deps/mozilla/js/src/jsparse.cpp b/deps/mozilla/js/src/jsparse.cpp
index c735970..b9cc2aa 100644
--- a/deps/mozilla/js/src/jsparse.cpp
+++ b/deps/mozilla/js/src/jsparse.cpp
@@ -3352,7 +3352,7 @@ Parser::functionDef(JSAtom *funAtom, FunctionType type, uintN lambda)
     if (!outertc->inFunction() && bodyLevel && funAtom && !lambda && outertc->compiling()) {
         JS_ASSERT(pn->pn_cookie.isFree());
         if (!DefineGlobal(pn, outertc->asCodeGenerator(), funAtom))
-            return false;
+            return NULL;
     }
 
     pn->pn_blockid = outertc->blockid();
diff --git a/deps/mozilla/js/src/jsstr.cpp b/deps/mozilla/js/src/jsstr.cpp
index cb5c87d..9632071 100644
--- a/deps/mozilla/js/src/jsstr.cpp
+++ b/deps/mozilla/js/src/jsstr.cpp
@@ -1734,7 +1734,7 @@ class RegExpGuard
         if (flat) {
             patstr = flattenPattern(cx, fm.patstr);
             if (!patstr)
-                return false;
+                return NULL;
         } else {
             patstr = fm.patstr;
         }
@@ -3400,7 +3400,7 @@ js_InitStringClass(JSContext *cx, JSObject *obj)
                                  UndefinedValue(), NULL, NULL,
                                  JSPROP_READONLY | JSPROP_PERMANENT | JSPROP_SHARED, 0, 0,
                                  NULL)) {
-        return JS_FALSE;
+        return NULL;
     }
 
     return proto;
diff --git a/deps/mozilla/js/src/jstypedarray.cpp b/deps/mozilla/js/src/jstypedarray.cpp
index 8975fb1..64e5e3f 100644
--- a/deps/mozilla/js/src/jstypedarray.cpp
+++ b/deps/mozilla/js/src/jstypedarray.cpp
@@ -1334,7 +1334,7 @@ class TypedArrayTemplate
         if (size != 0 && count >= INT32_MAX / size) {
             JS_ReportErrorNumber(cx, js_GetErrorMessage, NULL,
                                  JSMSG_NEED_DIET, "size and count");
-            return false;
+            return NULL;
         }
 
         int32 bytelen = size * count;
@@ -1668,7 +1668,7 @@ TypedArrayConstruct(JSContext *cx, jsint atype, uintN argc, Value *argv)
 
       default:
         JS_NOT_REACHED("shouldn't have gotten here");
-        return false;
+        return NULL;
     }
 }
 
diff --git a/deps/mozilla/js/src/jsxml.cpp b/deps/mozilla/js/src/jsxml.cpp
index 7fa3bc1..71bcc54 100644
--- a/deps/mozilla/js/src/jsxml.cpp
+++ b/deps/mozilla/js/src/jsxml.cpp
@@ -282,7 +282,7 @@ NewXMLNamespace(JSContext *cx, JSLinearString *prefix, JSLinearString *uri, JSBo
 
     obj = NewBuiltinClassInstanceXML(cx, &js_NamespaceClass);
     if (!obj)
-        return JS_FALSE;
+        return NULL;
     JS_ASSERT(JSVAL_IS_VOID(obj->getNamePrefixVal()));
     JS_ASSERT(JSVAL_IS_VOID(obj->getNameURIVal()));
     JS_ASSERT(JSVAL_IS_VOID(obj->getNamespaceDeclared()));
@@ -431,7 +431,7 @@ ConvertQNameToString(JSContext *cx, JSObject *obj)
         size_t length = str->length();
         jschar *chars = (jschar *) cx->malloc((length + 2) * sizeof(jschar));
         if (!chars)
-            return JS_FALSE;
+            return NULL;
         *chars = '@';
         const jschar *strChars = str->getChars(cx);
         if (!strChars) {
diff --git a/deps/mozilla/js/src/methodjit/InvokeHelpers.cpp b/deps/mozilla/js/src/methodjit/InvokeHelpers.cpp
index 59ac8c6..d017c2a 100644
--- a/deps/mozilla/js/src/methodjit/InvokeHelpers.cpp
+++ b/deps/mozilla/js/src/methodjit/InvokeHelpers.cpp
@@ -728,7 +728,7 @@ AtSafePoint(JSContext *cx)
 {
     JSStackFrame *fp = cx->fp();
     if (fp->hasImacropc())
-        return false;
+        return NULL;
 
     JSScript *script = fp->script();
     return script->maybeNativeCodeForPC(fp->isConstructing(), cx->regs->pc);
