diff --git a/slim-1.3.1/app.cpp b/slim-1.3.1/app.cpp
index 83ae947..e07f077 100644
--- a/slim-1.3.1/app.cpp
+++ b/slim-1.3.1/app.cpp
@@ -959,9 +959,9 @@ void App::StopServer() {
     signal(SIGALRM, SIG_DFL);
 
     // Catch X error
-    XSetIOErrorHandler(IgnoreXIO);
-    if(!setjmp(CloseEnv) && Dpy)
-        XCloseDisplay(Dpy);
+    //XSetIOErrorHandler(IgnoreXIO);
+    //if(!setjmp(CloseEnv) && Dpy)
+    //    XCloseDisplay(Dpy);
 
     // Send HUP to process group
     errno = 0;
