# Calculate format=diff
diff --git a/ace/os_include/os_sched.h b/ace/os_include/os_sched.h
index f06541b..6d0c07d 100644
--- a/ace/os_include/os_sched.h
+++ b/ace/os_include/os_sched.h
@@ -41,10 +41,6 @@ extern "C"
    typedef cpuset_t cpu_set_t;
 #else
 #  define ACE_CPU_SETSIZE 1024
-   typedef struct
-   {
-     ACE_UINT32 bit_array_[ACE_CPU_SETSIZE / (8 * sizeof (ACE_UINT32))];
-   } cpu_set_t;
 #endif
 #endif /* !ACE_HAS_CPU_SET_T || !__cpu_set_t_defined */
 
