diff --git desktop/cl_desktop.py desktop/cl_desktop.py
index e3130e0..6e3b904 100644
--- desktop/cl_desktop.py
+++ desktop/cl_desktop.py
@@ -80,8 +80,20 @@ class Desktop:
         """Apply templates for user"""
         if self.clTempl:
             self.closeClTemplate()
+
+        templates_locate = self.clVars.Get('cl_templates_locate')
+        self.clVars.Set("cl_template_path",
+            map(lambda x:x[1],
+            filter(lambda x:x[0] in templates_locate,
+            zip(self.clVars.Get('cl_template_location'),
+                self.clVars.Get('cl_template_path')))),
+            True)
+
         self.clTempl = ProgressTemplate(self.setProgress,self.clVars,
-                                        cltObj=False)
+                                        cltObj=False,
+                                        printERROR=self.printERROR,
+                                        printWARNING=self.printWARNING,
+                                        printSUCCESS=self.printSUCCESS)
         dirsFiles = self.clTempl.applyTemplates()
         if self.clTempl.getError():
             self.printERROR(self.clTempl.getError().strip())
diff --git desktop/cl_wsdl_desktop.py desktop/cl_wsdl_desktop.py
index e07295d..c1844dd 100644
--- desktop/cl_wsdl_desktop.py
+++ desktop/cl_wsdl_desktop.py
@@ -40,6 +40,7 @@ class DesktopInfo(ClassSerializer):
     """Parameters for method install"""
     ur_login = String
     cl_verbose_set = String
+    cl_templates_locate = Array(String)
 
     Default = Array(String)
     CheckOnly = Boolean
@@ -54,7 +55,7 @@ class Wsdl:
     """
     @rpc(Integer, DesktopInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Desktop'),title=__('Configure user'),
-        image='preferences-desktop',
+        image='user-desktop,preferences-desktop',
         gui=True,command='cl-desktop',
         rights=['userconfigure'])
     def desktop(self, sid, info):
@@ -69,7 +70,8 @@ class Wsdl:
             dv.flIniFile()
             dv.Set('cl_action','desktop',True)
         dv.addGroup(None,
-            normal=('ur_login','cl_verbose_set',),
+            normal=('ur_login',),
+            expert=('cl_verbose_set','cl_templates_locate'),
             next_label=_("Configure"))
         return dv
 
