diff --git desktop/utils/cl_desktop.py desktop/utils/cl_desktop.py
index 28cecb5..125a210 100644
--- desktop/utils/cl_desktop.py
+++ desktop/utils/cl_desktop.py
@@ -68,9 +68,10 @@ class ClDesktopAction(Action):
           'message':__("Mounting encrypted data"),
           'method':'Desktop.createCryptDir(ur_login,ur_uid,ur_gid,'
                                           'ur_home_path,False)',
-          'condition':lambda Get:(not isMount(Get('ur_home_path')) and \
-                      Get('ur_home_crypt_set') == 'on' and 
-                      Get('install.cl_autologin') != Get('ur_login'))
+          'condition':lambda dv:(not isMount(dv.Get('ur_home_path')) and \
+                      dv.Get('ur_home_crypt_set') == 'on' and 
+                      (not dv.isModuleInstalled("install") or
+                      dv.Get('install.cl_autologin') != dv.Get('ur_login')))
          },
          {'name':'user_profile',
           'message':__("Setting up the user profile"),
diff --git desktop/wsdl_desktop.py desktop/wsdl_desktop.py
index 3cbb78c..e9b883a 100644
--- desktop/wsdl_desktop.py
+++ desktop/wsdl_desktop.py
@@ -20,7 +20,6 @@ from calculate.core.server.func import WsdlBase
 from desktop import DesktopError
 from utils.cl_desktop import ClDesktopLogoutAction,ClDesktopAction
 import desktop
-import calculate.install.install as install
 
 from calculate.lib.cl_lang import setLocalTranslate,getLazyLocalTranslate
 setLocalTranslate('cl_desktop3',sys.modules[__name__])
@@ -81,7 +80,7 @@ class Wsdl(WsdlBase):
         # права для запуска метода 
         'rights':['userconfigure'],
         # объект содержащий модули для действия
-        'logic':{'Desktop':desktop.Desktop,'Install':install.Install},
+        'logic':{'Desktop':desktop.Desktop},
         # описание действия
         'action':ClDesktopAction,
         # объект переменных
