diff --git data/gtkbg data/gtkbg
deleted file mode 100644
index 7489965..0000000
--- data/gtkbg
+++ /dev/null
@@ -1,61 +0,0 @@
-#!/usr/bin/env python2
-# Copyright 2008-2010 Calculate Ltd. http://www.calculate-linux.org
-#
-#  Licensed under the Apache License, Version 2.0 (the "License");
-#  you may not use this file except in compliance with the License.
-#  You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-#  Unless required by applicable law or agreed to in writing, software
-#  distributed under the License is distributed on an "AS IS" BASIS,
-#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#  See the License for the specific language governing permissions and
-#  limitations under the License.
-
-import pygtk
-pygtk.require('2.0')
-import gtk
-import os
-import sys
-import re
-import subprocess
-
-def select_color():
-    try:
-        if filter(re.compile(r"(cld|cldx|cldg|cmc|cls)-themes-12").search,
-           os.listdir('/var/db/pkg/media-gfx')):
-            return "#73a363"
-    except:
-        pass
-    return '#30648b'
-
-if __name__ == "__main__":
-    if gtk.gdk.get_display():
-        window = gtk.Window(gtk.WINDOW_TOPLEVEL)
-        
-        # Here we connect the "destroy" event to a signal handler 
-        window.connect("destroy", lambda w: gtk.main_quit())
-        pipe = subprocess.Popen(["xdpyinfo"],
-                    stdout=subprocess.PIPE, env=os.environ)
-        reRes = re.compile("dimensions:\s+(\d+)x(\d+)\s+pixels")
-        cx, cy = 1024, 768
-        if pipe.wait() == 0:
-            for line in pipe.stdout:
-             searchRes = reRes.search(line)
-             if searchRes:
-                 cx, cy = int(searchRes.group(1)), int(searchRes.group(2))
-                 break
-        pipe.stdout.close()
-        window.set_size_request(cx,cy)
-        window.maximize()
-        
-        window.set_border_width(0)
-        window.modify_bg(gtk.STATE_NORMAL,
-            gtk.gdk.rgb_get_colormap().alloc_color(select_color()))
-        
-        window.show()
-        pid = os.fork()
-        if not pid:
-            gtk.main()
-    sys.exit(0)
diff --git data/login.d/00init data/login.d/00init
index 2e235d4..5cbebf4 100644
--- data/login.d/00init
+++ data/login.d/00init
@@ -14,11 +14,16 @@
 #  limitations under the License.
 
 # set background color
-if ls /var/db/pkg/media-gfx/cld[-gx]*themes-12* &>/dev/null
+if [[ -f /usr/share/wallpapers/dm-background.png ]]
 then
-    xsetroot -solid rgb:73/a3/63
+    feh --no-fehbg --bg-scale /usr/share/wallpapers/dm-background.png
 else
-    xsetroot -solid rgb:30/64/8b
+    if ls /var/db/pkg/media-gfx/cld[-gx]*themes-12* &>/dev/null
+    then
+        xsetroot -solid rgb:73/a3/63
+    else
+        xsetroot -solid rgb:30/64/8b
+    fi
 fi
 
 if [[ "`ps axeo command | grep 'xdm/xdm --logout' | grep -v grep | \
diff --git data/logout.d/00init data/logout.d/00init
index f959375..deaa12b 100644
--- data/logout.d/00init
+++ data/logout.d/00init
@@ -14,10 +14,15 @@
 #  limitations under the License.
 
 # set background color
-if ls /var/db/pkg/media-gfx/cld[-gx]*themes-12* &>/dev/null
+if [[ -f /usr/share/wallpapers/dm-background.png ]]
 then
-    xsetroot -solid rgb:73/a3/63
+    feh --no-fehbg --bg-scale /usr/share/wallpapers/dm-background.png
 else
-    xsetroot -solid rgb:30/64/8b
+    if ls /var/db/pkg/media-gfx/cld[-gx]*themes-12* &>/dev/null
+    then
+        xsetroot -solid rgb:73/a3/63
+    else
+        xsetroot -solid rgb:30/64/8b
+    fi
 fi
 exit 0
diff --git data/xdm data/xdm
index 209f191..3f764c2 100755
--- data/xdm
+++ data/xdm
@@ -63,14 +63,6 @@ die_xmes() {
 	die "${1}"
 }
 
-gtk_background(){
-	if [[ -n $(env | grep RUNNING_UNDER_GDM=true) ]] && ls /var/db/pkg/gnome-base/gdm-2* &>/dev/null;
-	then
-		/usr/bin/env python2 /usr/share/calculate/xdm/gtkbg
-		BG_PID=$(ps ax | sed -nr "s/^\s*([0-9]+)\s.*gtkbg$/\1/p")
-	fi
-}
-
 umount_user_res(){
 	if [[ -n $USER && -z `who | awk '{ print $1, $2 }' | grep -P ":\d+" | awk '{ print $1}' | grep $USER` ]];
 	then
@@ -146,12 +138,17 @@ fi
 FILELIST=$(find ${EXECDIR} -type f | sort)
 [[ $FILELIST ]] || warning "${EXECDIR} hasn't scripts"
 
-gtk_background
 DOMAIN_USER="`desktop_variable_value desktop.ur_domain_set`"
 for script in $FILELIST
 do
 	FILE_LOG=$FILE_LOG DOMAIN_USER=$DOMAIN_USER /bin/bash $script || die_xmes "failed on execute $script"
 done
 
-[[ -n $BG_PID ]] && kill $BG_PID
+sleep_kill()
+{
+	sleep 2
+	kill $1
+}
+
+[[ -n $BG_PID ]] && sleep_kill $BG_PID &
 exit 0
diff --git setup.py setup.py
index e0575de..3179831 100755
--- setup.py
+++ setup.py
@@ -18,7 +18,7 @@
 #  limitations under the License.
 
 __app__ = "calculate-desktop"
-__version__ = "3.0.0"
+__version__ = "3.1.0"
 
 import os
 import stat
@@ -34,7 +34,6 @@ data_files = []
 
 data_files += [('/usr/share/calculate/xdm', ['data/cmd_login',
                                              'data/functions',
-                                             'data/gtkbg',
                                              'data/xdm'])] + \
               [('/usr/share/calculate/xdm/login.d',
                                             ['data/login.d/00init',
@@ -77,8 +76,7 @@ def create_data_files(data_dirs, prefix="", replace_dirname=""):
 class cl_install_data(install_data):
     def run (self):
         install_data.run(self)
-        data_file = [("/usr/share/calculate/xdm/gtkbg",0755),
-                     ("/usr/share/calculate/xdm/cmd_login",0755),
+        data_file = [("/usr/share/calculate/xdm/cmd_login",0755),
                      ("/usr/share/calculate/xdm/xdm",0755)]
         fileNames = map(lambda x: os.path.split(x[0])[1], data_file)
         listNames = map(lambda x: filter(lambda y: y, x[0].split("/")),
