diff --git pym/desktop/desktop.py pym/desktop/desktop.py
index 766582c..183f81e 100644
--- pym/desktop/desktop.py
+++ pym/desktop/desktop.py
@@ -84,7 +84,8 @@ class Desktop:
                 if not mountEcryptfs(userName,userPwd,userDir):
                     error = _("Failed to mount ecrypted data")
             except CommonError as e:
-                error = _("Failed to mount ecrypted data")+": \"%s\""%str(e)
+                error = (_("Failed to mount ecrypted data")+
+                         _(": ")+'":%s"'%str(e))
         # если при подключении произошли ошибки
         if error:
             # заархивировать текущий профиль и удалить его
@@ -114,7 +115,8 @@ class Desktop:
                 if isBootstrapDataOnly(userDir):
                     if childMounts(userDir):
                         raise DesktopError(
-                            _("Failed to create the encrypted user profile")+":"+
+                            _("Failed to create the encrypted user profile")+
+                            _(": ")+
                             _("User home directory contains mount points"))
                     # поместить данные во временный tarfile
                     calculateName = ".calculate"
