commit ec1208cd226309f03bc7db10584d592f16226906
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jun 15 17:42:32 2012 +0400

    Login background

diff --git a/data/login.d/00init b/data/login.d/00init
index 7076476..2e235d4 100644
--- a/data/login.d/00init
+++ b/data/login.d/00init
@@ -14,7 +14,12 @@
 #  limitations under the License.
 
 # set background color
-xsetroot -solid rgb:30/64/8b
+if ls /var/db/pkg/media-gfx/cld[-gx]*themes-12* &>/dev/null
+then
+    xsetroot -solid rgb:73/a3/63
+else
+    xsetroot -solid rgb:30/64/8b
+fi
 
 if [[ "`ps axeo command | grep 'xdm/xdm --logout' | grep -v grep | \
 	sed -n -r 's/.* USER=([^ ]+) .*/\1/p'`" == "$USER" ]];
