commit 2e35f1af2ccd8ee1ab6a5a34d32264499c4ff1e2
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Jun 18 09:32:06 2012 +0400

    Fix other colors

diff --git a/data/gtkbg b/data/gtkbg
index 741dc15..7489965 100644
--- a/data/gtkbg
+++ b/data/gtkbg
@@ -21,6 +21,15 @@ import sys
 import re
 import subprocess
 
+def select_color():
+    try:
+        if filter(re.compile(r"(cld|cldx|cldg|cmc|cls)-themes-12").search,
+           os.listdir('/var/db/pkg/media-gfx')):
+            return "#73a363"
+    except:
+        pass
+    return '#30648b'
+
 if __name__ == "__main__":
     if gtk.gdk.get_display():
         window = gtk.Window(gtk.WINDOW_TOPLEVEL)
@@ -43,7 +52,7 @@ if __name__ == "__main__":
         
         window.set_border_width(0)
         window.modify_bg(gtk.STATE_NORMAL,
-            gtk.gdk.rgb_get_colormap().alloc_color('#30648b'))
+            gtk.gdk.rgb_get_colormap().alloc_color(select_color()))
         
         window.show()
         pid = os.fork()
diff --git a/data/logout.d/00init b/data/logout.d/00init
index 52c0006..f959375 100644
--- a/data/logout.d/00init
+++ b/data/logout.d/00init
@@ -14,5 +14,10 @@
 #  limitations under the License.
 
 # set background color
-xsetroot -solid rgb:30/64/8b
+if ls /var/db/pkg/media-gfx/cld[-gx]*themes-12* &>/dev/null
+then
+    xsetroot -solid rgb:73/a3/63
+else
+    xsetroot -solid rgb:30/64/8b
+fi
 exit 0
