diff --git a/data/gtkbg b/data/gtkbg
index 56ebcea..79b8872 100644
--- a/data/gtkbg
+++ b/data/gtkbg
@@ -43,7 +43,7 @@ if __name__ == "__main__":
         
         window.set_border_width(0)
         window.modify_bg(gtk.STATE_NORMAL,
-            gtk.gdk.rgb_get_colormap().alloc_color('#170b0a'))
+            gtk.gdk.rgb_get_colormap().alloc_color('#30648b'))
         
         window.show()
         pid = os.fork()
diff --git a/data/login.d/00init b/data/login.d/00init
index 876c171..7076476 100644
--- a/data/login.d/00init
+++ b/data/login.d/00init
@@ -14,7 +14,7 @@
 #  limitations under the License.
 
 # set background color
-xsetroot -solid rgb:17/0b/0a
+xsetroot -solid rgb:30/64/8b
 
 if [[ "`ps axeo command | grep 'xdm/xdm --logout' | grep -v grep | \
 	sed -n -r 's/.* USER=([^ ]+) .*/\1/p'`" == "$USER" ]];
diff --git a/data/logout.d/00init b/data/logout.d/00init
index b73ae14..52c0006 100644
--- a/data/logout.d/00init
+++ b/data/logout.d/00init
@@ -14,5 +14,5 @@
 #  limitations under the License.
 
 # set background color
-xsetroot -solid rgb:17/0b/0a
+xsetroot -solid rgb:30/64/8b
 exit 0
