diff --git desktop/cl_wsdl_desktop.py desktop/cl_wsdl_desktop.py
index e07295d..e71f13f 100644
--- desktop/cl_wsdl_desktop.py
+++ desktop/cl_wsdl_desktop.py
@@ -54,7 +54,7 @@ class Wsdl:
     """
     @rpc(Integer, DesktopInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Desktop'),title=__('Configure user'),
-        image='preferences-desktop',
+        image='user-desktop,preferences-desktop',
         gui=True,command='cl-desktop',
         rights=['userconfigure'])
     def desktop(self, sid, info):
