diff -urNp dleyna-server-0.7.0.orig/libdleyna/server/meson.build dleyna-server-0.7.0/libdleyna/server/meson.build
--- dleyna-server-0.7.0.orig/libdleyna/server/meson.build	2021-04-08 14:37:37.000000000 +0300
+++ dleyna-server-0.7.0/libdleyna/server/meson.build	2021-05-29 19:57:37.249535899 +0300
@@ -30,6 +30,7 @@ libdleyna_server = library(
         gupnp_dlna,
         soup,
         math,
+        libxml2,
         config_h
     ],
     install: true,
diff -urNp dleyna-server-0.7.0.orig/meson.build dleyna-server-0.7.0/meson.build
--- dleyna-server-0.7.0.orig/meson.build	2021-05-29 19:43:13.084924886 +0300
+++ dleyna-server-0.7.0/meson.build	2021-05-29 19:56:31.435711449 +0300
@@ -73,6 +73,7 @@ gupnp_av = dependency('gupnp-av-1.0', ve
 gupnp_dlna = dependency('gupnp-dlna-2.0', version: '>= 0.9.4')
 soup = dependency('libsoup-2.4', version: '>= 2.28.2')
 dleyna_core = dependency('dleyna-core-1.0', version: '>= 0.6.0')
+libxml2 = dependency('libxml-2.0', version: '>= 2.9.10')
 
 cc = meson.get_compiler('c')
 math = cc.find_library('m', required: false)
