# Calculate format=diff
diff --git a/wayland_server.cpp b/wayland_server.cpp
index 3f273ca..837be8b 100644
--- a/wayland_server.cpp
+++ b/wayland_server.cpp
@@ -34,6 +34,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // Server
 #include <KWayland/Server/compositor_interface.h>
 #include <KWayland/Server/datadevicemanager_interface.h>
+#include <KWayland/Server/datasource_interface.h>
 #include <KWayland/Server/display.h>
 #include <KWayland/Server/dpms_interface.h>
 #include <KWayland/Server/idle_interface.h>
