--- a/wgetpaste	2023-11-05 01:14:42.819088466 +0300
+++ b/wgetpaste	2023-11-06 14:31:32.400944620 +0300
@@ -41,7 +41,14 @@
 }
 
 ### services
-SERVICES="0x0 bpaste codepad dpaste gists ix_io pgz snippets calculate"
+SERVICES="0x0 bpaste codepad dpaste gists ix_io pgz snippets calculate orgru"
+# orgru
+ENGINE_orgru=orgru
+URL_orgru="http://paste.org.ru/"
+SIZE_orgru="100000 100%kB"
+REGEX_URL_orgru="s|<a href=\"\(/\?.*\)\">refresh</a>|http://paste.org.ru\1|p"
+REGEX_RAW_orgru="s|<a href=\"\(/\?.*\)\">refresh</a>|http://paste.org.ru\1|"
+DEFAULT_LANGUAGE_orgru="AutoDetect"
 # calculate
 ENGINE_calculate=calculate
 if [[ "$LANG" =~ 'ru_RU' ]]; then
@@ -97,6 +104,19 @@
 REGEX_RAW_tinyurl='s|^\(http://[^/]*/\)\([[:alnum:]]*\)$|\1\2|'
 
 ### engines
+# orgru
+LANGUAGES_orgru="AutoDetect C C++ ASM Perl Java IDL Pascal C# JS.NET VB.NET Forth Fortran Visual%Basic html css css%for%html \
+css%for%svg jsp php xhtml%transitional xhtml%strict xhtml%frameset SVG%1.0 ColdFusion JavaScript ActionScript VBScript xml \
+dtd xslt%1.0 XML%Schema Relax%NG Clarion Clipper FoxPro SQLJ%(Java%sql) Paradox SQL,%PL/SQL MySQL Batch/Config.sys/NTcmd \
+sh/ksh/bash%script Apache%httpd.conf Config,%INI%and%CTL Java%properties Lex YACC makefile TeX VRML 1C Ada ABAP/4 AutoIt%2.x \
+AWK Dssp ADSP-21xx%Asm Cobol Cache/Open-M Eiffel Erlang Icon Lisp MatLab Modula2%and%Oberon2 PicAsm Python Rexx Ruby Standard%ML \
+OCaml Tcl/Tk Verilog%HDL VHDL Diff/Patch xslfo%1.0 DocBook%4.2 MathML2 wml rss mason PostScript RTF%text M4 VIM"
+LANGUAGE_VALUES_orgru="1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 27 28 29 30 31 32 33 34 35 36 38 39 40 41 42 \
+43 44 45 46 47 48 53 54 55 56 59 61 66 67 68 69 70 71 72 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 91 92 98 111 112 113 114 \
+120 133 135 137 156 158"
+POST_orgru(){
+	POST_generic "submit % % type % % code" "$1" "$2" "$3" "$4" "$5" "$6"
+}
 # 0x0
 escape_description_0x0() { echo "$*"; }
 escape_input_0x0() { echo "$*"; }
