diff --git a/ktranslatoid.cpp b/ktranslatoid.cpp
index b3f5921..e28ca66 100644
--- a/ktranslatoid.cpp
+++ b/ktranslatoid.cpp
@@ -26,6 +26,8 @@
 #include <QTextCodec>
 #include <QToolButton>
 #include <QTreeView>
+#include <QApplication>
+#include <QTextEdit>
 
 #include <KConfigDialog>
 #include <KDebug>
@@ -408,12 +410,13 @@ void KTranslatoid::copy()
 {
     m_destText->nativeWidget()->selectAll();
     m_destText->nativeWidget()->copy();
+    QApplication::clipboard()->setText(m_destText->nativeWidget()->toPlainText(),QClipboard::Selection);
 }
 //=========================================================================================
 void KTranslatoid::paste()
 {
     m_sourceText->nativeWidget()->clear();
-    m_sourceText->nativeWidget()->paste();
+    m_sourceText->nativeWidget()->setText(QApplication::clipboard()->text(QClipboard::Selection));
 }
 //==========================================================================================
 void KTranslatoid::tts()
@@ -535,7 +538,7 @@ void KTranslatoid::readConfig()
 {
     KConfigGroup cg = config();
 
-    QString a = cg.readEntry("lsource", QString("fr"));
+    QString a = cg.readEntry("lsource", QString("ru"));
     QString b = cg.readEntry("lresult", QString("en"));
     m_autoPaste = cg.readEntry("autoPaste", true);
     m_autoTranslate = cg.readEntry("autoTranslate", true);
