diff -ruN translatoid.orig/ktranslatoid.cpp translatoid/ktranslatoid.cpp
--- translatoid.orig/ktranslatoid.cpp	2009-04-21 18:55:47.000000000 +0400
+++ translatoid/ktranslatoid.cpp	2009-05-10 14:00:58.510050532 +0400
@@ -26,6 +26,9 @@
 #include <QToolButton>
 #include <KMessageBox>
 #include <KPushButton>
+#include <QApplication>
+#include <QClipboard>
+#include <QTextEdit>
 #include <QLabel>
 #include <KConfigDialog>
 #include <KStandardDirs>
@@ -328,12 +331,13 @@
 {
     m_textEditB->nativeWidget()->selectAll();
     m_textEditB->nativeWidget()->copy();
+    QApplication::clipboard()->setText(m_textEditB->nativeWidget()->toPlainText(),QClipboard::Selection);
 }
 //=========================================================================================
 void KTranslatoid::paste()
 {
     m_textEditA->nativeWidget()->clear();
-    m_textEditA->nativeWidget()->paste();
+    m_textEditA->nativeWidget()->setText(QApplication::clipboard()->text(QClipboard::Selection));
 }
 
 //==========================================================================================
@@ -489,7 +493,7 @@
 
     KConfigGroup cg = config();
 
-    QString a = cg.readEntry("lsource", QString("fr"));
+    QString a = cg.readEntry("lsource", QString("ru"));
     QString b = cg.readEntry("lresult", QString("en"));
     m_command = cg.readEntry("tts", QString());
     m_autoPaste = cg.readEntry("autoPaste", true);
