diff -uNr translatoid2.old/ktranslatoid.cpp translatoid2/ktranslatoid.cpp
--- translatoid2.old/ktranslatoid.cpp	2009-01-31 15:57:29.000000000 +0300
+++ translatoid2/ktranslatoid.cpp	2009-02-11 12:53:30.000000000 +0300
@@ -34,6 +34,8 @@
 #include <QGraphicsRectItem>
 #include <QToolButton>
 #include <KPushButton>
+#include <QApplication>
+#include <QClipboard>
  #include <QLabel>
 KTranslatoid ::KTranslatoid (QObject *parent, const QVariantList &args)
     :  Plasma::PopupApplet(parent, args)
@@ -68,9 +70,9 @@
     setAspectRatioMode(Plasma::IgnoreAspectRatio);
 
 
-    m_languageA.code = "fr";
-    m_languageA.name= "french";
-    m_languageA.icon = "kbflags/fr.png";
+    m_languageA.code = "ru";
+    m_languageA.name= "russe";
+    m_languageA.icon = "kbflags/ru.png";
 
     m_languageB.code = "en";
     m_languageB.name= "english";
@@ -279,7 +281,7 @@
   void KTranslatoid::paste()
 {
   m_textEditA->nativeWidget()->clear();
-  m_textEditA->nativeWidget()->paste();
+  m_textEditA->nativeWidget()->setText(QApplication::clipboard()->text(QClipboard::Selection));
 }
 
 //==========================================================================================
