# Calculate format=diff merge(dev-libs/libqtxdg)=>3.2.0,3.2.1
diff --git a/qtxdg/xdgmenuwidget.cpp b/qtxdg/xdgmenuwidget.cpp
index 3756b34..b4be855 100644
--- a/qtxdg/xdgmenuwidget.cpp
+++ b/qtxdg/xdgmenuwidget.cpp
@@ -170,7 +170,7 @@ void XdgMenuWidgetPrivate::mouseMoveEvent(QMouseEvent *event)
         return;
 
     QList<QUrl> urls;
-    urls << QUrl(a->desktopFile().fileName());
+    urls << QUrl::fromLocalFile(a->desktopFile().fileName());
 
     QMimeData *mimeData = new QMimeData();
     mimeData->setUrls(urls);
