# Calculate format=diff merge(x11-misc/sddm)>=0.20.0
diff --git src/common/VirtualTerminal.cpp src/common/VirtualTerminal.cpp
index 2cac325..9808236 100644
--- src/common/VirtualTerminal.cpp
+++ src/common/VirtualTerminal.cpp
@@ -193,7 +193,7 @@ out:
                 return vtActive;
             }
 
-            return vt;
+            return 7;
         }
 
         void jumpToVt(int vt, bool vt_auto) {
