diff -ruN cuneiform-qt-0.1.1.orig/main.cpp cuneiform-qt-0.1.1/main.cpp
--- cuneiform-qt-0.1.1.orig/main.cpp	2009-04-09 15:11:56.000000000 +0400
+++ cuneiform-qt-0.1.1/main.cpp	2009-04-09 18:00:35.000000000 +0400
@@ -21,7 +21,7 @@
 #include <iostream>
 #include "mainWindow.h"
 
-int main( int argc, char *argv[] ) {
+int main( int argc, char *argv[], char **env ) {
 
 	Q_INIT_RESOURCE( mainWindow );
 	
@@ -34,9 +34,22 @@
 	app.setOrganizationDomain( "http://www.altlinux.org/Cuneiform-Qt" );
 
 	// Load localization
-	QTranslator translator;
 	QString locale = QLocale::system().name().left( 2 );
+    int i = 0;
+    while(env[i]!=0)
+    {
+        QStringList keyval = QString(env[i++]).split("=");
+        if( keyval.count() > 1 && keyval[0] == QString("LANG") )
+        {
+            if( keyval[1].length() >= 2 )
+            {
+                locale = keyval[1].left(2);
+            }
+        }
+    }
+	QTranslator translator;
 	translator.load( QString( DATADIR ) + app.applicationName() + QString ( "_" ) + locale );
+
     app.installTranslator( &translator );
 	 
 	// Process command line arguments
