diff --git update/utils/cl_update_profile.py update/utils/cl_update_profile.py
index 20121a1..74ddaf7 100644
--- update/utils/cl_update_profile.py
+++ update/utils/cl_update_profile.py
@@ -39,8 +39,6 @@ class ClUpdateProfileAction(Action):
 
     # список задач для дейсвия
     tasks = [
-        {'name': 'check_run',
-         'method': 'Update.checkRun(cl_update_wait_another_set)'},
         {'name': 'migrate_repository',
          'method': 'Update.migrateCacheRepository('
                    'cl_update_profile_rep,cl_update_profile_branch)',
