diff --git pym/update/update_info.py pym/update/update_info.py
index 8ab9fca..f4cfd32 100644
--- pym/update/update_info.py
+++ pym/update/update_info.py
@@ -46,7 +46,11 @@ class UpdateInfo(object):
         """
         return SystemIni().getVar(cls.section, cls.varname) == u'on'
 
-    def set_update_ready(self, value):
+    @classmethod
+    def set_update_ready(cls, value):
+        """
+        Установить статус обновления
+        """
         SystemIni().setVar(self.section,
             {self.varname:"on" if value else "off"})
 
