diff --git update/cl_update.py update/cl_update.py
index f6f83a6..67ec6cb 100644
--- update/cl_update.py
+++ update/cl_update.py
@@ -55,13 +55,15 @@ class Update:
             self.closeClTemplate()
         self.clVars.Set("cl_chroot_path","/", True)
         self.clTempl = ProgressTemplate(self.setProgress,self.clVars,
-                                        cltObj=False)
+                                        cltObj=False,
+                                        printSUCCESS=self.printSUCCESS,
+                                        printWARNING=self.printWARNING,
+                                        printERROR=self.printERROR)
         dirsFiles = self.clTempl.applyTemplates()
         if self.clTempl.getError():
             self.printERROR(self.clTempl.getError().strip())
             return False
         else:
-            print dirsFiles
             return dirsFiles
 
     def initVars(self,datavars=None):
diff --git update/cl_wsdl_update.py update/cl_wsdl_update.py
index 784dcff..c3b4a13 100644
--- update/cl_wsdl_update.py
+++ update/cl_wsdl_update.py
@@ -53,7 +53,7 @@ class Wsdl:
     """
     @rpc(Integer, UpdateInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Update'),title=__('Update world'),
-        image='preferences-desktop',
+        image='software-properties,preferences-desktop',
         gui=True,command='cl-update',
         rights=['update'])
     def update(self, sid, info):
