diff --git pym/update/update.py pym/update/update.py
index 9387524..a5d2c63 100644
--- pym/update/update.py
+++ pym/update/update.py
@@ -284,7 +284,7 @@ class Update(object):
                 except OSError as e:
                     raise UpdateError(_("Failed to modify the "
                                         "{repname} repository").format(
-                        repname=repname)+":"+str(e))
+                        repname=repname)+_(": ")+str(e))
                 finally:
                     if path.exists(rpath_new):
                         removeDir(rpath_new)
