diff --git update/variables/action.py update/variables/action.py
index 9fe0c5c..cf4f42c 100644
--- update/variables/action.py
+++ update/variables/action.py
@@ -24,11 +24,11 @@ setLocalTranslate('cl_update3',sys.modules[__name__])
 
 class VariableAcUpdateSync(ReadonlyVariable):
     """
-    Action variable which has value "up" for package install and
+    Action variable which has value "on" for package install and
     install this package
     """
     def get(self):
         action = self.Get("cl_action")
         if action in ("sync",):
-            return "up"
+            return "on"
         return ""
