diff --git pym/update/update.py pym/update/update.py
index 3db40c6..b462210 100644
--- pym/update/update.py
+++ pym/update/update.py
@@ -617,13 +617,13 @@ class Update:
                 mult=mult, fullname=fullname, shortname=shortname, size=size,
                 flag=install_flag))
 
-    def _display_install_package(self, emerge):
+    def _display_install_package(self, emerge, emergelike=False):
         """
         Отобразить список устанавливаемых пакетов
         """
         # подробный список пакетов
         _print = self.color_print
-        if self.clVars.Get('cl_update_emergelist_set') == 'on':
+        if emergelike:
             self.printPre(str(emerge.install_packages))
         else:
             pkglist = emerge.install_packages.list
@@ -722,10 +722,12 @@ class Update:
                     self.printSUCCESS(_("The system is up to date"))
                     self.set_need_update(False)
                     return True
-                self._display_install_package(emerge)
+                emergelike = self.clVars.Get('cl_update_emergelist_set') == 'on'
+                self._display_install_package(emerge, emergelike)
             except EmergeError:
                 self.set_need_update(False)
                 self.emerge_cache.drop_cache("Emerge error")
+                self._display_install_package(emerge, emergelike=True)
                 self._display_error(emerge.prepare_error)
                 raise
             if self.clVars.Get('cl_update_pretend_set') == 'on':
