diff --git update/wsdl_update.py update/wsdl_update.py
index 6f95581..fea9c45 100644
--- update/wsdl_update.py
+++ update/wsdl_update.py
@@ -86,7 +86,7 @@ class Wsdl(WsdlBase):
             # категория метода
             'category': __('Update'),
             # заголовок метода
-            'title': __("Update the Profile"),
+            'title': __("Change the Profile"),
             # иконка для графической консоли
             'image': 'preferences-system',
             # метод присутствует в графической консоли
