diff --git update/cl_wsdl_update.py update/cl_wsdl_update.py
index 784dcff..c3b4a13 100644
--- update/cl_wsdl_update.py
+++ update/cl_wsdl_update.py
@@ -53,7 +53,7 @@ class Wsdl:
     """
     @rpc(Integer, UpdateInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Update'),title=__('Update world'),
-        image='preferences-desktop',
+        image='software-properties,preferences-desktop',
         gui=True,command='cl-update',
         rights=['update'])
     def update(self, sid, info):
