diff --git update/variables/update.py update/variables/update.py
index a0634ac..69845c0 100644
--- update/variables/update.py
+++ update/variables/update.py
@@ -901,9 +901,7 @@ class VariableClUpdateAutocheckInterval(Variable):
         self.help = _("set interval for the updates checking")
 
     def choice(self):
-        return [["1h",_("hourly")],
-            ["3h",_("every three hours")],
-            ["6h",_("every six hours")],
+        return [["6h",_("every six hours")],
             ["12h",_("every twenty hours")],
             ["1d",_("daily")]]
 
