commit 15160e92a4627b27a8f0e6e41962a57a7baaffb5
Author: Mike Khiretskiy <mh@calculate.ru>
Date:   Mon Jan 19 15:37:09 2015 +0300

    Исправлен вывод при удалении пакета

diff --git pym/update/update.py pym/update/update.py
index 2f1e9f4..1efd4ec 100644
--- pym/update/update.py
+++ pym/update/update.py
@@ -531,10 +531,13 @@ class Update(object):
         """
         self.endTask()
         _print = self.color_print
-        one = _print("{0}", num)
-        two = _print("{0}", max_num)
-        part = _(" ({current} of {maximum})").format(current=one,
-                                                     maximum=two)
+        if num and max_num:
+            one = _print("{0}", num)
+            two = _print("{0}", max_num)
+            part = _(" ({current} of {maximum})").format(current=one,
+                                                         maximum=two)
+        else:
+            part = ""
         _print = _print.foreground(Colors.LIGHT_RED)
 
         self.startTask(
