diff --git update/cl_update.py update/cl_update.py
index 4de3d68..36197c6 100644
--- update/cl_update.py
+++ update/cl_update.py
@@ -54,8 +54,9 @@ class Update:
         if self.clTempl:
             self.closeClTemplate()
         self.clVars.Set("cl_chroot_path","/", True)
+        self.clVars.Set("cl_merge_set","on",True)
         self.clTempl = ProgressTemplate(self.setProgress,self.clVars,
-                                        cltObj=False,
+                                        cltObj=True,cltFilter=True,
                                         printSUCCESS=self.printSUCCESS,
                                         printWARNING=self.printWARNING,
                                         askConfirm=self.askConfirm,
