# Calculate format=diff
diff --git a/microblogs/twitter/twitteraccount.cpp b/microblogs/twitter/twitteraccount.cpp
index de091dc..bf7eeac 100644
--- a/microblogs/twitter/twitteraccount.cpp
+++ b/microblogs/twitter/twitteraccount.cpp
@@ -33,8 +33,8 @@ public:
 //     QStringList lists;
 };
 
-const char *twitterConsumerKey = "VyXMf0O7CvciiUQjliYtYg";
-const char *twitterConsumerSecret = "uD2HvsOBjzt1Vs6SnouFtuxDeHmvOOVwmn3fBVyCw0";
+const char *twitterConsumerKey = "irARiVrv825Fl2oZaYiQA";
+const char *twitterConsumerSecret = "qYZa8Dszrow6vry4iwtSm1zTidnW7ii8IQztgXWZY";
 
 TwitterAccount::TwitterAccount(TwitterMicroBlog *parent, const QString &alias)
     : TwitterApiAccount(parent, alias), d(new Private)
