# Calculate format=diff merge(xfce-base/xfdesktop)<4.20.0
diff --git settings/main.c settings/main.c
index e3d234b..b0b7150 100644
--- settings/main.c
+++ settings/main.c
@@ -1222,26 +1222,16 @@ cb_xfdesktop_combo_color_changed(GtkComboBox *combo,
 static void
 xfdesktop_settings_update_iconview_folder(AppearancePanel *panel)
 {
-    gchar *current_folder, *dirname;
+    gchar *dirname = "/usr/share/backgrounds/xfce";
 
     /* If we haven't found our window return now and wait for that */
     if(panel->wnck_window == NULL)
         return;
 
-    TRACE("entering");
-
-    current_folder = xfdesktop_settings_get_backdrop_image(panel);
-    dirname = g_path_get_dirname(current_folder);
-
-    XF_DEBUG("current_folder %s, dirname %s", current_folder, dirname);
-
     gtk_file_chooser_set_current_folder((GtkFileChooser*)panel->btn_folder, dirname);
 
     /* Workaround for a bug in GTK */
     cb_folder_selection_changed(panel->btn_folder, panel);
-
-    g_free(current_folder);
-    g_free(dirname);
 }
 
 /* This function is to add or remove all the bindings for the background
