# Calculate format=diff
diff -ruN lightdm-1.8.5.orig/src/seat.c lightdm-1.8.5/src/seat.c
--- lightdm-1.8.5.orig/src/seat.c	2013-12-13 11:02:46.635945026 +0400
+++ lightdm-1.8.5/src/seat.c	2013-12-13 11:03:32.904947413 +0400
@@ -605,7 +605,7 @@
     }
 
     /* Cleanup */
-    if (!IS_GREETER (session))
+    if (!IS_GREETER (session) && session_get_is_authenticated(session))
     {
         const gchar *script;
         script = seat_get_string_property (seat, "session-cleanup-script");
