# Calculate format=diff
diff --git plymouth.c plymouth.c
index e657e82..bb97f56 100644
--- plymouth.c
+++ plymouth.c
@@ -85,7 +85,8 @@ int commandf(const char* cmd, ...)
 
 bool ply_message(const char* hook, const char* name)
 {
-    return (commandf("/bin/plymouth message --text=\"%s %s\"", hook, name) == 0);
+    return true;
+    //return (commandf("/bin/plymouth message --text=\"%s %s\"", hook, name) == 0);
 }
 
 
@@ -94,6 +95,11 @@ bool ply_ping()
     return (system("/bin/plymouth --ping") == 0);
 }
 
+void restart_console_font()
+{
+    if(RC_SERVICE_STARTED == rc_service_state("consolefont"))
+        command("/etc/init.d/consolefont restart");
+}
 
 bool ply_quit(int mode)
 {
@@ -127,9 +133,9 @@ bool ply_start(int mode)
 #define PLYD "/sbin/plymouthd --attach-to-session --pid-file=" PID_FILE \
     " --mode="
         if(mode == PLY_MODE_BOOT)
-            rv = command(PLYD "boot");
+            rv = command(PLYD "boot --tty=/dev/tty1");
         else if(mode == PLY_MODE_SHUTDOWN)
-            rv = command(PLYD "shutdown");
+            rv = command(PLYD "shutdown --tty=/dev/tty1");
         else
             assert(0 && "Unknown mode");
 #undef PLYD
@@ -145,7 +151,8 @@ bool ply_start(int mode)
 
 bool ply_update_status(int hook, const char* name)
 {
-    return (commandf("/bin/plymouth update --status=%d-%s", hook, name) == 0);
+    return true;
+    //return (commandf("/bin/plymouth update --status=%d-%s", hook, name) == 0);
 }
 
 
@@ -215,6 +222,7 @@ int rc_plugin_hook(RC_HOOK hook, const char *name)
             if(!ply_start(PLY_MODE_SHUTDOWN)
                     || !ply_update_rootfs_rw())
                 rv = 1;
+    	    command("/usr/bin/chvt 1");
         }
         break;
 
@@ -235,16 +243,7 @@ int rc_plugin_hook(RC_HOOK hook, const char *name)
             DBG("ply_quit(PLY_MODE_BOOT)");
             if(!ply_quit(PLY_MODE_BOOT))
                 rv = 1;
-        }
-        break;
-
-    case RC_HOOK_SERVICE_STOP_IN:
-        /* Quit Plymouth when we're going to lost write access to /var/... */
-        if(strcmp(name, "localmount") == 0 &&
-                strcmp(runlevel, RC_LEVEL_SHUTDOWN) == 0) {
-            DBG("ply_quit(PLY_MODE_SHUTDOWN)");
-            if(!ply_quit(PLY_MODE_SHUTDOWN))
-                rv = 1;
+	    restart_console_font();
         }
         break;
 
