# Calculate merge(sys-kernel/dracut)>=40 format=diff
diff --git dracut.sh dracut.sh
index 1c7e208..6327201 100755
--- dracut.sh
+++ dracut.sh
@@ -1485,7 +1485,10 @@ if [[ $UID = 0 ]] && [[ $PRELINK_BIN ]]; then
     elif [[ $do_prelink == yes ]]; then
         dinfo "*** Pre-linking files ***"
         inst_multiple -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf
+        [[ -d $initdir/proc ]] || mkdir $initdir/proc
+        mount -n -t proc none $initdir/proc
         chroot "$initdir" "$PRELINK_BIN" -a
+        umount -n $initdir/proc
         rm -f -- "$initdir/$PRELINK_BIN"
         rm -fr -- "$initdir"/etc/prelink.*
         dinfo "*** Pre-linking files done ***"
