# Calculate format=diff merge(app-emulation/bochs)=>2.6.9,2.7
Index: iodev/network/slirp/slirp.h
===================================================================
--- iodev/network/slirp/slirp.h	(revision 11848)
+++ iodev/network/slirp/slirp.h	(working copy)
@@ -45,7 +45,7 @@
 #endif

 #include <sys/types.h>
-#if defined(__OpenBSD__)
+#if defined(__OpenBSD__) || (__linux__)
 #include <stdint.h>
 #include <sys/wait.h>
 #endif
