diff --git a/exo-helper/exo-helper.c b/exo-helper/exo-helper.c
index a3a8c60..a37ed83 100644
--- a/exo-helper/exo-helper.c
+++ b/exo-helper/exo-helper.c
@@ -422,7 +422,7 @@ exo_helper_execute (ExoHelper   *helper,
                   succeed = FALSE;
                   break;
                 }
-              else if (result > 0 && status != 0)
+              else if (result > 0 && status != 0 && status != 5888)
                 {
                   /* the command failed */
                   err = g_error_new_literal (G_FILE_ERROR, g_file_error_from_errno (EIO), g_strerror (EIO));
