# Calculate format=diff
diff --git a/plugins/thunar-tpa/thunar-tpa.c b/plugins/thunar-tpa/thunar-tpa.c
index 184237c..2841622 100644
--- a/plugins/thunar-tpa/thunar-tpa.c
+++ b/plugins/thunar-tpa/thunar-tpa.c
@@ -259,7 +259,7 @@ thunar_tpa_size_changed (XfcePanelPlugin *panel_plugin,
 #if LIBXFCE4PANEL_CHECK_VERSION (4,13,0)
   image_size = xfce_panel_plugin_get_icon_size (panel_plugin);
 #else
-  image_size = size - 2; // fall-back for older panel versions
+  image_size = size - 6; // fall-back for older panel versions
 #endif
   gtk_image_set_pixel_size (GTK_IMAGE (plugin->image), image_size);
 
