diff -ur KGtk-0.9.5.orig/kdialogd4/kdialogd.cpp KGtk-0.9.5/kdialogd4/kdialogd.cpp
--- KGtk-0.9.5.orig/kdialogd4/kdialogd.cpp	2008-06-05 22:15:05.000000000 +0400
+++ KGtk-0.9.5/kdialogd4/kdialogd.cpp	2008-11-10 09:56:55.000000000 +0300
@@ -383,7 +383,7 @@
     // * the dir select dialog doesnt seem to set the QDialog result parameter
     //   when it is accepted - so for this reason if ok is clicked we store an
     //   'accepted' value there, and check for that after the dialog is finished.
-    kDebug() << "finished " << (int)itsDlg << itsAccepted << (itsDlg ? QDialog::Accepted==itsDlg->result() : false);
+    kDebug() << "finished " << (long)itsDlg << itsAccepted << (itsDlg ? QDialog::Accepted==itsDlg->result() : false);
 
     if(itsDlg && !(itsAccepted || QDialog::Accepted==itsDlg->result()))
         cancel();
