# Calculate format=diff merge(x11-misc/lightdm-gtk-greeter)>=2.0.1
diff --git a/configure.ac b/configure.ac
index eb0d53a..9d7196b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -136,7 +136,7 @@ dnl ###########################################################################
 AC_ARG_ENABLE([kill-on-sigterm],
     AC_HELP_STRING([--enable-kill-on-sigterm], [Kill greeter instance on SIGTERM, see LP1445461])
     AC_HELP_STRING([--disable-kill-on-sigterm], [Don't kill greeter on SIGTERM, see LP1445461']),
-            [], [enable_kill_on_sigterm=no])
+            [], [enable_kill_on_sigterm=yes])
 
 AS_IF([test "x$enable_kill_on_sigterm" = "xyes"],
 [
