commit 4217ac49db27f7b0a289b534abf7095da1140ab3
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Dec 26 11:48:53 2011 +0400

    Add nouveau.

diff --git a/src/pageconfiguration.cpp b/src/pageconfiguration.cpp
index a57d791..37d193b 100644
--- a/src/pageconfiguration.cpp
+++ b/src/pageconfiguration.cpp
@@ -272,6 +272,7 @@ void PageConfiguration::retranslateUi()
 	m_cmbboxVideoDrv->addItem("fglrx");
 	m_cmbboxVideoDrv->addItem("intel");
 	m_cmbboxVideoDrv->addItem("nv");
+	m_cmbboxVideoDrv->addItem("nouveau");
 	m_cmbboxVideoDrv->addItem("nvidia");
 	m_cmbboxVideoDrv->addItem("radeon");
 	m_cmbboxVideoDrv->addItem("vesa");
