Index: krusader/panelmanager.cpp
===================================================================
--- krusader/panelmanager.cpp	(revision 1154580)
+++ krusader/panelmanager.cpp	(revision 1154581)
@@ -121,6 +121,7 @@
 
 //     _stack->setUpdatesEnabled(false);
     _stack->setCurrentWidget(_self);
+    kapp->processEvents();
     // make sure the view is focused (this also causes ListPanel::slotFocusOnMe() to be called)
     _self->view->widget()->setFocus();
 //     _stack->setUpdatesEnabled(true);
