diff --git pym/core/setup_package.py pym/core/setup_package.py
index 32c3e54..34a703c 100644
--- pym/core/setup_package.py
+++ pym/core/setup_package.py
@@ -361,12 +361,13 @@ class PackageUpdater(UpdateConfigs): #,updateUserConfigs):
     def updateConfig(self):
         if not os.path.exists('/etc/calculate/calculate.env') and \
             not (dv.Get('cl_core_pkg_name') in ('calculate-utilities',
+                                                'calculate-utils',
                                                 'calculate-install') and
                  getTupleVersion(dv.Get('cl_core_pkg_version')) >=
                  getTupleVersion("3.1.0_alpha4")):
             self.printWARNING(
                 _("Configuration skipping until "
-                  "calculate-utilities are updated"))
+                  "calculate-utils are updated"))
             return True
         self.initVars(dv)
         clVars = DataVars()
