commit 02574df1e65896200ae663fddd62776f681072ce
Author: Mike Khiretskiy <mh@calculate.ru>
Date:   Fri Oct 10 11:09:53 2014 +0400

    Исправлен вызов утилит при отсутствии stdin

diff --git pym/core/result_viewer.py pym/core/result_viewer.py
index 8bda238..7bf8b51 100644
--- pym/core/result_viewer.py
+++ pym/core/result_viewer.py
@@ -268,7 +268,7 @@ class TaskState(object):
                 printer(full_message)
                 ask = getch()
                 printer("\n")
-                if ask == CTRC_C:
+                if ask in (CTRC_C,""):
                     raise KeyboardInterrupt
             else:
                 try:
diff --git pym/core/server/func.py pym/core/server/func.py
index b06ff53..b6941d3 100644
--- pym/core/server/func.py
+++ pym/core/server/func.py
@@ -62,19 +62,22 @@ class CommonMethods:
             out = []
             orig,data = data
             data = data[FIRST]
-            for i,s in enumerate(list(process("diff","-Nu",
-                       orig,data[CFGFNAME]))):
-                s = convert_console_to_xml(s)
-                if s.startswith('+') and i>1:
-                    out.append('<font color="green">%s</font>'%s)
-                elif s.startswith('-') and i>1:
-                    out.append('<font color="red">%s</font>'%s)
-                else:
-                    out.append(s)
-            self.printPre("<br/>".join(out))
-            self.printSUCCESS(_("({one} of {_all}) -- {fname}").format(
-                one=ind+1,_all=maxInfo,fname=orig))
-            answ = self.askChoice(_("Choose a configuration action:"),
+            if self.clVars.Get('cl_autoupdate_set') == 'on':
+                answ = "use new"
+            else:
+                for i,s in enumerate(list(process("diff","-Nu",
+                           orig,data[CFGFNAME]))):
+                    s = convert_console_to_xml(s)
+                    if s.startswith('+') and i>1:
+                        out.append('<font color="green">%s</font>'%s)
+                    elif s.startswith('-') and i>1:
+                        out.append('<font color="red">%s</font>'%s)
+                    else:
+                        out.append(s)
+                self.printPre("<br/>".join(out))
+                self.printSUCCESS(_("({one} of {_all}) -- {fname}").format(
+                    one=ind+1,_all=maxInfo,fname=orig))
+                answ = self.askChoice(_("Choose a configuration action:"),
                                   answers=(("zap new", _("Zap new")),
                                            ("use new", _("Use new")),
                                            ("next", _("Next"))))
