diff --git core/server/func.py core/server/func.py
index 1e0f968..040b715 100644
--- core/server/func.py
+++ core/server/func.py
@@ -21,6 +21,7 @@ from os import path
 import traceback
 from traceback import print_exc
 from calculate.lib.cl_log import log
+from calculate.lib.utils.colortext import convert_console_to_xml
 from api_types import ReturnProgress,ViewParams,Integer,ViewInfo
 from loaded_methods import LoadedMethods
 from calculate.lib.cl_lang import setLocalTranslate,getLazyLocalTranslate
@@ -61,6 +62,7 @@ class CommonMethods:
             data = data[FIRST]
             for i,s in enumerate(list(process("diff","-Nu",
                        orig,data[CFGFNAME]))):
+                s = convert_console_to_xml(s)
                 if s.startswith('+') and i>1:
                     out.append('<font color="green">%s</font>'%s)
                 elif s.startswith('-') and i>1:
