diff --git core/server/func.py core/server/func.py
index 0d5d358..19807d5 100644
--- core/server/func.py
+++ core/server/func.py
@@ -59,7 +59,7 @@ class CommonMethods:
             self.printPre("<br/>".join(out))
             self.printSUCCESS(_("({one} of {_all}) -- {fname}").format(
                 one=ind+1,_all=maxInfo,fname=orig))
-            answ = self.askChoice(_("Choice action with config:"),
+            answ = self.askChoice(_("Choose a configuration action:"),
                     answers=(("zap new",_("Zap new")),
                              ("use new",_("Use new")),
                              ("next",_("Next"))))
diff --git core/server/replace_class.py core/server/replace_class.py
index 379798b..536209d 100644
--- core/server/replace_class.py
+++ core/server/replace_class.py
@@ -332,7 +332,7 @@ class replaceClass():
                 ask = ask.lower()
                 likeAnswers = filter(lambda x:x[VALUE].startswith(ask),answers)
                 if not likeAnswers:
-                    colorPrint.printERROR(_('The answer is not understood'))
+                    colorPrint.printERROR(_('The answer is uncertain'))
                     continue
                 if len(likeAnswers) == 1:
                     return likeAnswers[VALUE][FIRSTLETTER]
diff --git core/server/setup_package.py core/server/setup_package.py
index d10cd18..36bd5da 100644
--- core/server/setup_package.py
+++ core/server/setup_package.py
@@ -502,8 +502,8 @@ class PackageUpdater(updateSystemConfigs,updateUserConfigs):
         nameProgram = self.clVars.Get('cl_core_pkg_name')
         clTempl.onFirstValue = lambda *args: \
                 self.startTask(
-                    _("Using patches for the {nameProgram} package by "
-                      "Calculate Utilities").format(
+                    _("Using Calculate patches for the {nameProgram} "
+                      "package").format(
                       nameProgram=nameProgram),
                       progress=True)
         dirsFiles = clTempl.applyTemplates()
