commit fd067f475a9b3ff2343e41056b51463283ed227d
Author: Хирецкий Михаил <mh@calculate.ru>
Date:   Thu Aug 2 16:50:40 2012 +0400

    Fix bool variable

diff --git a/core/server/methods_func.py b/core/server/methods_func.py
index fa5338d..66c225b 100644
--- a/core/server/methods_func.py
+++ b/core/server/methods_func.py
@@ -140,10 +140,6 @@ def get_method_argparser(view, args, cl_core = False):
                     data['action'] = "store_true"
                 else:
                     data['type'] = str
-                if field.element in ['check', 'check_tristate']:
-                    data['type'] = bool
-                elif field.element == 'radio' and field.type == 'bool':
-                    data['type'] = bool
                 if field.element == 'table' and field.type != 'steps':
                     data['action'] = 'append'
 
