diff --git core/server/baseClass.py core/server/baseClass.py
index d55eada..b82eac8 100644
--- core/server/baseClass.py
+++ core/server/baseClass.py
@@ -39,6 +39,7 @@ class Basic (DefinitionBase) :
     glob_progress_dict = {}
     glob_table_dict = {}
     glob_frame_list = {}
+    manager = None
 
     data_path = None
     certbase = None
diff --git core/server/func.py core/server/func.py
index 8e8f92d..834ed54 100644
--- core/server/func.py
+++ core/server/func.py
@@ -419,8 +419,8 @@ class CoreWsdl():
         self.add_sid_pid(sid, pid)
 
         import multiprocessing
-        if not hasattr(self, 'manager'):
-            self.manager = multiprocessing.Manager()
+        if self.manager is None:
+            self.__class__.manager = multiprocessing.Manager()
         # Manager for sending glob_process_dict between watcher and process
         #manager = multiprocessing.Manager()
         self.glob_process_dict[pid] = self.manager.dict()
diff --git core/variables/core.py core/variables/core.py
index a0d7708..50665e8 100644
--- core/variables/core.py
+++ core/variables/core.py
@@ -150,13 +150,6 @@ class VariableClCoreKey(Variable):
     def get(self):
         return path.join(self.Get('cl_core_cert_path'),"server.key")
 
-class VariableClCoreSidLive(Variable):
-    """
-    Max time live sessions (in minutes)
-    """
-    type = "int"
-    value = "1440"
-
 class VariableClCoreCertLive(Variable):
     """
     Max time live sessions (in minutes)
diff --git core/variables/variable.py core/variables/variable.py
index de0717f..a350d47 100644
--- core/variables/variable.py
+++ core/variables/variable.py
@@ -126,7 +126,7 @@ class VariableClVariableFullname(VarHelper,ReadonlyVariable):
             moduleSection = self.mapSection.get(moduleName,moduleName)
             if not moduleSection in self.parent.importedModules:
                 self.parent.importVariables('calculate.%s.variables'%moduleName)
-                self.parent.flIniFile(onlySection=moduleSection)
+                self.parent.flIniFile()
             for varname,vardata in sorted(self.parent.allVars.items()):
                 if vardata[0] == moduleSection and \
                     not varname.startswith('cl_variable'):
@@ -164,6 +164,7 @@ class VariableClVariableValue(VarHelper,Variable):
         def __init__(self,dv,var):
             self.var = var
             self.dv = dv
+            self.dv.flIniFile()
 
         def __call__(self):
             isListOrTuple = lambda x: type(x) in (list,tuple)
diff --git scripts/calculate-core scripts/calculate-core
index a029d27..ef10337 100644
--- scripts/calculate-core
+++ scripts/calculate-core
@@ -41,7 +41,7 @@ start() {
 
 stop(){
 	ebegin "Stop calculate core"
-	start-stop-daemon --stop --signal INT --quiet --pidfile /var/run/cl_core.pid
+	start-stop-daemon --stop --signal 2 --quiet --pidfile /var/run/cl_core.pid
 	eend 0
 }
 
