diff --git core/server/setup_package.py core/server/setup_package.py
index cb30bb6..2417478 100644
--- core/server/setup_package.py
+++ core/server/setup_package.py
@@ -575,7 +575,7 @@ class CoreWsdl:
                     'cl_core_pkg_slot',
                     'cl_core_pkg_path', 'cl_core_pkg_system_set',
                     'cl_core_pkg_desktop_set', 'cl_core_pkg_root_set',
-                    'cl_template_locate','cl_verbose_set','cl_dispatch_conf'),
+                    'cl_templates_locate','cl_verbose_set','cl_dispatch_conf'),
             next_label=_("Setup"))
         return dv
 
