diff --git a/core/server/func_metaclass.py b/core/server/func_metaclass.py
index c0891f4..0bd9863 100644
--- a/core/server/func_metaclass.py
+++ b/core/server/func_metaclass.py
@@ -39,8 +39,8 @@ class Table (ClassSerializer):
         body = Array (String)
         values = Array (Array (String))
 
-list_module = ["clean", "gen_pid", "check_rights", "cl_template", \
-               "sid_pid_file", "gen_sid", "func", "api_types", 'test', 'test2', \
+list_module = ["clean", "gen_pid", "check_rights", \
+               "sid_pid_file", "gen_sid", "func", "api_types", \
                'system_update', 'request', 'certificate', 'edit_groups']
 
 imported_modules = []
