diff --git core/server/methods_func.py core/server/methods_func.py
index d6d7556..1b11b8a 100644
--- core/server/methods_func.py
+++ core/server/methods_func.py
@@ -345,22 +345,26 @@ def check_result_msg(method_result, view, input_error_dict = {}):
             continue
 
         params_text = ''
-        for Group in groups:
-            if hasattr (Group.fields, 'Field'):
-                fields = Group.fields.Field
-            else:
-                fields = Group.fields
-            for field in fields:
-                if field.name == error.field:
-                    if field.opt.shortopt or field.opt.longopt:
-                        params_text += _('Wrong option ')
-                        params_text += ' '+', '.join(filter(None,
-                            [field.opt.shortopt, field.opt.longopt])) \
-                            + '. '
+        if error.type != "commonerror":
+            for Group in groups:
+                if hasattr (Group.fields, 'Field'):
+                    fields = Group.fields.Field
+                else:
+                    fields = Group.fields
+                for field in fields:
+                    if field.name == error.field:
+                        if field.opt.shortopt or field.opt.longopt:
+                            params_text += _('Wrong option ')
+                            params_text += ' '+', '.join(filter(None,
+                                [field.opt.shortopt, field.opt.longopt])) \
+                                + '. '
 
         sys.stdout.write('\r')
         sys.stdout.flush()
-        colorPrint.printERROR(params_text + error.message)
+        if error.type != "commonerror":
+            colorPrint.printERROR(params_text + error.message)
+        else:
+            colorPrint.printWARNING(params_text + error.message)
 
     if len(password_errors) < len(method_result):
         return None
