diff --git core/server/setup_package.py core/server/setup_package.py
index 2111b8d..366f061 100644
--- core/server/setup_package.py
+++ core/server/setup_package.py
@@ -456,7 +456,7 @@ class updateSystemConfigs(shareUpdateConfigs):
             self.copyConfigFiles(configPath,variables.Get('cl_config_protect'))
         if clTempl and clTempl.getWarning():
             for warn in clTempl.getWarning().split("\n"):
-                self.SUCCESS(warn)
+                self.WARNING(warn)
         return True
 
 class PackageUpdater(updateSystemConfigs,updateUserConfigs):
