diff --git core/utils/cl_core_custom.py core/utils/cl_core_custom.py
index f9eb76d..4133f57 100644
--- core/utils/cl_core_custom.py
+++ core/utils/cl_core_custom.py
@@ -21,7 +21,7 @@ from calculate.lib.datavars import VariableError
 from calculate.lib.cl_template import TemplatesError
 from calculate.lib.utils.files import FilesError
 
-setLocalTranslate('cl_install3',sys.modules[__name__])
+setLocalTranslate('cl_core3',sys.modules[__name__])
 __ = getLazyLocalTranslate(_)
 
 class ClCoreCustomAction(Action):
