diff --git core/server/func.py core/server/func.py
index 5a6b964..b06ff53 100644
--- core/server/func.py
+++ core/server/func.py
@@ -1706,9 +1706,9 @@ class WsdlMeta(type):
             if kwargs["groups"]:
                 def group(*args,**kwargs):
                     if isinstance(kwargs.get('normal',()), (unicode, str)):
-                        raise DataVarsError(_("Wrong normal varaiable list"))
+                        raise DataVarsError(_("Wrong normal varaiables list"))
                     if isinstance(kwargs.get('expert',()), (unicode, str)):
-                        raise DataVarsError(_("Wrong expert varaiable list"))
+                        raise DataVarsError(_("Wrong expert varaiables list"))
                 for gr in kwargs['groups']:
                     gr(group)
             if not dv:
diff --git core/server/gen_pid.py core/server/gen_pid.py
index e0915da..865d991 100644
--- core/server/gen_pid.py
+++ core/server/gen_pid.py
@@ -65,7 +65,7 @@ def clear_finished_pids(clVars):
             d = pickle.load(open(pidfile))
             if path.exists(path.join("/proc", str(d['os_pid']))):
                 continue
-        except (ValueError, KeyError, EOFError, OSError):
+        except BaseException:
             pass
         with ignore(OSError):
             os.unlink(pidfile)
