diff --git core/server/replace_class.py core/server/replace_class.py
index 34fd75b..379798b 100644
--- core/server/replace_class.py
+++ core/server/replace_class.py
@@ -287,7 +287,7 @@ class replaceClass():
                 return default
             while True:
                 try:
-                    ask = raw_input(message+ '? (yes/no): ')
+                    ask = raw_input(message+ ' (yes/no): ')
                 except KeyboardInterrupt:
                     ask = 'no'
                     print
diff --git core/server/setup_package.py core/server/setup_package.py
index 33c0d43..d39c4c2 100644
--- core/server/setup_package.py
+++ core/server/setup_package.py
@@ -202,7 +202,8 @@ class updateUserConfigs(shareUpdateConfigs):
                                printERROR=self.printERROR,
                                askConfirm=self.askConfirm,
                                printWARNING=self.printWARNING,
-                               printWarning=False)
+                               printWarning=False,
+                               userProfile=True)
             clTempl.onFirstValue = lambda *args: \
                     self.startTask(
                         _("User configuring the {nameProgram} package by "
diff --git core/variables/setup_package.py core/variables/setup_package.py
index 154d489..dd340d8 100644
--- core/variables/setup_package.py
+++ core/variables/setup_package.py
@@ -136,7 +136,10 @@ class VariableClCorePkgCategory(Variable):
     def get(self):
         category = isPkgInstalled(self.Get('cl_core_pkg_name'))
         if category:
-            category = category[0].rpartition('/')[2]
+            if type(category[0]) == dict:
+                category = ""
+            else:
+                category = category[0].rpartition('/')[2]
         else:
             choice = self.choice()
             if len(choice) == 1:
