commit 095486b3ecb2a62eb2dce1f61c4bfaa6a572e9f7
Author: root <root@dspiridonov.local.calculate.ru>
Date:   Mon Jun 18 09:54:17 2012 +0400

    fix methods name

diff --git a/core/server/edit_groups.py b/core/server/edit_groups.py
index 9c1527d..9ad6cf5 100644
--- a/core/server/edit_groups.py
+++ b/core/server/edit_groups.py
@@ -21,7 +21,7 @@ from soaplib.serializers.clazz import Array, ClassSerializer
 from soaplib.service import rpc
 from calculate.core.server.api_types import (Field, GroupField, ViewInfo,
                                              ViewParams, ReturnedMessage)
-from calculate.lib.datavars import VariableError
+from calculate.lib.datavars import VariableError, _
 from calculate.lib.utils.files import readLinesFile
 from calculate.lib.cl_lang import getLazyLocalTranslate, setLocalTranslate
 setLocalTranslate('cl_core',sys.modules[__name__])
@@ -120,8 +120,8 @@ class CoreWsdl:
 
             if body:
                 self.printTable(_("List of available groups"), head, body, \
-                                fields = fields, onClick = 'detail_group', \
-                                addAction = 'add_group')
+                            fields = fields, onClick = 'core_detail_group', \
+                            addAction = 'add_group')
                 if not page_offset:
                     return True
                 count_page = len(list_group_name) / page_count + 1
@@ -166,7 +166,6 @@ class CoreWsdl:
                                                 field = var, message = mess))
         return errors
 
-    from calculate.core.server.baseClass import Basic
     from calculate.core.server.decorators import Dec
 
     def groupCommon(self,sid,info,methodname):
diff --git a/core/server/request.py b/core/server/request.py
index dbf1b73..37926fd 100644
--- a/core/server/request.py
+++ b/core/server/request.py
@@ -128,7 +128,7 @@ class CoreWsdl:
 
             if body:
                 self.printTable(_("List of request"),head,body,fields = fields,
-                            onClick = 'detail_request')
+                            onClick = 'core_detail_request')
                 if not page_offset:
                     return True
                 count_page = len(list_req_id) / page_count + 1
