diff --git pym/core/server/func.py pym/core/server/func.py
index b6941d3..ec404b4 100644
--- pym/core/server/func.py
+++ pym/core/server/func.py
@@ -62,7 +62,11 @@ class CommonMethods:
             out = []
             orig,data = data
             data = data[FIRST]
-            if self.clVars.Get('cl_autoupdate_set') == 'on':
+
+            origdata = readFile(orig)
+            newdata = readFile(data[CFGFNAME])
+            if (self.clVars.Get('cl_autoupdate_set') == 'on' or 
+                origdata == newdata):
                 answ = "use new"
             else:
                 for i,s in enumerate(list(process("diff","-Nu",
