diff --git core/server/setup_package.py core/server/setup_package.py
index 2111b8d..fa63aaf 100644
--- core/server/setup_package.py
+++ core/server/setup_package.py
@@ -184,7 +184,7 @@ class updateUserConfigs(shareUpdateConfigs):
             clVars.Set("cl_root_path", '/', True)
             clVars.Set("ur_login", userName, True)
             clVars.Set("cl_action", "desktop", True)
-            clVars.Set("cl_merge_pkg", [nameProgram], True)
+            clVars.Set("cl_merge_pkg", ["%s/%s"%(category,nameProgram)], True)
             clTempl = ChainProgressTemplate( self.startTask,
                                self.endTask,
                                self.setProgress,
@@ -350,7 +350,7 @@ class updateSystemConfigs(shareUpdateConfigs):
         clVars.Set("cl_root_path", configPath, True)
         if variables.Get('cl_core_pkg_root_set') == 'on':
             clVars.Set("cl_root_path_next", '/', True)
-        clVars.Set("cl_merge_pkg", [nameProgram], True)
+        clVars.Set("cl_merge_pkg", ["%s/%s"%(category,nameProgram)], True)
         clVars.Set("cl_action", 'merge', True)
         configFiles = []
         nameProg = clVars.Get("cl_name")
@@ -456,7 +456,7 @@ class updateSystemConfigs(shareUpdateConfigs):
             self.copyConfigFiles(configPath,variables.Get('cl_config_protect'))
         if clTempl and clTempl.getWarning():
             for warn in clTempl.getWarning().split("\n"):
-                self.SUCCESS(warn)
+                self.WARNING(warn)
         return True
 
 class PackageUpdater(updateSystemConfigs,updateUserConfigs):
