diff --git core/variables/core.py core/variables/core.py
index a0d7708..50665e8 100644
--- core/variables/core.py
+++ core/variables/core.py
@@ -150,13 +150,6 @@ class VariableClCoreKey(Variable):
     def get(self):
         return path.join(self.Get('cl_core_cert_path'),"server.key")
 
-class VariableClCoreSidLive(Variable):
-    """
-    Max time live sessions (in minutes)
-    """
-    type = "int"
-    value = "1440"
-
 class VariableClCoreCertLive(Variable):
     """
     Max time live sessions (in minutes)
diff --git core/variables/variable.py core/variables/variable.py
index de0717f..a350d47 100644
--- core/variables/variable.py
+++ core/variables/variable.py
@@ -126,7 +126,7 @@ class VariableClVariableFullname(VarHelper,ReadonlyVariable):
             moduleSection = self.mapSection.get(moduleName,moduleName)
             if not moduleSection in self.parent.importedModules:
                 self.parent.importVariables('calculate.%s.variables'%moduleName)
-                self.parent.flIniFile(onlySection=moduleSection)
+                self.parent.flIniFile()
             for varname,vardata in sorted(self.parent.allVars.items()):
                 if vardata[0] == moduleSection and \
                     not varname.startswith('cl_variable'):
@@ -164,6 +164,7 @@ class VariableClVariableValue(VarHelper,Variable):
         def __init__(self,dv,var):
             self.var = var
             self.dv = dv
+            self.dv.flIniFile()
 
         def __call__(self):
             isListOrTuple = lambda x: type(x) in (list,tuple)
