diff --git core/server/func.py core/server/func.py
index f4c1321..c095483 100644
--- core/server/func.py
+++ core/server/func.py
@@ -26,6 +26,7 @@ setLocalTranslate('cl_core3',sys.modules[__name__])
 __ = getLazyLocalTranslate(_)
 
 from calculate.lib.utils.files import process,readFile,processProgress
+from calculate.lib.cl_template import TemplatesError
 from calculate.lib.utils.content import getCfgFiles
 from itertools import *
 
@@ -148,7 +149,7 @@ class CommonMethods:
         try:
             dirsFiles = clTempl.applyTemplates()
             if clTempl.getError():
-                raise InstallError(clTempl.getError())
+                raise TemplatesError(clTempl.getError())
         finally:
             if clTempl:
                 if clTempl.cltObj:
diff --git core/utils/cl_core_custom.py core/utils/cl_core_custom.py
index f9eb76d..4133f57 100644
--- core/utils/cl_core_custom.py
+++ core/utils/cl_core_custom.py
@@ -21,7 +21,7 @@ from calculate.lib.datavars import VariableError
 from calculate.lib.cl_template import TemplatesError
 from calculate.lib.utils.files import FilesError
 
-setLocalTranslate('cl_install3',sys.modules[__name__])
+setLocalTranslate('cl_core3',sys.modules[__name__])
 __ = getLazyLocalTranslate(_)
 
 class ClCoreCustomAction(Action):
