diff -ruN genkernel.orig/arch/x86/modules_load genkernel-3.4.10.903/arch/x86/modules_load
--- genkernel.orig/arch/x86/modules_load	2009-01-06 01:22:10.000000000 +0300
+++ genkernel-3.4.10.903/arch/x86/modules_load	2009-06-11 13:07:17.000000000 +0400
@@ -22,5 +22,5 @@
 MODULES_USB="ehci-hcd uhci usb-ohci hid usb-storage uhci-hcd ohci-hcd usbhid sl811-hcd"
 
 # Filesystems
-MODULES_FS="ext2 ext3 reiserfs jfs nfs xfs fuse"
+MODULES_FS="ext2 ext3 ext4 reiserfs jfs nfs xfs fuse"
 
diff -ruN genkernel.orig/arch/x86_64/modules_load genkernel-3.4.10.903/arch/x86_64/modules_load
--- genkernel.orig/arch/x86_64/modules_load	2009-01-06 01:22:10.000000000 +0300
+++ genkernel-3.4.10.903/arch/x86_64/modules_load	2009-06-11 13:07:17.000000000 +0400
@@ -22,4 +22,4 @@
 MODULES_USB="ehci-hcd uhci usb-ohci hid usb-storage uhci-hcd ohci-hcd usbhid sl811-hcd"
 
 # Filesystems
-MODULES_FS="ext2 ext3 reiserfs jfs nfs xfs fuse"
+MODULES_FS="ext2 ext3 ext4 reiserfs jfs nfs xfs fuse"
diff -ruN genkernel.orig/defaults/initrd.scripts genkernel-3.4.10.903/defaults/initrd.scripts
--- genkernel.orig/defaults/initrd.scripts	2009-01-06 01:22:10.000000000 +0300
+++ genkernel-3.4.10.903/defaults/initrd.scripts	2009-06-11 13:14:47.000000000 +0400
@@ -169,11 +169,24 @@
 			# TODO: Check the size of the image versus the size of our tmpfs
 			# along with the amount of available RAM and increase tmpfs size
 			# if necessary. (Not having awk sucks...)
-			# z=0
-			# for i in $(cat /proc/meminfo | grep -e ^MemFree -e ^Cached | \
-			# cut -d: -f2 | cut -dk -f1 | sed -e "s/^\s*//") ; do
-			# z=$(($z + $i)) ; done
-			# echo $z
+			
+			# getting memory
+			z=0
+			for i in $(cat /proc/meminfo | grep -e ^MemFree -e ^Cached | \
+			cut -d: -f2 | cut -dk -f1 | sed -e "s/^\s*//") ; do
+			z=$(($z + $i)) ; done
+
+			# getting size tmpfs of default
+			let curtfs=$z/2
+           
+			# getting size of image
+			let q=`ls -la /newroot/mnt/cdrom/livecd.squashfs | sed -r -n 's/(\S*\s*){4}(\S*).*/\2/p'`/5000*6
+
+			if [[ $((${curtfs}<${q})) = "1" -a $((${q}<${z})) = "1" ]];
+			then
+				good_msg "Increasing size of tmpfs for ${NEW_ROOT}"
+				mount -o remount,size=${q}000 -t tmpfs tmpfs /newroot
+			fi
 			good_msg "Copying loop file for caching..."
 			cp -a ${NEW_ROOT}/mnt/cdrom/${LOOP} ${NEW_ROOT}/mnt/${LOOP}
 			if [ $? -ne 0 ]
@@ -184,6 +197,8 @@
 				rm -rf ${NEW_ROOT}/mnt/zisofs 2>/dev/null
 			else
 				LOOPEXT='../'
+				good_msg "Unmounting media from ${NEW_ROOT}/mnt/cdrom"
+				umount ${NEW_ROOT}/mnt/cdrom
 			fi
 		fi
 	fi
@@ -998,10 +1012,13 @@
 		mkdir /tmp
 		mkdir -p ${UNION}
 #		mkdir -p $CHANGES
-#		mount -t unionfs -o dirs=$CHANGES=rw unionfs ${UNION}
-		good_msg "Creating union mount"
-		unionfs -o allow_other,cow,noinitgroups,suid,dev ${rw_dir}=RW:${ro_dir}=RO ${UNION} 2>/dev/null
+##		mount -t unionfs -o dirs=$CHANGES=rw unionfs ${UNION}
+		mount -t unionfs -o dirs=${rw_dir}:${ro_dir}=ro unionfs ${UNION}
 		ret=$?
+		chmod 1777 ${UNION}/tmp
+		good_msg "Creating union mount"
+#		unionfs -o allow_other,cow,noinitgroups,suid,dev,default_permissions,use_ino ${rw_dir}=RW:${ro_dir}=RO ${UNION} 2>/dev/null
+#		ret=$?
 		if [ "${ret}" -ne 0 ]
 		then
 			bad_msg "Can't setup union mount!"
