diff -uNr kdebase-workspace-4.2.0.old/kdm/kfrontend/kgapp.cpp kdebase-workspace-4.2.0/kdm/kfrontend/kgapp.cpp
--- kdebase-workspace-4.2.0.old/kdm/kfrontend/kgapp.cpp	2008-09-18 00:54:59.000000000 +0400
+++ kdebase-workspace-4.2.0/kdm/kfrontend/kgapp.cpp	2009-02-12 15:50:05.000000000 +0300
@@ -377,6 +377,8 @@
 	setCursor( dpy, app.desktop()->winId(), XC_left_ptr );
 
 	int rslt = ex_exit;
+	app.setStyle( _GUIStyle );
+	app.setFont( *_normalFont );
 	for (;;) {
 		int cmd = gRecvInt();
 
