--- kdelibs-4.3.1.orig/kdecore/services/kfoldermimetype.cpp
+++ kdelibs-4.3.1/kdecore/services/kfoldermimetype.cpp
@@ -77,11 +77,11 @@ QString KFolderMimeTypePrivate::iconName( const KUrl& _url ) const
   // There are also other directories with 0 size, such as /proc, that may
   // be mounted, but those are unlikely to contain .directory (and checking
   // this would require KMountPoint from kio).
-  KDE_struct_stat buff;
-  if (KDE_stat( QFile::encodeName( _url.toLocalFile()), &buff ) == 0
-      && S_ISDIR( buff.st_mode ) && buff.st_size == 0 ) {
-    return KMimeTypePrivate::iconName( _url );
-  }
+  //KDE_struct_stat buff;
+  //if (KDE_stat( QFile::encodeName( _url.toLocalFile()), &buff ) == 0
+  //    && S_ISDIR( buff.st_mode ) && buff.st_size == 0 ) {
+  //  return KMimeTypePrivate::iconName( _url );
+  //}
 
   KUrl u( _url );
   u.addPath( ".directory" );
