diff --git a/kopete/plugins/history/historylogger.cpp b/kopete/plugins/history/historylogger.cpp
index bf1d3ec..dc95afa 100644
--- a/kopete/plugins/history/historylogger.cpp
+++ b/kopete/plugins/history/historylogger.cpp
@@ -298,7 +298,7 @@ void HistoryLogger::appendMessage( const Kopete::Message &msg , const Kopete::Co
 	if ( msg.format() != Qt::PlainText )
 		msgNode = doc.createTextNode( msg.escapedBody() );
 	else
-		msgNode = doc.createTextNode( Qt::escape(msg.plainBody()) );
+		msgNode = doc.createTextNode( msg.plainBody() );
 
 	docElem.appendChild( msgElem );
 	msgElem.appendChild( msgNode );
