diff --git install/cl_install.py install/cl_install.py
index a225a7a..1b7af1b 100644
--- install/cl_install.py
+++ install/cl_install.py
@@ -154,7 +154,8 @@ class Install(color_print):
         """Check that system has grub2 in current and installed system"""
         if self.clVars.Get('os_grub2_path'):
             return bool(
-                   filter(lambda x:x.startswith('grub-1.99'),
+                   filter(lambda x:x.startswith('grub-1.99') or \
+                                   x.startswith('grub-2'),
                    listDirectory('/var/db/pkg/sys-boot')))
         return False
 
diff --git install/variables/disk.py install/variables/disk.py
index 1682186..f2f661e 100644
--- install/variables/disk.py
+++ install/variables/disk.py
@@ -338,7 +338,7 @@ class VariableOsDiskMount(ReadonlyVariable,DeviceHelper):
 
     def get(self):
         disk_hash = self.Get('os_disk_dev')
-        fstab = FStab('/etc/fstab')
+        fstab = FStab('/etc/fstab',devs=disk_hash)
         rootdev = self.Get('os_root_dev')
         return map(lambda x: '/' if x == rootdev else fstab.getBy(eq=x) or "",
                self.Get('os_disk_dev'))
@@ -364,7 +364,7 @@ class VariableOsDiskFormat(ReadonlyVariable):
 
     def get(self):
         """Get current disk filesystem"""
-        fstab = FStab('/etc/fstab')
+        fstab = FStab('/etc/fstab',devs=self.Get('os_disk_dev'))
         def getFormat(device):
             prop = getUdevDeviceInfo(name=device)
             return prop.get('FSTAB_TYPE') or \
@@ -543,7 +543,7 @@ class VariableOsDiskOptions(ReadonlyVariable):
     type = "list"
 
     def get(self):
-        fstab = FStab('/etc/fstab')
+        fstab = FStab('/etc/fstab',devs=self.Get('os_disk_dev'))
         def getFormat(device):
             return fstab.getBy(what=fstab.OPTS,eq=device)
         return map(getFormat,
@@ -566,7 +566,7 @@ class VariableOsBindPath(ReadonlyVariable):
     type = "list"
 
     def get(self):
-        fstab = FStab('/etc/fstab')
+        fstab = FStab('/etc/fstab',devs=self.Get('os_disk_dev'))
         return fstab.getBy(what=fstab.NAME,where=fstab.OPTS,
                 _in="bind",allentry=True)
 
@@ -577,7 +577,7 @@ class VariableOsBindMountpoint(ReadonlyVariable):
     type = "list"
 
     def get(self):
-        fstab = FStab('/etc/fstab')
+        fstab = FStab('/etc/fstab',devs=self.Get('os_disk_dev'))
         return fstab.getBy(what=fstab.DIR,where=fstab.OPTS,
                 _in="bind",allentry=True)
 
diff --git install/variables/system.py install/variables/system.py
index 04d6455..088169b 100644
--- install/variables/system.py
+++ install/variables/system.py
@@ -490,12 +490,12 @@ class VariableOsGrub2Path(Variable):
     """
     def get(self):
         # find grub2-install
-        grubInstall = getProgPath('/sbin/grub2-install')
+        grubInstall = getProgPath('/usr/sbin/grub2-install')
         if grubInstall:
             return grubInstall
         # find grub-install and check, that this is grub2-install (ver 1.99)
-        grubInstall = self.getProgPath('/sbin/grub-install')
-        if grubInstall and filter(lambda x:"1.99" in x,
+        grubInstall = getProgPath('/usr/sbin/grub-install')
+        if grubInstall and filter(lambda x:"1.99" in x or "2.00" in x,
                            process(grubInstall,'-v')):
             return grubInstall
         return ""
