commit 9ee4d4c266ecee458afb0ef492d3307e67d12d9e
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Dec 28 16:21:39 2011 +0400

    Fix distribute copy.

diff --git a/pym/cl_distr.py b/pym/cl_distr.py
index 4e3ca90..0e99639 100644
--- a/pym/cl_distr.py
+++ b/pym/cl_distr.py
@@ -496,7 +496,7 @@ class Distributive(object, SignalInterrupt):
         if self.flagSpinner and not hideSpin:
             def checkRsync():
                 try:
-                    return len(filter(lambda x:"rsync\x00-a\x00-x" in x,
+                    return len(filter(lambda x:"rsync\x00-a\x00-H\x00-x" in x,
                                         getRunCommands()))>2
                 except:
                     return False
@@ -507,7 +507,7 @@ class Distributive(object, SignalInterrupt):
         if not rsyncCmd:
             raise DistributiveError(_("Failed to find '%s' command")%"rsync")
         try:
-            rsyncProcess = process(rsyncCmd, "-a", "-x",
+            rsyncProcess = process(rsyncCmd, "-a","-H", "-x",
                     "%s/"%fromdir,"%s/"%todir,stderr=STDOUT)
             self.addInterruptProcess(rsyncProcess)
             res = rsyncProcess.success()
