diff --git install/variables/net.py install/variables/net.py
index 6485769..438745d 100644
--- install/variables/net.py
+++ install/variables/net.py
@@ -25,7 +25,8 @@ from calculate.lib.cl_lang import setLocalTranslate
 setLocalTranslate('cl_install3',sys.modules[__name__])
 
 from calculate.lib.utils.ip import (getInterfaces,getIp,getMask,getMac,
-    cidrToMask,maskToCidr,getIpNet,isDhcpIp,checkIp,checkMask)
+    cidrToMask,maskToCidr,getIpNet,isDhcpIp,checkIp,checkMask,
+    getOperState)
 from calculate.lib.utils.device import lspci
 from calculate.lib.utils.files import listDirectory,readLinesFile,process
 from calculate.lib.utils import ip
@@ -259,7 +260,10 @@ class VariableOsInstallNetStatus(NetHelper,Variable):
             elif ipaddr:
                 return ipaddr
             else:
-                return "off"
+                if getOperState(iface) == 'down':
+                    return "off"
+                else:
+                    "dhcp"
         rootDevNfs = self.Get('os_root_dev') == '/dev/nfs'
         return statusValue(getIp(iface),"on" \
                            if rootDevNfs or isDhcpIp(iface) else "off")
@@ -332,7 +336,7 @@ class VariableOsInstallNetCidr(NetHelper,ReadonlyVariable):
         """
         Get CIDR of ip,net (Example: 24)
         """
-        return map(lambda x:maskToCidr(x),
+        return map(lambda x:maskToCidr(x) if x else '',
                self.Get('os_install_net_mask'))
 
 class VariableOsInstallNetMask(NetHelper,Variable):
