commit 26a4bb866ca188b477ab05d150216a2dde98f4a4
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Jul 30 11:09:52 2012 +0400

    Fix migrate users

diff --git a/install/cl_migrate_users.py b/install/cl_migrate_users.py
index fc7f7a2..f7332fb 100644
--- a/install/cl_migrate_users.py
+++ b/install/cl_migrate_users.py
@@ -472,6 +472,8 @@ class migrate(color_print):
         """Migrate users ang groups to new system"""
         if not self.checkPermFiles():
             return False
+        if not any(addUsersList):
+            addUsersList = []
         migrateUsers = ["root"]+\
                        map(lambda x: x[0], addUsersList + pwdUsersList)
         for existMigrUser in existsMigrateUsers:
