diff --git install/variables/net.py install/variables/net.py
index 438745d..c8801c4 100644
--- install/variables/net.py
+++ install/variables/net.py
@@ -263,7 +263,7 @@ class VariableOsInstallNetStatus(NetHelper,Variable):
                 if getOperState(iface) == 'down':
                     return "off"
                 else:
-                    "dhcp"
+                    return "dhcp"
         rootDevNfs = self.Get('os_root_dev') == '/dev/nfs'
         return statusValue(getIp(iface),"on" \
                            if rootDevNfs or isDhcpIp(iface) else "off")
