diff --git install/variables/disk.py install/variables/disk.py
index 92f275f..d2d2ba9 100644
--- install/variables/disk.py
+++ install/variables/disk.py
@@ -986,8 +986,8 @@ class VariableOsLocationFormat(LocationHelper,Variable):
                     return "vfat"
                 if dev in diskFormat and diskFormat[dev] in availFS:
                     if mount.count('/') == 1:
-                        if FileSystemManager.checkFSForType(diskFormat[dev],
-                                                     osInstallRootType):
+                        if FileSystemManager.checkFSForTypeMount(diskFormat[dev],
+                                                     osInstallRootType,mount):
                             return diskFormat[dev]
                     else:
                         return diskFormat[dev]
diff --git install/variables/kernel.py install/variables/kernel.py
index ae2a8f0..0de0cd5 100644
--- install/variables/kernel.py
+++ install/variables/kernel.py
@@ -93,7 +93,11 @@ class VariableOsInstallKernelResume(ReadonlyVariable):
         for dev, install in zip(self.Get('os_install_disk_use'),
                                 self.Get('os_install_disk_mount')):
             if install == "swap":
-                return "real_resume=%s"%dev
+                if self.Get('os_install_linux_system') == 'desktop':
+                    return \
+                        "tuxonice tuxonice_resume=%s real_resume=%s"%(dev,dev)
+                else:
+                    return "real_resume=%s"%dev
         return ""
 
 class KernelHelper:
