diff --git install/variables/disk.py install/variables/disk.py
index 8ccc646..e65eac1 100644
--- install/variables/disk.py
+++ install/variables/disk.py
@@ -110,7 +110,10 @@ class VariableOsDeviceDev(ReadonlyVariable,DeviceHelper):
     def get(self):
         """Get device /dev name"""
         if self.master is None:
-            self.master,slave = pty.openpty()
+            try:
+                self.master, slave = pty.openpty()
+            except OSError:
+                raise VariableError('Failed to create PTY')
             udevAdm = getProgPath('/sbin/udevadm')
             self.monitor = Popen([udevAdm,"monitor","--kernel",
                                 "--subsystem-match=block"],stdout=slave,
diff --git install/variables/distr.py install/variables/distr.py
index c564c66..2e0d32c 100644
--- install/variables/distr.py
+++ install/variables/distr.py
@@ -448,7 +448,9 @@ class VariableClImagePath(ReadonlyVariable):
                 livedistr = ['/mnt/livecd']
             # standard livecd
             else:
-                livedistr = ['/mnt/cdrom']
+                livedistr = ['/run/initramfs/squashfs',
+                             '/run/initramfs/live',
+                             '/mnt/cdrom']
         else:
             livedistr = []
         # search all partition for source installation distributive
diff --git install/wsdl_install.py install/wsdl_install.py
index c9bc196..bff5954 100644
--- install/wsdl_install.py
+++ install/wsdl_install.py
@@ -65,6 +65,7 @@ class Wsdl(WsdlBase):
                 normal=('os_install_locale_lang','os_install_clock_timezone')),
             lambda group:group(_("Distribution"),
                 normal=('cl_image_filename',),
+                hide=('cl_image_linux_shortname','cl_image_arch_machine'),
                 expert=('cl_image_linux_shortname', 'cl_image_arch_machine',
                         'cl_image_new_only')),
             lambda group:group(_("Allocate drive space"),
