diff --git install/install.py install/install.py
index 3ae3eb9..982a80d 100644
--- install/install.py
+++ install/install.py
@@ -350,7 +350,7 @@ class Install:
             newModuleName = defaultGL
         curModuleName = map(lambda x:x.strip().rpartition('=')[-1].strip('"\''),
                    filter(lambda x: x.startswith("OPENGL_PROFILE="),
-                          open(openGLenv,'r')))
+                          readFile(openGLenv)))
         curModuleName = curModuleName[-1] if curModuleName else ""
         if curModuleName == newModuleName:
             return True
diff --git install/utils/cl_setup.py install/utils/cl_setup.py
index fab08b9..968680d 100644
--- install/utils/cl_setup.py
+++ install/utils/cl_setup.py
@@ -18,6 +18,7 @@ import sys
 from calculate.core.server.func import Action,Tasks
 from calculate.lib.cl_lang import setLocalTranslate,getLazyLocalTranslate
 from calculate.lib.utils.files import FilesError
+from calculate.lib.utils.portage import isPkgInstalled
 from calculate.install.install import (MigrationError, TemplatesError,
                                   InstallError,
                                   AutopartitionError, DistributiveError)
@@ -43,10 +44,12 @@ class ClSetupVideoAction(Action):
         {'name':'check_video',
         'message':__("Checking the video driver"),
         'method':'Install.checkVideoDriver()',
+        'condition': lambda:isPkgInstalled('xorg-server')
         },
         {'name':'setup_opengl',
         'message':__("Configuring OpenGL"),
         'method':'Install.setupOpenGL()',
+        'condition': lambda:isPkgInstalled('xorg-server')
         },
         {'name':'reboot',
          'warning':__("To apply the changes, reboot the system"),
diff --git install/variables/system.py install/variables/system.py
index 1d32d8c..46641e9 100644
--- install/variables/system.py
+++ install/variables/system.py
@@ -504,11 +504,13 @@ class VariableOsInstallMdadmSet(ReadonlyVariable):
 class VariableClChrootGrub(ReadonlyVariable):
     """
     Chroot for grub-mkconfig
-    TODO: check for install scratch system.
     """
     def get(self):
         if self.Get('os_install_scratch') == "on":
-            return path.join(self.Get('cl_chroot_path'),"mnt/scratch")
+            if self.Get('cl_action') == 'system':
+                return self.Get('cl_target').mdirectory
+            else:
+                return path.join(self.Get('cl_chroot_path'),"mnt/scratch")
         else:
             return self.Get('cl_chroot_path')
 
