diff --git pym/install/variables/system.py pym/install/variables/system.py
index 297960e..60e868e 100644
--- pym/install/variables/system.py
+++ pym/install/variables/system.py
@@ -26,6 +26,7 @@ from calculate.lib.utils.files import (readFile,getProgPath,process,
     readLinesFile)
 from calculate.lib.utils.common import getPasswdUsers,getUserGroups,getGroups
 from calculate.lib.utils.common import getValueFromConfig,getValueFromCmdLine
+from calculate.lib.utils.common import getUserPrimaryGroup
 from calculate.lib.utils.portage import isPkgInstalled
 from calculate.lib.utils.device import getUdevDeviceInfo
 from crypt import crypt
@@ -217,13 +218,19 @@ class VariableClMigrateUserGroups(UserHelper,Variable):
                value)
         return value
 
+    def getPrimaryGroup(self, username):
+        pg = getUserPrimaryGroup(username)
+        if pg:
+            return [pg]
+        return []
+
     def get(self):
         """
         User groups
         """
         passwdList = getPasswdUsers()
-        return map(lambda x:getUserGroups(x) \
-                   if x in passwdList else self.getDefaultGroups(),
+        return map(lambda x:(self.getPrimaryGroup(x) + (getUserGroups(x) 
+                   if x in passwdList else self.getDefaultGroups())),
                self.Get('cl_migrate_user'))
 
     def choice(self):
