commit 9ac03f444d5c360bb9e6bd59c41a0b1946deb6f5
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Aug 1 12:44:18 2011 +0400

    Fix --iso param. Add btrfs support.

diff --git a/pym/cl_distr.py b/pym/cl_distr.py
index 9045f76..3e3df63 100644
--- a/pym/cl_distr.py
+++ b/pym/cl_distr.py
@@ -722,6 +722,7 @@ class PartitionDistributive(Distributive):
                         'ext4':'/sbin/mkfs.ext4 %s %s',
                         'jfs':'/sbin/mkfs.jfs %s -f %s',
                         'reiserfs':'/sbin/mkfs.reiserfs %s -f %s',
+                        'btrfs':'/sbin/mkfs.btrfs %s %s',
                         'xfs':'/sbin/mkfs.xfs %s -f %s',
                         'vfat':'/usr/sbin/mkfs.vfat %s -F 32 %s',
                         'ntfs-3g':'/usr/sbin/mkfs.ntfs %s -FQ %s',
@@ -731,6 +732,7 @@ class PartitionDistributive(Distributive):
     labelForUtilities = { 'ext2':'-L %s',
                         'ext3':'-L %s',
                         'ext4':'-L %s',
+                        'btrfs':'-L %s',
                         'jfs':'-L %s',
                         'reiserfs':'-l %s',
                         'xfs':'-L %s',
@@ -743,6 +745,7 @@ class PartitionDistributive(Distributive):
                  'ext3' : '83',
                  'ext4' : '83',
                  'reiserfs' : '83',
+                 'btrfs' : '83',
                  'jfs' : '83',
                  'xfs' : '83',
                  'vfat' : '0b',
@@ -753,6 +756,7 @@ class PartitionDistributive(Distributive):
                     'ext3' : '0700',
                     'ext4' : '0700',
                     'reiserfs' : '0700',
+                    'btrfs' : '0700',
                     'jfs' : '0700',
                     'xfs' : '0700',
                     'vfat' : '0700',
diff --git a/pym/cl_install.py b/pym/cl_install.py
index 29865e1..8145976 100644
--- a/pym/cl_install.py
+++ b/pym/cl_install.py
@@ -301,6 +301,10 @@ class convertDictOpt:
                                       "incompatible":["uni_xlate"]}},
                               "incompatible":[],
                               "makefs":"/usr/sbin/mkfs.vfat"},
+                      "btrfs":{"options":[],
+                              "pair":{},
+                              "incompatible":[],
+                              "makefs":"/sbin/mkfs.btrfs"},
                       "ntfs":{"options":["uni_xlate","posix","nls",
                                          "utf8","iocharset",
                                          "uid","gid","umask"],
@@ -1613,12 +1617,22 @@ class cl_install(color_print, SignalInterrupt):
                             where="os_install_disk_mount", eq="/boot")
         rootDiskType = self.varSelect("os_disk_type",
                             where="os_install_disk_mount", eq="/")
+        bootDiskFormat = self.varSelect("os_install_disk_format",
+                            where="os_install_disk_mount", eq="/boot")
+        rootDiskFormat = self.varSelect("os_install_disk_format",
+                            where="os_install_disk_mount", eq="/")
         bootDiskType = bootDiskType or rootDiskType
+        bootDiskFormat = bootDiskFormat or rootDiskFormat
         if "lvm" in bootDiskType or "raid" in bootDiskType:
             self.printERROR(
               _("Legacy grub not support boot from raid or lvm without"
                 " separate /boot partition"))
             return False
+        if bootDiskFormat == "btrfs":
+            self.printERROR(
+              _("Legacy grub not support boot from btrfs without"
+                " separate /boot partition"))
+            return False
         return True
 
     def setUsers(self,listUsers):
@@ -2613,8 +2627,12 @@ the system") + " (yes/no)"
             "march" in imageData:
             self.clVars.Set('os_install_arch_machine',
                             imageData['march'],True)
+            self.clVars.Set('os_install_linux_build',
+                            imageData['build'],True)
+            self.clVars.Set('os_install_linux_ver',
+                            imageData['ver'],True)
             self.setLinuxName(imageData['name'].upper())
-            self.clVars.Set('cl_image',True)
+            self.clVars.Set('cl_image',isoimage,True)
             return True
         else:
             self.printERROR(_("Wrong image file"))
