diff --git data/calculate data/calculate
index 1ec60b3..dc36f4c 100644
--- data/calculate
+++ data/calculate
@@ -142,6 +142,10 @@ start() {
 		touch /* /bin/* /sbin/*
 		udevadm trigger --action="add" --subsystem-match=net
 	fi
+	if [[ $roottype != "livecd" ]]
+	then
+		/usr/bin/eix-checkout &>/dev/null &
+	fi
 
 	/sbin/rc-update --update
 	return 0
diff --git install/cl_distr.py install/cl_distr.py
index 9dcdcb3..2dd8c6c 100644
--- install/cl_distr.py
+++ install/cl_distr.py
@@ -132,7 +132,7 @@ class Distributive(object):
             elif "Squashfs filesystem" in ftype:
                 return SquashDistributive(filename)
             elif path.isdir(filename):
-                if path.exists(path.join(filename,"livecd")):
+                if path.isfile(path.join(filename,"livecd")):
                     return IsoDistributive(filename)
                 else:
                     return DirectoryDistributive(filename)
diff --git install/cl_install.py install/cl_install.py
index c42e0a7..e65b50b 100644
--- install/cl_install.py
+++ install/cl_install.py
@@ -601,6 +601,7 @@ class Install(color_print):
         self.cleanInitrd()
         self.endTask("")
 
+        self.remountNTFS()
         # join templates
         self.startTask(_("Updating the configuration"),progress=True)
         self.applyTemplates(targetDistr.getDirectory(),
@@ -661,6 +662,7 @@ class Install(color_print):
                            setupType).capitalize(),progress=True)
         else:
             self.startTask(_("System configuration"),progress=True)
+        refreshLVM()
         res = self.applyTemplatesStartup()
         self.endTask()
         if setupType and self.clVars.Get('cl_setup') == 'network':
@@ -784,3 +786,31 @@ class Install(color_print):
             self.prepareBoot(self.targetDistr)
         self.endTask()
         return True
+
+    def _getFreeDirectory(self,directory):
+        """
+        Get free directory name
+        """
+        newDirectoryName = directory
+        while path.exists(newDirectoryName):
+            newDirectoryName = "%s.%s"%(directory,self.rndString())
+        return newDirectoryName
+
+    def remountNTFS(self):
+        """
+        Remount NTFS partitions 
+        """
+        for disk in self.clVars.Select('os_disk_dev',
+                                where='os_disk_format',like='ntfs'):
+            mountDir = self._getFreeDirectory('/var/lib/calculate/mount.test')
+            try:
+                os.mkdir(mountDir)
+            except:
+                continue
+            if process('/bin/mount',disk,mountDir).success():
+                if process('/bin/umount',mountDir).failed():
+                    self.printWARNING(_("Unable umount %s")%mountDir)
+            try:
+                os.rmdir(mountDir)
+            except:
+                self.printWARNING(_("Unable remove directory %s")%mountDir)
diff --git install/variables/action.py install/variables/action.py
index d704e74..5416223 100644
--- install/variables/action.py
+++ install/variables/action.py
@@ -32,7 +32,8 @@ class VariableAcInstallMerge(ActionVariable):
         if cl_action in ("system","merge") and \
             self.Get('os_install_root_type') != 'flash' and \
             self.Get('os_install_pxe') == 'off' and \
-            self.Get('cl_live') == 'off':
+            self.Get('cl_live') == 'off' or \
+            cl_action in ("sync",) and self.Get('cl_merge_pkg'):
             return "on"
         return "off"
 
@@ -47,7 +48,8 @@ class VariableAcInstallLive(ActionVariable):
     def action(self,cl_action):
         if cl_action in ("system","merge") and \
             self.Get('os_install_root_type') != 'flash' and \
-            self.Get('os_install_pxe') == 'off':
+            self.Get('os_install_pxe') == 'off' or \
+            cl_action in ("sync",) and self.Get('cl_merge_pkg'):
             return "on"
         return "off"
 
@@ -97,6 +99,7 @@ class VariableAcInstallUnmerge(ActionVariable):
     Action variable which has value "up" on prerm ebuild phase
     """
     def action(self,cl_action):
-        if cl_action == "merge" and self.Get('cl_ebuild_phase') == 'prerm':
+        if cl_action == "merge" and \
+            self.Get('cl_ebuild_phase') in ('prerm','postrm'):
             return "on"
         return "off"
