diff --git install/variables/disk.py install/variables/disk.py
index ef0f9e2..03d6636 100644
--- install/variables/disk.py
+++ install/variables/disk.py
@@ -1834,7 +1834,7 @@ class VariableOsInstallFstabMountConf(ReadonlyVariable,DeviceHelper):
                                        'os_install_disk_mount',
                                        'os_install_disk_format',
                                        'os_install_disk_options',
-                                       'os_disk_dev'],
+                                       'os_install_disk_dev'],
                                where='os_install_disk_mount',
                                func=lambda x:x[0] != "" and x[0] != "swap"),
                           lambda x,y: cmp(self.separateDevice(x[1]),
