diff --git install/cl_wsdl_install.py install/cl_wsdl_install.py
index a007be4..d4c117d 100644
--- install/cl_wsdl_install.py
+++ install/cl_wsdl_install.py
@@ -192,7 +192,8 @@ class Wsdl:
 
     @rpc(Integer, InstallInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Installation'),title=__('Flash install'),
-        image='drive-removable-media-usb-pendrive,media-flash',
+        image='drive-removable-media-usb-pendrive,'
+              'drive-removable-media-usb,media-flash',
         gui=True, rights=['install'])
     def install_flash ( self, sid, info):
         """
@@ -220,7 +221,8 @@ class Wsdl:
 
     @rpc(Integer, InstallInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Installation'),title=__('PXE install'),
-        image='network-server',command='cl-install-pxe',
+        image='network-server,preferences-desktop-remote-desktop',
+        command='cl-install-pxe',
         gui=True, rights=['installpxe'])
     def install_pxe ( self, sid, info):
         """
@@ -251,7 +253,8 @@ class Wsdl:
 
     @rpc(Integer, InstallInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Configuration'),title=__('Network'),
-        image='network-idle,preferences-system-network',
+        image='gnome-network-properties,'
+              'network-idle,preferences-system-network',
         command="cl-setup-network",
         gui=True, rights=['setup'])
     def setup_network ( self, sid, info):
@@ -281,7 +284,7 @@ class Wsdl:
 
     @rpc(Integer, InstallInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Configuration'),title=__('Video'),
-        image='gnome-multimedia,video-display',
+        image='video-display,gnome-multimedia',
         command="cl-setup-video",
         gui=True, rights=['setup'])
     def setup_video ( self, sid, info):
@@ -369,7 +372,7 @@ class Wsdl:
 
     @rpc(Integer, InstallInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Configuration'),title=__('System'),
-        image='system,computer',command="cl-setup-system",
+        image='system-run,system,computer',command="cl-setup-system",
         gui=True, rights=['setup'])
     def setup_system ( self, sid, info):
         """
diff --git install/variables/distr.py install/variables/distr.py
index 71423b2..0e0693c 100644
--- install/variables/distr.py
+++ install/variables/distr.py
@@ -121,18 +121,21 @@ class DistroRepository(Linux):
             if not d:
                 return False
             # check filter conditions
-            if system and systemByName(d['name']) != system:
+            if system and systemByName(d['os_linux_shortname']) != system:
                 return False
-            if not "name" in d or not "ver" in d:
+            if not "os_linux_shortname" in d or not "os_linux_ver" in d:
                 return False
-            if shortname and d['name'].lower() != shortname.lower():
+            if shortname and \
+                d['os_linux_shortname'].lower() != shortname.lower():
                 return False
-            if march and d['march'] != march:
+            if march and d['os_arch_machine'] != march:
                 return False
-            if version and not verCmp(getTupleVersion(d['ver']), version):
+            if version and \
+                not verCmp(getTupleVersion(d['os_linux_ver']), version):
                 return False
-            if build and "build" in d and (not d['build'].isdigit() or 
-                                           not buildCmp(int(d['build']),build)):
+            if build and "os_linux_build" in d and \
+                (not d['os_linux_build'].isdigit() or 
+                 not buildCmp(int(d['os_linux_build']),build)):
                 return False
             return True
 
@@ -167,11 +170,11 @@ class DistroRepository(Linux):
 
     def sortdistrfunc(self,x,y):
         """Func of comparing two distributive"""
-        ver1, ver2 = x[1].get('ver',""), y[1].get('ver',"")
+        ver1, ver2 = x[1].get('os_linux_ver',""), y[1].get('os_linux_ver',"")
         if ver1 and ver2 and ver1 != "0" and ver2 != "0" and ver1 != ver2:
             return cmpVersion(ver1,ver2)
-        build1 = getTupleVersion(x[1].get('build',""))
-        build2 = getTupleVersion(y[1].get('build',""))
+        build1 = getTupleVersion(x[1].get('os_linux_build',""))
+        build2 = getTupleVersion(y[1].get('os_linux_build',""))
         if  build1 != build2:
             return cmp(build1,build2)
         else:
diff --git install/variables/system.py install/variables/system.py
index 130439a..5e0dd96 100644
--- install/variables/system.py
+++ install/variables/system.py
@@ -25,6 +25,7 @@ from calculate.lib.utils.files import (readFile,getProgPath,process,
     readLinesFile)
 from calculate.lib.utils.common import getPasswdUsers,getUserGroups,getGroups
 from calculate.lib.utils.portage import isPkgInstalled
+from calculate.lib.utils.device import getUdevDeviceInfo
 from crypt import crypt
 from calculate.lib.encrypt import encrypt
 import calculate.lib.cl_template as cl_template
@@ -371,6 +372,8 @@ class VariableOsInstallDevFrom(Variable):
         """
         If device in calculate3.env dev_from not exists set ''
         """
+        if value:
+            value = getUdevDeviceInfo(name=value).get('DEVNAME',value)
         if value in self.Get('os_disk_dev'):
             return value
         else:
