commit c55ad8a565de03a5db52ebccf47b12bcc4564bb3
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jul 1 16:02:00 2011 +0400

    Fix suffix in partition size.

diff --git a/pym/cl_fill_install.py b/pym/cl_fill_install.py
index fb9434e..1b2e27b 100644
--- a/pym/cl_fill_install.py
+++ b/pym/cl_fill_install.py
@@ -25,7 +25,7 @@ from os import readlink,access,R_OK
 from cl_utils import isMount,typeFile,getTupleVersion,pathJoin,isFstabMount,\
                      listDirectory, getAvailableX11Drivers, getUUIDDict, \
                      isPkgInstalled, process, checkUtils, readLinesFile, \
-                     lspci
+                     lspci, getPartitionSize
 from cl_distr import DistributiveRepository,PartitionDistributive
 from cl_fill import clLocale
 from operator import itemgetter
@@ -381,7 +381,7 @@ class fillVars(object, glob_attr):
                     disk_hash[partition]['part'] = parts[4]
                     if len(parts)>5 and "swap" in parts[5]:
                         disk_hash[partition]['format'] = 'swap'
-                disk_hash[partition]['size'] = parts[3]
+                disk_hash[partition]['size'] = getPartitionSize(partition)
         # fill format, name and uuid
         execStr = self.getProgPath('/sbin/blkid')
         res = self._runos(execStr)
