commit 153a2716e78e1f467c83d61797df950bc2ab59c5
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri May 6 17:32:25 2011 +0400

    Fix cl-install -v for emerge --info ERROR RET.

diff --git a/pym/cl_fill_install.py b/pym/cl_fill_install.py
index fbb071f..f602f23 100644
--- a/pym/cl_fill_install.py
+++ b/pym/cl_fill_install.py
@@ -906,7 +906,7 @@ class fillVars(object, glob_attr):
         curlanguage = self.Get('os_install_locale_language')
         return get_linguas(open(makeconf,'r')) or \
                " ".join(filter(lambda x:x=="en" or x==curlanguage,
-                      get_linguas(self._runos(infocommand)).split())) or \
+                      get_linguas(self._runos(infocommand) or []).split())) or \
                "bg en de es fr it pl pt_BR ru uk"
 
     def get_os_install_locale_consolefont(self):
