diff --git install/variables/system.py install/variables/system.py
index 04d6455..088169b 100644
--- install/variables/system.py
+++ install/variables/system.py
@@ -490,12 +490,12 @@ class VariableOsGrub2Path(Variable):
     """
     def get(self):
         # find grub2-install
-        grubInstall = getProgPath('/sbin/grub2-install')
+        grubInstall = getProgPath('/usr/sbin/grub2-install')
         if grubInstall:
             return grubInstall
         # find grub-install and check, that this is grub2-install (ver 1.99)
-        grubInstall = self.getProgPath('/sbin/grub-install')
-        if grubInstall and filter(lambda x:"1.99" in x,
+        grubInstall = getProgPath('/usr/sbin/grub-install')
+        if grubInstall and filter(lambda x:"1.99" in x or "2.00" in x,
                            process(grubInstall,'-v')):
             return grubInstall
         return ""
