diff --git install/variables/disk.py install/variables/disk.py
index ef0f9e2..14462e7 100644
--- install/variables/disk.py
+++ install/variables/disk.py
@@ -31,7 +31,7 @@ from calculate.lib.utils.device import (getUdevDeviceInfo,getDeviceType,
                         getUUIDDict,getCommonDeviceName)
 from calculate.install.variables.autopartition import Sizes
 from calculate.lib.utils.files import (listDirectory,pathJoin,readFile,FStab,
-                                       isMount)
+                                       isMount,getProgPath)
 from calculate.install.distr import PartitionDistributive
 from calculate.install.fs_manager import FileSystemManager
 
@@ -110,7 +110,8 @@ class VariableOsDeviceDev(ReadonlyVariable,DeviceHelper):
         """Get device /dev name"""
         if self.master is None:
             self.master,slave = pty.openpty()
-            self.monitor = Popen(["/sbin/udevadm","monitor","--kernel",
+            udevAdm = getProgPath('/sbin/udevadm')
+            self.monitor = Popen([udevAdm,"monitor","--kernel",
                                 "--subsystem-match=block"],stdout=slave,
                                 close_fds=True)
             os.close(slave)
@@ -1834,7 +1835,7 @@ class VariableOsInstallFstabMountConf(ReadonlyVariable,DeviceHelper):
                                        'os_install_disk_mount',
                                        'os_install_disk_format',
                                        'os_install_disk_options',
-                                       'os_disk_dev'],
+                                       'os_install_disk_dev'],
                                where='os_install_disk_mount',
                                func=lambda x:x[0] != "" and x[0] != "swap"),
                           lambda x,y: cmp(self.separateDevice(x[1]),
