diff --git data/calculate data/calculate
index dc36f4c..a1a2537 100644
--- data/calculate
+++ data/calculate
@@ -127,12 +127,11 @@ start() {
 	LANG=C /usr/sbin/cl-core --create-symlink &>/dev/null
 	install_video_driver
 	LANG=C create_nonroot_user
-	LANG=C /usr/sbin/cl-core --method setup_system --no-progress --live on
+	LANG=C /usr/sbin/cl-core --method setup_system --no-progress --replace-conf --live on
 	LANG=C configure_users
 
 	/usr/sbin/env-update
 	source /etc/profile
-	/sbin/update-modules
 
 	# reinit net.* services
 	local roottype=`variable_value main.os_root_type`
diff --git install/cl_install.py install/cl_install.py
index f0d8d64..f4a511b 100644
--- install/cl_install.py
+++ install/cl_install.py
@@ -192,13 +192,14 @@ class Install(color_print):
             True)
 
         self.clTempl = ProgressTemplate(self.setProgress, self.clVars,
-                                        cltObj=useClt,
-                                        cltFilter=False,
-                                        printSUCCESS=self.printSUCCESS,
-                                        printWARNING=self.printWARNING,
-                                        askConfirm=self.askConfirm,
-                                        dispatchConf=self.dispatchConf,
-                                        printERROR=self.printERROR)
+            cltObj=useClt,
+            cltFilter=True if self.clVars.Get('cl_merge_set') == "on" \
+                      else False,
+            printSUCCESS=self.printSUCCESS,
+            printWARNING=self.printWARNING,
+            askConfirm=self.askConfirm,
+            dispatchConf=self.dispatchConf,
+            printERROR=self.printERROR)
 
         dirsFiles = self.clTempl.applyTemplates()
         try:
diff --git install/cl_wsdl_install.py install/cl_wsdl_install.py
index e6b08e9..a8b23d9 100644
--- install/cl_wsdl_install.py
+++ install/cl_wsdl_install.py
@@ -243,7 +243,8 @@ class Wsdl:
         dv.importInstall()
         dv.flIniFile()
         dv.Set('cl_action','merge',True)
-        dv.Set('cl_belong_pkg',[None],True)
+        dv.Set('cl_merge_pkg',[None],True)
+        dv.Set('cl_merge_set',"on",True)
         dv.Set('cl_setup','network',True)
         dv.addGroup(None,
             normal=('os_install_net_conf','os_install_net_data',
@@ -276,7 +277,8 @@ class Wsdl:
         dv.importInstall()
         dv.flIniFile()
         dv.Set('cl_action','merge',True)
-        dv.Set('cl_belong_pkg',[None],True)
+        dv.Set('cl_merge_pkg',[None],True)
+        dv.Set('cl_merge_set',"on",True)
         dv.Set('cl_setup','video',True)
         dv.addGroup(None,
             normal=('os_install_x11_video_drv', 'os_install_x11_resolution',
@@ -306,7 +308,8 @@ class Wsdl:
         dv.importInstall()
         dv.flIniFile()
         dv.Set('cl_action','merge',True)
-        dv.Set('cl_belong_pkg',[None],True)
+        dv.Set('cl_merge_pkg',[None],True)
+        dv.Set('cl_merge_set',"on",True)
         dv.Set('cl_setup','locale',True)
         dv.addGroup(None,
             normal=('os_install_locale_lang',
@@ -336,7 +339,8 @@ class Wsdl:
         dv.importInstall()
         dv.flIniFile()
         dv.Set('cl_action','merge',True)
-        dv.Set('cl_belong_pkg',[None],True)
+        dv.Set('cl_merge_pkg',[None],True)
+        dv.Set('cl_merge_set',"on",True)
         dv.Set('cl_setup','boot',True)
         dv.addGroup(None,
             normal=('os_install_mbr',
diff --git install/variables/locale.py install/variables/locale.py
index 6c740e9..916985d 100644
--- install/variables/locale.py
+++ install/variables/locale.py
@@ -213,7 +213,8 @@ class VariableOsInstallClockTimezone(LocaleVariable):
                 "Atlantic/Azores", "Atlantic/Cape_Verde", "UTC",
                 "Africa/Casablanca", "Europe/Amsterdam", "Europe/Belgrade",
                 "Europe/Brussels", "Europe/Zagreb", "Africa/Tunis",
-                "Asia/Amman", "Europe/Istanbul", "Asia/Beirut", "Europe/Kiev",
+                "Asia/Amman", "Europe/Istanbul", "Asia/Beirut",
+                "Europe/Helsinki", "Europe/Kiev",
                 "Africa/Windhoek", "Asia/Jerusalem", "Africa/Cairo",
                 "Europe/Minsk", "Africa/Harare", "Asia/Baghdad", "Asia/Kuwait",
                 "Europe/Moscow", "Africa/Nairobi", "Asia/Tbilisi",
