commit 8596f83187eae9d3e446e3eaa841968f1de387c4
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Apr 20 16:33:41 2011 +0400

    Turn on by default composite for radeon.

diff --git a/pym/cl_fill_install.py b/pym/cl_fill_install.py
index ff84ee8..41dae0d 100644
--- a/pym/cl_fill_install.py
+++ b/pym/cl_fill_install.py
@@ -879,7 +879,7 @@ class fillVars(object, glob_attr):
     
     def get_os_install_x11_composite(self):
         """On/off composite"""
-        defaultCompositeOn = ("nvidia","intel","fglrx","nouveau")
+        defaultCompositeOn = ("nvidia","intel","fglrx","nouveau","radeon")
         composite = self.getValueFromCmdLine("calculate",5)
         videodrv = self.getValueFromCmdLine("calculate",4)
         if videodrv != "auto":
