diff --git a/pym/cl_fill_install.py b/pym/cl_fill_install.py
index 9e55112..3977c35 100644
--- a/pym/cl_fill_install.py
+++ b/pym/cl_fill_install.py
@@ -258,20 +258,29 @@ class fillVars(object, glob_attr):
                         key = blkid_hash[key]
                         disk_hash[dev][key] = value[1:-1]
         # discard comment and empty string
+        devuuid = '/dev/disk/by-uuid'
+        mapUuidDev = dict(
+            map(lambda x:("UUID=%s"%os.path.basename(x),
+                    os.path.normpath(os.path.join(devuuid,os.readlink(x)))),
+            filter(os.path.islink,
+            map(lambda x:os.path.join(devuuid,x),
+            os.listdir(devuuid)))))
+        mapDevUuid = dict(map(lambda x:(x[1],x[0]),mapUuidDev.items()))
         mountOptionsList = \
             filter(lambda x: x.strip() and not x.strip().startswith('#'),
                    open("/etc/fstab").readlines())
         # split all string by columns and discard string which has less that 4
         # columns
         mountOptionsList = \
+            map(lambda x:[mapUuidDev.get(x[0],x[0])]+x[1:4],
             filter(lambda x: len(x)>=4,
-                   map(lambda x: filter(lambda x: x ,
-                                        x.replace('\t',' ').split(' ')),
-                       mountOptionsList))
+            map(lambda x: filter(lambda x: x ,
+                          x.replace('\t',' ').split(' ')),
+            mountOptionsList)))
         # get dev which has ntfs filesystem
         mountFormatList = \
-            filter(lambda x: x[0] in disk_hash.keys() and \
-                             x[1] in ("ntfs","ntfs-3g"),
+            filter(lambda x: x[0] in disk_hash.keys() \
+                             and x[1] in ("ntfs","ntfs-3g"),
                    map(lambda y: (y[0],y[2]), mountOptionsList))
         # split options and discard rw opt
         mountOptionsList = \
