diff --git a/install/cl_wsdl_install.py b/install/cl_wsdl_install.py
index 47b64be..956194b 100644
--- a/install/cl_wsdl_install.py
+++ b/install/cl_wsdl_install.py
@@ -285,7 +285,7 @@ class Wsdl:
 
     @rpc(Integer, InstallInfo, _returns = Array(ReturnedMessage))
     @core_method(category=__('Installation'),title=__('PXE install'),
-        image='network-server',
+        image='network-server',command='cl-install-pxe',
         gui=True, rights=['installpxe'])
     def install_pxe ( self, sid, info):
         """
