diff --git data/calculate data/calculate
index 9b3c784..ea32c87 100644
--- data/calculate
+++ data/calculate
@@ -124,7 +124,11 @@ start() {
 
 	# reinit net.* services
 	local roottype=`variable_value os_root_type`
-	[[ $roottype == "livecd" ]] && udevadm trigger --action="add" --subsystem-match=net
+	if [[ $roottype == "livecd" ]]
+	then
+		touch /* /bin/* /sbin/*
+		udevadm trigger --action="add" --subsystem-match=net
+	fi
 
 	/sbin/rc-update --update
 	return 0
