commit 48365759344f699d27a5a12047c7ca0ef48e8600
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Jun 27 17:57:39 2012 +0400

    Fix print expected error in cl-configure

diff --git a/install/cl_install.py b/install/cl_install.py
index 04dfbe5..e01d344 100644
--- a/install/cl_install.py
+++ b/install/cl_install.py
@@ -830,6 +830,12 @@ class Install(color_print):
                         pass
             if not setupType:
                 self.installDesktopClient()
+        except (MigrationError,
+                TemplatesError,
+                InstallError,
+                AutopartitionError,
+                DistributiveError) as e:
+            error = str(e)
         except (BaseException),e:
             error = ""
             for i in apply(traceback.format_exception, sys.exc_info()):
@@ -841,6 +847,12 @@ class Install(color_print):
                 self.clVars.close()
                 if target:
                     target.close()
+            except (MigrationError,
+                    TemplatesError,
+                    InstallError,
+                    AutopartitionError,
+                    DistributiveError) as e:
+                error = str(e)
             except (BaseException),e:
                 error = ""
                 for i in apply(traceback.format_exception, sys.exc_info()):
