diff --git data/calculate data/calculate
index d720d40..32eda6a 100755
--- data/calculate
+++ data/calculate
@@ -6,60 +6,6 @@ depend() {
 	use swap hostname
 }
 
-xorg_video_driver()
-{
-	local xorgFile="/etc/X11/xorg.conf"
-	if [[ -f $xorgFile ]]
-	then
-		cat /etc/X11/xorg.conf | grep Driver | cut -d\" -f2
-	fi
-}
-
-cmdline_video_driver()
-{
-	local cmdline=`cat /proc/cmdline`
-	for x in $cmdline
-	do
-		case $x in
-			# parse cmdline param calculate
-			calculate\=*)
-				local value=`echo "$x" | cut -d\= -f2-`
-				# if calculate specified in new format
-				if [[ "$value" =~ : ]]
-				then
-					for i in ${value//,/ }
-					do
-						case ${i} in
-							video:*)
-								local value=`echo "$i" | cut -d: -f2-`
-								if [[ -n "$value" ]]
-								then
-									echo "$value"
-									return 0
-								fi
-								;;
-						esac
-					done
-				# if calculate specified in old format
-				else
-					local videodrv=`echo "$value" | cut -d, -f5`
-					if [[ -n $videodrv ]]
-					then
-						echo "$videodrv"
-						return 0
-					fi
-				fi
-				;;
-		esac
-	done
-	return 1
-}
-
-video_driver()
-{
-	cmdline_video_driver || xorg_video_driver
-}
-
 check_install_ability()
 {
 	local pkg=$1
@@ -92,7 +38,7 @@ variable_value()
 install_video_driver()
 {
 	ebegin "Checking video drivers"
-	case `video_driver` in
+	case `variable_value os_install_x11_video_drv` in
 		fglrx)
 			local drv="x11-drivers/ati-drivers"
 			;;
diff --git pym/cl_fill_install.py pym/cl_fill_install.py
index 1b2e27b..9fbf889 100644
--- pym/cl_fill_install.py
+++ pym/cl_fill_install.py
@@ -23,9 +23,9 @@ from cl_datavars import glob_attr
 from os import path
 from os import readlink,access,R_OK
 from cl_utils import isMount,typeFile,getTupleVersion,pathJoin,isFstabMount,\
-                     listDirectory, getAvailableX11Drivers, getUUIDDict, \
+                     listDirectory, getAvailableVideo, getUUIDDict, \
                      isPkgInstalled, process, checkUtils, readLinesFile, \
-                     lspci, getPartitionSize
+                     lspci, getPartitionSize, getInstalledVideo
 from cl_distr import DistributiveRepository,PartitionDistributive
 from cl_fill import clLocale
 from operator import itemgetter
@@ -739,7 +739,7 @@ class fillVars(object, glob_attr):
         """Kernel attributes"""
         # on usb-hdd install must be "delay=5"
         if self.Get('os_install_root_type') == 'usb-hdd':
-            return "delay=5"
+            return " scandelay=5"
         else:
             return ""
 
@@ -957,15 +957,14 @@ class fillVars(object, glob_attr):
     def get_os_install_x11_video_drv(self):
         """Video driver used by xorg"""
         if self.Get('os_install_root_type') == 'usb-hdd':
-            available_drivers = \
-                getAvailableX11Drivers(prefix=self.Get('cl_chroot_path'))
+            prefix = self.Get('cl_chroot_path')
             videoVal = self.getValueFromCmdLine("calculate","video")
             videoVal = {'i915':'intel'}.get(videoVal,videoVal)
             if not isPkgInstalled('xorg-server',
                                   prefix=self.Get('cl_chroot_path')) or \
-                videoVal in available_drivers:
+                videoVal in getAvailableVideo(prefix=prefix):
                 return videoVal
-            return self.getVideoByDefault(available_drivers)
+            return self.getVideoByDefault(getInstalledVideo(prefix=prefix))
         else:
             return self.Get('os_x11_video_drv')
     
diff --git pym/cl_install_cmd.py pym/cl_install_cmd.py
index d1424d3..61c717c 100644
--- pym/cl_install_cmd.py
+++ pym/cl_install_cmd.py
@@ -242,7 +242,7 @@ class install_cmd(share_cmd):
                                " " + _("route specifing error: '{route}'"
                                        "(example: '{example}'").format(
                                        route=route,
-                                       example="10.0.0.0/24:192.168.1.1:eth0"))
+                                       example="default:192.168.1.1"))
         if not (values.install or values.uninstall or values.live):
             if values.v is False and \
                values.d is None and \
