diff --git a/configure b/configure
index 487bb30..7c4aa65 100755
--- a/configure
+++ b/configure
@@ -15625,7 +15625,7 @@ fi # test "$_lt_caught_CXX_error" != yes
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS -pthread >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
