diff --git calculate-builder-3.4.1.1/pym/builder/variables/builder.py calculate-builder-3.4.1.1/pym/builder/variables/builder.py
index 4d91a3b..2a0f94e 100644
--- calculate-builder-3.4.1.1/pym/builder/variables/builder.py
+++ calculate-builder-3.4.1.1/pym/builder/variables/builder.py
@@ -257,7 +257,8 @@ class VariableClBuilderDiskDev(Variable):
                     dn = base_name
                 full_dn = path.join(default_path, dn)
                 if ((not path.exists(full_dn) or not isMount(full_dn)) and
-                    ds is None or ds.lock_device(full_dn)):
+                    ds is None or
+                    isinstance(ds, DriveSpool) and ds.lock_device(full_dn)):
                     return full_dn
             else:
                 return ""
