diff --git console/application/methods_func.py console/application/methods_func.py
index 89e6ce7..3f19efb 100644
--- console/application/methods_func.py
+++ console/application/methods_func.py
@@ -88,9 +88,6 @@ def parse():
         '--no-progress', action='store_true', default=False,
         dest = 'no_progress', help=_('do not display the progress bar'))
     parser.add_argument(
-        '-f', '--force', action='store_true', default=False,
-        dest = 'no_questions', help=_('silent during the process'))
-    parser.add_argument(
         '-P', action='store_true', default=False,
         dest = 'stdin_passwd',
         help=_("use passwords from standard input for users accounts"))
@@ -106,7 +103,6 @@ def get_view(client, method, sid, view_params):
 
 def call_method(client, args, unknown_args, wait_thread):
     method = args.method
-    no_questions = args.no_questions
     stdin_passwd = args.stdin_passwd
     view_params = get_view_params(client, method + '_view', step = None, \
                                   expert = True)
@@ -115,7 +111,9 @@ def call_method(client, args, unknown_args, wait_thread):
     method_parser = get_method_argparser(view, args)
     param_object = _create_obj(client, method)
     try:
+        unknown_args = method_parser.fixBoolVariables(unknown_args)
         args, unknown_args = method_parser.parse_known_args(unknown_args)
+        no_questions = args.no_questions
     except SystemExit:
         raise Exception(1)
     for i in unknown_args:
