diff --git pym/console/application/cert_verify.py pym/console/application/cert_verify.py
index 3300ef8..24c0eee 100644
--- pym/console/application/cert_verify.py
+++ pym/console/application/cert_verify.py
@@ -77,8 +77,8 @@ def verify(server_cert, crl_path, flag):
     for rvk in revoked_objects:
         if serverSerial == int(rvk.get_serial(), 16):
             print _("This certificate has been revoked!")
-            print _("Serial")+ ': %s\n' %rvk.get_serial() + _("Revoke date") + \
-                                                    ': %s' %rvk.get_rev_date()
+            print _("Serial")+ _(': %s\n') %rvk.get_serial() + _("Revoke date") + \
+                                                    _(': %s') %rvk.get_rev_date()
             
             raise VerifyError('CRL Exception')
     return 0
diff --git pym/console/application/client_class.py pym/console/application/client_class.py
index 3ab97f4..81dc024 100644
--- pym/console/application/client_class.py
+++ pym/console/application/client_class.py
@@ -15,6 +15,8 @@
 #  limitations under the License.
 
 import urllib2 as u2
+if hasattr(u2,"ssl"):
+    u2.ssl._create_default_https_context = u2.ssl._create_unverified_context
 import os, sys
 import socket, ssl
 import OpenSSL, hashlib, M2Crypto
