diff --git pym/cl_ldap.py pym/cl_ldap.py
index c7e0f23..732b4f8 100644
--- pym/cl_ldap.py
+++ pym/cl_ldap.py
@@ -99,7 +99,7 @@ class ldapUser(_error):
                                 dictData[name].append(line.rpartition(spl)[2].\
                                                       strip())
         except:
-            self.setError(_("Can not open %s")%fileName)
+            #self.setError(_("Can not open %s")%fileName)
             return False
         if set(dictData.keys()) == set(namesData):
             # Кеширование данных
@@ -238,4 +238,4 @@ class ldapUser(_error):
             # Устанавливаем у объекта соединение и объект LDAP функций
             self.ldapObj = ldapObj
             self.conLdap = ldapObj.conLdap
-        return True
\ No newline at end of file
+        return True
diff --git pym/cl_template.py pym/cl_template.py
index 66a2634..f282401 100644
--- pym/cl_template.py
+++ pym/cl_template.py
@@ -2506,7 +2506,7 @@ class templateFunction(_error, _shareTemplate, _shareTermsFunction):
             self.pathConfigIni = os.path.join(self.homeDir, ".calculate")
             self.modeConfigIni = 0640
         else:
-            self.pathConfigIni = pathJoin(self._baseDir,'/etc/calculate')
+            self.pathConfigIni = pathJoin(self._chrootDir,'/etc/calculate')
             self.modeConfigIni = 0644
         self.fileConfigIni = os.path.join(self.pathConfigIni,"ini.env")
         # Словарь времен модификации env файлов
