commit 428d968e1eb781715f1d6084fd9390995ee58ffb
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Dec 7 11:59:54 2011 +0400

    Fix apply clt templates for installed system.

diff --git a/pym/cl_template.py b/pym/cl_template.py
index 3624fae..4b4905c 100644
--- a/pym/cl_template.py
+++ b/pym/cl_template.py
@@ -5252,7 +5252,8 @@ class templateClt(scanDirectoryClt, template):
         if self.objVar.Get("cl_name") in applyPackages:
             self.flagApplyTemplates = True
         # Базовая директория переноса шаблонов "/mnt/calculate" или "/" и.т.д
-        self._baseDir = pathJoin(self.objVar.Get("cl_chroot_path"),
+        self._chrootDir = os.path.normpath(self.objVar.Get("cl_chroot_path"))
+        self._baseDir = pathJoin(self._chrootDir,
                                  self.objVar.Get("cl_root_path"))
         self._baseDir = os.path.normpath(self._baseDir)
 
@@ -5368,7 +5369,7 @@ class templateClt(scanDirectoryClt, template):
             self.numberAllTemplates(self.allTemplates)
         # Обрабатываем шаблоны
         for dirTemplate in dirsTemplates:
-            if self.scanningTemplates(dirTemplate, "/") is False:
+            if self.scanningTemplates(dirTemplate, self._chrootDir) is False:
                 return False
         return (self.createdDirs, self.filesApply)
 
