diff --git pym/cl_distr.py pym/cl_distr.py
index 4a7ebf2..0470779 100644
--- pym/cl_distr.py
+++ pym/cl_distr.py
@@ -507,7 +507,7 @@ class Distributive(object, SignalInterrupt):
         if not rsyncCmd:
             raise DistributiveError(_("Failed to find '%s' command")%"rsync")
         try:
-            rsyncProcess = process(rsyncCmd, "-a","-H", "-x",
+            rsyncProcess = process(rsyncCmd, "-a","-H", "-x","-XX",
                     "%s/"%fromdir,"%s/"%todir,stderr=STDOUT)
             self.addInterruptProcess(rsyncProcess)
             res = rsyncProcess.success()
