commit 80ca6c052625f937b2ef50441ea1452f939fa00c
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Thu May 3 11:03:22 2012 +0400

    Fix work with python 2.7.3 pyxml.

diff --git a/pym/cl_template.py b/pym/cl_template.py
index f8127ec..8c6f7b9 100644
--- a/pym/cl_template.py
+++ b/pym/cl_template.py
@@ -19,6 +19,9 @@ import os
 import stat
 import re
 import xml.dom.minidom
+import xml
+if hasattr(xml,"use_pyxml"):
+    xml.use_pyxml()
 from xml import xpath
 import subprocess
 import types
