commit 57ee00e91ae33e51d9b19eb7c9108be08852a512
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Jul 25 09:04:05 2011 +0400

    Fix detect x11_video_drv by xorg.log.

diff --git a/pym/cl_fill.py b/pym/cl_fill.py
index 2e2bb65..9068caf 100644
--- a/pym/cl_fill.py
+++ b/pym/cl_fill.py
@@ -394,7 +394,7 @@ class fillVars(varsShare):
         if display and list_available_drivers:
             reDriver = re.compile('|'.join(map(lambda x: "%s_drv.so"%x,
                                                list_available_drivers)))
-            display_number = re.search(r':(\d+)\..*', display)
+            display_number = re.search(r':(\d+)(\..*)?', display)
             reDriverName = re.compile(r'([^/]+)_drv.so')
             if display_number:
                 xorg_log_file = '/var/log/Xorg.%s.log' % \
@@ -403,7 +403,7 @@ class fillVars(varsShare):
                     matchStrs = \
                         map(lambda x:x.group(1),
                         filter(lambda x:x,
-                        map(lambda x:reDriverName.search,
+                        map(reDriverName.search,
                         filter(lambda x:"drv" in x and reDriver.search(x),
                         readLinesFile(xorg_log_file)))))
                     if matchStrs:
