diff --git calculate/lib/cl_progressbar.py calculate/lib/cl_progressbar.py
index 5b88a84..152d87b 100644
--- calculate/lib/cl_progressbar.py
+++ calculate/lib/cl_progressbar.py
@@ -132,3 +132,4 @@ try:
 except:
     ControlledProgressDialog = None
     ClProgressDialog = None
+    ClMessageBox = None
diff --git calculate/lib/cl_template.py calculate/lib/cl_template.py
index 721a1f8..db8ba8b 100644
--- calculate/lib/cl_template.py
+++ calculate/lib/cl_template.py
@@ -4205,7 +4205,11 @@ re.M|re.S)
                 return (applyDir, True, crDirs)
         try:
             FD = open(templateDirFile)
-            textTemplate = FD.read()
+            textTemplate = FD.readline().rstrip()
+            buf = textTemplate
+            while buf and textTemplate.endswith('\\'):
+                buf = FD.readline()
+                textTemplate = "%s %s"%(textTemplate[:-1],buf.rstrip())
             FD.close()
         except:
             self.setError(_("Failed to open the template") + ": " +\
@@ -4218,6 +4222,11 @@ re.M|re.S)
                           headerLine.split())
             if moduleParam:
                 self.objVar.defaultModule = moduleParam[0].partition('=')[2]
+                try:
+                    varModule = importlib.import_module(
+                        "calculate.%s.variables"%self.objVar.defaultModule)
+                except (ImportError, AttributeError),e:
+                    return ("", False, [])
         # Заменяем переменные на их значения
         textTemplate = self.applyVarsTemplate(textTemplate, templateDirFile)
 
@@ -4899,6 +4908,11 @@ re.M|re.S)
                           headerLine.split())
             if moduleParam:
                 self.objVar.defaultModule = moduleParam[0].partition('=')[2]
+                try:
+                    varModule = importlib.import_module(
+                        "calculate.%s.variables"%self.objVar.defaultModule)
+                except (ImportError, AttributeError),e:
+                    return []
 
         if templateFileType != "bin":
             # Вычисляем условные блоки
diff --git calculate/lib/variables/X11.py calculate/lib/variables/X11.py
index ed5c49e..389f9ad 100644
--- calculate/lib/variables/X11.py
+++ calculate/lib/variables/X11.py
@@ -21,7 +21,7 @@ from calculate.lib.datavars import Variable,VariableError,ReadonlyVariable
 from calculate.lib.utils.files import process, readLinesFile
 from calculate.lib.utils.common import (getAvailableVideo, getValueFromCmdLine,
                 getVideoFromXorgLog, getVideoFromXorgConf, getVideoFromCmdLine,
-                getCompositeFromXorgconf,
+                getCompositeFromXorgconf,getTupleVersion,
                 getVideoFromModules, getVideoFromVendor,getInstalledVideo)
 from calculate.lib.utils.portage import isPkgInstalled
 from math import sqrt
@@ -126,23 +126,25 @@ class VariableOsX11Standart(ReadonlyVariable):
                          'media-gfx/cldg-themes']:
             installed = isPkgInstalled(themePkg,
                                        prefix=self.Get('cl_chroot_path'))
-            if installed and installed[0]['PV'].startswith('12'):
-                res = [(1024,576), (1024,600),
-                       (1024,768), (1200,800), (1280,1024),
-                       (1280,720), (1280,768), (1280,800),
-                       (1360,768), (1366,768), (1368,768),
-                       (1400,1050), (1440,900), (1600,1200),
-                       (1600,768), (1600,900), (1680,1050),
-                       (1680,945), (1920,1080), (1920,1200),
-                       (2048,1152), (2560,1440), (2560,1600),
-                       (640,480), (800,480) ]
+            if installed and \
+                getTupleVersion(installed[0]['PV'].startswith('12')) < \
+                getTupleVersion("12"):
+                res = [(1024,600), (1024,768),
+                       (1280,1024), (1280,800),
+                       (1366,768), (1440,900),
+                       (1600,1200), (1680,1050),
+                       (1920,1200)] 
                 break
         else:
-            res = [(1024,600), (1024,768),
-                   (1280,1024), (1280,800),
-                   (1366,768), (1440,900),
-                   (1600,1200), (1680,1050),
-                   (1920,1200)] 
+            res = [(1024,576), (1024,600),
+                   (1024,768), (1200,800), (1280,1024),
+                   (1280,720), (1280,768), (1280,800),
+                   (1360,768), (1366,768), (1368,768),
+                   (1400,1050), (1440,900), (1600,1200),
+                   (1600,768), (1600,900), (1680,1050),
+                   (1680,945), (1920,1080), (1920,1200),
+                   (2048,1152), (2560,1440), (2560,1600),
+                   (640,480), (800,480) ]
         bestRes = min(res,
                   key=lambda x:(abs(x[0]/float(x[1])-k),
                                 abs(gep-sqrt(x[0]**2+x[1]**2))))
