commit eb389f55b59fc57342c1dd504a2184c766b27c1f
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jun 19 08:55:37 2015 +0300

    Добавлены дистрибутивы CLDM, CLSM

diff --git a/pym/cl_utils.py b/pym/cl_utils.py
index 2e1a607..b3182ea 100644
--- a/pym/cl_utils.py
+++ b/pym/cl_utils.py
@@ -1303,8 +1303,13 @@ def detectBuild(pathname,dictInfo):
                 dictInfo['name'] = "CLD"
             elif "GNOME" in issue:
                 dictInfo['name'] = "CLDG"
+            elif "MATE" in issue:
+                dictInfo['name'] = "CLDM"
         elif "Scratch" in issue:
-            dictInfo['name'] = "CLS"
+            if "MATE" in issue:
+                dictInfo['name'] = "CLSM"
+            else:
+                dictInfo['name'] = "CLS"
     else:
         dictInfo['type'] = ''
     return dictInfo
diff --git a/pym/cl_vars_share.py b/pym/cl_vars_share.py
index ec4981a..ead4d2c 100644
--- a/pym/cl_vars_share.py
+++ b/pym/cl_vars_share.py
@@ -252,19 +252,24 @@ class varsShare:
     dictLinuxName = {"CLD":"Calculate Linux Desktop",
                      "CLDX":"Calculate Linux Desktop",
                      "CLDG":"Calculate Linux Desktop",
+                     "CLDM":"Calculate Linux Desktop",
                      "CDS":"Calculate Directory Server",
                      "CLS":"Calculate Linux Scratch",
+                     "CLSM":"Calculate Linux Scratch",
                      "CSS":"Calculate Scratch Server",
                      "CMC":"Calculate Media Center",
                      "Gentoo":"Gentoo"}
 
-    dictLinuxSubName = {"CLD":"KDE", "CLDX":"XFCE", "CLDG":"GNOME"}
+    dictLinuxSubName = {"CLD":"KDE", "CLDX":"XFCE", "CLDG":"GNOME",
+                        "CLSM":"MATE", "CLDM":"MATE"}
 
     dictNameSystem = {'CDS':'server',
                       'CLD':'desktop',
                       'CLDG':'desktop',
+                      'CLDM':'desktop',
                       'CLDX':'desktop',
                       'CLS':'desktop',
+                      'CLSM':'desktop',
                       'CMC':'desktop',
                       'CSS':'server'}
 
