diff --git a/pym/cl_vars_share.py b/pym/cl_vars_share.py
index 7bd048d..52a114e 100644
--- a/pym/cl_vars_share.py
+++ b/pym/cl_vars_share.py
@@ -462,14 +462,17 @@ class varsShare:
 
     def getShortnameByMakeprofile(self,systemroot):
         """Get shortname by symlink of make.profile"""
-        makeprofile = path.join(systemroot,'etc/make.profile')
-        if path.exists(makeprofile):
-            link = os.readlink(makeprofile)
-            reMakeProfileLink = re.compile('/calculate/(desktop|server)/(%s)/'%
-                        "|".join(self.dictLinuxName.keys()),re.S)
-            shortnameSearch = reMakeProfileLink.search(link)
-            if shortnameSearch:
-                return shortnameSearch.groups()[1]
+        for makeprofile in (path.join(systemroot,'etc/portage/make.profile'),
+                            path.join(systemroot,'etc/make.profile')):
+            if path.exists(makeprofile):
+                link = os.readlink(makeprofile)
+                for pattern in ('/calculate/(?:desktop|server)/(%s)/',
+                                '/distros/profiles/(%s)/'):
+                    reMakeProfileLink = re.compile(pattern%
+                                "|".join(self.dictLinuxName.keys()),re.S)
+                    shortnameSearch = reMakeProfileLink.search(link)
+                    if shortnameSearch:
+                        return shortnameSearch.groups()[0]
         return None
 
     def getShortnameByIni(self,systemroot):
