diff --git calculate/lib/cl_print.py calculate/lib/cl_print.py
index 7c7d979..ff13ed4 100644
--- calculate/lib/cl_print.py
+++ calculate/lib/cl_print.py
@@ -115,7 +115,10 @@ class color_print(object):
 
 
     def defaultPrint(self, string):
-        self._printSysOut.write(string)
+        try:
+            self._printSysOut.write(string)
+        except UnicodeError:
+            self._printSysOut.write(string.encode('utf-8'))
         try:
             self._printSysOut.flush()
         except IOError:
diff --git calculate/lib/cl_template.py calculate/lib/cl_template.py
index c7d72e5..831705d 100644
--- calculate/lib/cl_template.py
+++ calculate/lib/cl_template.py
@@ -358,19 +358,16 @@ class _terms(_error, _shareTermsFunction):
                                     vals[1] = 0
                                 try:
                                     valFile = int(vals[1])
+                                    valVar = valVars
+                                    res = dictRuleFunc[sepF](valVar, valFile)
+                                    if res:
+                                        listEqual.append(True)
+                                    else:
+                                        listEqual.append(False)
+                                        break
                                 except:
-                                    self.setError("'%s'"%rpl(term) +" "+\
-                                                  _("incorrect"))
-                                    self.setError(textError)
-                                    return False
-                                valVar = valVars
-                                res = dictRuleFunc[sepF](valVar, valFile)
-                                if res:
-                                    listEqual.append(True)
-                                else:
-                                    listEqual.append(False)
-                                    break
-                            else:
+                                    flagIntTypeVar = False
+                            if not flagIntTypeVar:
                                 if sepF == "!=" or sepF == "==":
                                     if not vals[1].strip():
                                         vals[1] = ""
diff --git calculate/lib/datavars.py calculate/lib/datavars.py
index a2f2b0d..835b649 100644
--- calculate/lib/datavars.py
+++ calculate/lib/datavars.py
@@ -415,6 +415,9 @@ class TableVariable(Variable):
     """
     type = "table"
 
+    def set(self,value):
+        return value
+
     def get(self,hr=False):
         """Get table data"""
         for varname,value in ifilter(lambda x:type(x[1]) != list,
@@ -500,6 +503,7 @@ class TableVariable(Variable):
                         fieldname=self.source[0],variablename=self.name))
 
     def setValue(self,value):
+        value = self.set(value)
         self.untrusted = True
         oldvalue = self.Get(self.name)
         # get writable columns
