diff --git calculate/lib/cl_template.py calculate/lib/cl_template.py
index fe055a8..ab79671 100644
--- calculate/lib/cl_template.py
+++ calculate/lib/cl_template.py
@@ -4516,6 +4516,9 @@ gettext -d cl_template "$*"
             if self.objVar.Get('cl_verbose_set') == 'on' and \
                 self.filesApply:
                 self.verboseOutput(self.filesApply)
+        self.objVar.Set('cl_merge_pkg_pass',[],force=True)
+        self.objVar.Set('cl_merge_pkg_new',[],force=True)
+        self.objVar.Set('cl_merge_pkg',[],force=True)
         return (self.createdDirs, self.filesApply)
 
     def verboseOutput(self,filesApply):
diff --git calculate/lib/variables/__init__.py calculate/lib/variables/__init__.py
index 9d3cf79..a8e307b 100644
--- calculate/lib/variables/__init__.py
+++ calculate/lib/variables/__init__.py
@@ -41,4 +41,4 @@ class VariableClVer(ReadonlyVariable):
     """
     Package version
     """
-    value = "3.2.0.1"
+    value = "3.2.0.2"
