diff --git pym/cl_opt.py pym/cl_opt.py
index 1f1f840..4a019de 100644
--- pym/cl_opt.py
+++ pym/cl_opt.py
@@ -24,7 +24,7 @@ from cl_lang import lang
 tr = lang()
 tr.setLocalDomain('cl_lib')
 tr.setLanguage(sys.modules[__name__])
-optparse._ = _
+tr.setLanguage(optparse)
 
 def check_choice_regignore(option, opt, value):
     """Check choice with registry independence"""
diff --git pym/cl_utils.py pym/cl_utils.py
index 6e9d830..1fb3743 100644
--- pym/cl_utils.py
+++ pym/cl_utils.py
@@ -1018,6 +1018,20 @@ def isPkgInstalled(pkg,prefix='/'):
                                 listDirectory(x))),
                listDirectory(pkgDir,fullPath=True)))
 
+def getPkgInstalled(pkg,prefix='/'):
+    """Check is package installed"""
+    pkgDir = path.join(prefix,'var/db/pkg')
+    if "/" in pkg:
+        category,op,pkg = pkg.partition('/')
+        return filter(lambda x:x['PN'] == pkg,
+               map(reVerSplitToPV,
+               listDirectory(path.join(pkgDir,category))))
+    else:
+        return filter(lambda x: filter(lambda y:y['PN'] == pkg,
+                                map(reVerSplitToPV,
+                                listDirectory(x))),
+               listDirectory(pkgDir,fullPath=True))
+
 def getPkgActiveUses(fullpkg):
     """Get active uses from package"""
     res = getPkgUses(fullpkg)
