commit e6681bcd4838b73dd2cf2eeeb36ae9ef930487ae
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Aug 22 10:10:51 2012 +0400

    Fix infinite recursion

diff --git calculate/lib/datavars.py calculate/lib/datavars.py
index 835b649..4dce5a9 100644
--- calculate/lib/datavars.py
+++ calculate/lib/datavars.py
@@ -129,6 +129,7 @@ class Variable:
         self.parent = parent
         self.name = self.getVariableName()
         self.init()
+        self.processInvalidate = False
         for key,val in kwargs.items():
             setattr(self,key,val)
 
@@ -391,15 +392,22 @@ class Variable:
         force : boolean=False
           Drop value also for after manual set
         """
-        if force or not self.wasSet:
-            self.wasSet = False
-            self.invalid = True
-            for var in self.reqVars:
-                var.invalidate()
-            for var in self.reqCheck:
-                var.untrusted = True
-            self.reqVars = []
-            self.reqCheck = []
+        # fix possible infinite recursion on invalidating
+        if not self.processInvalidate and (force or not self.wasSet):
+            try:
+                # mark variable that in invalidation now
+                self.processInvalidate = True
+                self.wasSet = False
+                self.invalid = True
+                for var in self.reqVars:
+                    var.invalidate()
+                for var in self.reqCheck:
+                    var.untrusted = True
+                self.reqVars = []
+                self.reqCheck = []
+            finally:
+                # unmark process invalidate
+                self.processInvalidate = False
 
     @classmethod
     def isTrue(self,value):
