commit 19de404978dcd87f477af6fbbeb7340c2fb2141d
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jun 3 13:52:49 2011 +0400

    Add update-config for meta packages.

diff --git a/pym/update_config/cl_update_config.py b/pym/update_config/cl_update_config.py
index 7f9ba9f..1bd10f4 100644
--- a/pym/update_config/cl_update_config.py
+++ b/pym/update_config/cl_update_config.py
@@ -383,7 +383,8 @@ class updateSystemConfigs(shareUpdateConfigs):
         # флаг обновления и программы используемые для наложения шаблонов
         flagUpdate, mergePrograms = self.getFlagUpdAndInstPrograms()
         # Проверка есть ли в пакете защищенные конфигурационные файлы
-        if not self.isExistsProtectFiles(configPath):
+        if not self.isExistsProtectFiles(configPath) and \
+            not "-meta" in nameProgram:
             self.logger.info(_("Package %s") %nameProgram)
             self.logger.warn(_("Not found protected files"))
             return True
