commit 1ad2a516b2522cec2d3c757dfc6eed7c8a3eee1e
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Mar 16 11:27:32 2011 +0300

    Fix locale by default.

diff --git a/pym/cl_fill.py b/pym/cl_fill.py
index a72bc34..1d7b9b4 100644
--- a/pym/cl_fill.py
+++ b/pym/cl_fill.py
@@ -342,7 +342,7 @@ class fillVars(varsShare):
             localeVal = self.getValueFromConfig('/etc/env.d/02locale','LANG')
             if locale.isValueInFieldExists('locale',localeVal):
                 return localeVal
-        if os.environ.has_key("LANG"):
+        if os.environ.has_key("LANG") and os.environ["LANG"] != "C":
             return os.environ["LANG"]
         return locale.getFieldByLang("locale","default")
 
