commit d4b9968ecc9de15549c00386b2726cdee166ce06
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jun 22 14:28:59 2012 +0400

    Fix pkg function

diff --git a/pym/cl_template.py b/pym/cl_template.py
index dae7872..734ab92 100644
--- a/pym/cl_template.py
+++ b/pym/cl_template.py
@@ -2782,20 +2782,18 @@ class templateFunction(_error, _shareTemplate, _shareTermsFunction):
                     if os.path.exists(pathCategory):
                         pkgs = os.listdir(pathCategory)
                         pkgs = map(lambda x: os.path.join(category,x),
-                                    pkgs)
+                               pkgs)
                         installProg = self.sharePkg(pkgs)
                         replace = self.pkg(nameProg, installProg)
-                        oldInstallProg = self.installProg
-                        self.installProg = {}
-                        self.installProg.update(installProg)
-                        self.installProg.update(oldInstallProg)
+                        self.installProg.update(
+                            filter(lambda x:not x[0] in self.installProg,
+                            installProg.items()))
         else:
             if not self.flagAllPkgScan:
                 installProg = self.getInstallPkgGentoo()
-                oldInstallProg = self.installProg
-                self.installProg = {}
-                self.installProg.update(installProg)
-                self.installProg.update(oldInstallProg)
+                self.installProg.update(
+                    filter(lambda x:not x[0] in self.installProg,
+                    installProg.items()))
                 templateFunction.flagAllPkgScan = True
             replace = self.pkg(nameProg, self.installProg)
         textTemplateTmp = textTemplateTmp[:resS.start()] + replace +\
