diff --git pym/cl_distr.py pym/cl_distr.py
index 572e9dd..2d8e51e 100644
--- pym/cl_distr.py
+++ pym/cl_distr.py
@@ -1183,10 +1183,10 @@ class SquashDistributive(Distributive):
             raise DistributiveError(_("Failed to create squash") +
                 " : %s"%_("command '%s' not found")%mksquashfsUtil)
         cmd = [mksquashfsUtil, "%s/"%directory,file, "-no-progress"]
-        if self.exclude:
-            cmd += ["-e"] + self.exclude
         if self.compress:
             cmd += ["-comp",self.compress]
+        if self.exclude:
+            cmd += ["-e"] + self.exclude
         processMkSquash = process(*cmd)
         if processMkSquash.failed():
             raise DistributiveError(_("Failed to create squashfs") +
