commit d22ba198c61eccec92c1ec1309380fba19757c23
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri Jun 24 10:49:36 2011 +0400

    Remove from available X11 drivers drivers in /usr/portage/distfiles.

diff --git a/pym/cl_utils.py b/pym/cl_utils.py
index 24760c9..025b051 100644
--- a/pym/cl_utils.py
+++ b/pym/cl_utils.py
@@ -967,16 +967,10 @@ def getSquashList():
 
 def getAvailableX11Drivers(prefix="/"):
     """Get available x11 drivers"""
-    distfiles = path.join(prefix,'usr/portage/distfiles')
     xorg_modules_dir = path.join(prefix,'usr/lib/xorg/modules/drivers')
-    return list(set(
-           map(lambda x:'fglrx' if x.startswith('ati-driver') else "nvidia",
-           filter(lambda x:x.startswith('ati-driver-installer') or
-                           x.startswith('NVIDIA-Linux'),
-           listDirectory(distfiles))) + \
-           map(lambda x: x[:-7],
+    return map(lambda x: x[:-7],
            filter(lambda x:x.endswith('_drv.so'),
-           listDirectory(xorg_modules_dir)))))
+           listDirectory(xorg_modules_dir)))
 
 def lspci(filtername=None,shortInfo=False):
     """Get hash of lspci, filtred by filtername. If shortInfo, then
