diff --git a/pym/cl_vars_share.py b/pym/cl_vars_share.py
index 2f19eae..7c25103 100644
--- a/pym/cl_vars_share.py
+++ b/pym/cl_vars_share.py
@@ -232,6 +232,7 @@ class varsShare:
                      "CDS":"Calculate Directory Server",
                      "CLS":"Calculate Linux Scratch",
                      "CSS":"Calculate Scratch Server",
+                     "CMC":"Calculate Media Center",
                      "Gentoo":"Gentoo"}
 
     dictLinuxSubName = {"CLD":"KDE", "CLDX":"XFCE", "CLDG":"GNOME"}
@@ -241,6 +242,7 @@ class varsShare:
                       'CLDG':'desktop',
                       'CLDX':'desktop',
                       'CLS':'desktop',
+                      'CMC':'desktop',
                       'CSS':'server'}
 
     # data object from LDAP
