diff --git calculate/lib/cl_template.py calculate/lib/cl_template.py
index 8ee09af..4ff7745 100644
--- calculate/lib/cl_template.py
+++ calculate/lib/cl_template.py
@@ -4279,6 +4279,13 @@ gettext -d cl_template "$*"
                             if not filename in self.cltObj.filterApplyTemplates:
                                self.cltObj.filterApplyTemplates[filename] = []
                             self.cltObj.filterApplyTemplates[filename].append(pkg)
+                        for filename,pkgs in self.changedFiles.data.items():
+                            if not filename in self.cltObj.filterApplyTemplates:
+                               self.cltObj.filterApplyTemplates[filename] = []
+                            pkgs = filter(lambda x:not x in \
+                                          self.cltObj.filterApplyTemplates[filename],
+                                   map(lambda x:x[0],pkgs))
+                            self.cltObj.filterApplyTemplates[filename].extend(pkgs)
             if not self.cltObj.applyTemplates():
                 return False
         if (self.objVar.Get('cl_merge_pkg') or \
