commit d3ee1a0c56c0dd607fa8ba1572c8303227a47894
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Fri May 25 09:57:41 2012 +0400

    Fix ati video card detect.

diff --git a/calculate/lib/variables/hardware.py b/calculate/lib/variables/hardware.py
index 2ca9615..1aafdba 100644
--- a/calculate/lib/variables/hardware.py
+++ b/calculate/lib/variables/hardware.py
@@ -134,16 +134,16 @@ class VariableHrVideo(ReadonlyVariable):
 
     def get(self):
         """Videocard vendor"""
-        line = self.Get('hr_video_name')
-        if "nVidia" in line or "GeForce" in line:
+        line = self.Get('hr_video_name').lower()
+        if any(x in line for x in ("nvidia","geforce")):
             return "nvidia"
-        elif "ATI" in line:
+        if any(x in line for x in ("ati","radeon")):
             return "ati"
-        elif "Intel" in line:
+        elif "intel" in line:
             return "intel"
-        elif "VIA" in line:
+        elif "via" in line:
             return "via"
-        elif "VMware" in line:
+        elif "vmware" in line:
             return "vmware"
         else:
             return "other"
