commit 714b61d18dd9dd98522946e867cff243dad220e7
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Thu Jul 28 09:16:09 2011 +0400

    Fix getPartitionSize.

diff --git a/pym/cl_utils.py b/pym/cl_utils.py
index f477aea..23616bc 100644
--- a/pym/cl_utils.py
+++ b/pym/cl_utils.py
@@ -1164,7 +1164,7 @@ def getPartitionSize(dev):
         suffix = filter(lambda x:size >x[0],suffix)
         if suffix:
             suffix = suffix[-1]
-            printSize = size / (suffix[0]/10)
+            printSize = int(size / (float(suffix[0])/10))
             printSizeTail = printSize % 10
             printSize = printSize / 10
             if suffix[2] and printSizeTail:
