diff --git a/calculate/lib/format/patch.py b/calculate/lib/format/patch.py
index 3d63b1e..1affb3e 100644
--- a/calculate/lib/format/patch.py
+++ b/calculate/lib/format/patch.py
@@ -106,7 +106,7 @@ class patch(_error):
                             return False
                     else:
                         text = ""
-                    dataList.append((regex, text))
+                    dataList.append((regex, text.encode('utf-8')))
                     regex = False
         for regex, text in dataList:
             # Замены в тексте конфигурационного файла
