commit 5877f4b4f77ef766b9afdfdbef68b84cd79dcbe0
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Mon Mar 5 17:34:59 2012 +0400

    Fix detect driver from xorg.conf

diff --git a/pym/cl_fill.py b/pym/cl_fill.py
index d183877..fb21ae7 100644
--- a/pym/cl_fill.py
+++ b/pym/cl_fill.py
@@ -427,8 +427,8 @@ class fillVars(varsShare):
             matchSect = re.search(r'Section "Device".*?EndSection',
                                   open('/etc/X11/xorg.conf').read(),re.S)
             if matchSect:
-                resDriver = re.search(r'^\S*Driver\s*"([^"]+)"',
-                                      matchSect.group(0),re.S)
+                resDriver = re.search(r'^\s*Driver\s*"([^"]+)"',
+                                      matchSect.group(0),re.M)
                 if resDriver and resDriver.group(1) in list_available_drivers:
                     return resDriver.group(1)
 
