commit 1c1b5f8544ffa79eafc91d50515de1a8cd47121d
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Thu Mar 26 17:07:17 2015 +0300

    Пропуск шаблонов с указанным env параметром

diff --git pym/cl_template.py pym/cl_template.py
index d9bfdee..a822251 100644
--- pym/cl_template.py
+++ pym/cl_template.py
@@ -4233,6 +4233,9 @@ re.M|re.S)
                                templateDirFile)
             return ("", False, [])
 
+        if " env=" in textTemplate:
+            return ("", False, [])
+
         # Заменяем переменные на их значения
         textTemplate = self.applyVarsTemplate(textTemplate, templateDirFile)
 
@@ -4878,6 +4881,8 @@ re.M|re.S)
         self.closeTemplFile()
         # Флаг копирования шаблона в конфигурационный файл
         flagCopyTemplate = True
+        if " env=" in self.textTemplate.partition('\n')[0]:
+            return False
         # Тип шаблона бинарный или текстовый
         templateFileType = self.getTemplateType()
         if templateFileType != "bin":
