diff --git pym/cl_utils.py pym/cl_utils.py
index f477aea..11bf591 100644
--- pym/cl_utils.py
+++ pym/cl_utils.py
@@ -1164,7 +1164,7 @@ def getPartitionSize(dev):
         suffix = filter(lambda x:size >x[0],suffix)
         if suffix:
             suffix = suffix[-1]
-            printSize = size / (suffix[0]/10)
+            printSize = int(size / (float(suffix[0])/10))
             printSizeTail = printSize % 10
             printSize = printSize / 10
             if suffix[2] and printSizeTail:
@@ -1227,7 +1227,7 @@ def getRaidPartitions(raidpath):
 
 def getPartitionType(prop):
     """Get type of dos part table (primary,extended or logical)"""
-    if prop.get('ID_PART_TABLE_TYPE') == 'dos':
+    if prop.get('ID_PART_ENTRY_SCHEME') == 'dos':
         partId = prop.get('ID_PART_ENTRY_TYPE','')
         partNumber = prop.get('ID_PART_ENTRY_NUMBER','')
         if partId and partNumber:
@@ -1301,3 +1301,21 @@ def getOsProberHash(getContentFunc=None):
     else:
         osProberHash = {}
     return osProberHash
+
+def refreshLVM():
+    """Run command which refresh information about LVM"""
+    vgscan = getProgPath('/sbin/vgscan')
+    vgchange = getProgPath('/sbin/vgchange')
+    lvchange = getProgPath('/sbin/lvchange')
+    
+    if vgscan and vgchange and lvchange:
+        process(vgscan).success()
+        process(vgchange,'-ay','--refresh').success()
+        for group in getLvmGroups():
+            process(lvchange,'-ay','--refresh',group).success()
+
+def refreshUdev():
+    """Run command which refresh information about device in udev"""
+    udevadm = getProgPath('/sbin/udevadm')
+    if udevadm:
+        process(udevadm,"trigger","--subsystem-match","block").success()
