diff --git calculate/lib/cl_template.py calculate/lib/cl_template.py
index c42dfbb..cda5c00 100644
--- calculate/lib/cl_template.py
+++ calculate/lib/cl_template.py
@@ -5457,6 +5457,8 @@ gettext -d cl_template "$*"
                 return ([], False)
             if not objHeadNew.body.strip():
                 preReturn(pathProg)
+                if "protected" in objHeadNew.params:
+                    self.protectedFiles += applyFiles
                 return (applyFiles, False)
         else:
             applyFiles = [pathOldFile]
