diff --git pym/cl_distr.py pym/cl_distr.py
index 2d8e51e..631f673 100644
--- pym/cl_distr.py
+++ pym/cl_distr.py
@@ -1290,12 +1290,21 @@ class IsoDistributive(Distributive):
         if not path.exists(mkisoUtil):
             raise DistributiveError(_("Failed to create the iso image") +
                 " : %s"%_("command '%s' not found")%mkisoUtil)
+        efiImage = "boot/grub/efi.img"
+        if os.path.exists(os.path.join(directory,efiImage)):
+            params = ["-J","-R","-l","-no-emul-boot","-boot-load-size 4",
+                      "-udf","-boot-info-table","-b isolinux/isolinux.bin",
+                      "-c isolinux/boot.cat","-eltorito-alt-boot",
+                      "-no-emul-boot","-eltorito-platform efi",
+                      "-eltorito-boot %s"%efiImage]
+        else:
+            params = ["-b isolinux/isolinux.bin","-no-emul-boot",
+                    "-boot-load-size 4","-boot-info-table","-iso-level 4",
+                    "-hide boot.catalog"]
         res,errmes = self.runOsCommand(
             "%(progname)s %(params)s -o %(target)s %(source)s/"%
             {'progname':mkisoUtil,
-             'params':" ".join(["-b isolinux/isolinux.bin","-no-emul-boot",
-                    "-boot-load-size 4","-boot-info-table","-iso-level 4",
-                    "-hide boot.catalog"]),
+             'params':" ".join(params),
              'target':file,
              'source':directory})
         if res == 0:
