commit 42857b2f7ed9790b1e71f6f1cc0bcc4a5c1b2701
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Thu Aug 23 15:03:30 2012 +0400

    Fix patch utf-8

diff --git a/pym/format/patch.py b/pym/format/patch.py
index ba11f6b..bb55a00 100644
--- a/pym/format/patch.py
+++ b/pym/format/patch.py
@@ -109,7 +109,7 @@ class patch(_error):
                             return False
                     else:
                         text = ""
-                    dataList.append((regex, text))
+                    dataList.append((regex, text.encode('utf-8')))
                     regex = False
         for regex, text in dataList:
             # Замены в тексте конфигурационного файла
