commit 669ed469a69a601c8cc65b761820b0e860abc447
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Oct 24 09:33:52 2012 +0400

    Fix import ClMessage

diff --git a/calculate/lib/cl_progressbar.py b/calculate/lib/cl_progressbar.py
index 5b88a84..152d87b 100644
--- a/calculate/lib/cl_progressbar.py
+++ b/calculate/lib/cl_progressbar.py
@@ -132,3 +132,4 @@ try:
 except:
     ControlledProgressDialog = None
     ClProgressDialog = None
+    ClMessageBox = None
