diff --git calculate/lib/cl_template.py calculate/lib/cl_template.py
index 14912c6..6c98b35 100644
--- calculate/lib/cl_template.py
+++ calculate/lib/cl_template.py
@@ -3948,7 +3948,7 @@ re.M|re.S)
     def executeTemplate(self, code, execPath):
         """Execute template"""
         p = process(execPath,lang=self.objVar.Get('os_locale_locale'),
-                    envdict=os.environ)
+                    envdict=dict(os.environ))
         if "/bin/bash" in code.partition('\n')[0]:
             p.write("""function translate() {
 gettext -d cl_template "$*"
diff --git calculate/lib/datavars.py calculate/lib/datavars.py
index ffae995..bf36bdc 100644
--- calculate/lib/datavars.py
+++ calculate/lib/datavars.py
@@ -263,15 +263,15 @@ class Variable:
         section = self.parent.iniCache[self.name]['section']
         value = self.parent.unserialize(self.type,
                                  value.encode('utf-8'))
-        try:
-            if self.mode == READONLY:
-                raise VariableError(
-                    _("Attempt to rewrite readonly variable %s")%
-                    varname)
-            else:
+        if self.mode == READONLY:
+            raise VariableError(
+                _("Attempt to rewrite readonly variable %s")%
+                self.name)
+        else:
+            try:
                 self._set(value)
-        except Exception as e:
-            return False
+            except Exception as e:
+                return False
         return True
 
     def _get(self):
