diff --git pym/cl_fill.py pym/cl_fill.py
index c2300b7..31bbc61 100644
--- pym/cl_fill.py
+++ pym/cl_fill.py
@@ -715,3 +715,18 @@ class fillVars(varsShare):
     def get_cl_ca_path(self):
         """CA path"""
         return '/var/calculate/ssl/main'
+
+    def get_os_clock_timezone(self):
+        """Current clock timezone"""
+        zoneinfodir = "/usr/share/zoneinfo/"
+        localtimefile = "/etc/localtime"
+        timezonefile = "/etc/timezone"
+        # try get timezone from kernel calculate param
+        timezone = self.getValueFromCmdLine("calculate",2)
+        if timezone and \
+            path.exists(path.join(zoneinfodir,timezone)):
+            return timezone
+        # get timezone from /etc/timezone
+        if path.exists(timezonefile):
+            return open(timezonefile,"r").read().strip()
+        return "UTC"
diff --git pym/cl_template.py pym/cl_template.py
index b06ddee..dbf39cd 100644
--- pym/cl_template.py
+++ pym/cl_template.py
@@ -4773,9 +4773,9 @@ variable 'cl_name'"))
                     preReturn(pathProg)
                     return ([], False)
             if not "/" == pathOldFile[0]:
-                applyFiles = [prevOldFile,os.path.join(pathLink,pathOldFile)]
+                applyFiles = [prevOldFile]#,os.path.join(pathLink,pathOldFile)]
             else:
-                applyFiles = [prevOldFile,pathOldFile]
+                applyFiles = [prevOldFile]#,pathOldFile]
             try:
                 os.symlink(pathOldFile, prevOldFile)
             except:
diff --git pym/cl_utils.py pym/cl_utils.py
index 5fea955..6da750d 100644
--- pym/cl_utils.py
+++ pym/cl_utils.py
@@ -1339,3 +1339,17 @@ def getPasswdUsers(minId=1000,maxId=65000):
         sortUsers.sort()
         retList =  ["root"] + sortUsers
     return retList
+
+def getCmdLineParam(paramName):
+    """Get value of param /proc/cmdline. If param not found then empty.
+    """
+    cmdLine = '/proc/cmdline'
+    paramName = "%s="%paramName
+    params = \
+        map(lambda x:x.partition('=')[2],
+        filter(lambda x:x.startswith(paramName),
+        readFile(cmdLine).split(' ')))
+    if params:
+        return params[-1]
+    else:
+        return ""
diff --git pym/cl_vars.py pym/cl_vars.py
index 3194ff8..0466cc5 100644
--- pym/cl_vars.py
+++ pym/cl_vars.py
@@ -103,6 +103,9 @@ class Data:
     # system (desktop or server)
     os_linux_system = {'hide':True}
 
+    # current timezone
+    os_clock_timezone = {'hide':True}
+
     # motherboard model
     hr_board_model = {'hide':True}
 
