diff --git calculate/lib/utils/device.py calculate/lib/utils/device.py
index 4f89891..d9eeee6 100644
--- calculate/lib/utils/device.py
+++ calculate/lib/utils/device.py
@@ -405,7 +405,7 @@ def refreshUdev(onlyTrigger=False):
     """Run command which refresh information about device in udev"""
     if not onlyTrigger:
         getUdevDeviceInfo.clearCache()
-    udevadm = getProgPath('/sbin/udevadm')
+    udevadm = getProgPath('/bin/udevadm')
     if udevadm:
         blkidFile = '/etc/blkid.tab'
         try:
@@ -414,6 +414,7 @@ def refreshUdev(onlyTrigger=False):
         except:
             pass
         process(udevadm,"trigger","--subsystem-match","block").success()
+        process(udevadm,"settle","--timeout=15").success()
 
 def getCommonDeviceName(dev):
     """
