diff --git a/i18n/cl_lib_ru.mo b/i18n/cl_lib_ru.mo
index c6c3102..f57a6af 100644
Binary files a/i18n/cl_lib_ru.mo and b/i18n/cl_lib_ru.mo differ
diff --git a/pym/cl_fill.py b/pym/cl_fill.py
index 35ad6c6..08b61ab 100644
--- a/pym/cl_fill.py
+++ b/pym/cl_fill.py
@@ -499,9 +499,11 @@ class fillVars(varsShare):
             return ""
         width = int(widthVal)
         height = int(heightVal)
-        res = [(1024,768),
+        res = [(1024,600),
+               (1024,768),
                (1280,1024),
                (1280,800),
+               (1366,768),
                (1440,900),
                (1600,1200),
                (1680,1050),
