commit 2beec31a78dc395ac44943dfe3c5afd528bbc388
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Wed Jun 27 23:28:42 2012 +0400

    Fix import cl_template

diff --git a/calculate/lib/variables/env.py b/calculate/lib/variables/env.py
index a1ae628..849563f 100644
--- a/calculate/lib/variables/env.py
+++ b/calculate/lib/variables/env.py
@@ -20,6 +20,7 @@ from os import path
 from itertools import *
 from calculate.lib.datavars import (TableVariable,Variable,ReadonlyVariable,
                                    VariableError)
+import calculate.lib.cl_template as cl_template
 from calculate.lib.utils.files import listDirectory,process,PIPE
 
 from calculate.lib.cl_lang import setLocalTranslate
