diff --git calculate/lib/cl_template.py calculate/lib/cl_template.py
index 24ad307..08f796f 100644
--- calculate/lib/cl_template.py
+++ calculate/lib/cl_template.py
@@ -2799,6 +2799,36 @@ class templateFunction(_error, _warning, _shareTemplate, _shareTermsFunction):
         else:
             return ""
 
+    def funcPkgw(self, funArgv, resS, localVars, textTemplateTmp, nameTemp):
+        """Функция шаблона pkgw(), возвращает аргумент если программа установлена"""
+        # Название программы
+        import time
+        t = time.time()
+        nameProg = funArgv.replace(" ","")
+        origProg = nameProg
+        # Замена функции в тексте шаблона
+        replace = ""
+        if "/" in nameProg:
+            category, spl, nameProg = nameProg.partition("/")
+            nameProg, spl, slot = nameProg.partition(":")
+            if not category in self.installCategory:
+                self.getInstallPkgGentoo(category=category)
+                self.installCategory.append(category)
+            replace = self.pkg(nameProg, slot=slot or None)
+        else:
+            if not self.flagAllPkgScan:
+                self.getInstallPkgGentoo()
+                templateFunction.flagAllPkgScan = True
+            nameProg,spl,slot = nameProg.partition(":")
+            replace = self.pkg(nameProg, 
+                               slot=slot)
+        if replace:
+            replace = origProg
+        textTemplateTmp = textTemplateTmp[:resS.start()] + replace +\
+                textTemplateTmp[resS.end():]
+        print "TIME",time.time() -t
+        return textTemplateTmp
+
     def funcPkg(self, funArgv, resS, localVars, textTemplateTmp, nameTemp):
         """Функция шаблона pkg(), выдает номер версии программы"""
         # Название программы
@@ -2810,6 +2840,7 @@ class templateFunction(_error, _warning, _shareTemplate, _shareTermsFunction):
             nameProg, spl, slot = nameProg.partition(":")
             if not category in self.installCategory:
                 self.getInstallPkgGentoo(category=category)
+                self.installCategory.append(category)
             replace = self.pkg(nameProg, slot=slot or None)
         else:
             if not self.flagAllPkgScan:
