commit 267519aa6b09b24cb45e965bbbdb8448eb8dde6b
Author: Mike Hiretsky <mh@calculate.ru>
Date:   Tue May 17 09:57:50 2011 +0400

    Add fifo to scan files.

diff --git a/pym/cl_profile.py b/pym/cl_profile.py
index cad373a..618ef27 100644
--- a/pym/cl_profile.py
+++ b/pym/cl_profile.py
@@ -1940,6 +1940,7 @@ class _file(_error):
                 self.files = []
                 self.links = []
                 self.sockets = []
+                self.fifo = []
         def getFilesDir(dirP, dirname,names):
             for nameFile in names:
                 absNameFile = dirname + "/" + nameFile
@@ -1953,6 +1954,8 @@ class _file(_error):
                     dirP.dirs.append(absNameFile)
                 elif stat.S_ISSOCK(os.stat(absNameFile)[stat.ST_MODE]):
                     dirP.sockets.append(absNameFile)
+                elif stat.S_ISFIFO(os.stat(absNameFile)[stat.ST_MODE]):
+                    dirP.fifo.append(absNameFile)
         for profileDir in profilesDirs:
             if profileDir:
                 dirP = dirProf()
