diff --git calculate/lib/cl_template.py calculate/lib/cl_template.py
index 36d71ed..8c6fb2e 100644
--- calculate/lib/cl_template.py
+++ calculate/lib/cl_template.py
@@ -3047,8 +3047,6 @@ class templateFunction(_error, _warning, _shareTemplate, _shareTermsFunction):
         """
         Вывод успешного сообщения
         """
-        # TODO: need remove
-        return textTemplateTmp
         self.printSUCCESS(_(funArgv))
         textTemplateTmp = textTemplateTmp[:resS.start()] + \
                 textTemplateTmp[resS.end():]
@@ -3058,8 +3056,6 @@ class templateFunction(_error, _warning, _shareTemplate, _shareTermsFunction):
         """
         Вывод сообщения с предупреждением
         """
-        # TODO: need remove
-        return textTemplateTmp
         self.printWARNING(_(funArgv))
         textTemplateTmp = textTemplateTmp[:resS.start()] + \
                 textTemplateTmp[resS.end():]
@@ -3069,8 +3065,6 @@ class templateFunction(_error, _warning, _shareTemplate, _shareTermsFunction):
         """
         Вывод сообщения с ошибкой
         """
-        # TODO: need remove
-        return textTemplateTmp
         self.printERROR(_(funArgv))
         textTemplateTmp = textTemplateTmp[:resS.start()] + \
                 textTemplateTmp[resS.end():]
diff --git calculate/lib/utils/ip.py calculate/lib/utils/ip.py
index aa523c2..b704c76 100644
--- calculate/lib/utils/ip.py
+++ calculate/lib/utils/ip.py
@@ -222,6 +222,14 @@ def getMac(iface):
     sockfd.close()
     return ":".join(['%02X' % i for i in mac])
 
+def getOperState(iface):
+    """
+    Get interface state up or down
+    """
+    if readFile("/sys/class/net/%s/operstate"%iface) == "down":
+        return "down"
+    return "up"
+
 def isOpenPort(ip,port):
     """
     Test if an [ip:port] is open
