diff --git a/pym/cl_vars_share.py b/pym/cl_vars_share.py
index 3ec525b..7bd048d 100644
--- a/pym/cl_vars_share.py
+++ b/pym/cl_vars_share.py
@@ -543,7 +543,7 @@ class varsShare:
         gentooFile = path.join(systemroot,"etc/gentoo-release")
         systemVersion = ""
         flagGentoo = False
-        reVer=re.compile("^(\d+\.)*\d$",re.S)
+        reVer=re.compile("^(\d+\.)*\d+$",re.S)
         if path.exists(gentooFile):
             gentooLink = path.join(systemroot,"etc/make.profile")
             if path.islink(gentooLink):
