diff --git pym/cl_utils.py pym/cl_utils.py
index 6da750d..2450693 100644
--- pym/cl_utils.py
+++ pym/cl_utils.py
@@ -1353,3 +1353,14 @@ def getCmdLineParam(paramName):
         return params[-1]
     else:
         return ""
+
+def getSupportArch():
+    """Get supported architectures by processor.
+
+    Is support processor x86_64 else only i686.
+    """
+    if filter(lambda x:x.startswith('flags') and " lm " in x,
+        readLinesFile('/proc/cpuinfo')):
+        return ['i686','x86_64']
+    else:
+        return ['i686']
