diff --git a/pym/cl_fill.py b/pym/cl_fill.py
index c37b506..0bf69c3 100644
--- a/pym/cl_fill.py
+++ b/pym/cl_fill.py
@@ -291,7 +291,7 @@ class fillVars(glob_attr):
         makeprofile = '/etc/make.profile'
         if os.path.exists(makeprofile):
             link = os.readlink(makeprofile)
-            reMakeProfileLink = re.compile('calculate/(desktop|server)/(%s)'%
+            reMakeProfileLink = re.compile('/calculate/(desktop|server)/(%s)/'%
                         "|".join(self.dictLinuxName.keys()),re.S)
             shortnameSearch = reMakeProfileLink.search(link)
             if shortnameSearch:
