diff --git a/pym/update_config/cl_update_config.py b/pym/update_config/cl_update_config.py
index 69b8ae1..0878dfd 100644
--- a/pym/update_config/cl_update_config.py
+++ b/pym/update_config/cl_update_config.py
@@ -238,6 +238,7 @@ class updateSystemConfigs(shareUpdateConfigs):
             if os.path.exists(realPath):
                 scanObj.scanningDirectory(realPath)
         configFiles = map(lambda x: x.partition(configPath)[2], configFiles)
+        configFiles = map(lambda x: pathJoin('/',x), configFiles)
         return configFiles
 
     def createDir(self, configPath, dstDir):
diff --git a/setup.py b/setup.py
index 3313700..338fcdc 100755
--- a/setup.py
+++ b/setup.py
@@ -70,7 +70,8 @@ setup(
                 'calculate-lib.pym.client',
                 'calculate-lib.pym.update_config'],
     data_files = [("/usr/share/calculate/i18n",['i18n/cl_lib_ru.mo']),
-                  ("/var/calculate/remote",[])],
+                  ("/var/calculate/remote", []),
+                  ("/var/log/calculate", [])],
     scripts=["./scripts/cl-update-config"],
     cmdclass={'build_scripts':cl_build_scripts,
               'install_scripts':cl_install_scripts},
