diff --git a/microblogs/twitter/twittereditaccount.cpp b/microblogs/twitter/twittereditaccount.cpp
index 5b1c4d2..87744ed 100644
--- a/microblogs/twitter/twittereditaccount.cpp
+++ b/microblogs/twitter/twittereditaccount.cpp
@@ -41,8 +41,8 @@ along with this program; if not, see http://www.gnu.org/licenses/
 #include <KInputDialog>
 
 #ifdef OAUTH
-const char * twitterConsumerKey = "VyXMf0O7CvciiUQjliYtYg";
-const char * twitterConsumerSecret = "uD2HvsOBjzt1Vs6SnouFtuxDeHmvOOVwmn3fBVyCw0";
+const char * twitterConsumerKey = "irARiVrv825Fl2oZaYiQA";
+const char * twitterConsumerSecret = "qYZa8Dszrow6vry4iwtSm1zTidnW7ii8IQztgXWZY";
 #endif
 
 TwitterEditAccountWidget::TwitterEditAccountWidget(TwitterMicroBlog *microblog,
