# Calculate format=diff
diff -urpN gcc-3.3.6.orig/libstdc++-v3/Makefile.am gcc-3.3.6/libstdc++-v3/Makefile.am
--- gcc-3.3.6.orig/libstdc++-v3/Makefile.am	2003-07-04 12:53:54.000000000 -0700
+++ gcc-3.3.6/libstdc++-v3/Makefile.am	2018-06-07 20:33:51.747397861 -0700
@@ -25,7 +25,7 @@
 AUTOMAKE_OPTIONS = 1.3 cygnus
 MAINT_CHARSET = latin1
 
-SUBDIRS = include libio libmath libsupc++ src po testsuite 
+SUBDIRS = include libio libmath libsupc++ src po
 
 mkinstalldirs = $(SHELL) $(toplevel_srcdir)/mkinstalldirs
 
@@ -38,13 +38,11 @@ PWD_COMMAND = $${PWDCMD-pwd}
 # compile time, execution time, and binary size.
 check-script: $(top_builddir)/mkcheck
 	-(chmod + $(top_builddir)/mkcheck; \
-	  cd testsuite; \
-	  @glibcpp_builddir@/mkcheck 0)
+	  cd testsuite
 
 check-script-install: $(top_builddir)/mkcheck
 	-(chmod + $(top_builddir)/mkcheck; \
-	  cd testsuite; \
-	  @glibcpp_builddir@/mkcheck 1)
+	  cd testsuite
 
 # These rules are messy, but are hella worth it.
 doxygen:
Binary files gcc-3.3.6.orig/libstdc++-v3/.Makefile.am.swp and gcc-3.3.6/libstdc++-v3/.Makefile.am.swp differ
diff -urpN gcc-3.3.6.orig/libstdc++-v3/Makefile.in gcc-3.3.6/libstdc++-v3/Makefile.in
--- gcc-3.3.6.orig/libstdc++-v3/Makefile.in	2003-07-04 12:53:54.000000000 -0700
+++ gcc-3.3.6/libstdc++-v3/Makefile.in	2018-06-07 20:34:32.451011377 -0700
@@ -147,7 +147,7 @@ toplevel_srcdir = @toplevel_srcdir@
 AUTOMAKE_OPTIONS = 1.3 cygnus
 MAINT_CHARSET = latin1
 
-SUBDIRS = include libio libmath libsupc++ src po testsuite 
+SUBDIRS = include libio libmath libsupc++ src po
 
 mkinstalldirs = $(SHELL) $(toplevel_srcdir)/mkinstalldirs
 
@@ -488,13 +488,11 @@ mostlyclean distclean maintainer-clean
 # compile time, execution time, and binary size.
 check-script: $(top_builddir)/mkcheck
 	-(chmod + $(top_builddir)/mkcheck; \
-	  cd testsuite; \
-	  @glibcpp_builddir@/mkcheck 0)
+	  cd testsuite
 
 check-script-install: $(top_builddir)/mkcheck
 	-(chmod + $(top_builddir)/mkcheck; \
-	  cd testsuite; \
-	  @glibcpp_builddir@/mkcheck 1)
+	  cd testsuite
 
 # These rules are messy, but are hella worth it.
 doxygen:
